---
title: "19Q1"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the [Broad's Avana CRISPR](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017), the [Cancer Therapeutics Response Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (CTRP) drug screen dataset (v2) (Basu et al., 2013; M. G. Rees et al., 2016; B. Seashore-Ludlow et al., 2015), the [Genomics of Drug Sensitivity in Cancer](https://www.cancerrxgene.org/downloads) (GDSC) drug screen dataset (Garnett et al., 2012; Yang et al., 2012) , and the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) (CCLE) drug screen dataset (J. Barretina et al., 2012; T. C. C. L. E. Consortium & Consortium, 2015).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

Annotation files for mutation status, copy number, and gene expression of the CCLE and CTRP datasets (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/). Note that the group that conducted the GDSC screen also collected their own genomic annotation information for mutation status, copy number, and gene expression, but we did not use it.

# Set up

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

```{r}
library(crunch)
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(kableExtra)
library(naniar)
library(openxlsx)
library(reshape2)
library(tidyverse)
library(VennDiagram)

theme_set(theme_light())
```

```{r functions}
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) {
    df[[i]] = as.character(df[[i]])
  }
  return(df)
}

MergeOmicsG2P <- function(database, outfile) {
  # Merge mutation status, CN, and GE omics data with drugscreen results and metadata
  if(database %in% c("CCLE", "CTRP", "GDSC")) {
    filt <- auc %>% filter(Database == database) %>% select(Cell_line, Histology, TCGA_classification, CCLE_Name, Broad_ID, Drug, AUC_IC50)
    print(paste0("Processing ", database, " data."))
    colnames(filt)[colnames(filt) == "AUC_IC50"] <- "Score"

    # Get all G2Pgene-cell line combinations
    grid <- expand.grid("Cell_line" = unique(filt$Cell_line), "Gene" = g2p_genes)
    data <-  merge(filt, grid, by = "Cell_line", all.y = TRUE)
    print("Merged grid.")
    # Merge metadata
    data <-  merge(data, select(ccl_meta, Cell_line, Histology, CCLE_Name, Broad_ID, Primary.Disease), by = c("Cell_line", "Histology", "CCLE_Name", "Broad_ID"), all.x = TRUE)
    ## Add columns for G2P drug-gene and drug-gene-lineage sorting
    data$Drug_Gene <- paste0(data$Drug, "_", data$Gene)
    data$G2P_Drug_Gene <- ifelse(data$Drug_Gene %in% g2p_druggenes, "Yes", "No")
    data$Drug_Gene_Lin <- paste0(data$Drug, "_", data$Gene, "_", data$Primary.Disease)
    data$G2P_Drug_Gene_Lin <- ifelse(data$Drug_Gene_Lin %in% g2p_dgls, "Yes", "No")
    print("Merged metadata.")
  }
  if(database == "CRISPR") {
    print(paste0("Processing ", database, " data."))
    # Melt CRISPR dataset for merging
    data <- melt(crispr_all, id.vars = "Broad_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")
    # Merge cell line metadata
    data <- merge(data, select(ccl_meta, Cell_line, Histology, CCLE_Name, Broad_ID, Primary.Disease), by = "Broad_ID", all.x = TRUE)
    data$Drug_Gene <- paste0(data$Drug, "_", data$Gene)
    data$Drug_Gene_Lin <- paste0(data$Drug, "_", data$Gene, "_", data$Primary.Disease)
    print("Merged metadata.")
  }
  
  # Merge mutation calls
  data <- merge(data, maf_df_filt, by = c("Gene", "Broad_ID"), all.x = TRUE)
  data$Mutation_Status <- ifelse(is.na(data$Mutation_Status), 0, data$Mutation_Status)
  ## Identify point mutations
  data$PointMutation <- ifelse(is.na(data$PointMutation), 1, data$PointMutation)
  ## Summarize number of mutant and Wildtype cell lines
  data_muts_summ <- data %>% group_by(Gene) %>% summarize(N_Wildtype = length(Mutation_Status) - sum(Mutation_Status == 1), N_Mutant = sum(Mutation_Status == 1))
  ## Merge test results back into full dataset, which restores information lost in the summarization
  data <- merge(data_muts_summ, data, by = "Gene")
  ## Add color columns for plotting
  if(database == "CCLE") {
    data$Color <- ifelse(data$Mutation_Status == 0, "palegreen3", "springgreen4")
  }
  if(database == "CTRP") {
    data$Color <- ifelse(data$Mutation_Status == 0, "slategray3", "steelblue4")
  }
  if(database == "GDSC") {
    data$Color <- ifelse(data$Mutation_Status == 0, "paleturquoise3", "turquoise4")
  }
  if(database == "CRISPR") {
    data$Color <- ifelse(data$Mutation_Status == 0, "thistle", "darkorchid")
  }
  data$Color <- factor(data$Color)
  print("Merged mutation calls.")

  # Merge GE data
  data <- merge(data, ge_filt, by = c("Gene", "Broad_ID"), all.x = TRUE)
  print("Merged gene expression data.")
  
  # Merge CN data
  data <- merge(data, cn_filt, by = c("Gene", "Broad_ID"), all.x = TRUE)
  print("Merged copy number data.")
  
  # Drop cell lines not in DepMap
  data <- filter(data, !is.na(Broad_ID))
  
  # Factorize gene
  data$Gene <- factor(data$Gene)
  
  # Save and return dataframe
  saveRDS(data, outfile, compress = "xz")
  return(data)
}

FilterForPointMuts <- function(dataframe, outfile) {
  # Filter for point mutations
  ptmuts <- filter(dataframe, PointMutation == 1)
  # Remove columns that lead to duplicate mutation entries
  ptmuts <- subset(ptmuts, select = -c(Variant_Classification, Reference_Allele, Tumor_Seq_Allele1, PointMutation))
  ptmuts$Gene <- ordered(ptmuts$Gene, levels = g2p_genes)
  ptmuts <- unique(ptmuts)
  saveRDS(ptmuts, outfile, compress = "xz")
  return(ptmuts)
}

adj_signif <- function(df, alpha) {
  # Takes df from compare_means and creates a signif code column for FDR-adjusted p-vals
  df$p.signif.adj <- ifelse(df$p.adj <= alpha, "*", NA)
  df$p.signif <- ifelse(df$p.signif == "ns", NA, df$p.signif)
  df$p.short <- formatC(df$p, format = "g", digits =  2)
  df$p.adj.short <- formatC(df$p.adj, format = "g", digits =  2)
  return(df)
}

WilcoxonTestDF <- function(dataframe, measure_var, grouping, outfile) {
  require(ggpubr)
  signif <- compare_means(Score ~ Mutation_Status, group.by = grouping, data = dataframe, method = "wilcox.test", p.adjust.method = "BH")
  signif <- adj_signif(signif, alpha = 0.05)
  signif <- signif[order(signif$p),]
  saveRDS(signif, outfile)
  return(signif)
}
```

# Data

--------------------------------------------------------------------------------

Note that the new DepMap 19Q1 release made multiple changes to variable names:

- For their universal ccancer cell line nomenclature, they reverted back to `Broad_ID` from 18Q3 rather than `DepMap_ID` that was used in 19Q1
- The `Primary.Disease` column of the cell line metadata file is more consistently named (`Subtype.Disease` now has the specific disease annotations), though there are a couple capitalization inconsistencies
- MAF file no longer has `CCLE_Name` column and has a duplicate column of the Broad IDs (i.e. `DepMap_ID` and `Tumor_Sample_Barcode`)

And changes to the data processing pipelines (the following is taken verbatim from the [DepMap data portal](https://depmap.org/portal/download/)):

- **Expression**: We use the GTEx pipeline. Since last release, there have been some updates to the versions of different components of the pipeline and minor bug fixes on GTEx's end.
- **Mutation**: In this release, we additionally process DepMap WES samples through the CGA mutation calling pipeline adapted for cell lines and available on FireCloud. We have added a column for the allele counts from this pipeline run (CGA_WES_AC). Previously, the WES_AC column was generated from running the Van Allen mutation calling pipeline for SNPs + Mutect2 for INDELs. This column is now named VA_WES_AC.
- **Copy Number**: We have moved to using the newest version of the GATK pipeline.

## Cell line metadata

**D. Charytonowicz cell line converter**

This comprehensive cancer cell line information was curated by both Daniel Charytonowicz and myself.
```{r}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
lineage_converter <- unique(ccl_converter[, c("Broad_ID", "grouped_general_doid", "group_general_lineage_name")])
ccl_histologies <- read.delim("./data_munging/ccl_histologies_metadata.csv", sep = ",", header = TRUE, row.names = 1)
```

**DepMap metadata**

DepMap 19Q1 update fixed almost all of the inconsistencies of the Primary.Disease column, though there were come capitalization inconsistencies.
```{r}
ccl_meta <- read.delim("./data_munging/19Q1/DepMap-2019q1-celllines_v2_20190305.csv", sep = ",", header = TRUE, na.strings = c("", NA))
ccl_meta$Primary.Disease <- tolower(ccl_meta$Primary.Disease)
ccl_meta$Primary.Disease <- with(ccl_meta, ifelse(Primary.Disease == "colon/colorectal cancer", "colorectal cancer", Primary.Disease))
ccl_meta <- subset(ccl_meta, Primary.Disease != "unknown" & Primary.Disease != "primary cells" & !grepl("MERGED", CCLE_Name, fixed = TRUE))
ccl_meta$Cell_line <- as.character(lapply(ccl_meta$CCLE_Name, function(x) strsplit(as.character(x), split = "_")[[1]][1]))
ccl_meta$Histology <- tolower(gsub("^[a-z0-9]*_", "", ccl_meta$CCLE_Name, ignore.case = TRUE))
# Convert Primary.Disease entries to all lower case to fix factorization issue
ccl_meta$Primary.Disease <- as.factor(tolower(ccl_meta$Primary.Disease))
# Harmonize ID column name
colnames(ccl_meta)[colnames(ccl_meta) == "DepMap_ID"] <- "Broad_ID"

# Remove two rows that contain cell lines with two different histologies
# The histologies that are kept are verified through matching in Cellosaurus and Cell Model Passports
# KMH2, haematopoietic_and_lymphoid: https://web.expasy.org/cellosaurus/CVCL_1330, https://cellmodelpassports.sanger.ac.uk/passports/SIDM01018
# MS1, lung: https://web.expasy.org/cellosaurus/CVCL_1429, https://cellmodelpassports.sanger.ac.uk/passports/SIDM00245
ccl_meta <- subset(ccl_meta, CCLE_Name != "KMH2_THYROID" & CCLE_Name != "MS1_SKIN")
```

## G2P

```{r}
g2p <- read.delim("./data_munging/g2p_full_level_A_dataset_NEW_DEC_2_2018_withGroupedLineages.csv", sep = "\t", header = TRUE)
g2p[, 1:2] <- NULL
g2p <- merge(g2p, lineage_converter, by = "grouped_general_doid", all.x = TRUE)

g2p$Drug <- tolower(g2p$Drug)
g2p[g2p == -1] <- NA
g2p$Ref[g2p$Ref == "None"] <- NA
g2p$Alt[g2p$Alt == "None"] <- NA
g2p$Drug_Gene <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene), NA, paste0(g2p$Drug, "_", g2p$Gene))
g2p$Drug_Gene_Lin <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene) | is.na(g2p$group_general_lineage_name), NA, paste0(g2p$Drug, "_", g2p$Gene, "_", g2p$group_general_lineage_name))
g2p$position <- paste0("g.chr", g2p$Chromosome, ":", g2p$End, g2p$Ref, ">", g2p$Alt)
g2p$position <- ifelse(grepl("NA", g2p$position), NA, g2p$position)
g2p_filt <- subset(g2p, grepl("^CID", g2p$DrugID))

g2p_dgls <-  unique(g2p_filt$Drug_Gene_Lin[!is.na(g2p_filt$Drug_Gene_Lin)])
g2p_druggenes <- unique(g2p_filt$Drug_Gene[!is.na(g2p_filt$Drug_Gene)])
g2p_drugs <- unique(sapply(strsplit(g2p_dgls, "_", fixed = TRUE), function(x) x[1]))
g2p_genes <- unique(sapply(strsplit(g2p_dgls, "_", fixed = TRUE), function(x) x[2]))
g2p_lins <- unique(sapply(strsplit(g2p_dgls, "_", fixed = TRUE), function(x) x[3]))
g2p_druggenes
g2p_drugs
g2p_genes
g2p_lins

# Order and genes taken from pink heatmap from Daniel's analyses
# g2p_genes <- c("EGFR", "KRAS", "ERBB2", "ERBB3", "MET", "ALK", "ROS1", "PDGFRA", "ABL1", "KIT", "PDGFRB", "PDGFB", "BRAF", "G6PD", "RET", "BRCA1", "BRCA2", "UGT1A1", "TSC1", "TSC2", "AKT1", "IDH2", "DPYD", "ESR1", "CYP19A1", "TPMT", "FLT3", "NPM1", "DNMT3A", "PML", "PTCH1", "JAK2", "MYD88", "MGMT")
```

```{r, eval=FALSE}
# levels(g2p_filt$Phenotype.Description)
unique(g2p_filt$group_general_lineage_name)
levels(g2p_filt$group_general_lineage_name)

g2p_nonCID <- subset(g2p, !grepl("^CID", g2p$DrugID))
summ <- g2p_nonCID %>% group_by(Drug, DrugID) %>% tally() %>% summarize(DrugID = DrugID, n = n, percent = formatC(n / 402 * 100, digits = 6, format = "f"))
summ
```

## Omics

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Mutation calls (coding region, germline filtered):
```{r, eval=FALSE}
maf_raw <- read.delim("./data_munging/19Q1/depmap_19Q1_mutation_calls_20190305.csv.gz", header = TRUE, sep = ",", row.names = 1)
# Harmonize ID column name
colnames(maf_raw)[colnames(maf_raw) == "DepMap_ID"] <- "Broad_ID"
colnames(maf_raw)[colnames(maf_raw) == "Hugo_Symbol"] <- "Gene"
# Select columns
maf_df <- subset(maf_raw, select = c("Gene", "Broad_ID", "Variant_Classification", "Reference_Allele", "Tumor_Seq_Allele1"))
maf_df$Reference_Allele <- ifelse(maf_df$Reference_Allele == "-", 0, nchar(as.character(maf_df$Reference_Allele)))
maf_df$Tumor_Seq_Allele1 <- nchar(as.character(maf_df$Tumor_Seq_Allele1))
maf_df <- maf_df[maf_df$Variant_Classification != "Silent",]
# Add Mutation_Status column
maf_df$Mutation_Status <- 1
maf_df$PointMutation <- with(maf_df, ifelse(Reference_Allele == 1 | is.na(Reference_Allele), TRUE, ifelse(Reference_Allele == 0 & Tumor_Seq_Allele1 == 1, 1, 0)))

saveRDS(maf_df, "./data_munging/19Q1/maf_df_19Q1.rds", compress = "xz")
```

DepMap WES CN Data:
```{r, eval=FALSE}
cn <- read.delim("./data_munging/19Q1/public_19Q1_gene_cn_20190305.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "Broad_ID"
# Melt
cn_melt <- melt(data = cn, id.vars = "Broad_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Gene", value.name = "CN")

saveRDS(cn_melt, "./data_munging/19Q1/cn_melt_19Q1.rds", compress = "xz")
```

19Q1, CCLE RNAseq gene expression data (log2(TPM+1)):
```{r, eval=FALSE}
ge <- read.delim("./../htscreens_giant_files/CCLE_depMap_19Q1_TPM_20190305.csv.gz", header = TRUE, sep = ",", check.names = FALSE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(ge) <- gsub(" .*", "", colnames(ge_19Q1))
colnames(ge)[1] <- "Broad_ID"
# Melt
ge_melt <- melt(ge, id.vars = "Broad_ID", measure.vars = colnames(ge_19Q1)[2:ncol(ge_19Q1)], variable.name = "Gene", value.name = "TPM")

saveRDS(ge_melt, "./data_munging/19Q1/ge_melt_19Q1.rds", compress = "xz")
```

Load Rds:
```{r}
maf_df <- readRDS("./data_munging/19Q1/maf_df_19Q1.rds")
cn_melt <- readRDS("./data_munging/19Q1/cn_melt_19Q1.rds")
ge_melt <- readRDS("./data_munging/19Q1/ge_melt_19Q1.rds")
```

## Drug screens

### Merge omics

```{r}
test <- filter(maf_df, Gene %in% g2p_genes) %>% count(Gene, Broad_ID)
test <- test[order(test$n, decreasing = TRUE),]
```

`r nrow(filter(test, n != 1))`/`r nrow(test)` gene-cell line combinations have more than one mutation in the MAF file. This is corrected in the point mutation filtering done below.

**NEED TO RUN CTRP OVER WEEKEND**
```{r, eval=FALSE}
# Load in Pozdeyev et al. (2016) data
ccle_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CCLE_drug_response_metrics.csv.gz", sep = ",")
ctrp_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CTRP_drug_response_metrics.csv.gz", sep = ",")
gdsc_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/GDSC_drug_response_metrics.csv.gz", sep = ",")

auc <- bind_rows(ccle_all, ctrp_all, gdsc_all)

# Harmonize Pozdeyev cell line names with DepMap
auc$Drug <- tolower(auc$Drug_name)
auc$Cell_line <- toupper(gsub("-", "", auc$Cell_line))
auc$Cell_line <- ifelse(auc$Cell_line == "K052", "KO52", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "TI73", "T173", auc$Cell_line)
auc <- merge(ccl_histologies, auc, by = "Cell_line", all.y = TRUE)

# Omics metadata
maf_df_filt <- filter(maf_df, Gene %in% g2p_genes)
ge_filt <- filter(ge_melt, Gene %in% g2p_genes)
cn_filt <- filter(cn_melt, Gene %in% g2p_genes)

ccle <- MergeOmicsG2P(database = "CCLE", outfile = "./data_munging/rds/ccle_data_19Q1_g2pgenes.rds")
ctrp <- MergeOmicsG2P(database = "CTRP", outfile = "./../htscreens_giant_files/ctrp_data_19Q1_g2pgenes.rds")
gdsc <- MergeOmicsG2P(database = "GDSC", outfile = "./data_munging/rds/gdsc_data_19Q1_g2pgenes.rds")

ccle_ptmuts <- FilterForPointMuts(dataframe = ccle, outfile = "./data_munging/rds/ccle_data_19Q1_g2pgenes_ptmuts.rds")
ctrp_ptmuts <- FilterForPointMuts(dataframe = ctrp, outfile = "./../htscreens_giant_files/ctrp_data_19Q1_g2pgenes_ptmuts.rds")
gdsc_ptmuts <- FilterForPointMuts(dataframe = gdsc, outfile = "./data_munging/rds/gdsc_data_19Q1_g2pgenes_ptmuts.rds")
```

### Summaries

```{r, eval=FALSE}
ccle <- readRDS("./data_munging/rds/ccle_data_19Q1_g2pgenes.rds")
ctrp <- readRDS("./../htscreens_giant_files/ctrp_data_19Q1_g2pgenes.rds")
gdsc <- readRDS("./data_munging/rds/gdsc_data_19Q1_g2pgenes.rds")
ccle_ptmuts <- readRDS("./data_munging/rds/ccle_data_19Q1_g2pgenes_ptmuts.rds")
ctrp_ptmuts <- readRDS("./../htscreens_giant_files/ctrp_data_19Q1_g2pgenes_ptmuts.rds")
gdsc_ptmuts <- readRDS("./data_munging/rds/gdsc_data_19Q1_g2pgenes_ptmuts.rds")

# Testing
test_ccle <- ccle_ptmuts %>% count(Gene, Mutation_Status)
test_ctrp <- ctrp_ptmuts %>% count(Gene, Mutation_Status)
test_gdsc <- gdsc_ptmuts %>% count(Gene, Mutation_Status)
```

How many datasets screen each G2P drug?
```{r, eval=FALSE}
drug_summ <- data.frame(Drug = character(), Dataset_Count = integer(), stringsAsFactors = FALSE)

for(drug in g2p_drugs) {
  counter = 0
  if(drug %in% as.character(unique(ccle_data_g2p$Drug))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(ctrp_data_g2p$Drug))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(gdsc_data_g2p$Drug))) { counter <- counter + 1 }
  drug_summ <- rbind(drug_summ, data.frame(Drug = drug, Dataset_Count = counter))
}
```

## CRISPR

### Merge CRISPR data and annotations

```{r, eval=FALSE}
crispr_all <- read.delim("./data_munging/19Q1/gene_effect_corrected_20190305.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr_all) <- gsub(" .*", "", colnames(crispr_all))
colnames(crispr_all)[1] <- "Broad_ID"

# Melt CRISPR dataset for merging
crispr_melt <- melt(crispr_all, id.vars = "Broad_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")

# Merge cell line metadata
crispr_melt <- merge(crispr_melt, ccl_meta, by = "Broad_ID", all.x = TRUE)

# Merge mutation annotations
maf_filt_crispr <- filter(maf_df, Gene %in% as.character(unique(crispr_melt$Gene)))
maf_filt_crispr <- subset(maf_filt_crispr, select = -c(Variant_Classification, Reference_Allele, Tumor_Seq_Allele1))
maf_filt_crispr <- unique(maf_filt_crispr)
crispr_muts <- merge(crispr_melt, maf_filt_crispr, by = c("Gene","Broad_ID"), all.x = TRUE)
crispr_muts$Gene <- factor(crispr_muts$Gene)
crispr_muts$Mutation_Status <- ifelse(is.na(crispr_muts$Mutation_Status), 0, crispr_muts$Mutation_Status)
## Identify point mutations and wildtypes for testing
crispr_muts$PointMutation <- ifelse(is.na(crispr_muts$PointMutation), 1, crispr_muts$PointMutation)

# Summarize number of mutant and Wildtype cell lines
crispr_muts_summ <- crispr_muts %>% group_by(Gene) %>% summarize(N_Wildtype = length(Mutation_Status) - sum(Mutation_Status == 1), N_Mutant = sum(Mutation_Status == 1))
## Merge test results back into full dataset, which restores information lost in the summarization
crispr_data <- merge(crispr_muts_summ, crispr_muts, by = "Gene")
## Add Color column
crispr_data$Color <- ifelse(crispr_data$Mutation_Status == 0, "thistle", "darkorchid")
crispr_data$Color <- factor(crispr_data$Color)

# Cell line lineages
crispr_data <- merge(crispr_data, cn_melt, by = c("Gene", "Broad_ID"), all.x = TRUE)

# Gene expression (RPKM)
ge_crispr <- filter(ge_melt, Gene %in% as.character(unique(crispr_melt$Gene)))
crispr_data <- merge(crispr_data, ge_crispr, by = c("Gene", "Broad_ID"), all.x = TRUE)
saveRDS(crispr_data, "./../htscreens_giant_files/crispr_data_19Q1.rds", compress = "xz")
```

```{r, eval=TRUE}
crispr_data <- readRDS("./../htscreens_giant_files/crispr_data_19Q1.rds")

# Filter G2P CRISPR for point mutations
crispr_ptmuts <- filter(crispr_data, PointMutation == 1)
crispr_ptmuts <- subset(crispr_ptmuts, select = -c(PointMutation))
# Filter CRISPR for G2P genes in G2P
crispr_g2p <- filter(crispr_ptmuts, Gene %in% g2p_genes)
crispr_g2p <- unique(crispr_g2p)
crispr_g2p$Gene <- factor(crispr_g2p$Gene)
crispr_g2p$Mutation_Status <- factor(crispr_g2p$Mutation_Status)
crispr_g2p$Primary.Disease <- factor(crispr_g2p$Primary.Disease)
```

Test merging and filtering functions on CRISPR data:
```{r, eval=FALSE}
# crispr_all <- read.delim("./data_munging/19Q1/gene_effect_corrected_20190305.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)
# # Remove Entrez gene IDs from colnames
# colnames(crispr_all) <- gsub(" .*", "", colnames(crispr_all))
# colnames(crispr_all)[1] <- "Broad_ID"

# Omics metadata
maf_df_filt <- filter(maf_df, Gene %in% g2p_genes)
ge_filt <- filter(ge_melt, Gene %in% g2p_genes)
cn_filt <- filter(cn_melt, Gene %in% g2p_genes)

crispr_fxn <- MergeOmicsG2P(database = "CRISPR", outfile = "./../htscreens_giant_files/crispr_data_19Q1_g2pgenes_fxn.rds")
crispr_ptmuts_fxn <- FilterForPointMuts(dataframe = crispr_fxn, outfile = "./../htscreens_giant_files/crispr_data_19Q1_g2pgenes_ptmuts_fxn.rds")
```


# Wilcoxon tests

--------------------------------------------------------------------------------

```{r, eval=TRUE}
crispr_signif_gene <- WilcoxonTestDF(dataframe = crispr_data_g2p, grouping = "Gene", outfile = "./data_munging/rds/crispr_signif_g2p_gene.rds")
crispr_signif_lineage <- WilcoxonTestDF(dataframe = crispr_data_g2p, grouping = c("Gene", "Primary.Disease"), outfile = "./data_munging/rds/crispr_signif_g2p_lineage.rds")

ccle_signif_gene <- WilcoxonTestDF(dataframe = ccle_ptmuts, grouping = "Drug_Gene", outfile = "./data_munging/rds/ccle_signif_g2p_gene.rds")
ccle_signif_gene$G2P_Drug_Gene <- ifelse(ccle_signif_gene$Drug_Gene %in% g2p_druggenes, TRUE, FALSE)
ccle_signif_lineage <- WilcoxonTestDF(dataframe = filter(ccle_ptmuts, Drug_Gene_Lin %in% g2p_dgls), grouping = "Drug_Gene_Lin", outfile = "./data_munging/rds/ccle_signif_g2p_lineage.rds")

ctrp_signif_gene <- WilcoxonTestDF(dataframe = ctrp_ptmuts, grouping = "Drug_Gene", outfile = "./data_munging/rds/ctrp_signif_g2p_gene.rds")
ctrp_signif_gene$G2P_Drug_Gene <- ifelse(ctrp_signif_gene$Drug_Gene %in% g2p_druggenes, TRUE, FALSE)
ctrp_signif_lineage <- WilcoxonTestDF(dataframe = filter(ctrp_ptmuts, Drug_Gene_Lin %in% g2p_dgls), grouping = "Drug_Gene_Lin", outfile = "./data_munging/rds/ctrp_signif_g2p_lineage.rds")

gdsc_signif_gene <- WilcoxonTestDF(dataframe = gdsc_ptmuts, grouping = "Drug_Gene", outfile = "./data_munging/rds/gdsc_signif_g2p_gene.rds")
gdsc_signif_gene$G2P_Drug_Gene <- ifelse(gdsc_signif_gene$Drug_Gene %in% g2p_druggenes, TRUE, FALSE)
gdsc_signif_lineage <- WilcoxonTestDF(dataframe = filter(gdsc_ptmuts, Drug_Gene_Lin %in% g2p_dgls), grouping = "Drug_Gene_Lin", outfile = "./data_munging/rds/gdsc_signif_g2p_lineage.rds")
```

# Spearman correlations

--------------------------------------------------------------------------------

Spearman correlations for gene expression and copy number:
```{r fig.height=8, fig.width=16}
crispr_spear_ge <- transform(crispr_data_g2p %>% group_by(Gene) %>% summarize(p = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$p.value, p_log10 = -log10(p), corr = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$estimate, Metric = "Gene Expression"), Gene = reorder(Gene, corr))
crispr_spear_cn <- transform(crispr_data_g2p %>% group_by(Gene) %>% summarize(p = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$p.value, p_log10 = -log10(p), corr = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$estimate, Metric = "Copy Number"), Gene = reorder(Gene, corr))

spearman_KRAS <- rbind(filter(crispr_data_g2p, Gene == "KRAS") %>% summarize(p = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$p.value, corr = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$estimate, Metric = "Gene Expression"), filter(crispr_data_g2p, Gene == "KRAS") %>% summarize(p = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$p.value, corr = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$estimate, Metric = "Copy Number"))

crispr_ge_spearman_corr_inset <- ggplot(filter(crispr_data_g2p, Gene == "KRAS"), aes(x = TPM, y = Score)) +
  geom_point(pch = 21, fill = "cadetblue3", alpha = 0.6) +
  labs(x = "KRAS Gene Expression (TPM)", y = "CERES Score", subtitle = paste0("Spearman correlation coefficient = ", round(spearman_KRAS$corr[1], 2), " (p = ", formatC(as.numeric(spearman_KRAS$p[1]), format = "e", digits = 1), ")"))

crispr_cn_spearman_corr_inset <- ggplot(filter(crispr_data_g2p, Gene == "KRAS"), aes(x = CN, y = Score)) +
  geom_point(pch = 21, fill = "palegreen2", alpha = 0.6) +
  labs(x = "KRAS Copy Number (log2 ratio)", y = "CERES Score", subtitle = paste0("Spearman correlation coefficient = ", round(spearman_KRAS$corr[2], 2), " (p = ", formatC(as.numeric(spearman_KRAS$p[2]), format = "e", digits = 1), ")"))

crispr_ge_spearman_corr_plot <- ggplot(data = crispr_spear_ge, aes(x = Gene, y = corr)) +
  geom_point(aes(fill = p_log10), pch = 21, size = 5) +
  scale_fill_gradient2(low = "white", mid = "white", high = "deepskyblue4", midpoint = 1.3010299957) +
  coord_cartesian(ylim = c(-0.45, 0.2)) +
  theme(legend.position = "bottom", axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.4, size = 10)) +
  labs(fill = "-log10(p)", x = "Gene", y = expression("Spearman Correlation Coefficient"~(r[s]))) +
  annotation_custom(ggplotGrob(crispr_ge_spearman_corr_inset), xmin = 12, xmax = 32, ymin = -0.15, ymax = -0.45) +
  geom_curve(aes(x = 3.5, y = -0.27, xend = 14, yend = -0.165), color = "lightgray", curvature = -0.2, arrow = arrow(length = unit(0.03, "npc")))

crispr_cn_spearman_corr_plot <- ggplot(data = crispr_spear_cn, aes(x = Gene, y = corr)) +
  geom_point(aes(fill = p_log10), pch = 21, size = 5) +
  scale_fill_gradient2(low = "white", mid = "white", high = "green4", midpoint = 1.3010299957) +
  coord_cartesian(ylim = c(-0.45, 0.2)) +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.4, size = 10)) +
  labs(fill = "-log10(p)", x = "Gene", y = "Spearman Correlation Coefficient (r_s)") +
  annotation_custom(ggplotGrob(crispr_cn_spearman_corr_inset), xmin = 12, xmax = 32, ymin = -0.15, ymax = -0.45) +
  geom_curve(aes(x = 2.5, y = -0.232, xend = 14, yend = -0.165), color = "lightgray", curvature = -0.2, arrow = arrow(length = unit(0.03, "npc")))

crispr_spearman_corr_plot <- plot_grid(crispr_ge_spearman_corr_plot, crispr_cn_spearman_corr_plot, align = "h")
crispr_spearman_corr_plot

# ggsave("./plots/manuscript/crispr_spearman_corr_plot.pdf", crispr_spearman_corr_plot, device = "pdf", dpi = 450, width = 16, height = 8, units = "in")
```

# DGL KS tests

--------------------------------------------------------------------------------

```{r KS_functions}
MakeCorrResSummaryTable <- function(df_list, dataset_name) {
  # Input: list of dataframes corresponding to Drug_Gene groupings in the drugscreen datasets filtered for point mutations
  CorrResSummaryTable <- data.frame(matrix(ncol = 6, nrow = 0))
  for(i in 1:length(df_list)) {
    corr_res <- df_list[[i]] %>% group_by(G2P_Drug_Gene_Lin) %>% summarize(Dataset = dataset_name,
                Drug_Gene = names(df_list)[i],
                Lineages = paste(as.character(unique(Primary.Disease)), collapse = ", "),
                N_Lineages = length(as.character(unique(Primary.Disease))),
                Spearman_Corr_GE = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$estimate,
                Spearman_Corr_CN = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$estimate)
    dgl_signif <- compare_means(Score ~ Mutation_Status, data = df_list[[i]], group.by = "G2P_Drug_Gene_Lin", method = "wilcox.test")[, c("G2P_Drug_Gene_Lin", "p")]
    corr_res <- merge(corr_res, dgl_signif, by = "G2P_Drug_Gene_Lin", all = TRUE)
    corr_res$Mut_Signif <- ifelse(corr_res$p < 0.05, 1, 0)
    CorrResSummaryTable <- rbind(CorrResSummaryTable, corr_res)
  }
  return(CorrResSummaryTable)
}

G2PCorrKStest <- function(dataset, dataset_name) {
  # Input: CorrResSummaryTable created by the MakeCorrResSummaryTable function
  Metric <- colnames(dataset)[6:9]
  D_stat <- NULL
  pval <- NULL
  signif <- NULL
  
  yes <- dataset[dataset$G2P_Drug_Gene_Lin == "Yes",]
  no <- dataset[dataset$G2P_Drug_Gene_Lin == "No",]
  
  for(i in 6:9) {
    test <- ks.test(yes[[colnames(yes)[i]]], no[[colnames(no)[i]]])
    D_stat <- c(D_stat, test$statistic)
    pval <- c(pval, test$p.value)
    signif <- c(signif, ifelse(test$p.value < 0.001, "***", ifelse(test$p.value < 0.01, "**", ifelse(test$p.value < 0.05, "*", NA))))
  }
  df <- data.frame(Metric, D_stat, pval, signif)
  df$Dataset <- dataset_name
  return(df)
}

G2PCorrTables <- function(table_type) {
  # CCLE
  ccle_split <- split(ccle_ptmuts, f = ccle_ptmuts$Drug_Gene)
  ccle_corr_all <- MakeCorrResSummaryTable(ccle_split, dataset_name = "CCLE")
  # CTRP
  ctrp_split <- split(ctrp_ptmuts, f = ctrp_ptmuts$Drug_Gene)
  ctrp_corr_all <- MakeCorrResSummaryTable(ctrp_split, dataset_name = "CTRP")
  # GDSC
  gdsc_split <- split(gdsc_ptmuts, f = gdsc_ptmuts$Drug_Gene)
  gdsc_corr_all <- MakeCorrResSummaryTable(gdsc_split, dataset_name = "GDSC")
  
  g2p_corr_all <- rbind(ccle_corr_all, ctrp_corr_all, gdsc_corr_all)
  
  if(table_type == "correlation" | table_type == "corr" ) {
    return(g2p_corr_all)
  }  
  if(table_type == "mutation" | table_type == "mut" ) {
    g2p_mut_stats <- g2p_corr_all %>% group_by(Dataset, G2P_Drug_Gene_lin) %>% drop_na(Mut_Signif) %>% summarize(N_Signif_Muts = sum(Mut_Signif),   Total_Tests = length(Mut_Signif), Perc_Signif = N_Signif_Muts / Total_Tests * 100)
    return(g2p_mut_stats)
  }
  if(table_type == "KS" | table_type == "ks") {
    ccle_corr_stats <- G2PCorrKStest(ccle_corr_all, "CCLE")
    ctrp_corr_stats <- G2PCorrKStest(ctrp_corr_all, "CTRP")
    gdsc_corr_stats <- G2PCorrKStest(gdsc_corr_all, "GDSC")
    g2p_corr_stats <- rbind(ccle_corr_stats, ctrp_corr_stats, gdsc_corr_stats)
    g2p_corr_stats <- g2p_corr_stats %>% separate(col = Metric, into = c("Correlation_Method", "corr", "Metric"), sep = "_")
    g2p_corr_stats$Metric <- ifelse(g2p_corr_stats$Metric == "GE", "Gene Expression", "Copy Number")
    return(g2p_corr_stats)
  }
}
```

Kolmogorov-Smirnov tests comparing DGLs that are in a G2P DGL and not.

**RUN THIS WHEN ALL DRUGSCREEN DATA HAS BEEN MERGED**
```{r, eval=FALSE, fig.height=6, fig.width=12}
g2p_mut_stats <- G2PCorrTables(table_type = "mutation")
g2p_corr_all <- G2PCorrTables(table_type = "correlation")
g2p_corr_stats <- G2PCorrTables(table_type = "KS")

g2p_corr_4plot <- g2p_corr_all
g2p_corr_4plot$p <- NULL
g2p_corr_4plot$Mut_Signif <- NULL
g2p_corr_4plot <- melt(g2p_corr_4plot, id.vars = colnames(g2p_corr_4plot)[1:5], measure.vars = colnames(g2p_corr_4plot)[6:9], value.name = "Correlation")
g2p_corr_4plot <- g2p_corr_4plot %>% separate(col = variable, into = c("Correlation_Method", "corr", "Metric"), sep = "_", remove = FALSE)
g2p_corr_4plot$Metric <- ifelse(g2p_corr_4plot$Metric == "GE", "Gene Expression", "Copy Number")

g2p_ks_res_plot_spearman <- ggplot(data = filter(g2p_corr_4plot, Correlation_Method == "Spearman")) +
  facet_grid(Metric ~ Dataset, labeller = labeller(yfacet = c('CN' = "Copy Number", 'GE' = "Gene Expression"))) +
  geom_violin(aes(y = Correlation, x = G2P_Drug_Gene_Lin, fill = Dataset, color = Dataset, alpha = G2P_Drug_Gene_Lin)) +
  geom_boxplot(width = 0.075, aes(y = Correlation, x = G2P_Drug_Gene_Lin, color = Dataset)) +
  geom_jitter(width = 0.3, alpha = 0.8, aes(y = Correlation, x = G2P_Drug_Gene_Lin, color = Dataset)) +
  theme(legend.position = "none") +
  scale_alpha_manual(values = c("Yes" = 0.6, "No" = 0.2)) +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  scale_color_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  coord_cartesian(ylim = c(-1, 1)) +
  geom_text(data = filter(g2p_corr_stats, Correlation_Method == "Spearman"), aes(x = 1.5, y = 0.8, label = paste0("D = ", formatC(D_stat, digits = 2, format = "f"), ", p = ", ifelse(pval > 0.1, round(pval, 2), round(pval, 3)), "\n", ifelse(is.na(signif), "", signif)))) +
  labs(x = "Is the lineage in a G2P DGL?", y = "Spearman Correlation Coefficient")
g2p_ks_res_plot_spearman

# ggsave("./plots/manuscript/g2p_spearman_ks_res_plot.pdf", g2p_ks_res_plot_spearman, device = "pdf", dpi = 500, width = 12, height = 6, units = "in")
```

# Sandbox

--------------------------------------------------------------------------------

Filter for BRCA1/2 mutations to see if we catch non-germline mutations/CN variants:
```{r}
test <- filter(maf_df, Gene == "BRCA1" | Gene == "BRCA2")
test2 <- filter(cn_melt, Gene == "BRCA1" | Gene == "BRCA2")
```

Test out standardization/normalization methods on Pozdeyev data. This yields evidence that the AUC values are already normalized.
```{r, eval=FALSE}
ccle_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CCLE_drug_response_metrics.csv.gz", sep = ",")
ccle_all$AUC_IC50_standardized <- scale(ccle_all$AUC_IC50, center = TRUE, scale = TRUE)
ccle_all$AUC_IC50_normalized <- scale(ccle_all$AUC_IC50, center = min(ccle_all$AUC_IC50[!is.na(ccle_all$AUC_IC50)]), scale = max(ccle_all$AUC_IC50[!is.na(ccle_all$AUC_IC50)]) - min(ccle_all$AUC_IC50[!is.na(ccle_all$AUC_IC50)]))
ctrp_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CTRP_drug_response_metrics.csv.gz", sep = ",")
ctrp_all$AUC_IC50_standardized <- scale(ctrp_all$AUC_IC50, center = TRUE, scale = TRUE)
ctrp_all$AUC_IC50_normalized <- scale(ctrp_all$AUC_IC50, center = min(ctrp_all$AUC_IC50[!is.na(ctrp_all$AUC_IC50)]), scale = max(ctrp_all$AUC_IC50[!is.na(ctrp_all$AUC_IC50)]) - min(ctrp_all$AUC_IC50[!is.na(ctrp_all$AUC_IC50)]))
gdsc_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/GDSC_drug_response_metrics.csv.gz", sep = ",")
gdsc_all$AUC_IC50_standardized <- scale(gdsc_all$AUC_IC50, center = TRUE, scale = TRUE)
gdsc_all$AUC_IC50_normalized <- scale(gdsc_all$AUC_IC50, center = min(gdsc_all$AUC_IC50[!is.na(gdsc_all$AUC_IC50)]), scale = max(gdsc_all$AUC_IC50[!is.na(gdsc_all$AUC_IC50)]) - min(gdsc_all$AUC_IC50[!is.na(gdsc_all$AUC_IC50)]))

auc_test <- bind_rows(ccle_all, ctrp_all, gdsc_all)

ggplot(data = auc_test) +
  geom_histogram(aes(x = AUC_IC50)) +
  facet_wrap(~ Database, ncol = 3, scales = "free_y")
ggplot(data = auc_test) +
  geom_histogram(aes(x = AUC_IC50_normalized)) +
  facet_wrap(~ Database, ncol = 3, scales = "free_y")
ggplot(data = auc_test) +
  geom_histogram(aes(x = AUC_IC50_standardized)) +
  facet_wrap(~ Database, ncol = 3, scales = "free_y")
```


# Session information

--------------------------------------------------------------------------------

```{r}
print(sessionInfo())
```