---
title: "19Q1"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the [Broad's Avana CRISPR](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017), the [Cancer Therapeutics Response Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (CTRP) drug screen dataset (v2) (Basu et al., 2013; M. G. Rees et al., 2016; B. Seashore-Ludlow et al., 2015), the [Genomics of Drug Sensitivity in Cancer](https://www.cancerrxgene.org/downloads) (GDSC) drug screen dataset (Garnett et al., 2012; Yang et al., 2012) , and the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) (CCLE) drug screen dataset (J. Barretina et al., 2012; T. C. C. L. E. Consortium & Consortium, 2015).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

All annotation files (copy number, mutation status, and gene expression) (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/).

# SET UP

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

Edited `.Renviron` to avoid "Error: memory exhausted (limit reached?)" when melting the GE dataset, as per the suggestion from [StackOverflow](https://stackoverflow.com/questions/51295402/r-on-macos-error-vector-memory-exhausted-limit-reached):Ã¥
```
cd ~
touch .Renviron
open .Renviron
# Added following line to .Renviron:
R_MAX_VSIZE=100Gb
```

```{r}
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(kableExtra)
library(reshape2)
library(tidyverse)
library(naniar)
theme_set(theme_light())
```

# DATA MANAGEMENT {.tabset .tabset-fade .tabset-pills}

--------------------------------------------------------------------------------

Note that the new 19Q1 release made multiple changes to variable names:

- They reverted back to "Broad_ID" from 18Q3 rather than "DepMap_ID" that was used in 18Q4
- Primary.Disease column of the cell line metadata file is more consistently named (Subtype.Disease now has the specific disease annotations), though there are a couple capitalization inconsistencies
- MAF file no longer has "CCLE_Name" column and has a duplicate column of the Broad IDs (DepMap_ID and Tumor_Sample_Barcode)

And data (the following is taken from the [data portal](https://depmap.org/portal/download/)):

- **Expression**: We use the GTEx pipeline. Since last release, there have been some updates to the versions of different components of the pipeline and minor bug fixes on GTEx's end.
- **Mutation**: In this release, we additionally process DepMap WES samples through the CGA mutation calling pipeline adapted for cell lines and available on FireCloud. We have added a column for the allele counts from this pipeline run (CGA_WES_AC). Previously, the WES_AC column was generated from running the Van Allen mutation calling pipeline for SNPs + Mutect2 for INDELs. This column is now named VA_WES_AC.
- **Copy Number**: We have moved to using the newest version of the GATK pipeline.

## D. Charytonowicz cell line converter

This comprehensive cancer cell line information was curated by both Daniel Charytonowicz and myself.
```{r}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
lineage_converter <- unique(ccl_converter[, c("grouped_general_doid", "group_general_lineage_name")])
```

## DepMap cell line metadata

19Q1 update fixed almost all of the inconsistencies of the Primary.Disease column, though there were come capitalization inconsistencies.
```{r}
ccl_info <- read.delim("./data_munging/19Q1/DepMap-2019q1-celllines_v2_20190305.csv", sep = ",", header = TRUE, na.strings = c("", NA))
# Convert Primary.Disease entries to all lower case to fix factorization
ccl_info$Primary.Disease <- as.factor(tolower(ccl_info$Primary.Disease))
# Harmonize ID column name
colnames(ccl_info)[colnames(ccl_info) == "DepMap_ID"] <- "Broad_ID"

# Sample info from figshare
crispr_meta <- read.delim("./data_munging/19Q1/sample_info_20190306.csv", sep = ",", header = TRUE, na.strings = c("", NA))
colnames(crispr_meta)[colnames(crispr_meta) == "CCLE_name"] <- "CCLE_Name"
```

## CCLE genomic annotations

### Mutation status (Mut)

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Mutation calls (coding region, germline filtered):
```{r, eval=FALSE}
maf_raw <- read.delim("./data_munging/19Q1/depmap_19Q1_mutation_calls_20190305.csv.gz", header = TRUE, sep = ",", row.names = 1)
# Harmonize ID column name
colnames(maf_raw)[colnames(maf_raw) == "DepMap_ID"] <- "Broad_ID"
# Select columns
maf_df <- subset(maf_raw, select = c("Hugo_Symbol", "Broad_ID", "Variant_Classification", "Reference_Allele", "Tumor_Seq_Allele1", "Genome_Change"))
maf_df$Reference_Allele <- ifelse(maf_df$Reference_Allele == "-", 0, nchar(as.character(maf_df$Reference_Allele)))
maf_df$Tumor_Seq_Allele1 <- nchar(as.character(maf_df$Tumor_Seq_Allele1))
maf_df <- maf_df[maf_df$Variant_Classification != "Silent",]
# Add Mutation_Status column
maf_df$Mutation_Status_Nonsilent <- "Mutant"

saveRDS(maf_df, "./data_munging/19Q1/maf_df_19Q1.rds", compress = "xz")
```

### Copy number (CN)

DepMap WES CN Data:
```{r, eval=FALSE}
cn <- read.delim("./data_munging/19Q1/public_19Q1_gene_cn_20190305.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
# Remove Entrez gene IDs from colnames, format is "Hugo_Symbol (Entrez ID)"
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "Broad_ID"
# Melt
cn_melt <- melt(data = cn, id.vars = "Broad_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Hugo_Symbol", value.name = "CN")
saveRDS(cn_melt, "./data_munging/19Q1/cn_melt_19Q1.rds", compress = "xz")
```

### Gene expression (GE)

19Q1, CCLE RNAseq gene expression data (log2(TPM+1)):
```{r, eval=FALSE}
ge_19Q1 <- read.delim("./../htscreens_giant_files/CCLE_depMap_19Q1_TPM_20190305.csv.gz", header = TRUE, sep = ",", check.names = FALSE)
# Remove Entrez gene IDs from colnames, format is "Hugo_Symbol (Entrez ID)"
colnames(ge_19Q1) <- gsub(" .*", "", colnames(ge_19Q1))
colnames(ge_19Q1)[1] <- "Broad_ID"
# Melt
ge_melt_19Q1 <- melt(ge_19Q1, id.vars = "Broad_ID", measure.vars = colnames(ge_19Q1)[2:ncol(ge_19Q1)], variable.name = "Hugo_Symbol", value.name = "TPM")
# ge_melt_19Q1 <- unique(ge_melt_19Q1_all)
saveRDS(ge_melt_19Q1, "./data_munging/19Q1/ge_melt_19Q1.rds", compress = "xz")
```

## Denise's GCN dataframes

Code for Denise's graph convolutional networks analyses (see meeting notes from 6 March 2019).
- Convert drug/compound names to all capitalized for harmonization.

```{r, eval=TRUE}
maf_df <- readRDS("./data_munging/19Q1/maf_df_19Q1.rds")
cn_melt <- readRDS("./data_munging/19Q1/cn_melt_19Q1.rds")
ge_melt <- readRDS("./data_munging/19Q1/ge_melt_19Q1.rds")

ccle <- read.delim("./data_munging/pharmacogenomics_raw_files/CCLE_NP24.2009_Drug_data_2015.02.24_20190307.csv", sep = ",", header  = TRUE, check.names = FALSE)
ctrp <- read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.data.curves_post_qc.txt.gz", sep = "\t", header  = TRUE, check.names = FALSE)
gdsc <- read.delim("./data_munging/pharmacogenomics_raw_files/GDSC_v17.3_fitted_dose_response_20190306.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)


## CCLE
# Note that ActArea is NOT an AUC value
ccle_filt <- ccle[, c("CCLE Cell Line Name", "ActArea", "Compound")]
ccle_filt$source_ccl_name <- sapply(strsplit(as.character(ccle_filt$`CCLE Cell Line Name`), split = "_"), function(x) x[1])
ccle_filt$source_histology <- tolower(gsub("^[a-z0-9]*_", "", ccle_filt$`CCLE Cell Line Name`, ignore.case = TRUE))
ccle_filt$source_histology <- ifelse(as.character(ccle_filt$source_histology) == "", NA, as.character(ccle_filt$source_histology))
ccle_filt$compound_name <- toupper(ccle_filt$Compound)
colnames(ccle_filt)[colnames(ccle_filt) == "ActArea"] <- "AUC"
ccle4harm_long <- select(ccle_filt, source_ccl_name, source_histology, compound_name, AUC)
ccle4harm_long$source <- "CCLE"
ccle4harm_long$sample_id <- with(ccle4harm_long, paste(source, source_ccl_name, source_histology, sep = "_"))
ccle4harm <- dcast(ccle4harm_long, source + source_ccl_name + source_histology + sample_id ~ compound_name, value.var = "AUC", fun.aggregate = mean, fill = 10000)
ccle4harm[, 5:ncol(ccle4harm)] <- ccle4harm[, 5:ncol(ccle4harm)] %>% replace_with_na_all(condition = ~.x == 10000)

## CTRP
# area_under_curve: integrated area under the sigmoid-fit concentration-response curve
ctrp_filt <- ctrp[, c("experiment_id", "area_under_curve", "master_cpd_id")]
ctrp_meta_ccl <- select(read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.meta.per_cell_line_20190306.txt"), master_ccl_id, ccl_name, ccle_primary_site)
ctrp_meta_exp <- select(read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.meta.per_experiment_20190306.txt"), experiment_id, master_ccl_id)
ctrp_meta_cpd <- select(read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.meta.per_compound_20190306.txt"), master_cpd_id, cpd_name)
ctrp_keep <- merge(merge(ctrp_meta_ccl, ctrp_meta_exp, by = "master_ccl_id", all = TRUE), merge(ctrp_meta_cpd, ctrp_filt, by = "master_cpd_id", all = TRUE), by = "experiment_id", all = TRUE)
ctrp_keep$cpd_name <- toupper(ctrp_keep$cpd_name)
ctrp_keep$ccle_primary_site <- ifelse(as.character(ctrp_keep$ccle_primary_site) == "", NA, as.character(ctrp_keep$ccle_primary_site))
ctrp4harm_long <- select(ctrp_keep, ccl_name, ccle_primary_site, cpd_name, area_under_curve)
colnames(ctrp4harm_long) <- c("source_ccl_name", "source_histology", "compound_name", "AUC")
ctrp4harm_long$source <- "CTRP"
ctrp4harm_long$sample_id <- with(ctrp4harm_long, paste(source, source_ccl_name, source_histology, sep = "_"))
ctrp4harm <- dcast(ctrp4harm_long, source + source_ccl_name + source_histology + sample_id ~ compound_name, value.var = "AUC", fun.aggregate = mean, fill = 10000)
ctrp4harm[, 5:ncol(ctrp4harm)] <- ctrp4harm[, 5:ncol(ctrp4harm)] %>% replace_with_na_all(condition = ~.x == 10000)

## GDSC
gdsc_meta_ccl <- select(read.delim("./data_munging/pharmacogenomics_raw_files/GDSC_Cell_Lines_Details.csv", sep = ","), Line, Site)
gdsc_filt <- gdsc[, c("CELL_LINE_NAME", "DRUG_NAME", "AUC")]
gdsc_filt$DRUG_NAME <- toupper(gdsc_filt$DRUG_NAME)
colnames(gdsc_filt)[colnames(gdsc_filt) ==  "CELL_LINE_NAME"] <- "Line"
gdsc_filt <- merge(gdsc_filt, gdsc_meta_ccl, by = "Line", all = TRUE)
gdsc4harm_long <- select(gdsc_filt, Line, Site, DRUG_NAME, AUC)
colnames(gdsc4harm_long) <- c("source_ccl_name", "source_histology", "compound_name", "AUC")
gdsc4harm_long$source <- "GDSC"
gdsc4harm_long$sample_id <- with(gdsc4harm_long, paste(source, source_ccl_name, source_histology, sep = "_"))
gdsc4harm <- dcast(gdsc4harm_long, source + source_ccl_name + source_histology + sample_id ~ compound_name, value.var = "AUC", fun.aggregate = mean, fill = 10000)
gdsc4harm[, 5:ncol(gdsc4harm)] <- gdsc4harm[, 5:ncol(gdsc4harm)] %>% replace_with_na_all(condition = ~.x == 10000)

## BIND ALL
pgx_harm_list <- list(ccle4harm, ctrp4harm, gdsc4harm)
pgx_harm_names <- unique(unlist(lapply(pgx_harm_list, names)))
pgx_harm <- do.call(rbind,
                    c(lapply(pgx_harm_list,
                             function(x) data.frame(c(x,
                                                      sapply(setdiff(pgx_harm_names, names(x)),
                                                             function(y) NA)
                                                      )
                                                    )
                             )
                      )
                    )
```







