---
title: "19Q2"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the [Broad's Avana CRISPR](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017), the [Cancer Therapeutics Response Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (CTRP) drug screen dataset (v2) (Basu et al., 2013; M. G. Rees et al., 2016; B. Seashore-Ludlow et al., 2015), the [Genomics of Drug Sensitivity in Cancer](https://www.cancerrxgene.org/downloads) (GDSC) drug screen dataset (Garnett et al., 2012; Yang et al., 2012) , and the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) (CCLE) drug screen dataset (J. Barretina et al., 2012; T. C. C. L. E. Consortium & Consortium, 2015).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

Annotation files for mutation status, copy number, and gene expression of the CCLE, GDSC, and CTRP datasets (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/). Note that the group that conducted the GDSC screen also collected their own genomic annotation information for mutation status, copy number, and gene expression, but we did not use it.

# SET UP

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

```{r libraries}
# install_github("jokergoo/ComplexHeatmap")
library(devtools)
library(circlize)
library(ComplexHeatmap)
library(cowplot)
library(crunch)
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(jsonlite)
library(kableExtra)
library(magrittr)
library(naniar)
library(openxlsx)
library(reshape2)
library(scales)
library(stringr)
library(tidyverse)
library(VennDiagram)

theme_set(theme_light())
```

```{r general_functions}
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) {
    df[[i]] <- as.character(df[[i]])
  }
  return(df)
}
```

# DATA

--------------------------------------------------------------------------------

## Cell line metadata

A comprehensive cancer cell line information was curated by both Daniel Charytonowicz and myself.

DepMap 19Q2 update fixed almost all of the inconsistencies of the Primary.Disease column, though there were come capitalization inconsistencies.
```{r}
ccle_meta_test <- read.delim("./data_munging/19Q2/sample_info.csv", sep = ",", header = TRUE, na.strings = c("", NA))
```

```{r depmap_metadata, eval=FALSE}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
colnames(ccl_converter)[colnames(ccl_converter) == "Broad_ID"] <- "DepMap_ID"
lineage_converter <- unique(ccl_converter[, c("DepMap_ID", "grouped_general_doid", "group_general_lineage_name")])

# OLD CCL_META
ccl_histologies <- read.delim("./data_munging/ccl_histologies_metadata.csv", sep = ",", header = TRUE, row.names = 1)
colnames(ccl_histologies) <- c("Cell_line", "CCLE_histology", "TCGA_classification", "CCLE_name", "DepMap_ID")
ccl_meta_orig <- read.delim("./data_munging/19Q2/sample_info.csv", sep = ",", header = TRUE, na.strings = c("", NA))
ccl_meta_orig_filt <- select(ccl_meta_orig, DepMap_ID, stripped_cell_line_name, CCLE_name, disease)
colnames(ccl_meta_orig_filt) <- c("DepMap_ID", "Cell_line", "CCLE_name", "disease")
ccl_meta <- subset(ccl_meta_orig_filt, !grepl("MERGED", CCLE_name, fixed = TRUE))
ccl_diseases <- as.character(unique(ccl_meta$disease))[as.character(unique(ccl_meta$disease)) != " "]
CCLE_histology <- tolower(gsub("^[a-z0-9]*_", "", ccl_meta$CCLE_name, ignore.case = TRUE))
ccl_meta <- add_column(ccl_meta, CCLE_histology = CCLE_histology, .after = "CCLE_name")
ccl_meta$disease <- with(ccl_meta, ifelse(disease == " ", as.character(CCLE_histology), as.character(disease)))
ccl_meta$disease <- with(ccl_meta, ifelse(disease == " " & CCLE_histology == "vulva", "vulva",
                                   ifelse(disease == " " & CCLE_histology == "upper_aerodigestive_tract", "upper_aerodigestive",
                                   ifelse(disease == " " & CCLE_histology == "large_intestine", "colorectal",
                                   ifelse(disease == " " & CCLE_histology == "biliary_tract", "bile_duct",
                                   ifelse(disease == " " & CCLE_histology == "oesophagus", "esophagus",
                                   ifelse(disease == "other" & CCLE_histology == "haematopoietic_and_lymphoid_tissue", "haematopoietic_and_lymphoid", disease)))))))

ccl_meta <- merge(merge(ccl_meta, ccl_histologies, by = c("Cell_line", "CCLE_histology", "CCLE_name", "DepMap_ID"), all = TRUE), lineage_converter, by = "DepMap_ID", all = TRUE)
ccl_meta$disease <- with(ccl_meta, ifelse(is.na(disease) & !is.na(CCLE_histology), CCLE_histology, disease))
ccl_meta$disease <- with(ccl_meta, ifelse(CCLE_histology == "large_intestine", "colorectal",
                                   ifelse(CCLE_histology == "haematopoietic_and_lymphoid_tissue", "haematopoietic_and_lymphoid",
                                   ifelse(CCLE_histology == "upper_aerodigestive_tract", "upper_aerodigestive",
                                   ifelse(CCLE_histology == "oesophagus", "esophagus",
                                   ifelse(CCLE_histology == "biliary_tract", "bile_duct",
                                   ifelse(Cell_line == "GT3TKB", "lung",
                                   ifelse(disease == "stomach" | CCLE_histology == "stomach", "gastric", disease))))))))
# Drop matched_normal_tissue cell lines
ccl_meta <- filter(ccl_meta, disease != "matched_normal_tissue")
# Filter out cell lines without a DepMap ID or a cell line
ccl_meta <- filter(ccl_meta, !is.na(DepMap_ID) & !is.na(Cell_line))
# Filter out cell lines that have a CCLE_name not present in the original sample info from DepMap
ccl_meta <- filter(ccl_meta, CCLE_name %in% ccl_meta_orig$CCLE_name & Cell_line %in% ccl_meta_orig_filt$Cell_line)
# Remove cell line not used in Pozdeyev
ccl_meta <- filter(ccl_meta, DepMap_ID != "ACH-002397")
write.csv(ccl_meta, "./data_munging/19Q2/sample_info_edit_20191008.csv")
```

ACH-002397 (KMH2_THYROID) was removed to align with Pozdeyev et al (2016) cell line name harmonization. CTRP only screened KMH2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE, whereas GDSC screend both KMH2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE (i.e. KM-H2) and KMH2_THYROID (i.e. KMH-2). During their harmonization process, Pozdeyev et al only kept KM-H2 from GDSC. Thus, we removed KMH2_THYROID from further analyses.

```{r}
ccl_meta <- read.delim("./data_munging/19Q2/sample_info_edit_20191008.csv", sep = ",", header = TRUE, row.names = 1)
ccl_meta <- ccl_meta %>% select(DepMap_ID, Cell_line, disease)
ccl_unique <- data.frame(disease = unique(ccl_meta$disease))
```

## G2P

```{r g2p_parsing_functions}
ExtractBiomarkerType <- function(json_string) {
  json_string <- as.character(json_string)
  biomarker_type <- ((str_extract(string = json_string, pattern = "biomarker_type.+") %>% strsplit(split = ","))[[1]][1] %>% str_extract(pattern = "u'.+") %>% str_match("'(.*?)'"))[,2]
  biomarker_type <- ifelse(is.na(biomarker_type), "", biomarker_type)
  return(biomarker_type)
}
```

```{r g2p_parsing, eval=FALSE}
g2p <- read.delim("./data_munging/g2p_full_level_A_dataset_NEW_DEC_2_2018_withGroupedLineages.csv", sep = "\t", header = TRUE)
g2p[, 1:2] <- NULL
g2p <- unfactorize(g2p)
g2p$Drug <- tolower(g2p$Drug)
g2p[g2p == -1] <- ""
g2p$Gene <- ifelse(g2p$Gene == "" & g2p$Drug == "daunorubicin", "MLL", as.character(g2p$Gene))
g2p$Direction <- ifelse(g2p$Direction == 1 | g2p$Direction == "", NA, as.character(g2p$Direction))
g2p$Ref[g2p$Ref == "None" | g2p$Ref == "-" | is.na(g2p$Ref)] <- ""
g2p$Alt[g2p$Alt == "None" | g2p$Alt == "-" | is.na(g2p$Alt)] <- ""
g2p <- g2p[, c("Drug", "DrugID", "Direction", "Gene", "Ref", "Alt", "Start", "End", "Chromosome", "Feature.Names", "Mutation", "JSON", "Evidence.Level", "Phenotype.Description", "Phenotype.Family", "Phenotype.ID", "Sequence.ID", "grouped_general_doid")]
g2p <- add_column(g2p, Ref_Length = nchar(as.character(g2p$Ref)), .after = "Ref")
g2p <- add_column(g2p, Alt_Length = nchar(as.character(g2p$Alt)), .after = "Alt")
g2p <- add_column(g2p, Variant_Length = NA, .after = "Alt_Length")
g2p$Variant_Length <-
  ifelse(g2p$Alt_Length == 1 & g2p$Ref_Length == 1, 1,
  ifelse(g2p$Alt_Length == 2 & g2p$Ref_Length == 2, 2,
  ifelse(g2p$Alt_Length == 3 & g2p$Ref_Length == 3, 3,
  ifelse(g2p$Ref_Length == g2p$Alt_Length & g2p$Ref_Length > 3, g2p$Ref_Length,
  ifelse(g2p$Ref_Length != g2p$Alt_Length, abs(g2p$Ref_Length - g2p$Alt_Length), NA)))))
g2p <- add_column(g2p, Variant_Type = NA, .after = "Alt_Length")
g2p$Variant_Type <-
  ifelse(g2p$Alt_Length == 1 & g2p$Ref_Length == 1, "SNP",
  ifelse(g2p$Alt_Length == 2 & g2p$Ref_Length == 2, "DNP",
  ifelse(g2p$Alt_Length == 3 & g2p$Ref_Length == 3, "TNP",
  ifelse(g2p$Ref_Length == g2p$Alt_Length & g2p$Ref_Length > 3, "ONP",
  ifelse(g2p$Ref_Length > g2p$Alt_Length, "DEL",
  ifelse(g2p$Ref_Length < g2p$Alt_Length, "INS", "None"))))))
g2p <- subset(g2p, !grepl("^SID|^CHEMBL", g2p$DrugID))
g2p <- subset(g2p, !grepl("germline|homozygosity", tolower(g2p$Feature.Names)))
g2p <- add_column(g2p, biomarker_type = tolower(sapply(g2p$JSON, ExtractBiomarkerType, simplify = TRUE)), .after = "Feature.Names")
g2p <- add_column(g2p, biomarker_type_curated = NA, .after = "biomarker_type")

g2p <- data.frame(apply(g2p, MARGIN = 2, function(x) gsub("^$", NA, trimws(x))))

# Manually curate rucaparib-BRCA/BRCA1/BRCA2 entry
g2p$Gene <- ifelse(g2p$Drug == "rucaparib" & is.na(g2p$Gene), "BRCA2;BRCA1;BRCA", as.character(g2p$Gene))
g2p <- separate_rows(g2p, Gene, sep = ";")

# Manually assign biomarker_types based on a combination of biomarker_type derived from the JSON string and patterns in the Feature.Names
biomarker_type_curated <- 
  ifelse(!is.na(str_match(string = tolower(g2p$biomarker_type), pattern = "copy number variant|amplification")[,1]) | !is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "copy number variant|amplification")), "Amplification/CNV",
  ifelse(!is.na(str_match(string = toupper(g2p$Feature.Names), pattern = "[A-Z0-9|A-Z]+-[A-Z0-9|A-Z]+|FUSION|ALK POSITIVE")[,1]) & is.na(str_match(string = g2p$Feature.Names, pattern = "[0-9]+-[0-9]+")) | g2p$biomarker_type == "fusion" | g2p$biomarker_type == "rearrange", "Fusion",
  ifelse(!is.na(str_match(string = g2p$Feature.Names, pattern = "[A-Z][0-9]+([A-Z]|[*])|[0-9A-Z]+ [A-Z][0-9]+|[A-Z][0-9]+$")[,1]), "Mutation",
  ifelse(!is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "expression|methylation|erbb2  positive")[,1]), "Expression",
  ifelse(!is.na(str_match(string = g2p$biomarker_type, pattern = "intron|insertion|deletion|mutant")[,1]), "Mutation",
  ifelse(!is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "codon|positive|biallelic inactivation|del|ins|dup|mut|frameshift|missense|variant|[0-9]+fs|>")[,1]), "Mutation",
  ifelse(!is.na(str_match(string = g2p$Feature.Names, pattern = "^KIT$|^BRCA$")[,1]), "Mutation", NA)))))))
g2p$biomarker_type_curated <- biomarker_type_curated[,1]

# Manually curate direction column
g2p <- add_column(g2p, Direction_Curated = NA, .after = "Direction")
g2p$Direction_Curated <- with(g2p, ifelse(!is.na(str_match(string = tolower(Direction), pattern = "adverse|unfavorable|^resistant")[,1]), "resistance",
                         ifelse(!is.na(str_match(string = tolower(Direction), pattern = "confers resistance")[,1]), "resistance",
                         ifelse(!is.na(str_match(string = tolower(Direction), pattern = "sensitive|sensitivity|responsive|better")[,1]), "sensitivity", 
                         ifelse(is.na(Direction) & Drug == "gefitinib" & Feature.Names == "L858R" & Gene == "EGFR", "sensitivity",
                         ifelse(is.na(Direction) & Drug == "trametinib" & Feature.Names %in% c("V600E", "V600K") & Gene == "BRAF", "sensitivity",
                         ifelse(is.na(Direction) & Drug == "cabozantinib" & Feature.Names == "RET C634W", "sensitivity", NA)))))))

# Curate protein change column
g2p <- add_column(g2p, Protein_Change = NA, .after = "Mutation")
g2p <- separate_rows(g2p, Mutation, sep = ",")
g2p$Protein_Change <- gsub("u'|'| |,|\\[|\\]", "", g2p$Mutation)
g2p$Protein_Change <- ifelse(is.na(g2p$Protein_Change), str_match(string = g2p$Feature.Names, pattern = "^[A-Z][0-9]+([A-Z]|[*])| [A-Z][0-9]+([A-Z]|[*])|:[A-Z][0-9]+([A-Z]|[*])")[,1], as.character(g2p$Protein_Change))
g2p$Protein_Change <- gsub(":| |c\\..*", "", g2p$Protein_Change)
g2p$Protein_Change <- ifelse(!is.na(str_match(string = tolower(trimws(g2p$Protein_Change)), pattern = "^$|exon|amplification|expression|frameshift|fusion|methylation|mutation|-")[,1]), NA, as.character(g2p$Protein_Change))
g2p$Protein_Change <- ifelse(g2p$Feature.Names == "ALK inframe insertion (1151T)", "1151T",
                      ifelse(g2p$Feature.Names == "MET M995Ifs*19", "M995Ifs*19",
                      ifelse(g2p$Feature.Names == "RET (618,620,634,768,791,891,918,C634W,M918T)", "C634W;M918T",
                      ifelse(g2p$Protein_Change == "V600E/K", "V600E;V600K", as.character(g2p$Protein_Change)))))
g2p <- separate_rows(g2p, Protein_Change, sep = ";")
g2p$Protein_Change <- gsub("^([0-9]+)([A-Z])$|^KRAS ([0-9]+)([A-Z])$", "\\2\\1\\2", g2p$Protein_Change) # Fix synonymous mut formatting, eg 13G
g2p$Protein_Change <- gsub("^([A-Z])([0-9]+)$|^KRAS ([A-Z])([0-9]+)$", "\\1\\2\\1", g2p$Protein_Change) # Fix synonymous mut formatting, eg G13
g2p <- add_column(g2p, Gene_Protein_Change = NA, .after = "Protein_Change")
g2p$Gene_Protein_Change <- ifelse(!is.na(g2p$Gene) & !is.na(g2p$Protein_Change), paste0(as.character(g2p$Gene), ":", as.character(g2p$Protein_Change)), NA)

# Curate lineage column
g2p <- separate_rows(g2p, Phenotype.Description, sep = ";")
g2p_lineages <- cbind(unique(select(g2p, Phenotype.Description, Phenotype.Family)), disease = NA)
g2p_lineages$disease <- with(g2p_lineages, ifelse(!is.na(str_match(string = tolower(Phenotype.Family), pattern = "astrocytoma|central nervous system|glioma")[,1]) | !is.na(str_match(string = tolower(Phenotype.Description), pattern = "astrocytoma|glioblastoma|craniopharyngioma|medulloblastoma")[,1]), "central_nervous_system",
                                           ifelse(Phenotype.Family %in% c("breast cancer") | Phenotype.Description %in% c("breast cancer", "Her2-receptor positive breast cancer", "invasive ductal carcinoma"), "breast",
                                           ifelse(!is.na(str_match(string = tolower(Phenotype.Family), pattern = "lymphoma|bone marrow|hematologic|immune system|leukemia")[,1]) | !is.na(str_match(string = tolower(Phenotype.Description), pattern = "leukemia|lymphoproliferative|myelodisplasic|systemic mastocytosis|hyper eosinophilic|waldenström macroglobulinemia|langerhans")[,1]), "haematopoietic_and_lymphoid",
                                           ifelse(Phenotype.Family %in% c("kidney cancer", "Nephrotic syndrome") | !is.na(str_match(string = tolower(Phenotype.Description), pattern = "renal|kidney")[,1]), "kidney",
                                           ifelse(Phenotype.Family %in% c("liver cancer"), "liver",
                                           ifelse(Phenotype.Family %in% c("lung cancer", "respiratory system cancer"), "lung",
                                           ifelse(Phenotype.Description %in% c("congenital fibrosarcoma", "synovial sarcoma", "myxoid liposarcoma", "liposarcoma"), "soft_tissue",
                                           ifelse(Phenotype.Family %in% c("integumentary system cancer", "skin cancer") | Phenotype.Description %in% c("basal cell carcinoma", "Basal cell carcinoma", "Dermatofibrosarcoma", "dermatofibrosarcoma protuberans", "melanoma", "skin melanoma", "squamous cell carcinoma"), "skin",
                                           ifelse(Phenotype.Family %in% c("peripheral nervous system neoplasm"), "peripheral_nervous_system",
                                           ifelse(Phenotype.Family %in% c("urinary bladder cancer"), "urinary_tract", 
                                           ifelse(!is.na(str_match(string = tolower(Phenotype.Description), pattern = "thyroid|papillary carcinoma")[,1]), "thyroid", 
                                           ifelse(Phenotype.Description %in% c("ovarian cancer"), "ovary", 
                                           ifelse(Phenotype.Description %in% c("rectal neoplasm"), "colorectal",
                                           ifelse(Phenotype.Description %in% c("oropharynx cancer"), "upper_aerodigestive", 
                                           ifelse(!is.na(str_match(string = tolower(Phenotype.Description), pattern = "gastrointestinal")[,1]), "gastric", NA))))))))))))))))

g2p <- merge(g2p, g2p_lineages, by = c("Phenotype.Description", "Phenotype.Family"), all = TRUE)

# Add association columns
g2p$Drug_Gene <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene), NA, paste0(g2p$Drug, "_", g2p$Gene))
# g2p <- merge(unique(select(ccl_meta, grouped_general_doid, disease)), g2p, by = "grouped_general_doid", all.y = TRUE)
g2p$Drug_Gene_Lin <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene) | is.na(g2p$disease), NA, paste0(g2p$Drug, "_", g2p$Gene, "_", g2p$disease))
g2p <- unique(g2p)
write.csv(g2p, "./data_munging/g2p_db_edited_20191225.csv")
```

```{r g2p}
g2p <- read.delim("./data_munging/g2p_db_edited_20191225.csv", sep = ",", header = TRUE, row.names = 1)

g2p_dgl <-  as.character(unique(g2p$Drug_Gene_Lin[!is.na(g2p$Drug_Gene_Lin)]))
g2p_dg <- as.character(unique(g2p$Drug_Gene[!is.na(g2p$Drug_Gene)]))
g2p_drugs <- as.character(unique(g2p$Drug[!is.na(g2p$Drug)]))
g2p_genes <- as.character(unique(g2p$Gene[!is.na(g2p$Gene)]))
g2p_dg_genes <- unique(sapply(strsplit(g2p_dgl, "_", fixed = TRUE), function(x) x[2]))
g2p_lins <- as.character(unique(g2p$group_general_lineage_name[!is.na(g2p$group_general_lineage_name)]))

g2p$biomarker_type_curated <- ifelse(is.na(g2p$biomarker_type_curated), "None", as.character(g2p$biomarker_type_curated))
g2p_split <- filter(g2p, !is.na(Drug)) %>% split(g2p$biomarker_type_curated)
g2p_amp <- g2p_split[[1]]
g2p_expr <- g2p_split[[2]]
g2p_fus <- g2p_split[[3]]
g2p_fus$FusionPair <- sapply(X = g2p_fus$Feature.Names, function(x) strsplit(as.character(x)," ")[[1]][1])
g2p_fus$FusionPair <- ifelse(g2p_fus$FusionPair == "Fusions", as.character(g2p_fus$Gene), as.character(g2p_fus$FusionPair))
g2p_fus$FusionPair <- ifelse(g2p_fus$FusionPair == "BCR-ABL", "BCR-ABL1", as.character(g2p_fus$FusionPair))
g2p_fus <- g2p_fus[grep("-", g2p_fus$FusionPair), ]
g2p_fus$Drug_FusionPair <- ifelse(is.na(g2p_fus$Drug) | is.na(g2p_fus$FusionPair), NA, paste0(g2p_fus$Drug, "_", g2p_fus$FusionPair))
g2p_fus$Drug_Fusion_Lineage <- ifelse(is.na(g2p_fus$Drug) | is.na(g2p_fus$FusionPair) | is.na(g2p_fus$disease), NA, paste0(g2p_fus$Drug, "_", g2p_fus$FusionPair, "_", g2p_fus$disease))
g2p_mut <- g2p_split[[4]] %>% drop_na(Direction_Curated)
g2p_mut_genes <- unique(sapply(strsplit(as.character(g2p_mut$Drug_Gene)[!is.na(g2p_mut$Drug_Gene)], "_", fixed = TRUE), function(x) x[2]))

g2p_prot <- unique(filter(g2p, !is.na(Gene_Protein_Change) & !is.na(Drug)) %>% select(disease, Drug, Direction_Curated, Gene, Protein_Change, Gene_Protein_Change))
g2p_prot$Drug_Gene_Protein_Change <- with(g2p_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change), NA, paste0(Drug, "_", Gene_Protein_Change)))
g2p_prot$Drug_Gene_Protein_Change_Lineage <- with(g2p_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Drug, "_", Gene_Protein_Change, "_", disease)))

# Counts
g2p_dg_counts <- g2p %>% count(Drug, Gene) %>% filter(!is.na(Drug), !is.na(Gene))
g2p_dg_counts <- dcast(g2p_dg_counts, Drug ~ Gene, value.var = "n", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
g2p_dg_counts <- g2p_dg_counts[order(rowSums(-g2p_dg_counts)), order(colSums(-g2p_dg_counts))]
g2p_gene_order <- colnames(g2p_dg_counts)
g2p_drug_order <- rownames(g2p_dg_counts)
```

## Omics

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Mutation calls (coding region, germline filtered):
```{r mutation_status, eval=FALSE}
maf_raw <- read.delim("./data_munging/19Q2/CCLE_mutations_20190514.csv.gz", header = TRUE, sep = ",")
maf_raw[, 1:2] <- NULL
colnames(maf_raw)[colnames(maf_raw) == "Hugo_Symbol"] <- "Gene"
maf_raw <- unfactorize(maf_raw)

# Select columns for maf_changes_df
maf_changes <- select(maf_raw, Gene, DepMap_ID, Protein_Change) %>% drop_na(Protein_Change) %>% filter(Protein_Change != "" & Gene %in% g2p_genes)
maf_changes <- unique(maf_changes)
maf_changes$Protein_Change <- gsub(pattern = "^p.", replacement = "", x = maf_changes$Protein_Change)
maf_changes$Gene_Protein_Change <- with(maf_changes, paste0(Gene, ":", Protein_Change))
saveRDS(maf_changes, "./data_munging/19Q2/maf_changes_df_19Q2.rds", compress = "xz")

# Select columns for maf_df
maf_raw_edit <- filter(maf_raw, Variant_Classification != "Silent" & !is.na(Variant_Classification))
maf <- subset(maf_raw_edit, select = c("Gene", "DepMap_ID"))
maf$Mutation <- 1
maf_grid <- expand.grid("Gene" = unique(maf$Gene), "DepMap_ID" = unique(maf$DepMap_ID))
maf_df <- merge(maf, maf_grid, by = c("Gene", "DepMap_ID"), all = TRUE)
maf_df[is.na(maf_df)] <- 0
saveRDS(maf_df, "./data_munging/19Q2/maf_df_19Q2_nosilent.rds", compress = "xz")
# saveRDS(maf_df, "./data_munging/19Q2/maf_df_19Q2.rds", compress = "xz")
```

DepMap 19Q2 fusion calls derived using the [STAR-Fusion pipeline](https://github.com/STAR-Fusion/STAR-Fusion/wiki). Column descriptions:

- *JunctionReads*: the number of RNA-Seq fragments containing a read that aligns as a split read at the site of the putative fusion junction
- *SpanningFrags*: the number of RNA-Seq fragments that encompass the fusion junction such that one read of the pair aligns to a different gene than the other paired-end read of that fragment
    - Those predictions that have very few JunctionReads and/or SpanningReads are going to be enriched for false positives. Note, depending on the site of the fusion breakpoint and length of the reads, it may not be possible to have SpanningFragments and all evidence may show up in the form of JunctionReads
- *FFPM (fusion fragments per million total reads)*: normalized measure of the fusion-supporting RNA-seq fragmentS; a filter of 0.1 sum FFPM (meaning at least 1 fusion-supporting RNA-seq fragment per 10M total reads) tends to be effective at excluding fusion artifacts, and is the current default for filtering fusions from the final output
- *LargeAnchorSupport*: whether there are split reads that provide 'long' (set to length of 25 bases) alignments on both sides of the putative breakpoint
    - Those fusions supported only by split reads (no spanning fragments) and lack LargeAnchorSupport are often highly suspicious and tend to be false positives
    - Those with LargeAnchorSupport are labeled as 'YES_LDAS' (where LDAS = long double anchor support)
- *SpliceType*: whether the proposed breakpoint occurs at reference exon junctions as provided by the reference transcript structure annotations (ex. gencode)
- *LeftBreakEntropy* and *RightBreakEntropy*: the Shannon entropy of the 15 exonic bases flanking the breakpoint.
    - The maximum entropy is 2, representing highest complexity, and the lowest would be zero (involving a 15 base mononucleotide run)
    - Low entropy sites should generally be treated as less confident breakpoints.

```{r fusion_calls, eval=FALSE}
fusions <- read.delim("./data_munging/19Q2/CCLE_fusions.csv.gz", header = TRUE, sep = ",")
fusions_bcr_abl1 <- filter(fusions, X.FusionName %in% c("BCR--ABL1", "ABL1--BCR"))
fusions$LeftGene <- gsub(" .*", "", fusions$LeftGene)
fusions$RightGene <- gsub(" .*", "", fusions$RightGene)
fusions$X.FusionName <- gsub("--", "-", fusions$X.FusionName)
fusions <- select(fusions, DepMap_ID, X.FusionName, LeftGene, RightGene)
fusions <- as.data.frame(lapply(fusions, function(x) gsub("IGH@|IGH-@", "IGH", x)))
colnames(fusions) <- c("DepMap_ID", "FusionPair", "LeftGene", "RightGene")
fusions <- merge(fusions, ccl_meta, by = "DepMap_ID", all.x = TRUE)
write.csv.gz(fusions, "./data_munging/19Q2/CCLE_fusions_edit_20191008.csv.gz")
```

```{r}
fusions <- read.delim("./data_munging/19Q2/CCLE_fusions_edit_20191008.csv.gz", header = TRUE, sep = ",")
```

DepMap WES CN Data:

- gene_cn - NumericalMatrix
- Gene level copy number data, log2 transformed with a pseudo count 1. This is generated by mapping genes onto the segment level calls.
- Rows: cell lines (Broad IDs)
- Columns: genes (HGNC symbol and Entrez ID)

```{r copy_number, eval=FALSE}
cn <- read.delim("./data_munging/19Q2/CCLE_gene_cn_20190514.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "DepMap_ID"
# Melt
cn_melt <- melt(data = cn, id.vars = "DepMap_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Gene", value.name = "CN")
cn_melt <- unfactorize(cn_melt)

saveRDS(cn_melt, "./data_munging/19Q2/cn_melt_19Q2.rds", compress = "xz")
```

19Q2, CCLE RNAseq gene expression data:

- expression - NumericalMatrix
- RNAseq TPM gene expression data for just protein coding genes using RSEM. Log2 transformed, using a pseudo-count of 1.
- Rows: cell lines (Broad IDs)
- Columns: genes (HGNC symbol and Entrez ID)

```{r gene_expression, eval=FALSE}
ge <- read.delim("./data_munging/19Q2/CCLE_expression_20190514.csv.gz", header = TRUE, sep = ",", check.names = FALSE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(ge) <- gsub(" .*", "", colnames(ge))
colnames(ge)[1] <- "DepMap_ID"
# Melt
ge_melt <- melt(ge, id.vars = "DepMap_ID", measure.vars = colnames(ge)[2:ncol(ge)], variable.name = "Gene", value.name = "TPM")
# ge_melt$TPM <- log2(ge_melt$TPM + 1)
ge_melt <- unfactorize(ge_melt)

saveRDS(ge_melt, "./data_munging/19Q2/ge_melt_19Q2.rds", compress = "xz")
```

Load Rds:
```{r load_omics_rds}
maf_df <- readRDS("./data_munging/19Q2/maf_df_19Q2_nosilent.rds")
maf_changes_df <- readRDS("./data_munging/19Q2/maf_changes_df_19Q2.rds")
cn_melt <- readRDS("./data_munging/19Q2/cn_melt_19Q2.rds")
ge_melt <- readRDS("./data_munging/19Q2/ge_melt_19Q2.rds")
```


# MERGE OMICS

--------------------------------------------------------------------------------

```{r omics_functions}
MergeOmicsG2P <- function(database, dataframe, maf_df, ge_df, cn_df, gene_list, outfile) {
  # Merge mutation status, CN, and GE omics data with drugscreen results and metadata
  if(database %in% c("CCLE", "CTRP", "GDSC")) {
    print(paste0("Processing ", database, " data."))

    grid <- expand.grid("Cell_line" = unique(dataframe$Cell_line), "Gene" = gene_list)
    data <-  merge(dataframe, grid, by = "Cell_line", all.y = TRUE)
    print("Merged grid.")
    # Merge metadata
    ## Add columns for G2P drug-gene and drug-gene-lineage sorting
    data$Drug_Gene <- paste0(data$Drug, "_", data$Gene)
    data$G2P_Drug_Gene <- ifelse(data$Drug_Gene %in% g2p_dg, "Yes", "No")
    data$Drug_Gene_Lin <- paste0(data$Drug, "_", data$Gene, "_", data$disease)
    data$G2P_Drug_Gene_Lin <- ifelse(data$Drug_Gene_Lin %in% g2p_dgl, "Yes", "No")
    print("Merged metadata.")
    # Merge mutation calls
    data <- unfactorize(data)
    data <- merge(data, maf_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
    data$Mutation <- ifelse(is.na(data$Mutation), 0, data$Mutation)
  }
  if(database == "CRISPR") {
    print(paste0("Processing ", database, " data."))
    # Melt CRISPR dataset for merging
    data <- dataframe[, names(dataframe) %in% c("DepMap_ID", gene_list)]
    data <- melt(dataframe, id.vars = "DepMap_ID", measure.vars = colnames(data)[2:ncol(data)], variable.name = "Gene", value.name = "Score")
    # Merge cell line metadata
    data <- merge(data, ccl_meta, by = "DepMap_ID", all.y = TRUE)
    data$Drug <- "CRISPR"
    data$Drug_Gene <- paste0(data$Drug, "_", data$Gene)
    data$G2P_Drug_Gene <- "No"
    data$Drug_Gene_Lin <- paste0(data$Drug, "_", data$Gene, "_", data$disease)
    data$G2P_Drug_Gene_Lin <- "No"
    print("Merged metadata.")
    # Merge mutation calls
    data <- unfactorize(data)
    data <- merge(data, maf_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
    data$Mutation <- ifelse(is.na(data$Mutation), 0, data$Mutation)
  }
  print("Merged mutation calls.")
  # Merge GE data
  data <- merge(data, ge_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
  print("Merged gene expression data.")
  # Merge CN data
  data <- merge(data, cn_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
  print("Merged copy number data.")
  # Drop cell lines not in DepMap
  data <- filter(data, !is.na(DepMap_ID) & !(DepMap_ID %in% c("ACH-001182", "ACH-001451")))
  # Save and return dataframe
  data <- unfactorize(data)
  saveRDS(data, outfile, compress = "xz")
  return(data)
}
```

## Drug screens

```{r adjust_drugscreen_data, eval=FALSE}
ccle_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CCLE_drug_response_metrics.csv.gz", sep = ",")
ctrp_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CTRP_drug_response_metrics.csv.gz", sep = ",")
gdsc_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/GDSC_drug_response_metrics.csv.gz", sep = ",")

auc <- bind_rows(ccle_all, ctrp_all, gdsc_all) %>% drop_na(Drug_name)
auc$Drug <- tolower(auc$Drug_name)
auc$Cell_line <- toupper(gsub("-", "", auc$Cell_line))
auc$Cell_line <- ifelse(auc$Cell_line == "K052", "KO52", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "TI73", "T173", auc$Cell_line)
auc <- merge(ccl_meta, auc, by = "Cell_line", all.y = TRUE)
colnames(auc)[colnames(auc) == "AUC_IC50"] <- "Score"
saveRDS(auc, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_auc_all.rds", compress = "xz")

ccle_all <- auc %>% filter(Database == "CCLE") %>% select(Cell_line, disease, DepMap_ID, Drug, Score)
saveRDS(ccle_all, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds", compress = "xz")
ctrp_all <- auc %>% filter(Database == "CTRP") %>% select(Cell_line, disease, DepMap_ID, Drug, Score)
saveRDS(ctrp_all, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds", compress = "xz")
gdsc_all <- auc %>% filter(Database == "GDSC") %>% select(Cell_line, disease, DepMap_ID, Drug, Score)
saveRDS(gdsc_all, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds", compress = "xz")
```

`r nrow(filter(test, n != 1))`/`r nrow(test)` gene-cell line combinations have more than one mutation in the MAF file. This is corrected in the point mutation filtering done below.
```{r merge_drug_data, eval=FALSE}
ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")

ccle <- MergeOmicsG2P(database = "CCLE", dataframe = ccle_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/ccle_data_19Q2.rds")
# write.csv.gz(ccle %>% drop_na(Score), "./data_munging/19Q2/ccle_data_19Q2.csv.gz")

ctrp <- MergeOmicsG2P(database = "CTRP", dataframe = ctrp_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./../htscreens_giant_files/ctrp_data_19Q2.rds")
# write.csv.gz(ctrp %>% drop_na(Score), "./../htscreens_giant_files/ctrp_data_19Q2.csv.gz")

gdsc <- MergeOmicsG2P(database = "GDSC", dataframe = gdsc_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/gdsc_data_19Q2.rds")
# write.csv.gz(gdsc %>% drop_na(Score), "./../htscreens_giant_files/gdsc_data_19Q2.csv.gz")
```

```{r load_drug_rds}
# ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
# ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
# gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")

# ccle <- readRDS("./data_munging/19Q2/ccle_data_19Q2_all.rds") %>% drop_na(Drug)
# ctrp <- readRDS("./../htscreens_giant_files/ctrp_data_19Q2_all.rds") %>% drop_na(Drug)
# gdsc <- readRDS("./data_munging/19Q2/gdsc_data_19Q2_all.rds") %>% drop_na(Drug)
# saveRDS(ccle, "./data_munging/19Q2/ccle_data_19Q2.rds", compress = "xz")
# saveRDS(ctrp, "./../htscreens_giant_files/ctrp_data_19Q2.rds", compress = "xz")
# saveRDS(gdsc, "./data_munging/19Q2/gdsc_data_19Q2.rds", compress = "xz")

ccle <- readRDS("./data_munging/19Q2/ccle_data_19Q2.rds") %>% drop_na(Drug)
ctrp <- readRDS("./../htscreens_giant_files/ctrp_data_19Q2.rds") %>% drop_na(Drug)
gdsc <- readRDS("./data_munging/19Q2/gdsc_data_19Q2.rds") %>% drop_na(Drug)

# Testing
test_ccle <- ccle %>% count(Drug)
test_ctrp <- ctrp %>% count(Drug)
test_gdsc <- gdsc %>% count(Drug)
```

How many datasets screen each G2P drug?
```{r drug_testing, eval=FALSE}
drug_summ <- data.frame(Drug = character(), Dataset_Count = integer(), stringsAsFactors = FALSE)

for(drug in g2p_drugs) {
  counter = 0
  if(drug %in% as.character(unique(ccle$Drug[!is.na(ccle$Drug)]))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(ctrp$Drug[!is.na(ctrp$Drug)]))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(gdsc$Drug[!is.na(gdsc$Drug)]))) { counter <- counter + 1 }
  drug_summ <- rbind(drug_summ, data.frame(Drug = drug, Dataset_Count = counter))
}
```

Merge fusions:
```{r merge_fusions, eval=FALSE}
ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")
fusions_filt <- filter(fusions, FusionPair %in% unique(as.character(g2p_fus$FusionPair)))

ccle_fus_grid <- expand.grid("DepMap_ID" = unique(ccle_all$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair))
ccle_fus <- merge(merge(ccle_all, ccle_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "Cell_line", "FusionPair", "disease"), all = TRUE)
ccle_fus$Fusion_Status <- ifelse(!is.na(ccle_fus$LeftGene), "Yes", "No")
ccle_fus$Drug_FusionPair <- with(ccle_fus, ifelse(is.na(Drug) | is.na(FusionPair), NA, paste0(Drug, "_", FusionPair)))
ccle_fus$Drug_Fusion_Lineage <- with(ccle_fus, ifelse(is.na(Drug) | is.na(FusionPair) | is.na(disease), NA, paste0(Drug, "_", FusionPair, "_", disease)))
ccle_fus$G2P_Drug_FusionPair <- ifelse(ccle_fus$Drug_FusionPair %in% unique(as.character(g2p_fus$Drug_FusionPair)), "Yes", "No")
ccle_fus$G2P_Fusion <- ifelse(ccle_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
ccle_fus$G2P_Drug_Fusion_Lineage <- ifelse(ccle_fus$Drug_Fusion_Lineage %in% unique(as.character(g2p_fus$Drug_Fusion_Lineage)), "Yes", "No")
ccle_fus$Database <- "CCLE"

ctrp_fus_grid <- expand.grid("DepMap_ID" = unique(ctrp_all$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair))
ctrp_fus <- merge(merge(ctrp_all, ctrp_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "Cell_line", "FusionPair", "disease"), all = TRUE)
ctrp_fus$Fusion_Status <- ifelse(!is.na(ctrp_fus$LeftGene), "Yes", "No")     
ctrp_fus$Drug_FusionPair <- with(ctrp_fus, ifelse(is.na(Drug) | is.na(FusionPair), NA, paste0(Drug, "_", FusionPair)))
ctrp_fus$Drug_Fusion_Lineage <- with(ctrp_fus, ifelse(is.na(Drug) | is.na(FusionPair) | is.na(disease), NA, paste0(Drug, "_", FusionPair, "_", disease)))
ctrp_fus$G2P_Drug_FusionPair <- ifelse(ctrp_fus$Drug_FusionPair %in% unique(as.character(g2p_fus$Drug_FusionPair)), "Yes", "No")
ctrp_fus$G2P_Fusion <- ifelse(ctrp_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
ctrp_fus$G2P_Drug_Fusion_Lineage <- ifelse(ctrp_fus$Drug_Fusion_Lineage %in% unique(as.character(g2p_fus$Drug_Fusion_Lineage)), "Yes", "No")
ctrp_fus$Database <- "CTRP"

gdsc_fus_grid <- expand.grid("DepMap_ID" = unique(gdsc_all$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair))
gdsc_fus <- merge(merge(gdsc_all, gdsc_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "Cell_line", "FusionPair", "disease"), all = TRUE)
gdsc_fus$Fusion_Status <- ifelse(!is.na(gdsc_fus$LeftGene), "Yes", "No")
gdsc_fus$Drug_FusionPair <- with(gdsc_fus, ifelse(is.na(Drug) | is.na(FusionPair), NA, paste0(Drug, "_", FusionPair)))
gdsc_fus$Drug_Fusion_Lineage <- with(gdsc_fus, ifelse(is.na(Drug) | is.na(FusionPair) | is.na(disease), NA, paste0(Drug, "_", FusionPair, "_", disease)))
gdsc_fus$G2P_Drug_FusionPair <- ifelse(gdsc_fus$Drug_FusionPair %in% unique(as.character(g2p_fus$Drug_FusionPair)), "Yes", "No")
gdsc_fus$G2P_Fusion <- ifelse(gdsc_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
gdsc_fus$G2P_Drug_Fusion_Lineage <- ifelse(gdsc_fus$Drug_Fusion_Lineage %in% unique(as.character(g2p_fus$Drug_Fusion_Lineage)), "Yes", "No")
gdsc_fus$Database <- "GDSC"

fus <- bind_rows(ccle_fus, ctrp_fus, gdsc_fus)
colnames(fus)[colnames(fus) == "AUC_IC50"] <- "Score"
saveRDS(fus, "./data_munging/rds/fusions_merged.rds")
```

```{r load_fusion_data}
fus <- readRDS("./data_munging/rds/fusions_merged.rds") %>% drop_na(Drug)
fus_summary <- fus %>% count(Database, Drug, FusionPair, Fusion_Status)
```

Merge protein changes
```{r merge_protein_changes, eval=FALSE}
ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")
maf_changes_df <- readRDS("./data_munging/19Q2/maf_changes_df_19Q2.rds")
maf_changes_filt <- filter(maf_changes_df, Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)))

# CCLE
ccle_prot_grid <- expand.grid("DepMap_ID" = unique(ccle_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
ccle_prot <- merge(ccle_all, ccle_prot_grid, by = "DepMap_ID", all = TRUE)
ccle_prot <- merge(ccle_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
ccle_prot$Change_Status <- ifelse(!is.na(ccle_prot$Protein_Change), "Yes", "No")
ccle_prot$Drug_Gene_Protein_Change <- with(ccle_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change), NA, paste0(Drug, "_", Gene_Protein_Change)))
ccle_prot$Drug_Gene_Protein_Change_Lineage <- with(ccle_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Drug, "_", Gene_Protein_Change, "_", disease)))
ccle_prot$G2P_Drug_Gene_Protein_Change <- ifelse(ccle_prot$Drug_Gene_Protein_Change %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change)), "Yes", "No")
ccle_prot$G2P_Protein_Change <- ifelse(ccle_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
ccle_prot$G2P_Drug_Gene_Protein_Change_Lineage <- ifelse(ccle_prot$Drug_Gene_Protein_Change_Lineage %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change_Lineage)), "Yes", "No")
ccle_prot <- merge(ccle_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
ccle_prot$Database <- "CCLE"
ccle_prot <- ccle_prot %>% drop_na(Drug, DepMap_ID)

# CTRP
ctrp_prot_grid <- expand.grid("DepMap_ID" = unique(ctrp_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
ctrp_prot <- merge(ctrp_all, ctrp_prot_grid, by = "DepMap_ID", all = TRUE)
ctrp_prot <- merge(ctrp_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
ctrp_prot$Change_Status <- ifelse(!is.na(ctrp_prot$Protein_Change), "Yes", "No")
ctrp_prot$Drug_Gene_Protein_Change <- with(ctrp_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change), NA, paste0(Drug, "_", Gene_Protein_Change)))
ctrp_prot$Drug_Gene_Protein_Change_Lineage <- with(ctrp_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Drug, "_", Gene_Protein_Change, "_", disease)))
ctrp_prot$G2P_Drug_Gene_Protein_Change <- ifelse(ctrp_prot$Drug_Gene_Protein_Change %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change)), "Yes", "No")
ctrp_prot$G2P_Protein_Change <- ifelse(ctrp_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
ctrp_prot$G2P_Drug_Gene_Protein_Change_Lineage <- ifelse(ctrp_prot$Drug_Gene_Protein_Change_Lineage %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change_Lineage)), "Yes", "No")
ctrp_prot <- merge(ctrp_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
ctrp_prot$Database <- "CTRP"
ctrp_prot <- ctrp_prot %>% drop_na(Drug, DepMap_ID)

# GDSC
gdsc_prot_grid <- expand.grid("DepMap_ID" = unique(gdsc_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
gdsc_prot <- merge(gdsc_all, gdsc_prot_grid, by = "DepMap_ID", all = TRUE)
gdsc_prot <- merge(gdsc_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
gdsc_prot$Change_Status <- ifelse(!is.na(gdsc_prot$Protein_Change), "Yes", "No")
gdsc_prot$Drug_Gene_Protein_Change <- with(gdsc_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change), NA, paste0(Drug, "_", Gene_Protein_Change)))
gdsc_prot$Drug_Gene_Protein_Change_Lineage <- with(gdsc_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Drug, "_", Gene_Protein_Change, "_", disease)))
gdsc_prot$G2P_Drug_Gene_Protein_Change <- ifelse(gdsc_prot$Drug_Gene_Protein_Change %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change)), "Yes", "No")
gdsc_prot$G2P_Protein_Change <- ifelse(gdsc_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
gdsc_prot$G2P_Drug_Gene_Protein_Change_Lineage <- ifelse(gdsc_prot$Drug_Gene_Protein_Change_Lineage %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change_Lineage)), "Yes", "No")
gdsc_prot <- merge(gdsc_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
gdsc_prot$Database <- "GDSC"
gdsc_prot <- gdsc_prot %>% drop_na(Drug, DepMap_ID)

prot <- bind_rows(ccle_prot, ctrp_prot, gdsc_prot)
prot <- separate(data = prot, col = Gene_Protein_Change, into = c("Gene", "Protein_Change"), remove = FALSE, sep = ":")
prot <- add_column(prot, Drug_Gene = paste0(prot$Drug, "_", prot$Gene), .after = "Drug")
prot$Direction_Curated <- ifelse(is.na(prot$Direction_Curated), "none", as.character(prot$Direction_Curated))
prot$Direction_Grouper <- with(prot, paste0(Change_Status, "_", Direction_Curated))
saveRDS(prot, "./data_munging/rds/protein_changes_merged.rds")
```

```{r load_prot_data}
prot <- readRDS("./data_munging/rds/protein_changes_merged.rds")
# prot_summary <- prot %>% count(Database, Drug, Gene_Protein_Change, Change_Status)
```

## CRISPR

```{r merge_crispr_data, eval=FALSE}
crispr_all <- read.delim("./data_munging/19Q2/Achilles_gene_effect_20190514.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr_all) <- gsub(" .*", "", colnames(crispr_all))
colnames(crispr_all)[1] <- "DepMap_ID"

# crispr_allgenes <- MergeOmicsG2P(database = "CRISPR", dataframe = crispr_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = colnames(crispr_all)[2:length(colnames(crispr_all))], outfile = "./../htscreens_giant_files/crispr_data_19Q2_allgenes.rds")

crispr <- MergeOmicsG2P(database = "CRISPR", dataframe = crispr_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/crispr_data_19Q2.rds")
# write.csv.gz(crispr %>% drop_na(Score), "./../htscreens_giant_files/crispr_data_19Q2.csv.gz")
```

```{r read_crispr_data}
crispr <- readRDS("./data_munging/19Q2/crispr_data_19Q2.rds") %>% drop_na(Gene)
```

```{r merge_crispr_fus, eval=FALSE}
# crispr_allgenes <- readRDS("./../htscreens_giant_files/crispr_data_19Q2_allgenes.rds")
# fusions_filt <- unfactorize(unique(filter(fusions, FusionPair %in% unique(as.character(g2p_fus$FusionPair)))) %>% select(-c("Cell_line", "disease")))
# crispr_fus_grid <- unfactorize(expand.grid("DepMap_ID" = unique(crispr_allgenes$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair)))
# crispr_fus <- merge(merge(crispr_allgenes, crispr_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "FusionPair"), all.x = TRUE)
# crispr_fus$Fusion_Status <- ifelse(!is.na(crispr_fus$LeftGene), "Yes", "No")
# crispr_fus$Drug_FusionPair <- with(crispr_fus, ifelse(is.na(Drug) | is.na(FusionPair), NA, paste0(Drug, "_", FusionPair)))
# crispr_fus$Drug_Fusion_Lineage <- with(crispr_fus, ifelse(is.na(Drug) | is.na(FusionPair) | is.na(disease), NA, paste0(Drug, "_", FusionPair, "_", disease)))
# crispr_fus$G2P_Drug_FusionPair <- "No"
# crispr_fus$G2P_Fusion <- ifelse(crispr_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
# crispr_fus$G2P_Drug_Fusion_Lineage <- "No"
# crispr_fus$Database <- "CRISPR"
# crispr_fus <- crispr_fus %>% drop_na(Score)
# saveRDS(crispr_fus, "./../htscreens_giant_files/crispr_fusions_allgenes.rds", compress = "xz")

fusions_filt <- unfactorize(unique(filter(fusions, FusionPair %in% unique(as.character(g2p_fus$FusionPair)))) %>% select(-c("Cell_line", "disease")))
crispr_fus_grid <- unfactorize(expand.grid("DepMap_ID" = unique(crispr$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair)))
crispr_fus <- merge(merge(crispr, crispr_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "FusionPair"), all.x = TRUE)
crispr_fus$Fusion_Status <- ifelse(!is.na(crispr_fus$LeftGene), "Yes", "No")
crispr_fus$Drug_FusionPair <- with(crispr_fus, ifelse(is.na(Drug) | is.na(FusionPair), NA, paste0(Drug, "_", FusionPair)))
crispr_fus$Drug_Fusion_Lineage <- with(crispr_fus, ifelse(is.na(Drug) | is.na(FusionPair) | is.na(disease), NA, paste0(Drug, "_", FusionPair, "_", disease)))
crispr_fus$G2P_Drug_FusionPair <- "No"
crispr_fus$G2P_Fusion <- ifelse(crispr_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
crispr_fus$G2P_Drug_Fusion_Lineage <- "No"
crispr_fus$Database <- "CRISPR"
crispr_fus <- crispr_fus %>% drop_na(Score)
saveRDS(crispr_fus, "./data_munging/19Q2/crispr_fusions.rds", compress = "xz")
```

```{r merge_crispr_prot, eval=FALSE}
crispr_all <- read.delim("./data_munging/19Q2/Achilles_gene_effect_20190514.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr_all) <- gsub(" .*", "", colnames(crispr_all))
colnames(crispr_all)[1] <- "DepMap_ID"

crispr_all <- crispr_all[, names(crispr_all) %in% c("DepMap_ID", g2p_genes)]

maf_changes_df <- readRDS("./data_munging/19Q2/maf_changes_df_19Q2.rds")
maf_changes_filt <- filter(maf_changes_df, Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change))) %>% select(-Protein_Change)
maf_changes_filt$Change_Status <- "Yes"

# Melt CRISPR dataset for merging
crispr_melt <- melt(crispr_all, id.vars = "DepMap_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")
crispr_melt <- filter(crispr_melt, Gene %in% unique(maf_changes_filt$Gene))

grid <- unfactorize(expand.grid("DepMap_ID" = unique(crispr_melt$DepMap_ID), "Gene_Protein_Change" = unique(g2p_prot$Gene_Protein_Change))) 
grid <- separate(grid, Gene_Protein_Change, c("Gene", "Protein_Change"), sep = ":", remove = FALSE) %>% select(-Protein_Change)

crispr_prot <- merge(crispr_melt, grid, by = c("DepMap_ID", "Gene"), all = TRUE)
crispr_prot <- merge(crispr_prot, maf_changes_filt, by = c("DepMap_ID", "Gene", "Gene_Protein_Change"), all = TRUE)  %>% drop_na(Score)
crispr_prot <- merge(crispr_prot, ccl_meta, by = "DepMap_ID", all.x = TRUE)
crispr_prot$Change_Status <- ifelse(is.na(crispr_prot$Change_Status), "No", crispr_prot$Change_Status)
crispr_prot$Drug <- "CRISPR"
crispr_prot$Gene_Protein_Change_Lineage <- with(crispr_prot, ifelse(is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Gene_Protein_Change, "_", disease)))
crispr_prot$Drug_Gene_Protein_Change <- with(crispr_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change), NA, paste0(Drug, "_", Gene_Protein_Change)))
crispr_prot$Drug_Gene_Protein_Change_Lineage <- with(crispr_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Drug, "_", Gene_Protein_Change, "_", disease)))
crispr_prot$G2P_Drug_Gene_Protein_Change <- "No"
crispr_prot$G2P_Protein_Change <- ifelse(crispr_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
crispr_prot$G2P_Drug_Gene_Protein_Change_Lineage <- "No"
crispr_prot <- merge(crispr_prot, unique(select(g2p_prot, Gene_Protein_Change, Direction_Curated)), by = "Gene_Protein_Change", all.x = TRUE)
crispr_prot$Database <- "CRISPR"

saveRDS(crispr_prot, "./../htscreens_giant_files/crispr_protein_changes_20191223.rds", compress = "xz")
```

```{r, eval = TRUE}
crispr_fus <- readRDS("./data_munging/19Q2/crispr_fusions.rds")
# crispr_prot <- readRDS("./../htscreens_giant_files/crispr_protein_changes.rds")
crispr_prot <- readRDS("./../htscreens_giant_files/crispr_protein_changes_20191223.rds")
```

```{r, eval = FALSE}
crispr_allgenes <- readRDS("./../htscreens_giant_files/crispr_data_19Q2_allgenes.rds")
crispr_fus_allgenes <- readRDS("./../htscreens_giant_files/crispr_fusions_allgenes.rds")
crispr_prot_allgenes <- readRDS("./../htscreens_giant_files/crispr_protein_changes_allgenes.rds")
```

# EXPLORATORY FIGURES

--------------------------------------------------------------------------------

## Mutation

## Cell line and drug overlaps

```{r ccl_venn, eval=FALSE}
ccle_ccls <- unique(ccle$DepMap_ID)
ctrp_ccls <- unique(ctrp$DepMap_ID)
gdsc_ccls <- unique(gdsc$DepMap_ID)
crispr_ccls <- unique(crispr$DepMap_ID)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_ccls, ctrp_ccls, gdsc_ccls, crispr_ccls),
             filename = "./plots/manuscript/all_venn_ccls.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1,
             category.names = c("CCLE", "CTRP", "GDSC", "CRISPR"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             # cat.dist = c(0.06, 0.06, 0.035),
             fill = c("springgreen4", "steelblue4", "turquoise4", "darkorchid"))
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_ccls, ctrp_ccls, gdsc_ccls),
             filename = "./plots/manuscript/drug_venn_ccls.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1.4,
             category.names = c("CCLE", "CTRP", "GDSC"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             cat.dist = c(0.06, 0.06, 0.035),
             fill = c("springgreen4", "steelblue4", "turquoise4"))
```

```{r drug_venn, eval=FALSE}
ccle_drugs <- unique(ccle$Drug)
ctrp_drugs <- unique(ctrp$Drug)
gdsc_drugs <- unique(gdsc$Drug)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_drugs, ctrp_drugs, gdsc_drugs),
             filename = "./plots/manuscript/drug_venn_drugs.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1.4,
             category.names = c("CCLE", "CTRP", "GDSC"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             cat.dist = c(0.06, 0.06, 0.035),
             fill = c("palegreen3", "slategray3", "paleturquoise3"))
```

## Histology distributions

```{r histology_dists_data, eval=FALSE}
hist_count <- function(df, dataset) {
  hists <- unique(select(df, DepMap_ID, disease)) %>% count(disease)
  hists$Percent <- hists$n / sum(hists$n) * 100
  hists$Dataset <- dataset
  hists$disease <- gsub(" cancer", "", hists$disease)
  hists <- hists[order(hists$n, decreasing = T),]
  return(hists)
}

ccle_hists <- hist_count(ccle, "CCLE")
ctrp_hists <- hist_count(ctrp, "CTRP")
gdsc_hists <- hist_count(gdsc, "GDSC")
crispr_hists <- hist_count(crispr, "CRISPR")

drug_hists <- rbind(ccle_hists, ctrp_hists, gdsc_hists, crispr_hists) %>% drop_na(disease)
drug_hists$Dataset <- factor(drug_hists$Dataset, levels = c("CCLE", "CTRP", "GDSC", "CRISPR"))
# drug_hists$disease <- as.character(drug_hists$disease)
saveRDS(drug_hists, "./data_munging/rds/histology_distributions.rds")
```

```{r histology_dists, fig.height=4.5, fig.width=4, eval=FALSE}
drug_hists <- readRDS("./data_munging/rds/histology_distributions.rds")

png(filename = "./plots/manuscript/all_histology_distributions.png", width = 8, height = 9, units = "in", res = 500)
ggplot(drug_hists, aes(x = Dataset, y = n, fill = disease)) +
  geom_bar(stat = "identity", color = "black") +
  scale_y_continuous(breaks = seq(0, 850, by = 50), labels = seq(0, 850, by = 50)) +
  guides(fill = guide_legend(title = "Histology", ncol = 1)) +
  labs(y = "Count")
dev.off()
```

```{r, eval=FALSE}
ccle_ccls <- unique(ccle$DepMap_ID)
ctrp_ccls <- unique(ctrp$DepMap_ID)
gdsc_ccls <- unique(gdsc$DepMap_ID)
crispr_ccls <- unique(crispr$DepMap_ID)

ccls <- data.frame(DepMap_ID = union(ccle_ccls, union(ctrp_ccls, gdsc_ccls)))
all_ccls <- merge(ccls, ccl_meta, by = "DepMap_ID")
all_ccls$disease <- gsub(" cancer", "", all_ccls$disease)

png(filename = "./plots/manuscript/drug_histology_distributions.png", width = 8, height = 4, units = "in", res = 500)
ggplot(all_ccls, aes(x = fct_infreq(factor(disease)), fill = disease)) +
  geom_bar(color = "black") +
  scale_y_continuous(breaks = seq(0, 220, by = 40), labels = seq(0, 220, by = 40)) +
  guides(fill = FALSE) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, vjust = 1)) +
  labs(y = "Number of Cell Lines", x = "Histology", caption = paste0(formatC(nrow(all_ccls), format = "d", big.mark = ","), " cell lines represented."))
dev.off()
```

## G2P summary heatmap

```{r dg_heatmap, fig.height=6, fig.width=8, warning=FALSE, eval=FALSE}
# Create annotation layer
hta_data <- data.frame(Drug = rownames(g2p_dg_counts), CCLE = NA, CTRP = NA, GDSC = NA)
hta_data$CCLE <- ifelse(hta_data$Drug %in% unique(ccle$Drug), "Screened", "Not screened")
hta_data$CTRP <- ifelse(hta_data$Drug %in% unique(ctrp$Drug), "Screened", "Not screened")
hta_data$GDSC <- ifelse(hta_data$Drug %in% unique(gdsc$Drug), "Screened", "Not screened")
rownames(hta_data) <- hta_data$Drug
hta_data$Drug <- NULL

hta_annot <- HeatmapAnnotation(text = anno_text(colnames(hta_data), offset = unit(0.4, "in"), gp = gpar(fontsize = 10)))
hta_legend_list <- list(legend_direction = "horizontal", border = "gray70", nrow = 2)
hta <- Heatmap(hta_data,
               name = "Functional screen status           ",
               col = c("Screened" = "purple4", "Not screened" = "gray97"), 
               rect_gp = gpar(col = "gray70"),
               top_annotation = hta_annot,
               heatmap_legend_param = hta_legend_list,
               row_names_side = "left",
               row_names_gp = gpar(fontsize = 10),
               column_names_side = "top",
               column_names_gp = gpar(fontsize = 10),
               show_column_names = FALSE,
               cluster_rows = FALSE, 
               cluster_columns = FALSE,
               row_title = "Drugs",
               row_title_gp = gpar(fontface = "bold", fontsize = 14),
               column_title = "Dataset",
               column_title_gp = gpar(fontface = "bold", fontsize = 14),
               width = unit(1.5, "in"))

# Define color scale
col_fun <- colorRamp2(c(0, 1, 100), c("gray97", "mistyrose", "purple4"))
# Legend properties list
ht_legend_list <- list(legend_direction = "horizontal", legend_width = unit(2.5, "in"), border = "gray70")
ht <- Heatmap(g2p_dg_counts, 
              name = "Number of G2P interpretations",
              col = col_fun,
              na_col = "gray97",
              rect_gp = gpar(col = "gray70"),
              heatmap_legend_param = ht_legend_list,
              cluster_rows = FALSE, 
              cluster_columns = FALSE, 
              show_row_names = FALSE,
              row_names_side = "left",
              row_names_gp = gpar(fontsize = 10),
              column_names_side = "bottom",
              show_column_names = FALSE,
              bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(g2p_dg_counts),
                                   rot = 45,
                                   gp = gpar(fontsize = 8),
                                   offset = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.45, "in")),
              row_title = "Drugs",
              row_title_gp = gpar(fontface = "bold", fontsize = 14),
              column_title = "Genes", 
              column_title_gp = gpar(fontface = "bold", fontsize = 14),
              column_title_side = "bottom")

png(filename = "./plots/manuscript/g2p_summary_heatmap.png", width = 10, height = 9, units = "in", res = 500)
draw(hta + ht, gap = unit(5, "mm"), heatmap_legend_side = "top")
dev.off()
```

## Clustered AUC heatmap

```{r auc_clustered_ht, eval=FALSE}
ccle_auc_grid <- dcast(ccle, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
ctrp_auc_grid <- dcast(ctrp, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
gdsc_auc_grid <- dcast(gdsc, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)

ccle_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(ccle_auc_grid)), ccl_meta, by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)
ctrp_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(ctrp_auc_grid)), ccl_meta, by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)
gdsc_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(gdsc_auc_grid)), ccl_meta, by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)

ht_legend_list <- list(direction = "horizontal")

ccle_col_fun <- colorRamp2(c(0, 1), c("gray97", "darkgreen"))
ccle_auc_ha <- HeatmapAnnotation(Lineage = ccle_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 5)))
ccle_auc_ht <- Heatmap(ccle_auc_grid,
                       na_col = "black",
                       col = ccle_col_fun,
                       top_annotation = ccle_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

ctrp_col_fun <- colorRamp2(c(0, 1), c("gray97", "steelblue4"))
ctrp_auc_ha <- HeatmapAnnotation(Lineage = ctrp_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 3)))
ctrp_auc_ht <- Heatmap(ctrp_auc_grid,
                       na_col = "black",
                       col = ctrp_col_fun,
                       top_annotation = ctrp_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

gdsc_col_fun <- colorRamp2(c(0, 1), c("gray97", "turquoise4"))
gdsc_auc_ha <- HeatmapAnnotation(Lineage = gdsc_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 3)))
gdsc_auc_ht <- Heatmap(gdsc_auc_grid,
                       na_col = "black",
                       col = gdsc_col_fun,
                       top_annotation = gdsc_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

png(filename = "./plots/manuscript/ccle_clustered_heatmap.png", width = 10, height = 8, units = "in", res = 500)
draw(ccle_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()

png(filename = "./plots/manuscript/ctrp_clustered_heatmap.png", width = 22, height = 40, units = "in", res = 500)
draw(ctrp_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()

png(filename = "./plots/manuscript/gdsc_clustered_heatmap.png", width = 18, height = 20, units = "in", res = 500)
draw(gdsc_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()
```


# WILCOXON: Mutations

--------------------------------------------------------------------------------

ggpubr's convention for symbols indicating statistical significance:

- ns: p > 0.05
- *: p <= 0.05
- **: p <= 0.01
- ***: p <= 0.001
- ****: p <= 0.0001

```{r wilcoxon_functions}
adj_signif <- function(df, alpha) {
  # Takes df from compare_means and creates a signif code column for FDR-adjusted p-vals
  df$p.signif.adj <- ifelse(df$p.adj <= alpha, "*", NA)
  df$p.signif <- ifelse(df$p.signif == "ns", NA, df$p.signif)
  df$p.short <- formatC(df$p, format = "g", digits =  2)
  df$p.adj.short <- formatC(df$p.adj, format = "g", digits =  2)
  return(df)
}

WilcoxonTestDF <- function(dataframe, grouping, g2p_list, type, outfile) {
  require(ggpubr)
  
  # Mutation status
  if(tolower(type) == "drug_gene") {
    # Filter out DGs with too few samples in at least one mutation status category
    # Require at least 3 samples in each category to run the test, else filter out that DG from testing df
    count_n <- dataframe %>% count(Drug_Gene, Mutation)
    count_n_wide <- dcast(count_n, Drug_Gene ~ Mutation, value.var = "n", fill = 0) %>% separate(Drug_Gene, sep = "_", c("Drug", "Gene"), remove = FALSE)
    colnames(count_n_wide) <- c("Drug_Gene", "Drug", "Gene", "Mutation_0_count", "Mutation_1_count")
    count_n_filt <- filter(count_n, n >= 3)
    count_category <- count_n_filt %>% count(Drug_Gene) %>% filter(n == 2)
    dg_keep <- intersect(count_n_filt$Drug_Gene, count_category$Drug_Gene)
    dataframe <- filter(dataframe, Drug_Gene %in% dg_keep)
    dataframe$Drug_Gene <- as.character(dataframe$Drug_Gene)
    signif <- compare_means(Score ~ Mutation, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif <- merge(signif, count_n_wide, by = grouping, all = TRUE)
    if(is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- ifelse(signif$Drug_Gene %in% g2p_list, "Yes", "No") }
    if(!is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- "No" }
  }
  
  # Mutation status and lineage
  if(tolower(type) == "drug_gene_lineage") {
    # Filter out DGLs with too few samples in at least one mutation status category
    # Require at least 3 samples in each category to run the test, else filter out that DGL from testing df
    count_n <- dataframe %>% count(Drug_Gene_Lin, Mutation)
    count_n_wide <- dcast(count_n, Drug_Gene_Lin ~ Mutation, value.var = "n", fill = 0) %>% separate(Drug_Gene_Lin, sep = "_", c("Drug", "Gene", "disease"), remove = FALSE)
    colnames(count_n_wide) <- c("Drug_Gene_Lin", "Drug", "Gene", "disease", "Mutation_0_count", "Mutation_1_count")
    count_n_wide <- add_column(count_n_wide, Drug_Gene = paste0(count_n_wide$Drug, "_", count_n_wide$Gene), .after = "Gene")
    count_n_filt <- filter(count_n, n >= 3)
    count_category <- count_n_filt %>% count(Drug_Gene_Lin) %>% filter(n == 2)
    dg_keep <- intersect(count_n_filt$Drug_Gene_Lin, count_category$Drug_Gene_Lin)
    dataframe <- filter(dataframe, Drug_Gene_Lin %in% dg_keep)
    dataframe$Drug_Gene_Lin <- as.character(dataframe$Drug_Gene_Lin)
    signif <- compare_means(Score ~ Mutation, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif <- merge(signif, count_n_wide, by = grouping, all = TRUE)
    if(is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- ifelse(signif$Drug_Gene_Lin %in% g2p_list, "Yes", "No") }
    if(!is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- "No" }
  }
  
  # Protein changes
  if(tolower(type) == "protein_change") {
    # Filter out Gene:Protein_Changes with too few samples in at least one category
    # Require at least 3 samples in each category to run the test, else filter out that Gene_Protein_Change from testing df
    count_n <- dataframe %>% count(Drug_Gene_Protein_Change, Change_Status)
    count_n_wide <- dcast(count_n, Drug_Gene_Protein_Change ~ Change_Status, value.var = "n", fill = 0) %>% separate(Drug_Gene_Protein_Change, sep = "_", c("Drug", "Gene_Protein_Change"), remove = FALSE)
    colnames(count_n_wide) <- c("Drug_Gene_Protein_Change", "Drug", "Gene_Protein_Change", "Change_Status_0_count", "Change_Status_1_count")
    count_n_filt <- filter(count_n, n >= 3)
    count_category <- count_n_filt %>% count(Drug_Gene_Protein_Change) %>% filter(n == 2)
    dg_keep <- intersect(count_n_filt$Drug_Gene_Protein_Change, count_category$Drug_Gene_Protein_Change)
    dataframe <- filter(dataframe, Drug_Gene_Protein_Change %in% dg_keep)
    dataframe$Drug_Gene_Protein_Change <- as.character(dataframe$Drug_Gene_Protein_Change)
    signif <- compare_means(Score ~ Change_Status, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif <- merge(signif, count_n_wide, by = grouping, all = TRUE)
    if(is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- ifelse(signif$Drug_Gene_Protein_Change %in% g2p_list, "Yes", "No") }
    if(!is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- "No" }
  }
  
  # Protein changes and lineage
  if(tolower(type) == "protein_change_lineage") {
    # Filter out Gene:Protein_Changes with too few samples in at least one category
    # Require at least 3 samples in each category to run the test, else filter out that Drug_Gene_Protein_Change_Lineage from testing df
    count_n <- dataframe %>% count(Drug_Gene_Protein_Change_Lineage, Change_Status)
    count_n_wide <- dcast(count_n, Drug_Gene_Protein_Change_Lineage ~ Change_Status, value.var = "n", fill = 0) %>% separate(Drug_Gene_Protein_Change_Lineage, sep = "_", c("Drug", "Gene_Protein_Change", "disease"), remove = FALSE)
    colnames(count_n_wide) <- c("Drug_Gene_Protein_Change_Lineage", "Drug", "Gene_Protein_Change", "disease", "Change_Status_0_count", "Change_Status_1_count")
    count_n_wide <- add_column(count_n_wide, Drug_Gene_Protein_Change = paste0(count_n_wide$Drug, "_", count_n_wide$Gene_Protein_Change), .after = "Gene_Protein_Change")
    count_n_filt <- filter(count_n, n >= 3)
    count_category <- count_n_filt %>% count(Drug_Gene_Protein_Change_Lineage) %>% filter(n == 2)
    dg_keep <- intersect(count_n_filt$Drug_Gene_Protein_Change_Lineage, count_category$Drug_Gene_Protein_Change_Lineage)
    dataframe <- filter(dataframe, Drug_Gene_Protein_Change_Lineage %in% dg_keep)
    dataframe$Drug_Gene_Protein_Change_Lineage <- as.character(dataframe$Drug_Gene_Protein_Change_Lineage)
    signif <- compare_means(Score ~ Change_Status, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif <- merge(signif, count_n_wide, by = grouping, all = TRUE)
    if(is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- ifelse(signif$Drug_Gene_Protein_Change_Lineage %in% g2p_list, "Yes", "No") }
    if(!is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- "No" }
  }
  
  # Fusions
  if(tolower(type) == "fusion") {
    # Filter out fusions with too few samples in at least one category
    # Require at least 3 samples in each category to run the test, else filter out that Drug_FusionPair from testing df
    count_n <- dataframe %>% count(Drug_FusionPair, Fusion_Status)
    count_n_wide <- dcast(count_n, Drug_FusionPair ~ Fusion_Status, value.var = "n", fill = 0) %>% separate(Drug_FusionPair, sep = "_", c("Drug", "FusionPair"), remove = FALSE)
    colnames(count_n_wide) <- c("Drug_FusionPair", "Drug", "FusionPair", "Fusion_Status_0_count", "Fusion_Status_1_count")
    count_n_filt <- filter(count_n, n >= 3)
    count_category <- count_n_filt %>% count(Drug_FusionPair) %>% filter(n == 2)
    dg_keep <- intersect(count_n_filt$Drug_FusionPair, count_category$Drug_FusionPair)
    dataframe <- filter(dataframe, Drug_FusionPair %in% dg_keep)
    dataframe$Drug_FusionPair <- as.character(dataframe$Drug_FusionPair)
    signif <- compare_means(Score ~ Fusion_Status, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif <- merge(signif, count_n_wide, by = grouping, all = TRUE)
    if(is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- ifelse(signif$Drug_FusionPair %in% g2p_list, "Yes", "No")  }
    if(!is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- "No" }
  }
  
  # Fusions and lineage
  if(tolower(type) == "fusion_lineage") {
    # Filter out fusions with too few samples in at least one category
    # Require at least 3 samples in each category to run the test, else filter out that Drug_Fusion_Lineage from testing df
    count_n <- dataframe %>% count(Drug_Fusion_Lineage, Fusion_Status)
    count_n_wide <- dcast(count_n, Drug_Fusion_Lineage ~ Fusion_Status, value.var = "n", fill = 0) %>% separate(Drug_Fusion_Lineage, sep = "_", c("Drug", "FusionPair", "disease"), remove = FALSE)
    colnames(count_n_wide) <- c("Drug_Fusion_Lineage", "Drug", "FusionPair", "disease", "Fusion_Status_0_count", "Fusion_Status_1_count")
    count_n_wide <- add_column(count_n_wide, Drug_FusionPair = paste0(count_n_wide$Drug, "_", count_n_wide$FusionPair), .after = "FusionPair")
    count_n_filt <- filter(count_n, n >= 3)
    count_category <- count_n_filt %>% count(Drug_Fusion_Lineage) %>% filter(n == 2)
    dg_keep <- intersect(count_n_filt$Drug_Fusion_Lineage, count_category$Drug_Fusion_Lineage)
    dataframe <- filter(dataframe, Drug_Fusion_Lineage %in% dg_keep)
    dataframe$Drug_Fusion_Lineage <- as.character(dataframe$Drug_Fusion_Lineage)
    signif <- compare_means(Score ~ Fusion_Status, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif <- merge(signif, count_n_wide, by = grouping, all = TRUE)
    if(is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- ifelse(signif$Drug_Fusion_Lineage %in% g2p_list, "Yes", "No") }
    if(!is.na(str_match(pattern = "crispr", string = outfile))) { signif$InG2P <- "No" }
  }
  
  # Direction (old)
  if(tolower(type) == "direction") {
    signif <- compare_means(Score ~ Direction_Grouper, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
  }
  
  signif$p_log10 <- as.numeric(formatC(-log10(signif$p), digits = 3, format = "f"))
  saveRDS(signif, outfile)
  return(signif)
}
```


## Mutation status

```{r mut_wilcoxon_tests, eval=FALSE}
# CCLE
ccle_filt <- filter(ccle, Drug %in% g2p_drugs & Gene %in% g2p_dg_genes)
ccle_filt$Gene <- factor(ccle_filt$Gene, ordered = FALSE)
ccle_filt <- unfactorize(ccle_filt)
ccle_signif_g2p_dg <- WilcoxonTestDF(dataframe = ccle_filt, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/ccle_signif_g2p_dg.rds")
ccle_signif_g2p_dgl <- WilcoxonTestDF(dataframe = ccle_filt, grouping = c("Drug", "Gene", "Drug_Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", g2p_list = unique(as.character(g2p_mut$Drug_Gene_Lin[!is.na(g2p_mut$Drug_Gene_Lin)])), outfile = "./data_munging/rds/ccle_signif_g2p_dgl.rds")

# CTRP
ctrp_filt <- filter(ctrp, Drug %in% g2p_drugs & Gene %in% g2p_dg_genes)
ctrp_filt$Gene <- factor(ctrp_filt$Gene, ordered = FALSE)
ctrp_filt <- unfactorize(ctrp_filt)
ctrp_signif_g2p_dg <- WilcoxonTestDF(dataframe = ctrp_filt, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/ctrp_signif_g2p_dg.rds")
ctrp_signif_g2p_dgl <- WilcoxonTestDF(dataframe = ctrp_filt, grouping = c("Drug", "Gene", "Drug_Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", g2p_list = unique(as.character(g2p_mut$Drug_Gene_Lin[!is.na(g2p_mut$Drug_Gene_Lin)])), outfile = "./data_munging/rds/ctrp_signif_g2p_dgl.rds")

# GDSC
gdsc_filt <- filter(gdsc, Drug %in% g2p_drugs & Gene %in% g2p_dg_genes)
gdsc_filt$Gene <- factor(gdsc_filt$Gene, ordered = FALSE)
gdsc_filt <- unfactorize(gdsc_filt)
gdsc_signif_g2p_dg <- WilcoxonTestDF(dataframe = gdsc_filt, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/gdsc_signif_g2p_dg.rds")
gdsc_signif_g2p_dgl <- WilcoxonTestDF(dataframe = gdsc_filt, grouping = c("Drug", "Gene", "Drug_Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", g2p_list = unique(as.character(g2p_mut$Drug_Gene_Lin[!is.na(g2p_mut$Drug_Gene_Lin)])), outfile = "./data_munging/rds/gdsc_signif_g2p_dgl.rds")
```

## Protein changes

```{r protein_change_data}
ccle_prot <- filter(prot, Database == "CCLE")
ccle_prot <- unfactorize(ccle_prot)
# ccle_prot_count <- ccle_prot %>% count(Gene_Protein_Change, Drug, Change_Status)

ctrp_prot <- filter(prot, Database == "CTRP")
ctrp_prot <- unfactorize(ctrp_prot)
# ctrp_prot_count <- ctrp_prot %>% count(Gene_Protein_Change, Drug, Change_Status)

gdsc_prot <- filter(prot, Database == "GDSC")
gdsc_prot <- unfactorize(gdsc_prot)
# gdsc_prot_count <- gdsc_prot %>% count(Gene_Protein_Change, Drug, Change_Status)
```

```{r prot_wilcoxon_tests, eval=FALSE}
# CCLE
ccle_signif_prot <- WilcoxonTestDF(dataframe = ccle_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change[!is.na(g2p_prot$Drug_Gene_Protein_Change)])), type = "protein_change", outfile = "./data_munging/rds/ccle_signif_protein_changes.rds")
ccle_signif_prot_lin <- WilcoxonTestDF(dataframe = ccle_prot, grouping = c("Gene_Protein_Change", "Drug", "disease", "Drug_Gene_Protein_Change", "Drug_Gene_Protein_Change_Lineage"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change_Lineage[!is.na(g2p_prot$Drug_Gene_Protein_Change_Lineage)])), type = "protein_change_lineage", outfile = "./data_munging/rds/ccle_signif_protein_changes_lineage.rds")

# CTRP
ctrp_signif_prot <- WilcoxonTestDF(dataframe = ctrp_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change[!is.na(g2p_prot$Drug_Gene_Protein_Change)])), type = "protein_change", outfile = "./data_munging/rds/ctrp_signif_protein_changes.rds")
ctrp_signif_prot_lin <- WilcoxonTestDF(dataframe = ctrp_prot, grouping = c("Gene_Protein_Change", "Drug", "disease", "Drug_Gene_Protein_Change", "Drug_Gene_Protein_Change_Lineage"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change_Lineage[!is.na(g2p_prot$Drug_Gene_Protein_Change_Lineage)])), type = "protein_change_lineage", outfile = "./data_munging/rds/ctrp_signif_protein_changes_lineage.rds")

# GDSC
gdsc_signif_prot <- WilcoxonTestDF(dataframe = gdsc_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change[!is.na(g2p_prot$Drug_Gene_Protein_Change)])), type = "protein_change", outfile = "./data_munging/rds/gdsc_signif_protein_changes.rds")
gdsc_signif_prot_lin <- WilcoxonTestDF(dataframe = gdsc_prot, grouping = c("Gene_Protein_Change", "Drug", "disease", "Drug_Gene_Protein_Change", "Drug_Gene_Protein_Change_Lineage"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change_Lineage[!is.na(g2p_prot$Drug_Gene_Protein_Change_Lineage)])), type = "protein_change_lineage", outfile = "./data_munging/rds/gdsc_signif_protein_changes_lineage.rds")
```

## Fusions

```{r fusion_data}
ccle_fus <- filter(fus, Database == "CCLE")
ccle_fus <- unfactorize(ccle_fus)
# ccle_fus_summ <- unique(select(ccle_fus, DepMap_ID, FusionPair, Fusion_Status)) %>% count(FusionPair, Fusion_Status)

ctrp_fus <- filter(fus, Database == "CTRP")
ctrp_fus <- unfactorize(ctrp_fus)
# ctrp_fus_summ <- unique(select(ctrp_fus, DepMap_ID, FusionPair, Fusion_Status)) %>% count(FusionPair, Fusion_Status)

gdsc_fus <- filter(fus, Drug %in% g2p_drugs & Database == "GDSC")
gdsc_fus <- unfactorize(gdsc_fus)
# gdsc_fus_summ <- unique(select(gdsc_fus, DepMap_ID, FusionPair, Fusion_Status)) %>% count(FusionPair, Fusion_Status)
```

```{r fus_wilcoxon_tests, eval=FALSE}
ccle_signif_fus <- WilcoxonTestDF(dataframe = ccle_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), g2p_list = unique(as.character(g2p_fus$Drug_FusionPair[!is.na(g2p_fus$Drug_FusionPair)])), type = "fusion", outfile = "./data_munging/rds/ccle_signif_fusions.rds")
ccle_signif_fus_lin <- WilcoxonTestDF(dataframe = ccle_fus, grouping = c("FusionPair", "Drug", "disease", "Drug_FusionPair", "Drug_Fusion_Lineage"), g2p_list = unique(as.character(g2p_fus$Drug_Fusion_Lineage[!is.na(g2p_fus$Drug_Fusion_Lineage)])), type = "fusion_lineage", outfile = "./data_munging/rds/ccle_signif_fusions_lineage.rds")

ctrp_signif_fus <- WilcoxonTestDF(dataframe = ctrp_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), g2p_list = unique(as.character(g2p_fus$Drug_FusionPair[!is.na(g2p_fus$Drug_FusionPair)])), type = "fusion", outfile = "./data_munging/rds/ctrp_signif_fusions.rds")
ctrp_signif_fus_lin<- WilcoxonTestDF(dataframe = ctrp_fus, grouping = c("FusionPair", "Drug", "disease", "Drug_FusionPair", "Drug_Fusion_Lineage"), g2p_list = unique(as.character(g2p_fus$Drug_Fusion_Lineage[!is.na(g2p_fus$Drug_Fusion_Lineage)])), type = "fusion_lineage", outfile = "./data_munging/rds/ctrp_signif_fusions_lineage.rds")

gdsc_signif_fus <- WilcoxonTestDF(dataframe = gdsc_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), g2p_list = unique(as.character(g2p_fus$Drug_FusionPair[!is.na(g2p_fus$Drug_FusionPair)])), type = "fusion", outfile = "./data_munging/rds/gdsc_signif_fusions.rds")
gdsc_signif_fus_lin <- WilcoxonTestDF(dataframe = gdsc_fus, grouping = c("FusionPair", "Drug", "disease", "Drug_FusionPair", "Drug_Fusion_Lineage"), g2p_list = unique(as.character(g2p_fus$Drug_Fusion_Lineage[!is.na(g2p_fus$Drug_Fusion_Lineage)])), type = "fusion_lineage", outfile = "./data_munging/rds/gdsc_signif_fusions_lineage.rds")
```

```{r}
ccle_fus_means_df <- select(ccle_fus, FusionPair, Drug, Score) %>% group_by(FusionPair, Drug) %>% summarize(Agnostic_Mean_Score = mean(Score))
ccle_fus_means_lin_df <- select(ccle_fus, FusionPair, disease, Drug, Score) %>% drop_na(disease) %>% group_by(FusionPair, disease, Drug) %>% summarize(Mean_Score = mean(Score))
ccle_fus_means <- merge(ccle_fus_means_df, ccle_fus_means_lin_df, by = c("FusionPair", "Drug"), all = TRUE)
```

```{r}
# ggplot(ccle_fus_means_df) +
#   geom_histogram(aes(x = disease, y = Mean_Score)) +
#   facet_wrap(~ Drug)
```


## CRISPR

```{r, eval=FALSE}
# Mutation status
## All genes
# crispr_allgenes <- unfactorize(crispr_allgenes)
# crispr_signif_gene_allgenes <- WilcoxonTestDF(dataframe = crispr_allgenes, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", outfile = "./data_munging/rds/crispr_signif_g_allgenes.rds")
# crispr_signif_lineage_allgenes <- WilcoxonTestDF(dataframe = crispr_allgenes, grouping =  c("Drug", "Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", outfile = "./data_munging/rds/crispr_signif_gl_allgenes.rds")
## G2P genes
crispr <- unfactorize(crispr)
crispr_signif_gene <- WilcoxonTestDF(dataframe = crispr, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", outfile = "./data_munging/rds/crispr_signif_g2p_g.rds")
crispr_signif_lineage <- WilcoxonTestDF(dataframe = crispr, grouping =  c("Drug", "Gene", "disease", "Drug_Gene", "Drug_Gene_Lin"), type = "drug_gene_lineage", outfile = "./data_munging/rds/crispr_signif_g2p_gl.rds")

# Protein changes
## CRISPR: All genes
# crispr_prot_allgenes <- unfactorize(crispr_prot_allgenes)
# crispr_signif_prot_allgenes <- WilcoxonTestDF(dataframe = crispr_prot_allgenes, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), type = "protein_change", outfile = "./data_munging/rds/crispr_signif_protein_changes_allgenes.rds")
# crispr_signif_prot_lin_allgenes <- WilcoxonTestDF(dataframe = crispr_prot_allgenes, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change", "Drug_Gene_Protein_Change_Lineage", "disease"), type = "protein_change_lineage", outfile = "./data_munging/rds/crispr_signif_protein_changes_lineage_allgenes.rds")
## CRISPR: G2P genes
crispr_prot <- unfactorize(crispr_prot)
crispr_signif_prot <- WilcoxonTestDF(dataframe = crispr_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), type = "protein_change", outfile = "./data_munging/rds/crispr_signif_protein_changes.rds")
crispr_signif_prot_lin <- WilcoxonTestDF(dataframe = crispr_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change", "Drug_Gene_Protein_Change_Lineage", "disease"), type = "protein_change_lineage", outfile = "./data_munging/rds/crispr_signif_protein_changes_lineage.rds")

# Fusions
## CRISPR: All genes
# crispr_fus_allgenes <- unfactorize(crispr_fus_allgenes)
# crispr_signif_fus_allgenes <- WilcoxonTestDF(dataframe = crispr_fus_allgenes, grouping = c("FusionPair", "Drug", "Drug_FusionPair", "Drug_Gene"), type = "fusion", outfile = "./data_munging/rds/crispr_signif_fusions_allgenes.rds")
# crispr_signif_fus_lin_allgenes <- WilcoxonTestDF(dataframe = crispr_fus_allgenes, grouping = c("Drug_Gene", "FusionPair", "Drug", "Drug_FusionPair", "Drug_Fusion_Lineage"), type = "fusion_lineage", outfile = "./data_munging/rds/crispr_signif_fusions_lineage_allgenes.rds")
## CRISPR: G2P genes\
crispr_fus <- unfactorize(crispr_fus)
crispr_signif_fus <- WilcoxonTestDF(dataframe = crispr_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), type = "fusion", outfile = "./data_munging/rds/crispr_signif_fusions.rds")
crispr_signif_fus_lin <- WilcoxonTestDF(dataframe = crispr_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair", "disease", "Drug_Fusion_Lineage"), type = "fusion_lineage", outfile = "./data_munging/rds/crispr_signif_fusions_lineage.rds")
```

## Load and munge

```{r crispr_mut}
g2p_mut_summ_crispr <- g2p_mut %>% count(Gene, Direction_Curated) %>% select(-n)
g2p_mut_summ_crispr_n <- g2p_mut_summ_crispr %>% count(Gene)
g2p_mut_summ_crispr <- merge(g2p_mut_summ_crispr, g2p_mut_summ_crispr_n, by = c("Gene"))
g2p_mut_summ_crispr$Direction_Heatmap <- with(g2p_mut_summ_crispr, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
g2p_mut_summ_crispr <- unique(select(g2p_mut_summ_crispr, Gene, Direction_Heatmap))

g2p_mut_summ_lin_crispr <- g2p_mut %>% count(Gene, disease, Direction_Curated) %>% select(-n)
g2p_mut_summ_lin_crispr_n <- g2p_mut_summ_lin_crispr %>% count(Gene, disease)
g2p_mut_summ_lin_crispr <- merge(g2p_mut_summ_lin_crispr, g2p_mut_summ_lin_crispr_n, by = c("Gene", "disease"))
g2p_mut_summ_lin_crispr$Direction_Heatmap <- with(g2p_mut_summ_lin_crispr, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
g2p_mut_summ_lin_crispr <- unique(select(g2p_mut_summ_lin_crispr, Gene, disease, Direction_Heatmap))

crispr_signif_g2p_g <- readRDS("./data_munging/rds/crispr_signif_g2p_g.rds")
crispr_signif_g2p_g <- merge(crispr_signif_g2p_g, g2p_mut_summ_crispr, by = "Gene", all.x = TRUE)
crispr_signif_lin <- readRDS("./data_munging/rds/crispr_signif_g2p_gl.rds")
crispr_signif_lin <- merge(crispr_signif_lin, g2p_mut_summ_lin_crispr, by = c("Gene", "disease"), all.x = TRUE)

crispr_signif_allgenes <- readRDS("./data_munging/rds/crispr_signif_g_allgenes.rds")
crispr_signif_lin_allgenes <- readRDS("./data_munging/rds/crispr_signif_gl_allgenes.rds")
```

```{r load_mut_wilcoxon_rds}
g2p_mut_summ <- g2p_mut %>% count(Drug, Gene, Direction_Curated) %>% select(-n)
g2p_mut_summ_n <- g2p_mut_summ %>% count(Drug, Gene)
g2p_mut_summ <- merge(g2p_mut_summ, g2p_mut_summ_n, by = c("Drug", "Gene"))
g2p_mut_summ$Direction_Heatmap <- with(g2p_mut_summ, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
g2p_mut_summ <- unique(select(g2p_mut_summ, Drug, Gene, Direction_Heatmap))

g2p_mut_summ_lin <- g2p_mut %>% count(Drug, Gene, disease, Direction_Curated) %>% select(-n)
g2p_mut_summ_lin_n <- g2p_mut_summ_lin %>% count(Drug, Gene, disease)
g2p_mut_summ_lin <- merge(g2p_mut_summ_lin, g2p_mut_summ_lin_n, by = c("Drug", "Gene", "disease"))
g2p_mut_summ_lin$Direction_Heatmap <- with(g2p_mut_summ_lin, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
g2p_mut_summ_lin <- unique(select(g2p_mut_summ_lin, Drug, Gene, disease, Direction_Heatmap))

ccle_signif_g2p_dg <- readRDS("./data_munging/rds/ccle_signif_g2p_dg.rds")
ccle_signif_g2p_dg <- merge(ccle_signif_g2p_dg, g2p_mut_summ, by = c("Drug", "Gene"), all.x = TRUE)
ccle_signif_dgl <- readRDS("./data_munging/rds/ccle_signif_g2p_dgl.rds")
ccle_signif_dgl <- merge(ccle_signif_dgl, g2p_mut_summ_lin, by = c("Drug", "Gene", "disease"), all.x = TRUE)

ctrp_signif_g2p_dg <- readRDS("./data_munging/rds/ctrp_signif_g2p_dg.rds")
ctrp_signif_g2p_dg <- merge(ctrp_signif_g2p_dg, g2p_mut_summ, by = c("Drug", "Gene"), all.x = TRUE)
ctrp_signif_dgl <- readRDS("./data_munging/rds/ctrp_signif_g2p_dgl.rds")
ctrp_signif_dgl <- merge(ctrp_signif_dgl, g2p_mut_summ_lin, by = c("Drug", "Gene", "disease"), all.x = TRUE)

gdsc_signif_g2p_dg <- readRDS("./data_munging/rds/gdsc_signif_g2p_dg.rds")
gdsc_signif_g2p_dg <- merge(gdsc_signif_g2p_dg, g2p_mut_summ, by = c("Drug", "Gene"), all.x = TRUE)
gdsc_signif_dgl <- readRDS("./data_munging/rds/gdsc_signif_g2p_dgl.rds")
gdsc_signif_dgl <- merge(gdsc_signif_dgl, g2p_mut_summ_lin, by = c("Drug", "Gene", "disease"), all.x = TRUE)

ccle_signif_dgl %>% count(InG2P)
filter(ccle_signif_dgl, InG2P == "Yes" & p < 0.05) %>% select(Drug, Gene, disease, p.short, Direction_Heatmap)
ctrp_signif_dgl %>% count(InG2P)
filter(ctrp_signif_dgl, InG2P == "Yes" & p < 0.05) %>% select(Drug, Gene, disease, p.short, Direction_Heatmap)
gdsc_signif_dgl %>% count(InG2P)
filter(gdsc_signif_dgl, InG2P == "Yes" & p < 0.05) %>% select(Drug, Gene, disease, p.short, Direction_Heatmap)
```

```{r load_prot_wilcoxon_rds, eval=FALSE}
crispr_signif_prot <- readRDS("./data_munging/rds/crispr_signif_protein_changes.rds") %>% filter(Change_Status_1_count >= 3)
ccle_signif_prot <- readRDS("./data_munging/rds/ccle_signif_protein_changes.rds") %>% filter(Change_Status_1_count >= 3)
ctrp_signif_prot <- readRDS("./data_munging/rds/ctrp_signif_protein_changes.rds") %>% filter(Change_Status_1_count >= 3)
gdsc_signif_prot <- readRDS("./data_munging/rds/gdsc_signif_protein_changes.rds") %>% filter(Change_Status_1_count >= 3)

gene_prots <- union(union(unique(ccle_signif_prot$Gene_Protein_Change), union(unique(ctrp_signif_prot$Gene_Protein_Change), unique(gdsc_signif_prot$Gene_Protein_Change))), unique(crispr_signif_prot$Gene_Protein_Change))

# all_directions <- unique(rbind(select(crispr_prot, Gene_Protein_Change, Drug, Direction_Curated), select(ccle_prot, Gene_Protein_Change, Drug, Direction_Curated), select(ctrp_prot, Gene_Protein_Change, Drug, Direction_Curated), select(gdsc_prot, Gene_Protein_Change, Drug, Direction_Curated)))
# all_directions <- filter(all_directions, !is.na(Direction_Curated))
# saveRDS(all_directions, "./data_munging/rds/protein_changes_directions.rds")
all_directions <- readRDS("./data_munging/rds/protein_changes_directions.rds")

crispr_signif_prot_4plot <- merge(merge(merge(crispr_signif_prot, unique(select(crispr_prot, Drug, Gene_Protein_Change)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(crispr_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE), all_directions,  by = c("Drug", "Gene_Protein_Change"), all.x = TRUE)
ccle_signif_prot_4plot <- merge(merge(merge(ccle_signif_prot, unique(select(ccle_prot, Drug, Gene_Protein_Change)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(ccle_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE), all_directions,  by = c("Drug", "Gene_Protein_Change"), all.x = TRUE)
ctrp_signif_prot_4plot <- merge(merge(merge(ctrp_signif_prot, unique(select(ctrp_prot, Drug, Gene_Protein_Change)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(ctrp_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE), all_directions,  by = c("Drug", "Gene_Protein_Change"), all.x = TRUE)
gdsc_signif_prot_4plot <- merge(merge(merge(gdsc_signif_prot, unique(select(gdsc_prot, Drug, Gene_Protein_Change)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(gdsc_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE), all_directions,  by = c("Drug", "Gene_Protein_Change"), all.x = TRUE)

saveRDS(crispr_signif_prot_4plot, "./data_munging/rds/crispr_signif_protein_changes_4plot.rds")
saveRDS(ccle_signif_prot_4plot, "./data_munging/rds/ccle_signif_protein_changes_4plot.rds")
saveRDS(ctrp_signif_prot_4plot, "./data_munging/rds/ctrp_signif_protein_changes_4plot.rds")
saveRDS(gdsc_signif_prot_4plot, "./data_munging/rds/gdsc_signif_protein_changes_4plot.rds")
```

```{r prot_munging}
crispr_signif_prot <- readRDS("./data_munging/rds/crispr_signif_protein_changes.rds")
ccle_signif_prot <- readRDS("./data_munging/rds/ccle_signif_protein_changes.rds")
ctrp_signif_prot <- readRDS("./data_munging/rds/ctrp_signif_protein_changes.rds")
gdsc_signif_prot <- readRDS("./data_munging/rds/gdsc_signif_protein_changes.rds")

crispr_signif_prot_4plot <- readRDS("./data_munging/rds/crispr_signif_protein_changes_4plot.rds")
ccle_signif_prot_4plot <- readRDS("./data_munging/rds/ccle_signif_protein_changes_4plot.rds")
ctrp_signif_prot_4plot <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_4plot.rds")
gdsc_signif_prot_4plot <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_4plot.rds")

all_directions <- readRDS("./data_munging/rds/protein_changes_directions.rds")
crispr_signif_prot_lin <- readRDS("./data_munging/rds/crispr_signif_protein_changes_lineage.rds")
crispr_signif_prot_lin <- merge(crispr_signif_prot_lin, all_directions, by = c("Gene_Protein_Change", "Drug"), all.x = TRUE)
ccle_signif_prot_lin <- readRDS("./data_munging/rds/ccle_signif_protein_changes_lineage.rds")
ccle_signif_prot_lin <- merge(ccle_signif_prot_lin, all_directions, by = c("Gene_Protein_Change", "Drug"), all.x = TRUE)
ctrp_signif_prot_lin <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_lineage.rds")
ctrp_signif_prot_lin <- merge(ctrp_signif_prot_lin, all_directions, by = c("Gene_Protein_Change", "Drug"), all.x = TRUE)
gdsc_signif_prot_lin <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_lineage.rds")
gdsc_signif_prot_lin <- merge(gdsc_signif_prot_lin, all_directions, by = c("Gene_Protein_Change", "Drug"), all.x = TRUE)

ccle_signif_prot_lin %>% count(InG2P)
filter(ccle_signif_prot_lin, InG2P == "Yes" & p < 0.05) %>% select(Drug, Gene_Protein_Change, disease, p.short, Direction_Curated)
ctrp_signif_prot_lin %>% count(InG2P)
filter(ctrp_signif_prot_lin, InG2P == "Yes" & p < 0.05) %>% select(Drug, Gene_Protein_Change, disease, p.short, Direction_Curated)
gdsc_signif_prot_lin %>% count(InG2P)
filter(gdsc_signif_prot_lin, InG2P == "Yes" & p < 0.05) %>% select(Drug, Gene_Protein_Change, disease, p.short, Direction_Curated)
```

```{r load_fus_wilcoxon_rds}
# crispr_signif_fus_allgenes <- readRDS("./data_munging/rds/crispr_signif_fusions_allgenes.rds")
# crispr_signif_fus_lin_allgenes <- readRDS("./data_munging/rds/crispr_signif_fusions_lineage_allgenes.rds")
crispr_signif_fus <- readRDS("./data_munging/rds/crispr_signif_fusions.rds")
crispr_signif_fus_lin <- readRDS("./data_munging/rds/crispr_signif_fusions_lineage.rds")

ccle_signif_fus <- readRDS("./data_munging/rds/ccle_signif_fusions.rds")
ctrp_signif_fus <- readRDS("./data_munging/rds/ctrp_signif_fusions.rds")
gdsc_signif_fus <- readRDS("./data_munging/rds/gdsc_signif_fusions.rds")

ccle_signif_fus_lin <- readRDS("./data_munging/rds/ccle_signif_fusions_lineage.rds")
ctrp_signif_fus_lin <- readRDS("./data_munging/rds/ctrp_signif_fusions_lineage.rds")
gdsc_signif_fus_lin <- readRDS("./data_munging/rds/gdsc_signif_fusions_lineage.rds")
```

## Boxplots

```{r crispr_mut_boxplots, fig.height=5, fig.width=10, eval=FALSE}
crispr_filt <- filter(crispr, Gene %in% g2p_mut_genes)
crispr_filt$Gene <- factor(crispr_filt$Gene, levels = g2p_mut_genes)
crispr_filt$Mutation <- as.character(crispr_filt$Mutation)
crispr_signif_filt <- filter(crispr_signif_g2p_g, Gene %in% g2p_mut_genes)
crispr_signif_filt$Gene <- factor(crispr_signif_filt$Gene, levels = g2p_mut_genes)

crispr_gene_mut_plot <- ggplot(data = crispr_filt) +
  facet_wrap(~ Gene, nrow = 2, drop = FALSE) +
  scale_color_manual(name = "Mutation Status", labels = c("Wildtype", "Mutant"), values = c("1" = "darkorchid", "0" = "thistle")) +
  scale_x_discrete(name = "Mutation Status", labels = c("0" = "WT", "1" = "Mut")) +
  geom_jitter(aes(x = Mutation, y = Score, color = Mutation), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Mutation, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  geom_text(data = crispr_signif_filt, aes(x = 1.5, y = 2, label = p.signif)) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  theme(legend.position = "top", legend.direction = "horizontal") +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "CERES Score")
crispr_gene_mut_plot

ggsave("./plots/manuscript/crispr_gene_mut_plot.png", crispr_gene_mut_plot, width = 12, height = 6, units = "in")
```

```{r fusion_boxplots, fig.height=5, fig.width=12, eval=FALSE}
# CRISPR
crispr_fus_filt <- filter(crispr_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1", "PML-PARA") & Drug_Gene %in% c("CRISPR_ABL1"))
crispr_signif_fus_filt <- filter(crispr_signif_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1", "PML-PARA") & Drug_Gene %in% c("CRISPR_ABL1"))

crispr_fus_boxplot <- ggplot(data = crispr_fus_filt) +
  facet_grid(FusionPair ~ Drug_Gene) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "darkorchid4", "No" = "thistle")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  # coord_cartesian(ylim = c(0, 1.2)) +
  # scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = crispr_signif_fus_filt, aes(x = 1.5, y = 1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "Score", x = "Fusion Status")
crispr_fus_boxplot
ggsave("./plots/manuscript/crispr_fus_boxplots.png", crispr_fus_boxplot, width = 8, height = 5, units = "in")

# CCLE
ccle_fus_filt <- filter(ccle_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1") & Drug %in% g2p_drugs)
ccle_fus_filt$Drug_f <- factor(ccle_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ccle_fus$Drug)))
ccle_signif_fus_filt <- filter(ccle_signif_fus, Drug %in% g2p_drugs)
ccle_signif_fus_filt$Drug_f <- factor(ccle_signif_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ccle_signif_fus$Drug)))

ccle_fus_boxplot <- ggplot(data = ccle_fus_filt) +
  facet_grid(FusionPair ~ Drug_f) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "springgreen4", "No" = "palegreen3")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  coord_cartesian(ylim = c(0, 1.2)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = ccle_signif_fus_filt, aes(x = 1.5, y = 1.1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "AUC", x = "Fusion Status")
ccle_fus_boxplot
ggsave("./plots/manuscript/ccle_fus_boxplots.png", ccle_fus_boxplot, width = 8, height = 5, units = "in")

# CTRP
ctrp_fus_filt <- filter(ctrp_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1") & Drug %in% g2p_drugs)
ctrp_fus_filt$Drug_f <- factor(ctrp_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ctrp_fus$Drug)))
ctrp_signif_fus_filt <- filter(ctrp_signif_fus, Drug %in% g2p_drugs)
ctrp_signif_fus_filt$Drug_f <- factor(ctrp_signif_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ctrp_signif_fus$Drug)))

ctrp_fus_boxplot <- ggplot(data = ctrp_fus_filt) +
  facet_grid(FusionPair ~ Drug_f) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "steelblue4", "No" = "slategray3")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  coord_cartesian(ylim = c(0, 1.2)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = ctrp_signif_fus_filt, aes(x = 1.5, y = 1.1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "AUC", x = "Fusion Status")
ctrp_fus_boxplot
ggsave("./plots/manuscript/ctrp_fus_boxplots.png", ctrp_fus_boxplot, width = 19, height = 5, units = "in")

# GDSC
gdsc_fus_filt <- filter(gdsc_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1") & Drug %in% g2p_drugs)
gdsc_fus_filt$Drug_f <- factor(gdsc_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(gdsc_fus$Drug)))
gdsc_signif_fus_filt <- filter(gdsc_signif_fus, Drug %in% g2p_drugs)
gdsc_signif_fus_filt$Drug_f <- factor(gdsc_signif_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(gdsc_signif_fus$Drug)))

gdsc_fus_boxplot <- ggplot(data = gdsc_fus_filt) +
  facet_grid(FusionPair ~ Drug_f) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "turquoise4", "No" = "paleturquoise3")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  coord_cartesian(ylim = c(0, 1.2)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = gdsc_signif_fus_filt, aes(x = 1.5, y = 1.1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "AUC", x = "Fusion Status")
gdsc_fus_boxplot
ggsave("./plots/manuscript/gdsc_fus_boxplots.png", gdsc_fus_boxplot, width = 12, height = 5, units = "in")
```

## Heatmaps

```{r wilcoxon_heatmap_function}
MakeWilcoxonHeatmap <- function(signif_df, feature, row_names, col_title, color, legend_title, legend_breaks, row_title_txt, na_color, radius) {
  if(missing(na_color)) {
    na_color <- "white"
  }
  
  if(!is.na(str_match(string = tolower(feature), pattern = "crispr")) & is.na(str_match(string = tolower(feature), pattern = "prot|fus"))) {
    print("Making CRISPR mutation status grid and annotation dataframes.")
    signif_df <- filter(crispr_signif_g2p_g, Gene %in% unique(g2p_mut$Gene))
    signif_grid <- dcast(signif_df, Gene ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0)
    signif_grid <- signif_grid[match(g2p_gene_order, signif_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    
    signif_grid_annot1 <- signif_grid
    signif_grid_annot1[signif_grid_annot1 <= -log10(0.05) | is.na(signif_grid_annot1)] <- ""
    signif_grid_annot1[signif_grid_annot1 > -log10(0.05)] <- "*"
    
    signif_grid_annot <- dcast(signif_df, Gene ~ Drug, value.var = "Direction_Heatmap", fill = 0)
    signif_grid_annot <- signif_grid_annot[match(g2p_gene_order, signif_grid_annot$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid_annot[signif_grid_annot == "sensitivity"] <- "O"
    print("Done.")
  }
  if(!is.na(str_match(string = tolower(feature), pattern = "mut"))) {
    print("Making mutation status grid and annotation dataframes.")
    signif_grid <- dcast(signif_df, Gene ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid <- signif_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid <- signif_grid[complete.cases(signif_grid), ]
    
    signif_grid_annot1 <- signif_grid
    signif_grid_annot1[signif_grid_annot1 <= -log10(0.05) | is.na(signif_grid_annot1)] <- ""
    signif_grid_annot1[signif_grid_annot1 > -log10(0.05)] <- "*"
    
    signif_grid_annot <- dcast(signif_df, Gene ~ Drug, value.var = "Direction_Heatmap", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid_annot <- signif_grid_annot[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid_annot <- signif_grid_annot[complete.cases(signif_grid_annot), ]
    signif_grid_annot[signif_grid_annot == "sensitivity"] <- "O"
    print("Done.")
  }
  if(!is.na(str_match(string = tolower(feature), pattern = "prot"))) {
    if(is.na(str_match(string = tolower(feature), pattern = "crispr"))) {
      print("Making protein changes grid and annotation dataframes.")
      signif_grid <- dcast(signif_df, Gene_Protein_Change ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene_Protein_Change)
      signif_grid <- signif_grid[, intersect(g2p_drug_order, colnames(signif_grid))]
      
      signif_grid_annot1 <- signif_grid
      signif_grid_annot1[signif_grid_annot1 <= -log10(0.05) | is.na(signif_grid_annot1)] <- ""
      signif_grid_annot1[signif_grid_annot1 > -log10(0.05)] <- "*"
      
      signif_grid_annot <- dcast(signif_df, Gene_Protein_Change ~ Drug, value.var = "Direction_Curated", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene_Protein_Change)
      signif_grid_annot <- signif_grid_annot[, intersect(g2p_drug_order, colnames(signif_grid))]
      signif_grid_annot[signif_grid_annot == "resistance"] <- "X"
      signif_grid_annot[signif_grid_annot == "sensitivity"] <- "O"
      print("Done.")
    }
    if(!is.na(str_match(string = tolower(feature), pattern = "crispr"))) {
      print("Making CRISPR protein changes grid and annotation dataframes.")
      signif_grid <- dcast(signif_df, Gene_Protein_Change ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene_Protein_Change)
      
      signif_grid_annot1 <- signif_grid
      signif_grid_annot1[signif_grid_annot1 <= -log10(0.05) | is.na(signif_grid_annot1)] <- ""
      signif_grid_annot1[signif_grid_annot1 > -log10(0.05)] <- "*"
      
      signif_grid_annot <- dcast(signif_df, Gene_Protein_Change ~ Drug, value.var = "Direction_Curated", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene_Protein_Change)
      signif_grid_annot[signif_grid_annot == "resistance"] <- "X"
      signif_grid_annot[signif_grid_annot == "sensitivity"] <- "O"
      print("Done.")
    }
  }
  if(!is.na(str_match(string = tolower(feature), pattern = "fus"))) {
    if(is.na(str_match(string = tolower(feature), pattern = "crispr"))) {
      print("Making fusions grid and annotation dataframes.")
      signif_grid <- dcast(signif_df, FusionPair ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-FusionPair)
      signif_grid <- signif_grid[, intersect(g2p_drug_order, colnames(signif_grid))]
      signif_grid <- signif_grid[complete.cases(signif_grid), ]
      
      signif_grid_annot1 <- signif_grid
      signif_grid_annot1[signif_grid_annot1 <= -log10(0.05) | is.na(signif_grid_annot1)] <- ""
      signif_grid_annot1[signif_grid_annot1 > -log10(0.05)] <- "*"
      
      signif_grid_annot <- dcast(signif_df, FusionPair ~ Drug, value.var = "InG2P", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-FusionPair)
      signif_grid_annot <- signif_grid_annot[, intersect(g2p_drug_order, colnames(signif_grid))]
      signif_grid_annot <- signif_grid_annot[complete.cases(signif_grid_annot), ]
      signif_grid_annot[signif_grid_annot == "No"] <- ""
      signif_grid_annot[signif_grid_annot == "Yes"] <- "O"
      print("Done.")
    }
    if(!is.na(str_match(string = tolower(feature), pattern = "crispr"))) {
      print("Making CRISPR fusions grid and annotation dataframes.")
      signif_grid <- dcast(signif_df, FusionPair ~ Drug_Gene, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-FusionPair)
      
      signif_grid_annot1 <- signif_grid
      signif_grid_annot1[signif_grid_annot1 <= -log10(0.05) | is.na(signif_grid_annot1)] <- ""
      signif_grid_annot1[signif_grid_annot1 > -log10(0.05)] <- "*"
      
      signif_grid_annot <- dcast(signif_df, FusionPair ~ Drug, value.var = "InG2P", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-FusionPair)
      signif_grid_annot[signif_grid_annot == "No"] <- ""
      signif_grid_annot[signif_grid_annot == "Yes"] <- "O"
      print("Done.")
    }
  }
  
  signif_grid <- as.matrix(signif_grid)
  col_fun <- colorRamp2(c(legend_breaks[1], legend_breaks[length(legend_breaks)]), c("gray97", color))
  # Legend properties list
  ht_legend_list <- list(title = legend_title, title_position = "topleft", legend_direction = "horizontal", legend_width = unit(1.5, "in"), border = "gray70", at = legend_breaks, labels = legend_breaks)
  
  print("Making heatmap object.")
  ht <- Heatmap(signif_grid,
                col = col_fun,
                na_col = na_color,
                rect_gp = gpar(col = "gray70"),
                heatmap_legend_param = ht_legend_list,
                cluster_rows = FALSE, 
                cluster_columns = FALSE, 
                show_row_names = as.logical(row_names),
                row_names_side = "left",
                row_names_gp = gpar(fontsize = 10),
                column_names_side = "bottom",
                column_names_gp = gpar(fontsize = 10),
                show_column_names = FALSE,
                bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(signif_grid),
                                   rot = 45,
                                   gp = gpar(fontsize = 11),
                                   location = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.85, "in")),
                row_title = row_title_txt,
                row_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title = col_title, 
                column_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title_side = "bottom",
                cell_fun = function(j, i, x, y, width, height, fill) {
                  grid.text(signif_grid_annot1[i, j], x = x + width*0.25, y = y + height*0.065, gp = gpar(fontsize = 17))
                  
                  if(signif_grid_annot[i, j] == "O") {
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "white", col = "black"))
                  }
                  if(signif_grid_annot[i, j] == "X") {
                    grid.circle(x = x - width*0.20, y = y + height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "red", col = "black"))
                  }
                  if(signif_grid_annot[i, j] == "resistance") {
                    grid.circle(x = x - width*0.20, y = y + height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "red", col = "black"))
                  }
                  if(signif_grid_annot[i, j] == "both") {
                    grid.circle(x = x - width*0.20, y = y + height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "red", col = "black"))
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "white", col = "black"))
                  }
                }
                )
                
  print("Done.")
  return(ht)
}
```

### Pancan
**12/16 Need to make all heatmaps tomorrow**
```{r mut_heatmaps, eval=FALSE, fig.height=10, fig.width=3}
ht_crispr <- MakeWilcoxonHeatmap(signif_df = filter(crispr_signif_g2p_g, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "crispr", legend_title = "-log10(p-value)", legend_breaks = seq(0, 45, by = 9), row_names = TRUE, color = "darkorchid4", col_title = "", row_title_txt = "Genes", radius = 0.05)
png(filename = "./plots/manuscript/crispr_mut_heatmap.png", width = 3, height = 10, units = "in", res = 500)
draw(ht_crispr, heatmap_legend_side = "top")
dev.off()

ht_ccle <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "mutation", legend_title = "-log10(p-value)", legend_breaks = seq(0, 12, by = 3), row_names = TRUE, color = "darkgreen", col_title = "Drugs", row_title_txt = "Genes", radius = 0.04)
png(filename = "./plots/manuscript/ccle_mut_heatmap.png", width = 3, height = 10, units = "in", res = 500)
draw(ht_ccle, heatmap_legend_side = "top")
dev.off()

ht_ctrp <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "mutation", legend_title = "-log10(p-value)", legend_breaks = seq(0, 18, by = 3), row_names = TRUE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Genes", radius = 0.0075)
png(filename = "./plots/manuscript/ctrp_mut_heatmap.png", width = 9.5, height = 10, units = "in", res = 500)
draw(ht_ctrp, heatmap_legend_side = "top")
dev.off()

ht_gdsc <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "-log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = TRUE, color = "turquoise4", col_title = "Drugs", row_title_txt = "Genes", radius = 0.01)
png(filename = "./plots/manuscript/gdsc_mut_heatmap.png", width = 5, height = 10, units = "in", res = 500)
draw(ht_gdsc, heatmap_legend_side = "top")
dev.off()

ht_ccle2 <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "mutation", legend_title = "CCLE -log10(p-value)          ", legend_breaks = seq(0, 12, by = 3), row_names = FALSE, color = "darkgreen", col_title = "", row_title_txt = "Genes", radius = 0.023)
ht_ctrp2 <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "mutation", legend_title = "CTRP -log10(p-value)          ", legend_breaks = seq(0, 18, by = 3), row_names = FALSE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Genes", radius = 0.007)
ht_gdsc2 <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "mutation", legend_title = "GDSC -log10(p-value)          ", legend_breaks = seq(0, 5, by = 1), row_names = FALSE, color = "turquoise4", col_title = "", row_title_txt = "Genes", radius = 0.011)
ht_crispr2 <- MakeWilcoxonHeatmap(signif_df = filter(crispr_signif_g2p_g, Gene %in% unique(g2p_mut$Gene) & Mutation_1_Count >= 3), feature = "crispr", legend_title = "CRISPR -log10(p-value)", legend_breaks = seq(0, 45, by = 9), row_names = TRUE, color = "darkorchid4", col_title = "", row_title_txt = "Genes", radius = 0.15)

png(filename = "./plots/manuscript/all_mut_heatmap.png", width = 14, height = 7.5, units = "in", res = 500)
draw(ht_crispr2 + ht_ccle2 + ht_ctrp2 + ht_gdsc2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```

```{r prot_heatmaps, eval=FALSE, fig.height=10, fig.width=3}
ht_crispr <- MakeWilcoxonHeatmap(signif_df = crispr_signif_prot_4plot, feature = "protein changes CRISPR", legend_title = "-log10(p-value)", legend_breaks = seq(0, 25, by = 5), row_names = TRUE, color = "darkorchid4", col_title = "", row_title_txt = "Gene:Protein Change", radius = 0.03)
png(filename = "./plots/manuscript/crispr_prot_heatmap.png", width = 3, height = 5.5, units = "in", res = 500)
draw(ht_crispr, heatmap_legend_side = "top")
dev.off()

ht_ccle <- MakeWilcoxonHeatmap(signif_df = ccle_signif_prot_4plot, feature = "protein changes", legend_title = "-log10(p-value)", legend_breaks = seq(0, 15, by = 3), row_names = TRUE, color = "darkgreen", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.03)
png(filename = "./plots/manuscript/ccle_prot_heatmap.png", width = 3, height = 5.5, units = "in", res = 500)
draw(ht_ccle, heatmap_legend_side = "top")
dev.off()

ht_ctrp <- MakeWilcoxonHeatmap(signif_df = ctrp_signif_prot_4plot, feature = "protein changes", legend_title = "-log10(p-value)", legend_breaks = seq(0, 15, by = 3), row_names = TRUE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.01)
png(filename = "./plots/manuscript/ctrp_prot_heatmap.png", width = 9.5, height = 5.5, units = "in", res = 500)
draw(ht_ctrp, heatmap_legend_side = "top")
dev.off()

ht_gdsc <- MakeWilcoxonHeatmap(signif_df = gdsc_signif_prot_4plot, feature = "protein changes", legend_title = "-log10(p-value)", legend_breaks = seq(0, 8, by = 2), row_names = TRUE, color = "turquoise4", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.01)
png(filename = "./plots/manuscript/gdsc_prot_heatmap.png", width = 5, height = 5.5, units = "in", res = 500)
draw(ht_gdsc, heatmap_legend_side = "top")
dev.off()

ht_ccle2 <- MakeWilcoxonHeatmap(signif_df = ccle_signif_prot_4plot, feature = "protein changes", legend_title = "CCLE -log10(p-value)          ", legend_breaks = seq(0, 15, by = 3), row_names = FALSE, color = "darkgreen", col_title = "", row_title_txt = "Gene:Protein Change", na_color = "black", radius = 0.021)
ht_ctrp2 <- MakeWilcoxonHeatmap(signif_df = ctrp_signif_prot_4plot, feature = "protein changes", legend_title = "CTRP -log10(p-value)          ", legend_breaks = seq(0, 15, by = 3), row_names = FALSE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Gene:Protein Change", na_color = "black", radius = 0.0125)
ht_gdsc2 <- MakeWilcoxonHeatmap(signif_df = gdsc_signif_prot_4plot, feature = "protein changes", legend_title = "GDSC -log10(p-value)          ", legend_breaks = seq(0, 8, by = 2), row_names = FALSE, color = "turquoise4", col_title = "", row_title_txt = "Gene:Protein Change", na_color = "black", radius = 0.0125)
ht_crispr2 <- MakeWilcoxonHeatmap(signif_df = crispr_signif_prot_4plot, feature = "protein changes CRISPR", legend_title = "CRISPR -log10(p-value)", legend_breaks = seq(0, 25, by = 5), row_names = TRUE, color = "darkorchid4", col_title = "", row_title_txt = "Gene:Protein Change", na_color = "black", radius = 0.135)

png(filename = "./plots/manuscript/all_prot_heatmap.png", width = 14, height = 5, units = "in", res = 500)
draw(ht_crispr2 + ht_ccle2 + ht_ctrp2 + ht_gdsc2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```

```{r fus_heatmaps, eval=FALSE}
ht_crispr <- MakeWilcoxonHeatmap(signif_df = filter(crispr_signif_fus, Drug_Gene == "CRISPR_ABL1" & FusionPair %in% c("BCR-ABL1", "ABL1-BCR")), feature = "fusion crispr", legend_title = "-log10(p-value)",  legend_breaks = seq(0, 2, by = 0.5), row_names = TRUE, color = "darkorchid4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.5)
png(filename = "./plots/manuscript/crispr_fus_heatmap.png", width = 3, height = 3, units = "in", res = 500)
draw(ht_crispr, heatmap_legend_side = "top")
dev.off()

ht_ccle <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "-log10(p-value)",  legend_breaks = seq(0, 4, by = 1), row_names = TRUE, color = "darkgreen", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.01)
png(filename = "./plots/manuscript/ccle_fus_heatmap.png", width = 5, height = 3, units = "in", res = 500)
draw(ht_ccle, heatmap_legend_side = "top")
dev.off()

ht_ctrp <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "-log10(p-value)", legend_breaks = seq(0, 10, by = 2), row_names = TRUE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.01)
png(filename = "./plots/manuscript/ctrp_fus_heatmap.png", width = 13, height = 3, units = "in", res = 500)
draw(ht_ctrp, heatmap_legend_side = "top")
dev.off()

ht_gdsc <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "-log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = TRUE, color = "turquoise4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.01)
png(filename = "./plots/manuscript/gdsc_fus_heatmap.png", width = 9, height = 3, units = "in", res = 500)
draw(ht_gdsc, heatmap_legend_side = "top")
dev.off()

ht_crispr2 <- MakeWilcoxonHeatmap(signif_df = filter(crispr_signif_fus, Drug_Gene == "CRISPR_ABL1" & FusionPair %in% c("BCR-ABL1", "ABL1-BCR")), feature = "fusion crispr", legend_title = "CRISPR -log10(p-value)          ",  legend_breaks = seq(0, 2, by = 0.5), row_names = TRUE, color = "darkorchid4", col_title = "", row_title_txt = "Fusion Pairs", radius = 0.15)
ht_ccle2 <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "CCLE -log10(p-value)          ", legend_breaks = seq(0, 4, by = 1), row_names = FALSE, color = "darkgreen", col_title = "", row_title_txt = "Fusion Pairs", radius = 0.15)
ht_ctrp2 <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "CTRP -log10(p-value)          ", legend_breaks = seq(0, 10, by = 2), row_names = FALSE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.08)
ht_gdsc2 <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "GDSC -log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = FALSE, color = "turquoise4", col_title = "", row_title_txt = "Fusion Pairs", radius = 0.08)

png(filename = "./plots/manuscript/all_fus_heatmap.png", width = 15, height = 2.4, units = "in", res = 500)
draw(ht_crispr2 + ht_ccle2 + ht_ctrp2 + ht_gdsc2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```

### Lineage

```{r lineage_mut_heatmap_functions}
MakeWilcoxonLineageHeatmap <- function(df, database, lineage) {
  plot_colors <- c("CCLE" = "darkgreen", "CTRP" = "steelblue4", "GDSC" = "turquoise4")
  plot_breaks <- list("CCLE" = seq(0, 5, by = 1), "CTRP" = seq(0, 12, by = 2), "GDSC" = seq(0, 8, by = 2))
  
  df <- filter(df, disease == lineage)
  
  ht <- MakeWilcoxonHeatmap(signif_df = df, feature = "mutation", legend_title = paste0(database, "\n", lineage, "\n-log10(p-value)"), legend_breaks = plot_breaks[[database]], row_names = TRUE, color = plot_colors[[database]], col_title = "Drugs", row_title_txt = "Genes")
  
  return(ht)
}

SaveWilcoxonLineageHeatmap <- function(database, ht_list, lineage) {
  ht <- ht_list[[lineage]]
  out_height <- floor((nrow(ht@matrix) + 1) / 2) + 2
  out_width <- ncol(ht@matrix) / 2 + 1
  
  png(filename = paste0("./plots/manuscript/lineage_heatmaps/", tolower(database), "_", lineage, "_mut_heatmap.png"), width = out_width, height = out_height, units = "in", res = 500)
  draw(ht, heatmap_legend_side = "top")
  invisible(dev.off())
}
```

```{r lineage_mut_heatmaps, eval=FALSE, message=FALSE}
ccle_hts <- lapply(unique(ccle_signif_dgl$disease), MakeWilcoxonLineageHeatmap, df = ccle_signif_dgl, database = "CCLE")
names(ccle_hts) <- unique(ccle_signif_dgl$disease)
lapply(unique(ccle_signif_dgl$disease), SaveWilcoxonLineageHeatmap, ht_list = ccle_hts, database = "CCLE")

ctrp_hts <- lapply(unique(ctrp_signif_dgl$disease), MakeWilcoxonLineageHeatmap, df = ctrp_signif_dgl, database = "CTRP")
names(ctrp_hts) <- unique(ctrp_signif_dgl$disease)
lapply(unique(ctrp_signif_dgl$disease), SaveWilcoxonLineageHeatmap, ht_list = ctrp_hts, database = "CTRP")

gdsc_hts <- lapply(unique(gdsc_signif_dgl$disease), MakeWilcoxonLineageHeatmap, df = gdsc_signif_dgl, database = "GDSC")
names(gdsc_hts) <- unique(gdsc_signif_dgl$disease)
lapply(unique(gdsc_signif_dgl$disease), SaveWilcoxonLineageHeatmap, ht_list = gdsc_hts, database = "GDSC")
```

# SPEARMAN: GE and CN

--------------------------------------------------------------------------------

```{r}
MakeSpearmanCorrDF <- function(dataset, df, metric, lineage) {
  df$Gene <- factor(df$Gene, ordered = FALSE)
  df <- unfactorize(df)

  if(metric == "GE") {
    if(lineage) {
      df_count <- df %>% drop_na(TPM) %>% count(Drug, Gene, disease, Drug_Gene_Lin)
      df_count_filt <- filter(df_count, n >= 3)
      df_filt <- filter(df, Drug_Gene_Lin %in% df_count_filt$Drug_Gene_Lin)
      res <- df_filt %>% group_by(Drug, Gene, disease, Drug_Gene_Lin) %>%
        summarize(GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs", exact = FALSE)$estimate,
                  GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs", exact = FALSE)$p.value, 
                  GE_p_log10 = -log10(GE_p))
      res$InG2P <- ifelse(res$Drug_Gene_Lin %in% unique(as.character(g2p_expr$Drug_Gene_Lin[!is.na(g2p_expr$Drug_Gene_Lin)])), "Yes", "No")
      res <- merge(res, df_count, by = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), all = TRUE)
    }
    if(!lineage) {
      df_count <- df %>% drop_na(TPM) %>% count(Drug, Gene, Drug_Gene)
      res <- df %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>%
        summarize(GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs", exact = FALSE)$estimate,
                  GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs", exact = FALSE)$p.value, 
                  GE_p_log10 = -log10(GE_p))
      res$InG2P <- ifelse(res$Drug_Gene %in% unique(as.character(g2p_expr$Drug_Gene[!is.na(g2p_expr$Drug_Gene)])), "Yes", "No")
      res <- merge(res, df_count, by = c("Drug", "Gene", "Drug_Gene"), all = TRUE)
    }
  }
  if(metric == "CN") {
    if(lineage) {
      df_count <- df %>% drop_na(CN) %>% count(Drug, Gene, disease, Drug_Gene_Lin)
      df_count_filt <- filter(df_count, n >= 3)
      df_filt <- filter(df, Drug_Gene_Lin %in% df_count_filt$Drug_Gene_Lin)
      res <- df_filt %>% group_by(Drug, Gene, disease, Drug_Gene_Lin) %>%
        summarize(CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs", exact = FALSE)$estimate,
                  CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs", exact = FALSE)$p.value,
                  CN_p_log10 = -log10(CN_p))
      res$InG2P <- ifelse(res$Drug_Gene_Lin %in% unique(as.character(g2p_amp$Drug_Gene_Lin[!is.na(g2p_amp$Drug_Gene_Lin)])), "Yes", "No")
      res <- merge(res, df_count, by = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), all = TRUE)
    }
    if(!lineage) {
      df_count <- df %>% drop_na(CN) %>% count(Drug, Gene, Drug_Gene)
      res <- df %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>%
        summarize(CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs", exact = FALSE)$estimate,
                  CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs", exact = FALSE)$p.value, 
                  CN_p_log10 = -log10(CN_p))
      res$InG2P <- ifelse(res$Drug_Gene %in% unique(as.character(g2p_expr$Drug_Gene[!is.na(g2p_expr$Drug_Gene)])), "Yes", "No")
      res <- merge(res, df_count, by = c("Drug", "Gene", "Drug_Gene"), all = TRUE)
    }
  }
  res$Metric <- metric
  res$Dataset <- dataset
  return(res)
}
```


```{r spear_calc, warning=FALSE, eval=FALSE}
crispr_cn_res <- MakeSpearmanCorrDF(dataset = "CRISPR", df = crispr, metric = "CN", lineage = FALSE)
ccle_cn_res <- MakeSpearmanCorrDF(dataset = "CCLE", df = ccle, metric = "CN", lineage = FALSE)
ctrp_cn_res <- MakeSpearmanCorrDF(dataset = "CTRP", df = ctrp, metric = "CN", lineage = FALSE)
gdsc_cn_res <- MakeSpearmanCorrDF(dataset = "GDSC", df = gdsc, metric = "CN", lineage = FALSE)
cn_cor <- rbind(crispr_cn_res, ccle_cn_res, ctrp_cn_res, gdsc_cn_res)
saveRDS(cn_cor, "./data_munging/rds/cn_cor.rds", compress = "xz")

crispr_ge_res <- MakeSpearmanCorrDF(dataset = "CRISPR", df = crispr, metric = "GE", lineage = FALSE)
ccle_ge_res <- MakeSpearmanCorrDF(dataset = "CCLE", df = ccle, metric = "GE", lineage = FALSE)
ctrp_ge_res <- MakeSpearmanCorrDF(dataset = "CTRP", df = ctrp, metric = "GE", lineage = FALSE)
gdsc_ge_res <- MakeSpearmanCorrDF(dataset = "GDSC", df = gdsc, metric = "GE", lineage = FALSE)
ge_cor <- rbind(crispr_ge_res, ccle_ge_res, ctrp_ge_res, gdsc_ge_res)
saveRDS(ge_cor, "./data_munging/rds/ge_cor.rds", compress = "xz")
```

```{r load_spear}
# CN
g2p_amp_crispr_4merge <- unique(select(g2p_amp, Gene, Direction_Curated))
crispr_amp_summ <- g2p_amp_crispr_4merge %>% count(Gene)
crispr_amp_summ <- merge(crispr_amp_summ, g2p_amp_crispr_4merge, by = "Gene")
crispr_amp_summ$Drug_Gene <- paste0("CRISPR_", crispr_amp_summ$Gene)
crispr_amp_summ$Direction_Heatmap <- with(crispr_amp_summ, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
crispr_amp_summ <- unique(select(crispr_amp_summ, Drug_Gene, Direction_Heatmap))

g2p_amp_4merge <- select(g2p_amp, Drug_Gene, Direction_Curated)
amp_summ <- g2p_amp_4merge %>% count(Drug_Gene)
amp_summ <- merge(amp_summ, g2p_amp_4merge, by = "Drug_Gene")
amp_summ$Direction_Heatmap <- with(amp_summ, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
amp_summ <- unique(select(amp_summ, Drug_Gene, Direction_Heatmap))

amp_summ <- rbind(amp_summ, crispr_amp_summ)
cn_cor <- readRDS("./data_munging/rds/cn_cor.rds")
cn_cor <- merge(cn_cor, amp_summ, by = "Drug_Gene", all.x = TRUE)

# GE
g2p_expr_crispr_4merge <- unique(select(g2p_expr, Gene, Direction_Curated))
crispr_expr_summ <- g2p_expr_crispr_4merge %>% count(Gene)
crispr_expr_summ <- merge(crispr_expr_summ, g2p_expr_crispr_4merge, by = "Gene")
crispr_expr_summ$Drug_Gene <- paste0("CRISPR_", crispr_expr_summ$Gene)
crispr_expr_summ$Direction_Heatmap <- with(crispr_expr_summ, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
crispr_expr_summ <- unique(select(crispr_expr_summ, Drug_Gene, Direction_Heatmap))

g2p_expr_4merge <- select(g2p_expr, Drug_Gene, Direction_Curated)
expr_summ <- g2p_expr_4merge %>% count(Drug_Gene)
expr_summ <- merge(expr_summ, g2p_expr_4merge, by = "Drug_Gene")
expr_summ$Direction_Heatmap <- with(expr_summ, ifelse(n == 2, "both", ifelse(n == 1 & Direction_Curated == "resistance", "resistance", "sensitivity")))
expr_summ <- unique(select(expr_summ, Drug_Gene, Direction_Heatmap))

expr_summ <- rbind(expr_summ, crispr_expr_summ)
ge_cor <- readRDS("./data_munging/rds/ge_cor.rds")
ge_cor <- merge(ge_cor, expr_summ, by = "Drug_Gene", all.x = TRUE)
```

## Lineage

**12/16 FIX THIS **
```{r spear_calc_lin, eval=FALSE}
# Ran on 12/3
crispr_cn_res_lin <- MakeSpearmanCorrDF(dataset = "CRISPR", df = crispr, metric = "CN", lineage = TRUE)
ccle_cn_res_lin <- MakeSpearmanCorrDF(dataset = "CCLE", df = ccle, metric = "CN", lineage = TRUE)
ctrp_cn_res_lin <- MakeSpearmanCorrDF(dataset = "CTRP", df = ctrp, metric = "CN", lineage = TRUE)
gdsc_cn_res_lin <- MakeSpearmanCorrDF(dataset = "GDSC", df = gdsc, metric = "CN", lineage = TRUE)
cn_cor_lin <- rbind(crispr_cn_res_lin, ccle_cn_res_lin, ctrp_cn_res_lin, gdsc_cn_res_lin) %>% filter(CN_p_log10 != Inf)
saveRDS(cn_cor_lin, "./data_munging/rds/cn_cor_lineage.rds")

crispr_ge_res <- MakeSpearmanCorrDF(dataset = "CRISPR", df = crispr, metric = "GE", lineage = TRUE)
ccle_ge_res_lin <- MakeSpearmanCorrDF(dataset = "CCLE", df = ccle, metric = "GE", lineage = TRUE)
ctrp_ge_res_lin <- MakeSpearmanCorrDF(dataset = "CTRP", df = ctrp, metric = "GE", lineage = TRUE)
gdsc_ge_res_lin <- MakeSpearmanCorrDF(dataset = "GDSC", df = gdsc, metric = "GE", lineage = TRUE)
ge_cor_lin <- rbind(crispr_ge_res, ccle_ge_res_lin, ctrp_ge_res_lin, gdsc_ge_res_lin) %>% filter(GE_p_log10 != Inf)
saveRDS(ge_cor_lin, "./data_munging/rds/ge_cor_lineage.rds")
```

```{r}
cn_cor_lin <- readRDS("./data_munging/rds/cn_cor_lineage.rds")
cn_cor_lin_g2p <- filter(cn_cor_lin, InG2P == "Yes")
ge_cor_lin <- readRDS("./data_munging/rds/ge_cor_lineage.rds")
ge_cor_lin_g2p <- filter(ge_cor_lin, InG2P == "Yes")
```

## Plots

```{r crispr_spear_eda, warning=TRUE}
crispr_spear_cn <- filter(cn_cor, Dataset == "CRISPR" & Gene %in% g2p_dg_genes)
crispr_spear_cn$Gene <- with(crispr_spear_cn, reorder(Gene, CN_r, function(x) x))
crispr_spear_cn_plot <- ggplot(data = crispr_spear_cn) +
  geom_point(aes(x = Gene, y = CN_r, fill = CN_p_log10), pch = 21, size = 5) +
  theme(legend.position = "bottom") +
  scale_fill_gradientn(colours = c("white", "green4"),
                       breaks = seq(0, 14, by = 2),
                       labels = seq(0, 14, by = 2),
                       limits = c(0, 14)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10), legend.key.width = unit(0.5, "in")) +
  labs(title = "Copy Number",
       fill = expression(-log[10](p)),
       x = "Gene",
       y = expression("Spearman Correlation Coefficient"~(r[s])))

crispr_spear_ge <- filter(ge_cor, Dataset == "CRISPR" & Gene %in% g2p_dg_genes)
crispr_spear_ge$Gene <- with(crispr_spear_ge, reorder(Gene, GE_r, function(x) x))
crispr_spear_ge_plot <- ggplot(data = crispr_spear_ge) +
  geom_point(aes(x = Gene, y = GE_r, fill = GE_p_log10), pch = 21, size = 5) +
  theme(legend.position = "bottom") +
  scale_fill_gradientn(colours = c("white", "deepskyblue4"),
                       breaks = seq(0, 14, by = 2),
                       labels = seq(0, 14, by = 2),
                       limits = c(0, 14)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10), legend.key.width = unit(0.5 , "in")) +
  labs(title = "Gene Expression",
       fill = expression(-log[10](p)),
       x = "Gene",
       y = expression("Spearman Correlation Coefficient"~(r[s])))

crispr_spearman_corr_plot <- plot_grid(crispr_spear_cn_plot, crispr_spear_ge_plot, align = "h")
crispr_spearman_corr_plot

# ggsave("./plots/manuscript/crispr_spearman_corr_plot.png", crispr_spearman_corr_plot, device = "png", dpi = 450, width = 20, height = 10, units = "in")
```

```{r corr_heatmap_function}
MakeCorrHeatmap <- function(cor_res, row_names, col_title, color_range, legend_title, metric, radius) {
  if(toupper(metric) == "CN" | toupper(metric) == "COPY NUMBER") {
    if(col_title == "CRISPR") {
      signif_grid <- dcast(cor_res, Gene ~ Drug, value.var = "CN_p", fun.aggregate = mean, fill = 0)
      signif_grid <- signif_grid[match(g2p_gene_order, signif_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      signif_grid[signif_grid >= 0.05] <- ""
      signif_grid[signif_grid != ""] <- "*"
      
      corr_grid <- dcast(cor_res, Gene ~ Drug, value.var = "CN_r", fun.aggregate = mean, fill = 0)
      corr_grid <- corr_grid[match(g2p_gene_order, corr_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      
      annot_grid <- dcast(cor_res, Gene ~ Drug, value.var = "Direction_Heatmap", fill = 0)
      annot_grid <- annot_grid[match(g2p_gene_order, annot_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    }
    if(col_title != "CRISPR") {
      signif_grid <- dcast(cor_res, Gene ~ Drug, value.var = "CN_p", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      signif_grid <- signif_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
      signif_grid <- signif_grid[complete.cases(signif_grid), ]
      signif_grid[signif_grid >= 0.05] <- ""
      signif_grid[signif_grid != ""] <- "*"
      
      corr_grid <- dcast(cor_res, Gene ~ Drug, value.var = "CN_r", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      corr_grid <- corr_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(corr_grid))]
      corr_grid <- corr_grid[complete.cases(corr_grid), ]
      
      annot_grid <- dcast(cor_res, Gene ~ Drug, value.var = "Direction_Heatmap", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      annot_grid <- annot_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
      annot_grid <- annot_grid[complete.cases(annot_grid), ]
    }
  }
  
  if(toupper(metric) == "GE" | toupper(metric) == "GENE EXPRESSION") {
    if(col_title == "CRISPR") {
      signif_grid <- dcast(cor_res, Gene ~ Drug, value.var = "GE_p", fun.aggregate = mean, fill = 0)
      signif_grid <- signif_grid[match(g2p_gene_order, signif_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      signif_grid[signif_grid >= 0.05] <- ""
      signif_grid[signif_grid != ""] <- "*"
      
      corr_grid <- dcast(cor_res, Gene ~ Drug, value.var = "GE_r", fun.aggregate = mean, fill = 0)
      corr_grid <- corr_grid[match(g2p_gene_order, corr_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      
      annot_grid <- dcast(cor_res, Gene ~ Drug, value.var = "Direction_Heatmap", fill = 0)
      annot_grid <- annot_grid[match(g2p_gene_order, annot_grid$Gene),] %>% drop_na(Gene) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    }
    if(col_title != "CRISPR") {
      signif_grid <- dcast(cor_res, Gene ~ Drug, value.var = "GE_p", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      signif_grid <- signif_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
      signif_grid <- signif_grid[complete.cases(signif_grid), ]
      signif_grid[signif_grid >= 0.05] <- ""
      signif_grid[signif_grid != ""] <- "*"
      
      corr_grid <- dcast(cor_res, Gene ~ Drug, value.var = "GE_r", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      corr_grid <- corr_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(corr_grid))]
      corr_grid <- corr_grid[complete.cases(corr_grid), ]
      
      annot_grid <- dcast(cor_res, Gene ~ Drug, value.var = "Direction_Heatmap", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
      annot_grid <- annot_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
      annot_grid <- annot_grid[complete.cases(annot_grid), ]
    }
  }
  
  corr_grid <- as.matrix(corr_grid)
  
  col_fun <- colorRamp2(color_range, c("forestgreen", "white", "blue3"))
  
  # Legend properties list
  if(metric == "CN") { legend_sep <- 0.1 }
  if(metric == "GE") { legend_sep <- 0.2 }
  ht_legend_list <- list(title_position = "topleft",
                         legend_direction = "horizontal",
                         legend_width = unit(2, "in"),
                         border = "gray70",
                         at = seq(color_range[1], color_range[3], by = legend_sep),
                         labels = seq(color_range[1], color_range[3], by = legend_sep))
  
  ht <- Heatmap(corr_grid,
                col = col_fun,
                name = legend_title,
                na_col = "gray97",
                rect_gp = gpar(col = "gray70"),
                heatmap_legend_param = ht_legend_list,
                cluster_rows = FALSE, 
                cluster_columns = FALSE, 
                show_row_names = as.logical(row_names),
                row_names_side = "left",
                row_names_gp = gpar(fontsize = 10),
                column_names_side = "bottom",
                column_names_gp = gpar(fontsize = 10),
                show_column_names = FALSE,
                bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(signif_grid),
                                   rot = 45,
                                   gp = gpar(fontsize = 11),
                                   location = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.85, "in")),
                row_title = "Genes",
                row_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title = col_title, 
                column_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title_side = "top",
                cell_fun = function(j, i, x, y, width, height, fill) {
                  grid.text(signif_grid[i, j], x = x + width*0.20, y = y + height*0.025, gp = gpar(fontsize = 18))
                  if(annot_grid[i, j] == "sensitivity") {
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "white", col = "black"))
                  }
                  if(annot_grid[i, j] == "resistance") {
                    grid.circle(x = x - width*0.20, y = y + height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "red", col = "black"))
                  }
                  if(annot_grid[i, j] == "both") {
                    grid.circle(x = x - width*0.20, y = y + height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "red", col = "black"))
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "white", col = "black"))
                  }
                }
                )
  return(ht)
}
```

```{r spear_plots, eval=FALSE}
# Copy number
crispr_cn_cor <- unfactorize(filter(cn_cor, Dataset == "CRISPR" & Gene %in% unique(g2p_amp$Gene)))
crispr_cn_ht <- MakeCorrHeatmap(cor_res = crispr_cn_cor, row_names = TRUE, col_title = "CRISPR", color_range = c(-0.25, 0, 0.25), radius = 0.03, legend_title = "CRISPR Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/crispr_cn_heatmap.png", width = 3, height = 3, units = "in", res = 500)
draw(crispr_cn_ht, heatmap_legend_side = "top")
dev.off()

ccle_cn_cor <- unfactorize(filter(cn_cor, Dataset == "CCLE" & Gene %in% unique(g2p_amp$Gene)))
ccle_cn_ht <- MakeCorrHeatmap(cor_res = ccle_cn_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.25, 0, 0.25), radius = 0.02, legend_title = "CCLE Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/ccle_cn_heatmap.png", width = 3, height = 4, units = "in", res = 500)
draw(ccle_cn_ht, heatmap_legend_side = "top")
dev.off()

ctrp_cn_cor <- unfactorize(filter(cn_cor, Dataset == "CTRP" & Gene %in% unique(g2p_amp$Gene)))
ctrp_cn_ht <- MakeCorrHeatmap(cor_res = ctrp_cn_cor, row_names = TRUE, col_title = "CTRP", color_range = c(-0.25, 0, 0.25), radius = 0.02, legend_title = "CTRP Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/ctrp_cn_heatmap.png", width = 8, height = 3, units = "in", res = 500)
draw(ctrp_cn_ht, heatmap_legend_side = "top")
dev.off()

gdsc_cn_cor <- unfactorize(filter(cn_cor, Dataset == "GDSC" & Gene %in% unique(g2p_amp$Gene)))
gdsc_cn_ht <- MakeCorrHeatmap(cor_res = gdsc_cn_cor, row_names = TRUE, col_title = "GDSC", color_range = c(-0.25, 0, 0.25), radius = 0.02, legend_title = "GDSC Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/gdsc_cn_heatmap.png", width = 5, height = 3, units = "in", res = 500)
draw(gdsc_cn_ht, heatmap_legend_side = "top")
dev.off()

crispr_cn_ht2 <- MakeCorrHeatmap(cor_res = crispr_cn_cor, row_names = TRUE, col_title = "CRISPR", color_range = c(-0.25, 0, 0.25), radius = 0.15, legend_title = "Copy num. Spearman correlation", metric = "CN")
ccle_cn_ht2 <- MakeCorrHeatmap(cor_res = ccle_cn_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.25, 0, 0.25), radius = 0.0215, legend_title = "Copy num. Spearman correlation", metric = "CN")
ctrp_cn_ht2 <- MakeCorrHeatmap(cor_res = ctrp_cn_cor, row_names = FALSE, col_title = "CTRP", color_range = c(-0.25, 0, 0.25), radius = 0.0215, legend_title = "Copy num. Spearman correlation", metric = "CN")
gdsc_cn_ht2 <- MakeCorrHeatmap(cor_res = gdsc_cn_cor, row_names = FALSE, col_title = "GDSC", color_range = c(-0.25, 0, 0.25), radius = 0.0215, legend_title = "Copy num. Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/all_cn_heatmap.png", width = 13, height = 3.5, units = "in", res = 500)
draw(crispr_cn_ht2 +ccle_cn_ht2 + ctrp_cn_ht2 + gdsc_cn_ht2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()

# Gene expression
crispr_ge_cor <- unfactorize(filter(ge_cor, Dataset == "CRISPR" & Gene %in% unique(g2p_expr$Gene)))
crispr_ge_ht <- MakeCorrHeatmap(cor_res = crispr_ge_cor, row_names = TRUE, col_title = "CRISPR", color_range = c(-0.5, 0, 0.5), radius = 0.1, legend_title = "CRISPR Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/crispr_ge_heatmap.png", width = 3, height = 3, units = "in", res = 500)
draw(crispr_ge_ht, heatmap_legend_side = "top")
dev.off()

ccle_ge_cor <- unfactorize(filter(ge_cor, Dataset == "CCLE" & Gene %in% unique(g2p_expr$Gene)))
ccle_ge_ht <- MakeCorrHeatmap(cor_res = ccle_ge_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "CCLE Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/ccle_ge_heatmap.png", width = 3, height = 2.5, units = "in", res = 500)
draw(ccle_ge_ht, heatmap_legend_side = "top")
dev.off()

ctrp_ge_cor <- unfactorize(filter(ge_cor, Dataset == "CTRP" & Gene %in% unique(g2p_expr$Gene)))
ctrp_ge_ht <- MakeCorrHeatmap(cor_res = ctrp_ge_cor, row_names = TRUE, col_title = "CTRP", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "CTRP Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/ctrp_ge_heatmap.png", width = 8, height = 2.5, units = "in", res = 500)
draw(ctrp_ge_ht, heatmap_legend_side = "top")
dev.off()

gdsc_ge_cor <- unfactorize(filter(ge_cor, Dataset == "GDSC" & Gene %in% unique(g2p_expr$Gene)))
gdsc_ge_ht <- MakeCorrHeatmap(cor_res = gdsc_ge_cor, row_names = TRUE, col_title = "GDSC", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "GDSC Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/gdsc_ge_heatmap.png", width = 5, height = 2.5, units = "in", res = 500)
draw(gdsc_ge_ht, heatmap_legend_side = "top")
dev.off()

crispr_ge_ht2 <- MakeCorrHeatmap(cor_res = crispr_ge_cor, row_names = TRUE, col_title = "CRISPR", color_range = c(-0.5, 0, 0.5), radius = 0.155, legend_title = "Gene expr. Spearman correlation", metric = "GE")
ccle_ge_ht2 <- MakeCorrHeatmap(cor_res = ccle_ge_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.5, 0, 0.5), radius = 0.035, legend_title = "Gene expr. Spearman correlation", metric = "GE")
ctrp_ge_ht2 <- MakeCorrHeatmap(cor_res = ctrp_ge_cor, row_names = FALSE, col_title = "CTRP", color_range = c(-0.5, 0, 0.5), radius = 0.035, legend_title = "Gene expr. Spearman correlation", metric = "GE")
gdsc_ge_ht2 <- MakeCorrHeatmap(cor_res = gdsc_ge_cor, row_names = FALSE, col_title = "GDSC", color_range = c(-0.5, 0, 0.5), radius = 0.035, legend_title = "Gene expr. Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/all_ge_heatmap.png", width = 13, height = 2.9, units = "in", res = 500)
draw(crispr_ge_ht2 + ccle_ge_ht2 + ctrp_ge_ht2 + gdsc_ge_ht2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```


# OLD CODE

--------------------------------------------------------------------------------

## Mut: Sen vs. Res tests

```{r SvR_direction_datasets, eval=FALSE}
# CCLE
ccle_prot_dir <- unique(select(ccle_prot, DepMap_ID, Gene, Cell_line, CCLE_histology, TCGA_classification, CCLE_name, Drug, Drug_Gene, Score, Change_Status, Direction_Curated, Direction_Grouper))
## Resistance
ccle_prot_res <- filter(ccle_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_resistance", "Yes_resistance"))
ccle_prot_res$Direction_Grouper <- with(ccle_prot_res, ifelse(Direction_Grouper == "Yes_resistance", "Mutated", "Not mutated"))
## Sensitivity
ccle_prot_sen <- filter(ccle_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_sensitivity", "Yes_sensitivity"))
ccle_prot_sen$Direction_Grouper <- with(ccle_prot_sen, ifelse(Direction_Grouper == "Yes_sensitivity", "Mutated", "Not mutated"))
saveRDS(ccle_prot_res, "./data_munging/rds/ccle_protein_changes_resistance.rds")
saveRDS(ccle_prot_sen, "./data_munging/rds/ccle_protein_changes_sensitivity.rds")

# CTRP
ctrp_prot_dir <- unique(select(ctrp_prot, DepMap_ID, Gene, Cell_line, CCLE_histology, TCGA_classification, CCLE_name, Drug, Drug_Gene, Score, Change_Status, Direction_Curated, Direction_Grouper))
## Resistance
ctrp_prot_res <- filter(ctrp_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_resistance", "Yes_resistance"))
ctrp_prot_res$Direction_Grouper <- with(ctrp_prot_res, ifelse(Direction_Grouper == "Yes_resistance", "Mutated", "Not mutated"))
## Sensitivity
ctrp_prot_sen <- filter(ctrp_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_sensitivity", "Yes_sensitivity"))
ctrp_prot_sen$Direction_Grouper <- with(ctrp_prot_sen, ifelse(Direction_Grouper == "Yes_sensitivity", "Mutated", "Not mutated"))
saveRDS(ctrp_prot_res, "./data_munging/rds/ctrp_protein_changes_resistance.rds")
saveRDS(ctrp_prot_sen, "./data_munging/rds/ctrp_protein_changes_sensitivity.rds")

# GDSC
gdsc_prot_dir <- unique(select(gdsc_prot, DepMap_ID, Gene, Cell_line, CCLE_histology, TCGA_classification, CCLE_name, Drug, Drug_Gene, Score, Change_Status, Direction_Curated, Direction_Grouper))
## Resistance
gdsc_prot_res <- filter(gdsc_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_resistance", "Yes_resistance"))
gdsc_prot_res$Direction_Grouper <- with(gdsc_prot_res, ifelse(Direction_Grouper == "Yes_resistance", "Mutated", "Not mutated"))
## Sensitivity
gdsc_prot_sen <- filter(gdsc_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_sensitivity", "Yes_sensitivity"))
gdsc_prot_sen$Direction_Grouper <- with(gdsc_prot_sen, ifelse(Direction_Grouper == "Yes_sensitivity", "Mutated", "Not mutated"))
saveRDS(gdsc_prot_res, "./data_munging/rds/gdsc_protein_changes_resistance.rds")
saveRDS(gdsc_prot_sen, "./data_munging/rds/gdsc_protein_changes_sensitivity.rds")
```

```{r prot_SvR_wilcoxon_tests, eval=FALSE}
# CCLE
ccle_signif_prot_res <- WilcoxonTestDF(dataframe = ccle_prot_res, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ccle_signif_protein_changes_resistance.rds")
ccle_signif_prot_sen <- WilcoxonTestDF(dataframe = ccle_prot_sen, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ccle_signif_protein_changes_sensitivity.rds")

# CTRP
ctrp_signif_prot_res <- WilcoxonTestDF(dataframe = ctrp_prot_res, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ctrp_signif_protein_changes_resistance.rds")
ctrp_signif_prot_sen <- WilcoxonTestDF(dataframe = ctrp_prot_sen, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ctrp_signif_protein_changes_sensitivity.rds")

# GDSC
gdsc_signif_prot_res <- WilcoxonTestDF(dataframe = gdsc_prot_res, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/gdsc_signif_protein_changes_resistance.rds")
gdsc_signif_prot_sen <- WilcoxonTestDF(dataframe = gdsc_prot_sen, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/gdsc_signif_protein_changes_sensitivity.rds")
```

```{r load_prot_SvR_wilcoxon_tests}
ccle_signif_prot_res <- readRDS("./data_munging/rds/ccle_signif_protein_changes_resistance.rds")
ccle_signif_prot_res$Direction_Test <- "Resistance"
ccle_signif_prot_sen <- readRDS("./data_munging/rds/ccle_signif_protein_changes_sensitivity.rds")
ccle_signif_prot_sen$Direction_Test <- "Sensitivity"
ccle_signif_prot_dir <- bind_rows(ccle_signif_prot_res, ccle_signif_prot_sen)

ctrp_signif_prot_res <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_resistance.rds")
ctrp_signif_prot_res$Direction_Test <- "Resistance"
ctrp_signif_prot_sen <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_sensitivity.rds")
ctrp_signif_prot_sen$Direction_Test <- "Sensitivity"
ctrp_signif_prot_dir <- bind_rows(ctrp_signif_prot_res, ctrp_signif_prot_sen)

gdsc_signif_prot_res <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_resistance.rds")
gdsc_signif_prot_res$Direction_Test <- "Resistance"
gdsc_signif_prot_sen <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_sensitivity.rds")
gdsc_signif_prot_sen$Direction_Test <- "Sensitivity"
gdsc_signif_prot_dir <- bind_rows(gdsc_signif_prot_res, gdsc_signif_prot_sen)

ccle_prot_res <- readRDS("./data_munging/rds/ccle_protein_changes_resistance.rds")
ccle_prot_sen <- readRDS("./data_munging/rds/ccle_protein_changes_sensitivity.rds")
ctrp_prot_res <- readRDS("./data_munging/rds/ctrp_protein_changes_resistance.rds")
ctrp_prot_sen <- readRDS("./data_munging/rds/ctrp_protein_changes_sensitivity.rds")
gdsc_prot_res <- readRDS("./data_munging/rds/gdsc_protein_changes_resistance.rds")
gdsc_prot_sen <- readRDS("./data_munging/rds/gdsc_protein_changes_sensitivity.rds")
```

```{r direction_boxplots, eval=FALSE}
# CCLE
ccle_res_box <- ggplot(data = filter(ccle_prot_res, Drug_Gene %in% ccle_signif_prot_res$Drug_Gene)) +
  facet_wrap(~ Drug * Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "springgreen4", "Not mutated" = "palegreen3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = ccle_signif_prot_res, aes(x = 1.5, y = 1.05, label = p.signif), size = 7) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  labs(y = "AUC", x = "Resistance mutation?")

ccle_sen_box <- ggplot(data = filter(ccle_prot_sen, Drug_Gene %in% ccle_signif_prot_sen$Drug_Gene)) +
  facet_wrap(~ Drug * Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "springgreen4", "Not mutated" = "palegreen3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = ccle_signif_prot_sen, aes(x = 1.5, y = 1.05, label = p.signif), size = 7) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  labs(y = "AUC", x = "Sensitivity mutation?")
ccle_dir_boxes <- plot_grid(ccle_res_box, ccle_sen_box, labels = c("A", "B"), rel_widths = c(3, 1.1))
ggsave("./plots/manuscript/ccle_direction_boxplots.png", ccle_dir_boxes, device = "png", dpi = 450, width = 12, height = 5, units = "in")

# CTRP
ctrp_res_box <- ggplot(data = filter(ctrp_prot_res, Drug_Gene %in% ctrp_signif_prot_res$Drug_Gene)) +
  facet_wrap(~ Drug * Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "steelblue4", "Not mutated" = "slategray3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = ctrp_signif_prot_res, aes(x = 1.5, y = 1.05, label = p.signif), size = 7) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  labs(y = "AUC", x = "Resistance mutation?")

ctrp_sen_box <- ggplot(data = filter(ctrp_prot_sen, Drug_Gene %in% ctrp_signif_prot_sen$Drug_Gene)) +
  facet_wrap(~ Drug * Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "steelblue4", "Not mutated" = "slategray3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = ctrp_signif_prot_sen, aes(x = 1.5, y = 1.05, label = p.signif), size = 7) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  labs(y = "AUC", x = "Sensitivity mutation?")
ctrp_dir_boxes <- plot_grid(ctrp_res_box, ctrp_sen_box, labels = c("A", "B"), ncol = 1)
ggsave("./plots/manuscript/ctrp_direction_boxplots.png", ctrp_dir_boxes, device = "png", dpi = 450, width = 12, height = 8, units = "in")

# GDSC
gdsc_res_box <- ggplot(data = filter(gdsc_prot_res, Drug_Gene %in% gdsc_signif_prot_res$Drug_Gene)) +
  facet_wrap(~ Drug * Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "turquoise4", "Not mutated" = "paleturquoise3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = gdsc_signif_prot_res, aes(x = 1.5, y = 1.05, label = p.signif), size = 7) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  labs(y = "AUC", x = "Resistance mutation?")

gdsc_sen_box <- ggplot(data = filter(gdsc_prot_sen, Drug_Gene %in% gdsc_signif_prot_sen$Drug_Gene)) +
  facet_wrap(~ Drug * Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "turquoise4", "Not mutated" = "paleturquoise3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = gdsc_signif_prot_sen, aes(x = 1.5, y = 1.05, label = p.signif), size = 7) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  labs(y = "AUC", x = "Sensitivity mutation?")
gdsc_dir_boxes <- plot_grid(gdsc_res_box, gdsc_sen_box, labels = c("A", "B"), rel_widths = c(2, 1.1))
ggsave("./plots/manuscript/gdsc_direction_boxplots.png", gdsc_dir_boxes, device = "png", dpi = 450, width = 12, height = 5, units = "in")
```

# SESSION INFORMATION

--------------------------------------------------------------------------------

```{r session_info}
print(sessionInfo())
```