---
title: "19Q2"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the [Broad's Avana CRISPR](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017), the [Cancer Therapeutics Response Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (CTRP) drug screen dataset (v2) (Basu et al., 2013; M. G. Rees et al., 2016; B. Seashore-Ludlow et al., 2015), the [Genomics of Drug Sensitivity in Cancer](https://www.cancerrxgene.org/downloads) (GDSC) drug screen dataset (Garnett et al., 2012; Yang et al., 2012) , and the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) (CCLE) drug screen dataset (J. Barretina et al., 2012; T. C. C. L. E. Consortium & Consortium, 2015).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

Annotation files for mutation status, copy number, and gene expression of the CCLE, GDSC, and CTRP datasets (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/). Note that the group that conducted the GDSC screen also collected their own genomic annotation information for mutation status, copy number, and gene expression, but we did not use it.

# SET UP

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

```{r libraries}
library(circlize)
library(ComplexHeatmap)
library(cowplot)
library(crunch)
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(jsonlite)
library(kableExtra)
library(naniar)
library(openxlsx)
library(reshape2)
library(scales)
library(stringr)
library(tidyverse)
library(VennDiagram)

theme_set(theme_light())
```

```{r general_functions}
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) {
    df[[i]] = as.character(df[[i]])
  }
  return(df)
}
```

# DATA

--------------------------------------------------------------------------------

## Cell line metadata

**D. Charytonowicz cell line converter**

This comprehensive cancer cell line information was curated by both Daniel Charytonowicz and myself.
```{r dan_ccl_database}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
colnames(ccl_converter)[colnames(ccl_converter) == "Broad_ID"] <- "DepMap_ID"
ccl_lineage_converter <- unique(ccl_converter[, c("DepMap_ID", "grouped_general_doid", "group_general_lineage_name")])
lineage_converter <- unique(select(ccl_lineage_converter, grouped_general_doid, group_general_lineage_name))
ccl_histologies <- read.delim("./data_munging/ccl_histologies_metadata.csv", sep = ",", header = TRUE, row.names = 1)
colnames(ccl_histologies) <- c("Cell_line", "CCLE_histology", "TCGA_classification", "CCLE_name", "DepMap_ID")
```

**DepMap metadata**

DepMap 19Q2 update fixed almost all of the inconsistencies of the Primary.Disease column, though there were come capitalization inconsistencies.
```{r depmap_metadata}
ccl_meta <- read.delim("./data_munging/19Q2/sample_info.csv", sep = ",", header = TRUE, na.strings = c("", NA))
ccl_meta <- select(ccl_meta, DepMap_ID, stripped_cell_line_name, CCLE_name, disease)
colnames(ccl_meta) <- c("DepMap_ID", "Cell_line", "CCLE_name", "disease")
# Remove two rows that contain cell lines with two different histologies
# The histologies that are kept are verified through matching in Cellosaurus and Cell Model Passports
# KMH2, haematopoietic_and_lymphoid: https://web.expasy.org/cellosaurus/CVCL_1330, https://cellmodelpassports.sanger.ac.uk/passports/SIDM01018
# MS1, lung: https://web.expasy.org/cellosaurus/CVCL_1429, https://cellmodelpassports.sanger.ac.uk/passports/SIDM00245
ccl_meta <- subset(ccl_meta, CCLE_name != "KMH2_THYROID" & CCLE_name != "MS1_SKIN")
ccl_meta <- subset(ccl_meta, !grepl("MERGED", CCLE_name, fixed = TRUE))
ccl_diseases <- as.character(unique(ccl_meta$disease))[as.character(unique(ccl_meta$disease)) != " "]
CCLE_histology <- tolower(gsub("^[a-z0-9]*_", "", ccl_meta$CCLE_name, ignore.case = TRUE))
ccl_meta <- add_column(ccl_meta, CCLE_histology = CCLE_histology, .after = "CCLE_name")
ccl_meta$disease <- with(ccl_meta, ifelse(disease == " " & CCLE_histology %in% ccl_diseases, as.character(CCLE_histology), as.character(disease)))
ccl_meta$disease <- with(ccl_meta, ifelse(disease == "ovary", "ovarian",
                                   ifelse(disease == " " & CCLE_histology == "vulva", "vulva",
                                   ifelse(disease == " " & CCLE_histology == "stomach", "gastric",
                                   ifelse(disease == " " & CCLE_histology == "upper_aerodigestive_tract", "upper_aerodigestive_tract",
                                   ifelse(disease == " " & CCLE_histology == "large_intestine", "colorectal",
                                   ifelse(disease == " " & CCLE_histology == "biliary_tract", "bile_duct",
                                   ifelse(disease == " " & CCLE_histology == "oesophagus", "esophagus",
                                   ifelse(disease == " " & CCLE_histology == "testis", "embryo", disease)))))))))
```

## G2P

```{r g2p_parsing_functions}
ExtractBiomarkerType <- function(json_string) {
  json_string <- as.character(json_string)
  biomarker_type <- ((str_extract(string = json_string, pattern = "biomarker_type.+") %>% strsplit(split = ","))[[1]][1] %>% str_extract(pattern = "u'.+") %>% str_match("'(.*?)'"))[,2]
  biomarker_type <- ifelse(is.na(biomarker_type), "", biomarker_type)
  return(biomarker_type)
}
```

```{r g2p_parsing, eval=FALSE}
g2p <- read.delim("./data_munging/g2p_full_level_A_dataset_NEW_DEC_2_2018_withGroupedLineages.csv", sep = "\t", header = TRUE)
g2p[, 1:2] <- NULL
g2p <- unfactorize(g2p)
g2p$Drug <- tolower(g2p$Drug)
g2p[g2p == -1] <- ""
g2p$Gene <- ifelse(g2p$Gene == "" & g2p$Drug == "daunorubicin", "MLL", as.character(g2p$Gene))
g2p$Direction <- ifelse(g2p$Direction == 1 | g2p$Direction == "", NA, as.character(g2p$Direction))
g2p$Ref[g2p$Ref == "None" | g2p$Ref == "-" | is.na(g2p$Ref)] <- ""
g2p$Alt[g2p$Alt == "None" | g2p$Alt == "-" | is.na(g2p$Alt)] <- ""
g2p <- g2p[, c("Drug", "DrugID", "Direction", "Gene", "Ref", "Alt", "Start", "End", "Chromosome", "Feature.Names", "Mutation", "JSON", "Evidence.Level", "Phenotype.Description", "Phenotype.Family", "Phenotype.ID", "Sequence.ID", "grouped_general_doid")]
g2p <- add_column(g2p, Ref_Length = nchar(as.character(g2p$Ref)), .after = "Ref")
g2p <- add_column(g2p, Alt_Length = nchar(as.character(g2p$Alt)), .after = "Alt")
g2p <- add_column(g2p, Variant_Length = NA, .after = "Alt_Length")
g2p$Variant_Length <-
  ifelse(g2p$Alt_Length == 1 & g2p$Ref_Length == 1, 1,
  ifelse(g2p$Alt_Length == 2 & g2p$Ref_Length == 2, 2,
  ifelse(g2p$Alt_Length == 3 & g2p$Ref_Length == 3, 3,
  ifelse(g2p$Ref_Length == g2p$Alt_Length & g2p$Ref_Length > 3, g2p$Ref_Length,
  ifelse(g2p$Ref_Length != g2p$Alt_Length, abs(g2p$Ref_Length - g2p$Alt_Length), NA)))))
g2p <- add_column(g2p, Variant_Type = NA, .after = "Alt_Length")
g2p$Variant_Type <-
  ifelse(g2p$Alt_Length == 1 & g2p$Ref_Length == 1, "SNP",
  ifelse(g2p$Alt_Length == 2 & g2p$Ref_Length == 2, "DNP",
  ifelse(g2p$Alt_Length == 3 & g2p$Ref_Length == 3, "TNP",
  ifelse(g2p$Ref_Length == g2p$Alt_Length & g2p$Ref_Length > 3, "ONP",
  ifelse(g2p$Ref_Length > g2p$Alt_Length, "DEL",
  ifelse(g2p$Ref_Length < g2p$Alt_Length, "INS", "None"))))))
g2p <- subset(g2p, !grepl("^SID|^CHEMBL", g2p$DrugID))
g2p <- subset(g2p, !grepl("germline|homozygosity", tolower(g2p$Feature.Names)))
g2p <- add_column(g2p, biomarker_type = tolower(sapply(g2p$JSON, ExtractBiomarkerType, simplify = TRUE)), .after = "Feature.Names")
g2p <- add_column(g2p, biomarker_type_curated = NA, .after = "biomarker_type")

# Manually assign biomarker_types based on a combination of biomarker_type derived from the JSON string and patterns in the Feature.Names
biomarker_type_curated <- 
  ifelse(!is.na(str_match(string = tolower(g2p$biomarker_type), pattern = "copy number variant|amplification")[,1]) | !is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "copy number variant|amplification")), "Amplification/CNV",
  ifelse(!is.na(str_match(string = toupper(g2p$Feature.Names), pattern = "[A-Z0-9|A-Z]+-[A-Z0-9|A-Z]+|FUSION")[,1]) & is.na(str_match(string = g2p$Feature.Names, pattern = "[0-9]+-[0-9]+")) | g2p$biomarker_type == "fusion" | g2p$biomarker_type == "rearrange", "Fusion",
  ifelse(!is.na(str_match(string = g2p$Feature.Names, pattern = "[A-Z][0-9]+([A-Z]|[*])|[0-9A-Z]+ [A-Z][0-9]+|[A-Z][0-9]+$")[,1]), "Mutation",
  ifelse(!is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "expression|methylation")[,1]), "Expression",
  ifelse(!is.na(str_match(string = g2p$biomarker_type, pattern = "intron|insertion|deletion|mutant")[,1]), "Mutation",
  ifelse(!is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "codon|positive|biallelic inactivation|del|ins|dup|mut|frameshift|missense|variant|[0-9]+fs|>")[,1]), "Mutation",
  ifelse(!is.na(str_match(string = g2p$Feature.Names, pattern = "^KIT$|^BRCA$")[,1]), "Mutation", NA)))))))
g2p$biomarker_type_curated <- biomarker_type_curated[,1]

# Edit direction column
g2p <- add_column(g2p, Direction_Curated = NA, .after = "Direction")
g2p$Direction_Curated <- with(g2p, ifelse(!is.na(str_match(string = tolower(Direction), pattern = "adverse|unfavorable|^resistant")[,1]), "resistance",
                         ifelse(!is.na(str_match(string = tolower(Direction), pattern = "confers resistance")[,1]), "resistance",
                         ifelse(!is.na(str_match(string = tolower(Direction), pattern = "sensitive|sensitivity|responsive|better")[,1]), "sensitivity", 
                         ifelse(is.na(Direction) & Drug == "gefitinib" & Feature.Names == "L858R" & Gene == "EGFR", "sensitivity",
                         ifelse(is.na(Direction) & Drug == "trametinib" & Feature.Names %in% c("V600E", "V600K") & Gene == "BRAF", "sensitivity",
                         ifelse(is.na(Direction) & Drug == "cabozantinib" & Feature.Names == "RET C634W", "sensitivity", NA)))))))

# Edit mutation column
g2p <- add_column(g2p, Protein_Change = NA, .after = "Mutation")
g2p$Protein_Change <- gsub("u'|'| |,|\\[|\\]", "", g2p$Mutation)
g2p$Protein_Change <- ifelse(g2p$Protein_Change == "", str_match(string = g2p$Feature.Names, pattern = "^[A-Z][0-9]+([A-Z]|[*])| [A-Z][0-9]+([A-Z]|[*])|:[A-Z][0-9]+([A-Z]|[*])")[,1], as.character(g2p$Protein_Change))
g2p$Protein_Change <- gsub(":| |c\\..*", "", g2p$Protein_Change)
g2p$Protein_Change <- ifelse(!is.na(str_match(string = tolower(g2p$Protein_Change), pattern = "exon|expression|frameshift|fusion|methylation|mutation|-")[,1]), NA, as.character(g2p$Protein_Change))
g2p$Protein_Change <- gsub("^([0-9]+)([A-Z])$", "\\2\\1\\2", g2p$Protein_Change)
g2p$Protein_Change <- gsub("^([A-Z])([0-9]+)$", "\\1\\2\\1", g2p$Protein_Change)
g2p <- add_column(g2p, Gene_Protein_Change = NA, .after = "Protein_Change")
g2p$Gene_Protein_Change <- ifelse(!is.na(g2p$Gene) & !is.na(g2p$Protein_Change), paste0(as.character(g2p$Gene), ":", as.character(g2p$Protein_Change)), NA)

# Add association columns
g2p[g2p == ""] <- NA
g2p$Drug_Gene <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene), NA, paste0(g2p$Drug, "_", g2p$Gene))
g2p <- merge(lineage_converter, g2p, by = "grouped_general_doid", all.y = TRUE)
g2p$Drug_Gene_Lin <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene) | is.na(g2p$group_general_lineage_name), NA, paste0(g2p$Drug, "_", g2p$Gene, "_", g2p$group_general_lineage_name))
g2p$Gene_Lin <- ifelse(is.na(g2p$Drug) | !is.na(g2p$Gene) | !is.na(g2p$group_general_lineage_name), paste0(g2p$Gene, "_", g2p$group_general_lineage_name), NA)

write.csv(g2p, "./data_munging/g2p_db_edited.csv")
```

```{r g2p}
g2p <- read.delim("./data_munging/g2p_db_edited.csv", sep = ",", header = TRUE, row.names = 1)
g2p$Drug_Gene_Lin <- gsub(pattern = " cancer", "", g2p$Drug_Gene_Lin)

g2p_dgl <-  as.character(unique(g2p$Drug_Gene_Lin[!is.na(g2p$Drug_Gene_Lin)]))
g2p_dg <- as.character(unique(g2p$Drug_Gene[!is.na(g2p$Drug_Gene)]))
g2p_gl <- as.character(unique(g2p$Gene_Lin[!is.na(g2p$Gene_Lin)]))
g2p_drugs <- as.character(unique(g2p$Drug[!is.na(g2p$Drug)]))
g2p_genes <- as.character(unique(g2p$Gene[!is.na(g2p$Gene)]))
g2p_dg_genes <- unique(sapply(strsplit(g2p_dgl, "_", fixed = TRUE), function(x) x[2]))
g2p_lins <- as.character(unique(g2p$group_general_lineage_name[!is.na(g2p$group_general_lineage_name)]))

g2p$biomarker_type_curated <- ifelse(is.na(g2p$biomarker_type_curated), "None", as.character(g2p$biomarker_type_curated))
g2p_split <- filter(g2p, !is.na(Drug)) %>% group_split(biomarker_type_curated)
g2p_amp <- g2p_split[[1]]
g2p_expr <- g2p_split[[2]]
g2p_fus <- g2p_split[[3]]
g2p_fus$FusionPair <- sapply(X = g2p_fus$Feature.Names, function(x) strsplit(as.character(x)," ")[[1]][1])
g2p_fus$FusionPair <- ifelse(g2p_fus$FusionPair == "Fusions", as.character(g2p_fus$Gene), as.character(g2p_fus$FusionPair))
g2p_fus$FusionPair <- ifelse(g2p_fus$FusionPair == "BCR-ABL", "BCR-ABL1", as.character(g2p_fus$FusionPair))
g2p_fus$Drug_FusionPair <- ifelse(is.na(g2p_fus$Drug) | is.na(g2p_fus$FusionPair), NA, paste0(g2p_fus$Drug, "_", g2p_fus$FusionPair))
g2p_mut <- g2p_split[[4]] %>% drop_na(Direction_Curated)
g2p_mut_genes <- unique(sapply(strsplit(as.character(g2p_mut$Drug_Gene)[!is.na(g2p_mut$Drug_Gene)], "_", fixed = TRUE), function(x) x[2]))
g2p_nodrug <- filter(g2p, is.na(Drug))
g2p_prot <- unique(filter(g2p, !is.na(Gene_Protein_Change) & !is.na(Drug)) %>% select(group_general_lineage_name, Drug, Direction_Curated, Gene, Protein_Change, Gene_Protein_Change))
g2p_prot$Drug_Gene_Protein_Change <- with(g2p_prot, paste0(Drug, "_", Gene_Protein_Change))
g2p_prot$Drug_Gene_Protein_Change_Direction <- with(g2p_prot, paste0(Drug_Gene_Protein_Change, "_", Direction_Curated))
g2p_mut_summ <- g2p_mut %>% count(Drug_Gene, Direction_Curated)
# g2p_mut_test_n <- g2p_mut_summ %>% count(Drug_Gene) %>% filter(n > 1)
# g2p_mut_test <- filter(g2p_mut, Drug_Gene %in% as.character(g2p_mut_test_n$Drug_Gene))
```

## Omics

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Mutation calls (coding region, germline filtered):
```{r mutation_status, eval=FALSE}
maf_raw <- read.delim("./data_munging/19Q2/CCLE_mutations_20190514.csv.gz", header = TRUE, sep = ",")
maf_raw[, 1:2] <- NULL
colnames(maf_raw)[colnames(maf_raw) == "Hugo_Symbol"] <- "Gene"
maf_raw <- unfactorize(maf_raw)

# Select columns for maf_changes_df
maf_changes <- select(maf_raw, Gene, DepMap_ID, Protein_Change) %>% drop_na(Protein_Change) %>% filter(Protein_Change != "" & Gene %in% g2p_genes)
maf_changes <- unique(maf_changes)
maf_changes$Protein_Change <- gsub(pattern = "^p.", replacement = "", x = maf_changes$Protein_Change)
maf_changes$Gene_Protein_Change <- with(maf_changes, paste0(Gene, ":", Protein_Change))
saveRDS(maf_changes, "./data_munging/19Q2/maf_changes_df_19Q2.rds", compress = "xz")

# Select columns for maf_df
maf <- subset(maf_raw, select = c("Gene", "DepMap_ID"))
maf$Mutation <- 1
maf_grid <- expand.grid("Gene" = unique(maf$Gene), "DepMap_ID" = unique(maf$DepMap_ID))
maf_df <- merge(maf, maf_grid, by = c("Gene", "DepMap_ID"), all = TRUE)
maf_df[is.na(maf_df)] <- 0
saveRDS(maf_df, "./data_munging/19Q2/maf_df_19Q2.rds", compress = "xz")
```

DepMap 19Q2 fusion calls derived using the [STAR-Fusion pipeline](https://github.com/STAR-Fusion/STAR-Fusion/wiki). Column descriptions:

- *JunctionReads*: the number of RNA-Seq fragments containing a read that aligns as a split read at the site of the putative fusion junction
- *SpanningFrags*: the number of RNA-Seq fragments that encompass the fusion junction such that one read of the pair aligns to a different gene than the other paired-end read of that fragment
    - Those predictions that have very few JunctionReads and/or SpanningReads are going to be enriched for false positives. Note, depending on the site of the fusion breakpoint and length of the reads, it may not be possible to have SpanningFragments and all evidence may show up in the form of JunctionReads
- *FFPM (fusion fragments per million total reads)*: normalized measure of the fusion-supporting RNA-seq fragmentS; a filter of 0.1 sum FFPM (meaning at least 1 fusion-supporting RNA-seq fragment per 10M total reads) tends to be effective at excluding fusion artifacts, and is the current default for filtering fusions from the final output
- *LargeAnchorSupport*: whether there are split reads that provide 'long' (set to length of 25 bases) alignments on both sides of the putative breakpoint
    - Those fusions supported only by split reads (no spanning fragments) and lack LargeAnchorSupport are often highly suspicious and tend to be false positives
    - Those with LargeAnchorSupport are labeled as 'YES_LDAS' (where LDAS = long double anchor support)
- *SpliceType*: whether the proposed breakpoint occurs at reference exon junctions as provided by the reference transcript structure annotations (ex. gencode)
- *LeftBreakEntropy* and *RightBreakEntropy*: the Shannon entropy of the 15 exonic bases flanking the breakpoint.
    - The maximum entropy is 2, representing highest complexity, and the lowest would be zero (involving a 15 base mononucleotide run)
    - Low entropy sites should generally be treated as less confident breakpoints.

```{r fusion_calls, eval=TRUE}
fusions <- read.delim("./data_munging/19Q2/CCLE_fusions.csv.gz", header = TRUE, sep = ",")
fusions_bcr_abl1 <- filter(fusions, X.FusionName %in% c("BCR--ABL1", "ABL1--BCR"))
fusions$LeftGene <- gsub(" .*", "", fusions$LeftGene)
fusions$RightGene <- gsub(" .*", "", fusions$RightGene)
fusions$X.FusionName <- gsub("--", "-", fusions$X.FusionName)
fusions <- select(fusions, DepMap_ID, X.FusionName, LeftGene, RightGene)
fusions <- as.data.frame(lapply(fusions, function(x) gsub("IGH@|IGH-@", "IGH", x)))
colnames(fusions) <- c("DepMap_ID", "FusionPair", "LeftGene", "RightGene")
fusions <- merge(fusions, ccl_meta, by = "DepMap_ID", all.x = TRUE)
```

DepMap WES CN Data:

- gene_cn - NumericalMatrix
- Gene level copy number data, log2 transformed with a pseudo count 1. This is generated by mapping genes onto the segment level calls.
- Rows: cell lines (Broad IDs)
- Columns: genes (HGNC symbol and Entrez ID)

```{r copy_number, eval=FALSE}
cn <- read.delim("./data_munging/19Q2/CCLE_gene_cn_20190514.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "DepMap_ID"
# Melt
cn_melt <- melt(data = cn, id.vars = "DepMap_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Gene", value.name = "CN")
cn_melt <- unfactorize(cn_melt)

saveRDS(cn_melt, "./data_munging/19Q2/cn_melt_19Q2.rds", compress = "xz")
```

19Q2, CCLE RNAseq gene expression data:

- expression - NumericalMatrix
- RNAseq TPM gene expression data for just protein coding genes using RSEM. Log2 transformed, using a pseudo-count of 1.
- Rows: cell lines (Broad IDs)
- Columns: genes (HGNC symbol and Entrez ID)

```{r gene_expression, eval=FALSE}
ge <- read.delim("./data_munging/19Q2/CCLE_expression_20190514.csv.gz", header = TRUE, sep = ",", check.names = FALSE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(ge) <- gsub(" .*", "", colnames(ge))
colnames(ge)[1] <- "DepMap_ID"
# Melt
ge_melt <- melt(ge, id.vars = "DepMap_ID", measure.vars = colnames(ge)[2:ncol(ge)], variable.name = "Gene", value.name = "TPM")
# ge_melt$TPM <- log2(ge_melt$TPM + 1)
ge_melt <- unfactorize(ge_melt)

saveRDS(ge_melt, "./data_munging/19Q2/ge_melt_19Q2.rds", compress = "xz")
```

Load Rds:
```{r load_omics_rds}
maf_df <- readRDS("./data_munging/19Q2/maf_df_19Q2.rds")
maf_changes_df <- readRDS("./data_munging/19Q2/maf_changes_df_19Q2.rds")
cn_melt <- readRDS("./data_munging/19Q2/cn_melt_19Q2.rds")
ge_melt <- readRDS("./data_munging/19Q2/ge_melt_19Q2.rds")
```

```{r}
test <- filter(maf_changes_df, Gene_Protein_Change %in% unique(g2p_mut$Gene_Protein_Change))
```


# MERGE OMICS

--------------------------------------------------------------------------------

```{r omics_functions}
MergeOmicsG2P <- function(database, dataframe, maf_df, ge_df, cn_df, gene_list, outfile) {
  # Merge mutation status, CN, and GE omics data with drugscreen results and metadata
  if(database %in% c("CCLE", "CTRP", "GDSC")) {
    print(paste0("Processing ", database, " data."))

    grid <- expand.grid("Cell_line" = unique(dataframe$Cell_line), "Gene" = gene_list)
    data <-  merge(dataframe, grid, by = "Cell_line", all.y = TRUE)
    print("Merged grid.")
    # Merge metadata
    data <-  merge(data, select(ccl_meta, Cell_line, CCLE_histology, CCLE_name, DepMap_ID, disease), by = c("Cell_line", "CCLE_histology", "CCLE_name", "DepMap_ID"), all.y = TRUE)
    ## Add columns for G2P drug-gene and drug-gene-lineage sorting
    data$Drug_Gene <- paste0(data$Drug, "_", data$Gene)
    data$G2P_Drug_Gene <- ifelse(data$Drug_Gene %in% g2p_dg, "Yes", "No")
    data$Drug_Gene_Lin <- paste0(data$Drug, "_", data$Gene, "_", data$disease)
    data$G2P_Drug_Gene_Lin <- ifelse(data$Drug_Gene_Lin %in% g2p_dgl, "Yes", "No")
    data$Gene_Lin <- paste0(data$Gene, "_", data$Gene, "_", data$disease)
    data$G2P_Gene_Lin <- ifelse(data$Gene_Lin %in% g2p_gl, "Yes", "No")
    print("Merged metadata.")
    # Merge mutation calls
    data <- unfactorize(data)
    data <- merge(data, maf_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
    data$Mutation <- ifelse(is.na(data$Mutation), 0, data$Mutation)
  }
  if(database == "CRISPR") {
    print(paste0("Processing ", database, " data."))
    # Melt CRISPR dataset for merging
    data <- melt(dataframe, id.vars = "DepMap_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")
    # Merge cell line metadata
    data <- merge(data, select(ccl_meta, Cell_line, CCLE_histology, CCLE_name, DepMap_ID, disease), by = "DepMap_ID", all.y = TRUE)
    print("Merged metadata.")
    # Merge mutation calls
    data <- unfactorize(data)
    data <- merge(data, maf_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
    data$Mutation <- ifelse(is.na(data$Mutation), 0, data$Mutation)
  }

  ## Summarize number of mutant and Wildtype cell lines
  data_muts_summ <- data %>% group_by(Gene) %>% summarize(N_Wildtype = length(Mutation) - sum(Mutation == 1), N_Mutant = sum(Mutation == 1))
  ## Merge test results back into full dataset, which restores information lost in the summarization
  data <- merge(data_muts_summ, data, by = "Gene")
  print("Merged mutation calls.")
  # Merge GE data
  data <- merge(data, ge_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
  print("Merged gene expression data.")
  # Merge CN data
  data <- merge(data, cn_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
  print("Merged copy number data.")
  # Drop cell lines not in DepMap
  data <- filter(data, !is.na(DepMap_ID) & !(DepMap_ID %in% c("ACH-001182", "ACH-001451")))
  # Save and return dataframe
  data <- unfactorize(data)
  saveRDS(data, outfile, compress = "xz")
  return(data)
}
```

## Drug screens

```{r adjust_drugscreen_data, eval=FALSE}
ccle_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CCLE_drug_response_metrics.csv.gz", sep = ",")
ctrp_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CTRP_drug_response_metrics.csv.gz", sep = ",")
gdsc_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/GDSC_drug_response_metrics.csv.gz", sep = ",")

auc <- bind_rows(ccle_all, ctrp_all, gdsc_all) %>% drop_na(Drug_name)
auc$Drug <- tolower(auc$Drug_name)
auc$Cell_line <- toupper(gsub("-", "", auc$Cell_line))
auc$Cell_line <- ifelse(auc$Cell_line == "K052", "KO52", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "TI73", "T173", auc$Cell_line)
auc <- merge(ccl_histologies, auc, by = "Cell_line", all.y = TRUE)
colnames(auc)[colnames(auc) == "AUC_IC50"] <- "Score"
saveRDS(auc, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_auc_all.rds", compress = "xz")

ccle_all <- auc %>% filter(Database == "CCLE") %>% select(Cell_line, CCLE_histology, TCGA_classification, CCLE_name, DepMap_ID, Drug, Score)
saveRDS(ccle_all, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds", compress = "xz")
ctrp_all <- auc %>% filter(Database == "CTRP") %>% select(Cell_line, CCLE_histology, TCGA_classification, CCLE_name, DepMap_ID, Drug, Score)
saveRDS(ctrp_all, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds", compress = "xz")
gdsc_all <- auc %>% filter(Database == "GDSC") %>% select(Cell_line, CCLE_histology, TCGA_classification, CCLE_name, DepMap_ID, Drug, Score)
saveRDS(gdsc_all, "./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds", compress = "xz")
```

`r nrow(filter(test, n != 1))`/`r nrow(test)` gene-cell line combinations have more than one mutation in the MAF file. This is corrected in the point mutation filtering done below.
```{r merge_drug_data, eval=FALSE}
ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")

ccle <- MergeOmicsG2P(database = "CCLE", dataframe = ccle_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/ccle_data_19Q2.rds")
write.csv.gz(ccle %>% drop_na(Score), "./data_munging/19Q2/ccle_data_19Q2.csv.gz")

ctrp <- MergeOmicsG2P(database = "CTRP", dataframe = ctrp_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./../htscreens_giant_files/ctrp_data_19Q2.rds")
write.csv.gz(ctrp %>% drop_na(Score), "./../htscreens_giant_files/ctrp_data_19Q2.csv.gz")


gdsc <- MergeOmicsG2P(database = "GDSC", dataframe = gdsc_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/gdsc_data_19Q2.rds")
write.csv.gz(gdsc %>% drop_na(Score), "./../htscreens_giant_files/gdsc_data_19Q2.csv.gz")

# all_genes <- intersect(unique(maf_df$Gene), intersect(unique(cn_melt$Gene), unique(ge_melt$Gene)))
# ccle <- MergeOmicsG2P(database = "CCLE", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = all_genes, outfile = "./../htscreens_giant_files/ccle_data_allgenes_19Q2.rds")
# ctrp <- MergeOmicsG2P(database = "CTRP", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = all_genes, outfile = "./../htscreens_giant_files/ctrp_data_allgenes_19Q2.rds")
# gdsc <- MergeOmicsG2P(database = "GDSC", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = all_genes, outfile = "./../htscreens_giant_files/gdsc_data_allgenes_19Q2.rds")
```

```{r load_drug_rds, eval=TRUE}
ccle <- readRDS("./data_munging/19Q2/ccle_data_19Q2.rds") %>% drop_na(Drug)
ctrp <- readRDS("./../htscreens_giant_files/ctrp_data_19Q2.rds") %>% drop_na(Drug)
gdsc <- readRDS("./data_munging/19Q2/gdsc_data_19Q2.rds") %>% drop_na(Drug)

# Testing
test_ccle <- ccle %>% count(Drug)
test_ctrp <- ctrp %>% count(Drug)
test_gdsc <- gdsc %>% count(Drug)
```

How many datasets screen each G2P drug?
```{r drug_testing, eval=FALSE}
drug_summ <- data.frame(Drug = character(), Dataset_Count = integer(), stringsAsFactors = FALSE)

for(drug in g2p_drugs) {
  counter = 0
  if(drug %in% as.character(unique(ccle$Drug[!is.na(ccle$Drug)]))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(ctrp$Drug[!is.na(ctrp$Drug)]))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(gdsc$Drug[!is.na(gdsc$Drug)]))) { counter <- counter + 1 }
  drug_summ <- rbind(drug_summ, data.frame(Drug = drug, Dataset_Count = counter))
}
```

Merge fusions:
```{r merge_fusions, eval=FALSE}
ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")
fusions_filt <- filter(fusions, FusionPair %in% unique(as.character(g2p_fus$FusionPair)))

ccle_fus_grid <- expand.grid("DepMap_ID" = unique(ccle_all$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair))
ccle_fus <- merge(merge(ccle_all, ccle_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "CCLE_name", "CCLE_histology", "Cell_line", "FusionPair"), all = TRUE)
ccle_fus$Fusion_Status <- ifelse(!is.na(ccle_fus$LeftGene), "Yes", "No")
ccle_fus$Drug_FusionPair <- ifelse(is.na(ccle_fus$Drug) | is.na(ccle_fus$FusionPair), NA, paste0(ccle_fus$Drug, "_", ccle_fus$FusionPair))
ccle_fus$G2P_Drug_FusionPair <- ifelse(ccle_fus$Drug_FusionPair %in% unique(as.character(g2p_fus$Drug_FusionPair)), "Yes", "No")
ccle_fus$G2P_Fusion <- ifelse(ccle_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
ccle_fus$Database <- "CCLE"

ctrp_fus_grid <- expand.grid("DepMap_ID" = unique(ctrp_all$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair))
ctrp_fus <- merge(merge(ctrp_all, ctrp_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "CCLE_name", "CCLE_histology", "Cell_line", "FusionPair"), all = TRUE)
ctrp_fus$Fusion_Status <- ifelse(!is.na(ctrp_fus$LeftGene), "Yes", "No")
ctrp_fus$Drug_FusionPair <- ifelse(is.na(ctrp_fus$Drug) | is.na(ctrp_fus$FusionPair), NA, paste0(ctrp_fus$Drug, "_", ctrp_fus$FusionPair))
ctrp_fus$G2P_Drug_FusionPair <- ifelse(ctrp_fus$Drug_FusionPair %in% unique(as.character(g2p_fus$Drug_FusionPair)), "Yes", "No")
ctrp_fus$G2P_Fusion <- ifelse(ctrp_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
ctrp_fus$Database <- "CTRP"

gdsc_fus_grid <- expand.grid("DepMap_ID" = unique(gdsc_all$DepMap_ID), "FusionPair" = unique(fusions_filt$FusionPair))
gdsc_fus <- merge(merge(gdsc_all, gdsc_fus_grid, by = "DepMap_ID", all = TRUE), fusions_filt, by = c("DepMap_ID", "CCLE_name", "CCLE_histology", "Cell_line", "FusionPair"), all = TRUE)
gdsc_fus$Fusion_Status <- ifelse(!is.na(gdsc_fus$LeftGene), "Yes", "No")
gdsc_fus$Drug_FusionPair <- ifelse(is.na(gdsc_fus$Drug) | is.na(gdsc_fus$FusionPair), NA, paste0(gdsc_fus$Drug, "_", gdsc_fus$FusionPair))
gdsc_fus$G2P_Drug_FusionPair <- ifelse(gdsc_fus$Drug_FusionPair %in% unique(as.character(g2p_fus$Drug_FusionPair)), "Yes", "No")
gdsc_fus$G2P_Fusion <- ifelse(gdsc_fus$FusionPair %in% unique(as.character(g2p_fus$FusionPair)), "Yes", "No")
gdsc_fus$Database <- "GDSC"

fus <- bind_rows(ccle_fus, ctrp_fus, gdsc_fus)
colnames(fus)[colnames(fus) == "AUC_IC50"] <- "Score"
saveRDS(fus, "./data_munging/rds/fusions_merged.rds")
```

```{r load_fusion_data}
fus <- readRDS("./data_munging/rds/fusions_merged.rds") %>% drop_na(Drug)
fus_summary <- fus %>% count(Database, Drug, FusionPair, Fusion_Status)
```

Merge protein changes
```{r merge_protein_changes, eval=FALSE}
ccle_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ccle_all_edited.rds")
ctrp_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_ctrp_all_edited.rds")
gdsc_all <- readRDS("./../htscreens_giant_files/pozdeyev_data/pozdeyev2016_gdsc_all_edited.rds")
maf_changes_df <- readRDS("./data_munging/19Q2/maf_changes_df_19Q2.rds")
maf_changes_filt <- filter(maf_changes_df, Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)))

# CCLE
ccle_prot_grid <- expand.grid("DepMap_ID" = unique(ccle_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
ccle_prot <- merge(ccle_all, ccle_prot_grid, by = "DepMap_ID", all = TRUE)
ccle_prot <- merge(ccle_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
ccle_prot$Change_Status <- ifelse(!is.na(ccle_prot$Protein_Change), "Yes", "No")
ccle_prot$Drug_Gene_Protein_Change <- ifelse(is.na(ccle_prot$Drug) | is.na(ccle_prot$Gene_Protein_Change), NA, paste0(ccle_prot$Drug, "_", ccle_prot$Gene_Protein_Change))
ccle_prot$G2P_Drug_Gene_Protein_Change <- ifelse(ccle_prot$Drug_Gene_Protein_Change %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change)), "Yes", "No")
ccle_prot$G2P_Protein_Change <- ifelse(ccle_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
ccle_prot <- merge(ccle_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
ccle_prot$Database <- "CCLE"
ccle_prot <- ccle_prot %>% drop_na(Drug)

# CTRP
ctrp_prot_grid <- expand.grid("DepMap_ID" = unique(ctrp_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
ctrp_prot <- merge(ctrp_all, ctrp_prot_grid, by = "DepMap_ID", all = TRUE)
ctrp_prot <- merge(ctrp_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
ctrp_prot$Change_Status <- ifelse(!is.na(ctrp_prot$Protein_Change), "Yes", "No")
ctrp_prot$Drug_Gene_Protein_Change <- ifelse(is.na(ctrp_prot$Drug) | is.na(ctrp_prot$Gene_Protein_Change), NA, paste0(ctrp_prot$Drug, "_", ctrp_prot$Gene_Protein_Change))
ctrp_prot$G2P_Drug_Gene_Protein_Change <- ifelse(ctrp_prot$Drug_Gene_Protein_Change %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change)), "Yes", "No")
ctrp_prot$G2P_Protein_Change <- ifelse(ctrp_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
ctrp_prot <- merge(ctrp_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
ctrp_prot$Database <- "CTRP"
ctrp_prot <- ctrp_prot %>% drop_na(Drug)

# GDSC
gdsc_prot_grid <- expand.grid("DepMap_ID" = unique(gdsc_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
gdsc_prot <- merge(gdsc_all, gdsc_prot_grid, by = "DepMap_ID", all = TRUE)
gdsc_prot <- merge(gdsc_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
gdsc_prot$Change_Status <- ifelse(!is.na(gdsc_prot$Protein_Change), "Yes", "No")
gdsc_prot$Drug_Gene_Protein_Change <- ifelse(is.na(gdsc_prot$Drug) | is.na(gdsc_prot$Gene_Protein_Change), NA, paste0(gdsc_prot$Drug, "_", gdsc_prot$Gene_Protein_Change))
gdsc_prot$G2P_Drug_Gene_Protein_Change <- ifelse(gdsc_prot$Drug_Gene_Protein_Change %in% unique(as.character(g2p_prot$Drug_Gene_Protein_Change)), "Yes", "No")
gdsc_prot$G2P_Protein_Change <- ifelse(gdsc_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
gdsc_prot <- merge(gdsc_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
gdsc_prot$Database <- "GDSC"
gdsc_prot <- gdsc_prot %>% drop_na(Drug)

prot <- bind_rows(ccle_prot, ctrp_prot, gdsc_prot)
prot <- separate(data = prot, col = Gene_Protein_Change, into = c("Gene", "Protein_Change"), remove = FALSE, sep = ":")
prot <- add_column(prot, Drug_Gene = paste0(prot$Drug, "_", prot$Gene), .after = "Drug")
prot$Direction_Curated <- ifelse(is.na(prot$Direction_Curated), "none", as.character(prot$Direction_Curated))
prot$Direction_Grouper <- with(prot, paste0(Change_Status, "_", Direction_Curated))
saveRDS(prot, "./data_munging/rds/protein_changes_merged.rds")
```

```{r load_prot_data}
prot <- readRDS("./data_munging/rds/protein_changes_merged.rds")
# prot_summary <- prot %>% count(Database, Drug, Gene_Protein_Change, Change_Status)
```


## CRISPR

```{r merge_crispr_data, eval=FALSE}
crispr_all <- read.delim("./data_munging/19Q2/Achilles_gene_effect_20190514.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr_all) <- gsub(" .*", "", colnames(crispr_all))
colnames(crispr_all)[1] <- "DepMap_ID"

# Melt CRISPR dataset for merging
# crispr_melt <- melt(crispr_all, id.vars = "DepMap_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")

crispr <- MergeOmicsG2P(database = "CRISPR", dataframe = crispr_all, maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/crispr_data_19Q2.rds")
write.csv.gz(crispr %>% drop_na(Score), "./../htscreens_giant_files/crispr_data_19Q2.csv.gz")
```

```{r read_crispr_data}
crispr <- readRDS("./data_munging/19Q2/crispr_data_19Q2.rds")
```

```{r, eval=FALSE}
maf_changes_df <- readRDS("./data_munging/19Q2/maf_changes_df_19Q2.rds")
maf_changes_filt <- filter(maf_changes_df, Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)))

# CCLE
crispr_prot_grid <- expand.grid("DepMap_ID" = unique(crispr_all$DepMap_ID), "Gene_Protein_Change" = unique(maf_changes_filt$Gene_Protein_Change))
crispr_prot <- merge(crispr_all, crispr_prot_grid, by = "DepMap_ID", all = TRUE)
crispr_prot <- merge(crispr_prot, maf_changes_filt, by = c("DepMap_ID", "Gene_Protein_Change"), all = TRUE)
crispr_prot$Change_Status <- ifelse(!is.na(crispr_prot$Protein_Change), "Yes", "No")
crispr_prot$G2P_Protein_Change <- ifelse(crispr_prot$Gene_Protein_Change %in% unique(as.character(g2p_prot$Gene_Protein_Change)), "Yes", "No")
crispr_prot <- merge(crispr_prot, unique(select(g2p_prot, Drug_Gene_Protein_Change, Direction_Curated)), by = "Drug_Gene_Protein_Change", all.x = TRUE)
crispr_prot$Database <- "CRISPR"

crispr_prot <- separate(data = crispr_prot, col = Gene_Protein_Change, into = c("Gene", "Protein_Change"), remove = FALSE, sep = ":")
crispr_prot <- add_column(crispr_prot, Drug_Gene = paste0("CRISPR_", crispr_prot$Gene), .after = "Gene")
crispr_prot$Direction_Curated <- ifelse(is.na(crispr_prot$Direction_Curated), "none", as.character(crispr_prot$Direction_Curated))
crispr_prot$Direction_Grouper <- with(crispr_prot, paste0(Change_Status, "_", Direction_Curated))

saveRDS(crispr_prot, "./data_munging/rds/protein_changes_merged.rds")
```


# EXPLORATORY FIGURES

--------------------------------------------------------------------------------

## Mutation

## Cell line and drug overlaps

```{r ccl_venn, eval=FALSE}
ccle_ccls <- unique(ccle$DepMap_ID)
ctrp_ccls <- unique(ctrp$DepMap_ID)
gdsc_ccls <- unique(gdsc$DepMap_ID)
crispr_ccls <- unique(crispr$DepMap_ID)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_ccls, ctrp_ccls, gdsc_ccls, crispr_ccls),
             filename = "./plots/manuscript/all_venn_ccls.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1,
             category.names = c("CCLE", "CTRP", "GDSC", "CRISPR"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             # cat.dist = c(0.06, 0.06, 0.035),
             fill = c("springgreen4", "steelblue4", "turquoise4", "darkorchid"))
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_ccls, ctrp_ccls, gdsc_ccls),
             filename = "./plots/manuscript/drug_venn_ccls.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1.4,
             category.names = c("CCLE", "CTRP", "GDSC"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             cat.dist = c(0.06, 0.06, 0.035),
             fill = c("springgreen4", "steelblue4", "turquoise4"))
```

```{r drug_venn, eval=FALSE}
ccle_drugs <- unique(ccle$Drug)
ctrp_drugs <- unique(ctrp$Drug)
gdsc_drugs <- unique(gdsc$Drug)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_drugs, ctrp_drugs, gdsc_drugs),
             filename = "./plots/manuscript/drug_venn_drugs.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1.4,
             category.names = c("CCLE", "CTRP", "GDSC"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             cat.dist = c(0.06, 0.06, 0.035),
             fill = c("palegreen3", "slategray3", "paleturquoise3"))
```

## Histology distributions

**RUN THIS WHEN CRISPR IS DONE MERGING (5/17)**
```{r histology_dists_data, eval=FALSE}
ccle_hists <- unique(select(ccle, DepMap_ID, disease)) %>% count(disease)
ccle_hists$Perc <- ccle_hists$n / sum(ccle_hists$n) * 100
ccle_hists$Dataset <- "CCLE"
ctrp_hists <- unique(select(ctrp, DepMap_ID, disease)) %>% count(disease)
ctrp_hists$Perc <- ctrp_hists$n / sum(ctrp_hists$n) * 100
ctrp_hists$Dataset <- "CTRP"
gdsc_hists <- unique(select(gdsc, DepMap_ID, disease)) %>% count(disease)
gdsc_hists$Perc <- gdsc_hists$n / sum(gdsc_hists$n) * 100
gdsc_hists$Dataset <- "GDSC"
crispr_hists <- unique(select(crispr, DepMap_ID, disease)) %>% count(disease)
crispr_hists$Perc <- crispr_hists$n / sum(crispr_hists$n) * 100
crispr_hists$Dataset <- "CRISPR"

drug_hists <- rbind(ccle_hists, ctrp_hists, gdsc_hists, crispr_hists) %>% drop_na(disease)
drug_hists$Dataset <- factor(drug_hists$Dataset, levels = c("CCLE", "CTRP", "GDSC", "CRISPR"))
drug_hists$disease <- as.character(drug_hists$disease)
drug_hists$disease <- gsub(" cancer", "", drug_hists$disease)
saveRDS(drug_hists, "./data_munging/rds/histology_distributions.rds")
```

```{r histology_dists, fig.height=4.5, fig.width=4, eval=FALSE}
drug_hists <- readRDS("./data_munging/rds/histology_distributions.rds")
png(filename = "./plots/manuscript/all_histology_distributions.png", width = 8, height = 9, units = "in", res = 500)
ggplot(drug_hists, aes(x = Dataset, y = n, fill = disease)) +
  geom_bar(stat = "identity", color = "black") +
  scale_y_continuous(breaks = seq(0, 850, by = 50), labels = seq(0, 850, by = 50)) +
  guides(fill = guide_legend(title = "Histology", ncol = 1)) +
  labs(y = "Count")
dev.off()
```

**RUN THIS WHEN CRISPR IS DONE MERGING (5/17)**
```{r, eval=FALSE}
ccle_ccls <- unique(ccle$DepMap_ID)
ctrp_ccls <- unique(ctrp$DepMap_ID)
gdsc_ccls <- unique(gdsc$DepMap_ID)
crispr_ccls <- unique(crispr$DepMap_ID)

ccls <- data.frame(DepMap_ID = union(ccle_ccls, union(ctrp_ccls, gdsc_ccls)))
all_ccls <- merge(ccls, ccl_meta, by = "DepMap_ID")
all_ccls$disease <- gsub(" cancer", "", all_ccls$disease)

png(filename = "./plots/manuscript/drug_histology_distributions.png", width = 8, height = 4, units = "in", res = 500)
ggplot(all_ccls, aes(x = fct_infreq(factor(disease)), fill = disease)) +
  geom_bar(color = "black") +
  scale_y_continuous(breaks = seq(0, 220, by = 40), labels = seq(0, 220, by = 40)) +
  guides(fill = FALSE) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
  labs(y = "Number of Cell Lines", x = "Histology", caption = paste0(formatC(nrow(all_ccls), format = "d", big.mark = ","), " cell lines represented."))
dev.off()
```

## G2P summary heatmap

```{r dg_heatmap, fig.height=6, fig.width=8, warning=FALSE}
g2p_dg_counts <- g2p %>% count(Drug, Gene) %>% filter(!is.na(Drug), !is.na(Gene))
g2p_dg_counts <- dcast(g2p_dg_counts, Drug ~ Gene, value.var = "n", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
g2p_dg_counts <- g2p_dg_counts[order(rowSums(-g2p_dg_counts)), order(colSums(-g2p_dg_counts))]
# g2p_dg_counts[g2p_dg_counts == 0] <- NA

g2p_gene_order <- colnames(g2p_dg_counts)
g2p_drug_order <- rownames(g2p_dg_counts)

# Create annotation layer
hta_data <- data.frame(Drug = rownames(g2p_dg_counts), CCLE = NA, CTRP = NA, GDSC = NA)
hta_data$CCLE <- ifelse(hta_data$Drug %in% unique(ccle$Drug), "Screened", "Not screened")
hta_data$CTRP <- ifelse(hta_data$Drug %in% unique(ctrp$Drug), "Screened", "Not screened")
hta_data$GDSC <- ifelse(hta_data$Drug %in% unique(gdsc$Drug), "Screened", "Not screened")
rownames(hta_data) <- hta_data$Drug
hta_data$Drug <- NULL

hta_annot <- HeatmapAnnotation(text = anno_text(colnames(hta_data), offset = unit(0.1, "in"), gp = gpar(fontsize = 10)))
hta_legend_list <- list(legend_direction = "horizontal", border = "gray70", nrow = 2)
hta <- Heatmap(hta_data,
               name = "Functional screen status           ",
               col = c("Screened" = "purple4", "Not screened" = "gray97"), 
               rect_gp = gpar(col = "gray70"),
               top_annotation = hta_annot,
               heatmap_legend_param = hta_legend_list,
               row_names_side = "left",
               row_names_gp = gpar(fontsize = 10),
               column_names_side = "top",
               column_names_gp = gpar(fontsize = 10),
               show_column_names = FALSE,
               cluster_rows = FALSE, 
               cluster_columns = FALSE,
               row_title = "Drugs",
               row_title_gp = gpar(fontface = "bold", fontsize = 14),
               column_title = "Dataset",
               column_title_gp = gpar(fontface = "bold", fontsize = 14),
               width = unit(1.5, "in"))

# Define color scale
col_fun <- colorRamp2(c(0, 1, 100), c("gray97", "mistyrose", "purple4"))
# Legend properties list
ht_legend_list <- list(legend_direction = "horizontal", legend_width = unit(2.5, "in"), border = "gray70")
ht <- Heatmap(g2p_dg_counts, 
              name = "Number of G2P interpretations",
              col = col_fun,
              na_col = "gray97",
              rect_gp = gpar(col = "gray70"),
              heatmap_legend_param = ht_legend_list,
              cluster_rows = FALSE, 
              cluster_columns = FALSE, 
              show_row_names = FALSE,
              row_names_side = "left",
              row_names_gp = gpar(fontsize = 10),
              column_names_side = "bottom",
              show_column_names = FALSE,
              bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(g2p_dg_counts),
                                   rot = 45,
                                   gp = gpar(fontsize = 8),
                                   offset = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.45, "in")),
              row_title = "Drugs",
              row_title_gp = gpar(fontface = "bold", fontsize = 14),
              column_title = "Genes", 
              column_title_gp = gpar(fontface = "bold", fontsize = 14),
              column_title_side = "bottom")

png(filename = "./plots/manuscript/g2p_summary_heatmap.png", width = 10, height = 9, units = "in", res = 500)
draw(hta + ht, gap = unit(5, "mm"), heatmap_legend_side = "top")
dev.off()
```

## Clustered AUC heatmap

```{r auc_clustered_ht, eval=FALSE}
ccle_auc_grid <- dcast(ccle, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
ctrp_auc_grid <- dcast(ctrp, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
gdsc_auc_grid <- dcast(gdsc, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)

ccle_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(ccle_auc_grid)), select(ccl_meta, DepMap_ID, disease), by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)
ctrp_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(ctrp_auc_grid)), select(ccl_meta, DepMap_ID, disease), by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)
gdsc_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(gdsc_auc_grid)), select(ccl_meta, DepMap_ID, disease), by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)

ht_legend_list <- list(direction = "horizontal")

ccle_col_fun <- colorRamp2(c(0, 1), c("gray97", "darkgreen"))
ccle_auc_ha <- HeatmapAnnotation(Lineage = ccle_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 5)))
ccle_auc_ht <- Heatmap(ccle_auc_grid,
                       na_col = "black",
                       col = ccle_col_fun,
                       top_annotation = ccle_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

ctrp_col_fun <- colorRamp2(c(0, 1), c("gray97", "steelblue4"))
ctrp_auc_ha <- HeatmapAnnotation(Lineage = ctrp_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 3)))
ctrp_auc_ht <- Heatmap(ctrp_auc_grid,
                       na_col = "black",
                       col = ctrp_col_fun,
                       top_annotation = ctrp_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

gdsc_col_fun <- colorRamp2(c(0, 1), c("gray97", "turquoise4"))
gdsc_auc_ha <- HeatmapAnnotation(Lineage = gdsc_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 3)))
gdsc_auc_ht <- Heatmap(gdsc_auc_grid,
                       na_col = "black",
                       col = gdsc_col_fun,
                       top_annotation = gdsc_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

png(filename = "./plots/manuscript/ccle_clustered_heatmap.png", width = 10, height = 8, units = "in", res = 500)
draw(ccle_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()

png(filename = "./plots/manuscript/ctrp_clustered_heatmap.png", width = 22, height = 40, units = "in", res = 500)
draw(ctrp_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()

png(filename = "./plots/manuscript/gdsc_clustered_heatmap.png", width = 18, height = 20, units = "in", res = 500)
draw(gdsc_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()
```


# WILCOXON: PanCan

--------------------------------------------------------------------------------

ggpubr's convention for symbols indicating statistical significance:

- ns: p > 0.05
- *: p <= 0.05
- **: p <= 0.01
- ***: p <= 0.001
- ****: p <= 0.0001

```{r wilcoxon_functions}
adj_signif <- function(df, alpha) {
  # Takes df from compare_means and creates a signif code column for FDR-adjusted p-vals
  df$p.signif.adj <- ifelse(df$p.adj <= alpha, "*", NA)
  df$p.signif <- ifelse(df$p.signif == "ns", NA, df$p.signif)
  df$p.short <- formatC(df$p, format = "g", digits =  2)
  df$p.adj.short <- formatC(df$p.adj, format = "g", digits =  2)
  return(df)
}

WilcoxonTestDF <- function(dataframe, grouping, g2p_list, type, outfile) {
  require(ggpubr)
  if(tolower(type) == "drug_gene_lineage") {
    # Filter out DGLs with too few samples in at least one mutation status category
    # Require at least 3 samples in each category to run the test, else filter out that DGL from testing df
    count_mut_n <- dataframe %>% count(Drug_Gene_Lin, Mutation) %>% filter(n >= 3)
    count_mut_category <- count_mut_n %>% count(Drug_Gene_Lin) %>% filter(n == 2)
    dgl_keep <- intersect(count_mut_n$Drug_Gene_Lin, count_mut_category$Drug_Gene_Lin)
    dataframe <- filter(dataframe, Drug_Gene_Lin %in% dgl_keep)
    dataframe$Drug_Gene_Lin <- as.character(dataframe$Drug_Gene_Lin)
    signif <- compare_means(Score ~ Mutation, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif$InG2P <- ifelse(signif$Drug_Gene_Lin %in% g2p_list, "Yes", "No")
  }
  if(tolower(type) == "fusion") {
    # Filter out fusions with too few samples in at least one category
    # Require at least 3 samples in each category to run the test, else filter out that Drug_FusionPair from testing df
    count_n <- dataframe %>% count(Drug, FusionPair, Drug_FusionPair, Fusion_Status) %>% filter(n >= 3)
    count_category <- count_n %>% count(Drug, FusionPair, Drug_FusionPair) %>% filter(n == 2)
    keep <- intersect(count_n$Drug_FusionPair, count_category$Drug_FusionPair)
    dataframe <- filter(dataframe, Drug_FusionPair %in% keep)
    dataframe$Drug_FusionPair <- as.character(dataframe$Drug_FusionPair)
    signif <- compare_means(Score ~ Fusion_Status, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif$InG2P <- ifelse(signif$Drug_FusionPair %in% g2p_list, "Yes", "No")
  }
  if(tolower(type) == "protein_change") {
    # Filter out Gene:Protein_Changes with too few samples in at least one category
    # Require at least 3 samples in each category to run the test, else filter out that Gene_Protein_Change from testing df
    count_n <- dataframe %>% count(Drug, Gene_Protein_Change, Drug_Gene_Protein_Change, Change_Status) %>% filter(n >= 3)
    count_category <- count_n %>% count(Drug, Gene_Protein_Change, Drug_Gene_Protein_Change) %>% filter(n == 2)
    keep <- intersect(count_n$Drug_Gene_Protein_Change, count_category$Drug_Gene_Protein_Change)
    dataframe <- filter(dataframe, Drug_Gene_Protein_Change %in% keep)
    dataframe$Drug_Gene_Protein_Change <- as.character(dataframe$Drug_Gene_Protein_Change)
    signif <- compare_means(Score ~ Change_Status, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif$InG2P <- ifelse(signif$Drug_Gene_Protein_Change %in% g2p_list, "Yes", "No")
  }
  if(tolower(type) == "direction") {
    signif <- compare_means(Score ~ Direction_Grouper, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    # signif$InG2P <- ifelse(signif$Drug_Gene_Protein_Change %in% g2p_list, "Yes", "No")
  }
  if(tolower(type) == "drug_gene") {
    # Filter out DGs with too few samples in at least one mutation status category
    # Require at least 3 samples in each category to run the test, else filter out that DG from testing df
    count_mut_n <- dataframe %>% count(Drug_Gene, Mutation) %>% filter(n >= 3)
    count_mut_category <- count_mut_n %>% count(Drug_Gene) %>% filter(n == 2)
    dg_keep <- intersect(count_mut_n$Drug_Gene, count_mut_category$Drug_Gene)
    dataframe <- filter(dataframe, Drug_Gene %in% dg_keep)
    dataframe$Drug_Gene <- as.character(dataframe$Drug_Gene)
    signif <- compare_means(Score ~ Mutation, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
    signif <- adj_signif(signif, alpha = 0.05)
    signif <- signif[order(signif$p),]
    signif$InG2P <- ifelse(signif$Drug_Gene %in% g2p_list, "Yes", "No")
  }
  signif$p_log10 <- as.numeric(formatC(-log10(signif$p), digits = 3, format = "f"))
  saveRDS(signif, outfile)
  return(signif)
}
```

## Mutation status

```{r mut_wilcoxon_tests, eval=FALSE}
# CRISPR
crispr <- unfactorize(crispr)
# crispr_signif_all_gene <- WilcoxonTestDF(dataframe = crispr, grouping = "Gene", outfile = "./data_munging/rds/crispr_signif_g.rds")
# crispr_filt <- filter(crispr, Gene %in% g2p_genes)
# crispr_signif_gene <- WilcoxonTestDF(dataframe = crispr_filt, grouping = "Gene", type = "drug_gene", outfile = "./data_munging/rds/crispr_signif_g2p_g.rds")
# crispr_signif_lineage <- WilcoxonTestDF(dataframe = crispr_filt, grouping = c("Gene", "disease"), type = "drug_gene_lineage", outfile = "./data_munging/rds/crispr_signif_g2p_gl.rds")

# CCLE
ccle_filt <- filter(ccle, Drug %in% g2p_drugs & Gene %in% g2p_dg_genes)
ccle_filt$Gene <- factor(ccle_filt$Gene, ordered = FALSE)
ccle_filt <- unfactorize(ccle_filt)
ccle_signif_dg <- WilcoxonTestDF(dataframe = ccle, grouping = c("Drug", "Gene", "Drug_Gene"), g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), type = "drug_gene", outfile = "./data_munging/rds/ccle_signif_dg.rds")
ccle_signif_g2p_dg <- WilcoxonTestDF(dataframe = ccle_filt, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/ccle_signif_g2p_dg.rds")
ccle_signif_g2p_dgl <- WilcoxonTestDF(dataframe = ccle_filt, grouping = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", g2p_list = unique(as.character(g2p_mut$Drug_Gene_Lin[!is.na(g2p_mut$Drug_Gene_Lin)])), outfile = "./data_munging/rds/ccle_signif_g2p_dgl.rds")

# CTRP
ctrp_filt <- filter(ctrp, Drug %in% g2p_drugs & Gene %in% g2p_dg_genes)
ctrp_filt$Gene <- factor(ctrp_filt$Gene, ordered = FALSE)
ctrp_filt <- unfactorize(ctrp_filt)
# ctrp_signif_dg <- WilcoxonTestDF(dataframe = ctrp, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/ctrp_signif_dg.rds")
ctrp_signif_g2p_dg <- WilcoxonTestDF(dataframe = ctrp_filt, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/ctrp_signif_g2p_dg.rds")
ctrp_signif_g2p_dgl <- WilcoxonTestDF(dataframe = ctrp_filt, grouping = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", g2p_list = unique(as.character(g2p_mut$Drug_Gene_Lin[!is.na(g2p_mut$Drug_Gene_Lin)])), outfile = "./data_munging/rds/ctrp_signif_g2p_dgl.rds")

# GDSC
gdsc_filt <- filter(gdsc, Drug %in% g2p_drugs & Gene %in% g2p_dg_genes)
gdsc_filt$Gene <- factor(gdsc_filt$Gene, ordered = FALSE)
gdsc_filt <- unfactorize(gdsc_filt)
# gdsc_signif_dg <- WilcoxonTestDF(dataframe = gdsc, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/gdsc_signif_dg.rds")
gdsc_signif_g2p_dg <- WilcoxonTestDF(dataframe = gdsc_filt, grouping = c("Drug", "Gene", "Drug_Gene"), type = "drug_gene", g2p_list = unique(as.character(g2p_mut$Drug_Gene[!is.na(g2p_mut$Drug_Gene)])), outfile = "./data_munging/rds/gdsc_signif_g2p_dg.rds")
gdsc_signif_g2p_dgl <- WilcoxonTestDF(dataframe = gdsc_filt, grouping = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), type = "drug_gene_lineage", g2p_list = unique(as.character(g2p_mut$Drug_Gene_Lin[!is.na(g2p_mut$Drug_Gene_Lin)])), outfile = "./data_munging/rds/gdsc_signif_g2p_dgl.rds")
```

```{r load_mut_wilcoxon_rds}
g2p_mut_4merge <- unique(select(g2p_mut, Drug_Gene, Direction_Curated))
test <- g2p_mut_4merge %>% filter(Direction_Curated != "none") %>% count(Drug_Gene) %>% filter(n == 2)
extended_mut_ht_dg <- unique(as.character(test$Drug_Gene))

crispr_signif_g <- readRDS("./data_munging/rds/crispr_signif_g.rds")
crispr_signif_g2p_g <- readRDS("./data_munging/rds/crispr_signif_g2p_g.rds")
crispr_signif_lin <- readRDS("./data_munging/rds/crispr_signif_g2p_gl.rds")

ccle_signif_dg <- readRDS("./data_munging/rds/ccle_signif_dg.rds")
ccle_signif_g2p_dg <- readRDS("./data_munging/rds/ccle_signif_g2p_dg.rds")
ccle_signif_dgl <- readRDS("./data_munging/rds/ccle_signif_g2p_dgl.rds")

# ctrp_signif_g <- readRDS("./data_munging/rds/ctrp_signif_dg.rds")
ctrp_signif_g2p_dg <- readRDS("./data_munging/rds/ctrp_signif_g2p_dg.rds")
ctrp_signif_dgl <- readRDS("./data_munging/rds/ctrp_signif_g2p_dgl.rds")

# gdsc_signif_g <- readRDS("./data_munging/rds/gdsc_signif_dg.rds")
gdsc_signif_g2p_dg <- readRDS("./data_munging/rds/gdsc_signif_g2p_dg.rds")
gdsc_signif_dgl <- readRDS("./data_munging/rds/gdsc_signif_g2p_dgl.rds")
```

## Protein changes

```{r protein_change_data}
ccle_prot <- filter(prot, Database == "CCLE")
ccle_prot <- unfactorize(ccle_prot)
# ccle_prot_count <- ccle_prot %>% count(Gene_Protein_Change, Drug, Change_Status)

ctrp_prot <- filter(prot, Database == "CTRP")
ctrp_prot <- unfactorize(ctrp_prot)
# ctrp_prot_count <- ctrp_prot %>% count(Gene_Protein_Change, Drug, Change_Status)

gdsc_prot <- filter(prot, Drug %in% g2p_drugs & Database == "GDSC")
gdsc_prot <- unfactorize(gdsc_prot)
# gdsc_prot_count <- gdsc_prot %>% count(Gene_Protein_Change, Drug, Change_Status)
```

### Sen vs. Res tests

```{r prot_SvR_wilcoxon_tests, eval=FALSE}
# CCLE
ccle_prot_dir <- unique(select(ccle_prot, DepMap_ID, Gene, Cell_line, CCLE_histology, TCGA_classification, CCLE_name, Drug, Drug_Gene, Score, Change_Status, Direction_Curated, Direction_Grouper))
## Resistance
ccle_prot_res <- filter(ccle_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_resistance", "Yes_resistance"))
ccle_prot_res$Direction_Grouper <- with(ccle_prot_res, ifelse(Direction_Grouper == "Yes_resistance", "Mutated", "Not mutated"))
ccle_signif_prot_res <- WilcoxonTestDF(dataframe = ccle_prot_res, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ccle_signif_protein_changes_resistance.rds")
## Sensitivity
ccle_prot_sen <- filter(ccle_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_sensitivity", "Yes_sensitivity"))
ccle_prot_sen$Direction_Grouper <- with(ccle_prot_sen, ifelse(Direction_Grouper == "Yes_sensitivity", "Mutated", "Not mutated"))
ccle_signif_prot_sen <- WilcoxonTestDF(dataframe = ccle_prot_sen, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ccle_signif_protein_changes_sensitivity.rds")

# CTRP
ctrp_prot_dir <- unique(select(ctrp_prot, DepMap_ID, Gene, Cell_line, CCLE_histology, TCGA_classification, CCLE_name, Drug, Drug_Gene, Score, Change_Status, Direction_Curated, Direction_Grouper))
## Resistance
ctrp_prot_res <- filter(ctrp_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_resistance", "Yes_resistance"))
ctrp_prot_res$Direction_Grouper <- with(ctrp_prot_res, ifelse(Direction_Grouper == "Yes_resistance", "Mutated", "Not mutated"))
ctrp_signif_prot_res <- WilcoxonTestDF(dataframe = ctrp_prot_res, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ctrp_signif_protein_changes_resistance.rds")
## Sensitivity
ctrp_prot_sen <- filter(ctrp_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_sensitivity", "Yes_sensitivity"))
ctrp_prot_sen$Direction_Grouper <- with(ctrp_prot_sen, ifelse(Direction_Grouper == "Yes_sensitivity", "Mutated", "Not mutated"))
ctrp_signif_prot_sen <- WilcoxonTestDF(dataframe = ctrp_prot_sen, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/ctrp_signif_protein_changes_sensitivity.rds")

# GDSC
gdsc_prot_dir <- unique(select(gdsc_prot, DepMap_ID, Gene, Cell_line, CCLE_histology, TCGA_classification, CCLE_name, Drug, Drug_Gene, Score, Change_Status, Direction_Curated, Direction_Grouper))
## Resistance
gdsc_prot_res <- filter(gdsc_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_resistance", "Yes_resistance"))
gdsc_prot_res$Direction_Grouper <- with(gdsc_prot_res, ifelse(Direction_Grouper == "Yes_resistance", "Mutated", "Not mutated"))
gdsc_signif_prot_res <- WilcoxonTestDF(dataframe = gdsc_prot_res, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/gdsc_signif_protein_changes_resistance.rds")
## Sensitivity
gdsc_prot_sen <- filter(gdsc_prot_dir, Direction_Grouper %in% c("No_none", "Yes_none", "No_sensitivity", "Yes_sensitivity"))
gdsc_prot_sen$Direction_Grouper <- with(gdsc_prot_sen, ifelse(Direction_Grouper == "Yes_sensitivity", "Mutated", "Not mutated"))
gdsc_signif_prot_sen <- WilcoxonTestDF(dataframe = gdsc_prot_sen, grouping = c("Drug", "Gene", "Drug_Gene"), type = "direction", outfile = "./data_munging/rds/gdsc_signif_protein_changes_sensitivity.rds")
```

```{r load_prot_SvR_wilcoxon_tests}
ccle_signif_prot_res <- readRDS("./data_munging/rds/ccle_signif_protein_changes_resistance.rds")
ccle_signif_prot_res$Direction_Test <- "Resistance"
ccle_signif_prot_sen <- readRDS("./data_munging/rds/ccle_signif_protein_changes_sensitivity.rds")
ccle_signif_prot_sen$Direction_Test <- "Sensitivity"
ccle_signif_prot_dir <- bind_rows(ccle_signif_prot_res, ccle_signif_prot_sen)

ctrp_signif_prot_res <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_resistance.rds")
ctrp_signif_prot_res$Direction_Test <- "Resistance"
ctrp_signif_prot_sen <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_sensitivity.rds")
ctrp_signif_prot_sen$Direction_Test <- "Sensitivity"
ctrp_signif_prot_dir <- bind_rows(ctrp_signif_prot_res, ctrp_signif_prot_sen)

gdsc_signif_prot_res <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_resistance.rds")
gdsc_signif_prot_res$Direction_Test <- "Resistance"
gdsc_signif_prot_sen <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_sensitivity.rds")
gdsc_signif_prot_sen$Direction_Test <- "Sensitivity"
gdsc_signif_prot_dir <- bind_rows(gdsc_signif_prot_res, gdsc_signif_prot_sen)
```

### Exact match tests

```{r prot_wilcoxon_tests, eval=FALSE}
ccle_prot <- unfactorize(ccle_prot)
ccle_signif_prot <- WilcoxonTestDF(dataframe = ccle_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change[!is.na(g2p_prot$Drug_Gene_Protein_Change)])), type = "protein_change", outfile = "./data_munging/rds/ccle_signif_protein_changes.rds")

ctrp_prot <- unfactorize(ctrp_prot)
ctrp_signif_prot <- WilcoxonTestDF(dataframe = ctrp_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change[!is.na(g2p_prot$Drug_Gene_Protein_Change)])), type = "protein_change", outfile = "./data_munging/rds/ctrp_signif_protein_changes.rds")

gdsc_prot <- unfactorize(gdsc_prot)
gdsc_signif_prot <- WilcoxonTestDF(dataframe = gdsc_prot, grouping = c("Gene_Protein_Change", "Drug", "Drug_Gene_Protein_Change"), g2p_list = unique(as.character(g2p_prot$Drug_Gene_Protein_Change[!is.na(g2p_prot$Drug_Gene_Protein_Change)])), type = "protein_change", outfile = "./data_munging/rds/gdsc_signif_protein_changes.rds")
```

```{r load_prot_wilcoxon_rds, eval=FALSE}
ccle_signif_prot <- readRDS("./data_munging/rds/ccle_signif_protein_changes.rds")
ctrp_signif_prot <- readRDS("./data_munging/rds/ctrp_signif_protein_changes.rds")
gdsc_signif_prot <- readRDS("./data_munging/rds/gdsc_signif_protein_changes.rds")

gene_prots <- union(unique(ccle_signif_prot$Gene_Protein_Change), union(unique(ctrp_signif_prot$Gene_Protein_Change), unique(gdsc_signif_prot$Gene_Protein_Change)))

ccle_signif_prot_4plot <- merge(merge(ccle_signif_prot, unique(select(ccle_prot, Drug, Gene_Protein_Change, Direction_Curated)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(ccle_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE)
ctrp_signif_prot_4plot <- merge(merge(ctrp_signif_prot, unique(select(ctrp_prot, Drug, Gene_Protein_Change, Direction_Curated)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(ctrp_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE)
gdsc_signif_prot_4plot <- merge(merge(gdsc_signif_prot, unique(select(gdsc_prot, Drug, Gene_Protein_Change, Direction_Curated)), by = c("Drug", "Gene_Protein_Change"), all.x = TRUE), expand.grid("Drug" = unique(gdsc_signif_prot$Drug), "Gene_Protein_Change" = gene_prots), by = c("Drug", "Gene_Protein_Change"), all = TRUE)

saveRDS(ccle_signif_prot_4plot, "./data_munging/rds/ccle_signif_protein_changes_4plot.rds")
saveRDS(ctrp_signif_prot_4plot, "./data_munging/rds/ctrp_signif_protein_changes_4plot.rds")
saveRDS(gdsc_signif_prot_4plot, "./data_munging/rds/gdsc_signif_protein_changes_4plot.rds")
```

```{r}
ccle_signif_prot_4plot <- readRDS("./data_munging/rds/ccle_signif_protein_changes_4plot.rds")
ctrp_signif_prot_4plot <- readRDS("./data_munging/rds/ctrp_signif_protein_changes_4plot.rds")
gdsc_signif_prot_4plot <- readRDS("./data_munging/rds/gdsc_signif_protein_changes_4plot.rds")
```

## Fusions

```{r fusion_data}
ccle_fus <- filter(fus, Database == "CCLE")
ccle_fus <- unfactorize(ccle_fus)

ctrp_fus <- filter(fus, Database == "CTRP")
ctrp_fus <- unfactorize(ctrp_fus)

gdsc_fus <- filter(fus, Drug %in% g2p_drugs & Database == "GDSC")
gdsc_fus <- unfactorize(gdsc_fus)
```

```{r fus_wilcoxon_tests, eval=FALSE}
ccle_signif_fus <- WilcoxonTestDF(dataframe = ccle_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), g2p_list = unique(as.character(g2p_fus$Drug_FusionPair[!is.na(g2p_fus$Drug_FusionPair)])), type = "fusion", outfile = "./data_munging/rds/ccle_signif_fusions.rds")

ctrp_signif_fus <- WilcoxonTestDF(dataframe = ctrp_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), g2p_list = unique(as.character(g2p_fus$Drug_FusionPair[!is.na(g2p_fus$Drug_FusionPair)])), type = "fusion", outfile = "./data_munging/rds/ctrp_signif_fusions.rds")

gdsc_signif_fus <- WilcoxonTestDF(dataframe = gdsc_fus, grouping = c("FusionPair", "Drug", "Drug_FusionPair"), g2p_list = unique(as.character(g2p_fus$Drug_FusionPair[!is.na(g2p_fus$Drug_FusionPair)])), type = "fusion", outfile = "./data_munging/rds/gdsc_signif_fusions.rds")
```

```{r load_fus_wilcoxon_rds}
g2p_fus_4merge <- unique(select(g2p_fus, Drug_FusionPair, Direction_Curated))

ccle_signif_fus <- readRDS("./data_munging/rds/ccle_signif_fusions.rds")
# ccle_signif_fus <- merge(ccle_signif_fus, g2p_fus_4merge, by = "Drug_FusionPair", all.x = TRUE)

ctrp_signif_fus <- readRDS("./data_munging/rds/ctrp_signif_fusions.rds")
# ctrp_signif_fus <- merge(ctrp_signif_fus, g2p_fus_4merge, by = "Drug_FusionPair", all.x = TRUE)

gdsc_signif_fus <- readRDS("./data_munging/rds/gdsc_signif_fusions.rds")
# gdsc_signif_fus <- merge(gdsc_signif_fus, g2p_fus_4merge, by = "Drug_FusionPair", all.x = TRUE)
```

## Boxplots

```{r crispr_mut_boxplots, fig.height=5, fig.width=10}
crispr_filt <- filter(crispr, Gene %in% g2p_mut_genes)
crispr_filt$Gene <- factor(crispr_filt$Gene, levels = g2p_mut_genes)
crispr_filt$Mutation <- as.character(crispr_filt$Mutation)
crispr_signif_filt <- filter(crispr_signif_g2p_g, Gene %in% g2p_mut_genes)
crispr_signif_filt$Gene <- factor(crispr_signif_filt$Gene, levels = g2p_mut_genes)

crispr_gene_mut_plot <- ggplot(data = crispr_filt) +
  facet_wrap(~ Gene, nrow = 2, drop = FALSE) +
  scale_color_manual(name = "Mutation Status", labels = c("Wildtype", "Mutant"), values = c("1" = "darkorchid", "0" = "thistle")) +
  scale_x_discrete(name = "Mutation Status", labels = c("0" = "WT", "1" = "Mut")) +
  geom_jitter(aes(x = Mutation, y = Score, color = Mutation), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Mutation, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  geom_text(data = crispr_signif_filt, aes(x = 1.5, y = 2, label = p.signif)) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  theme(legend.position = "top", legend.direction = "horizontal") +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "CERES Score")
crispr_gene_mut_plot

# ggsave("./plots/manuscript/crispr_gene_mut_plot.png", crispr_gene_mut_plot, width = 12, height = 6, units = "in")
```

```{r fusion_boxplots, fig.height=5, fig.width=12, eval=FALSE}
# CCLE
ccle_fus_filt <- filter(ccle_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1") & Drug %in% g2p_drugs)
ccle_fus_filt$Drug_f <- factor(ccle_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ccle_fus$Drug)))
ccle_signif_fus_filt <- filter(ccle_signif_fus, Drug %in% g2p_drugs)
ccle_signif_fus_filt$Drug_f <- factor(ccle_signif_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ccle_signif_fus$Drug)))

ccle_fus_boxplot <- ggplot(data = ccle_fus_filt) +
  facet_grid(FusionPair ~ Drug_f) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "springgreen4", "No" = "palegreen3")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  coord_cartesian(ylim = c(0, 1.2)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = ccle_signif_fus_filt, aes(x = 1.5, y = 1.1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "AUC", x = "Fusion Status")
ccle_fus_boxplot
# ggsave("./plots/manuscript/ccle_fus_boxplots.png", ccle_fus_boxplot, width = 8, height = 5, units = "in")

# CTRP
ctrp_fus_filt <- filter(ctrp_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1") & Drug %in% g2p_drugs)
ctrp_fus_filt$Drug_f <- factor(ctrp_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ctrp_fus$Drug)))
ctrp_signif_fus_filt <- filter(ctrp_signif_fus, Drug %in% g2p_drugs)
ctrp_signif_fus_filt$Drug_f <- factor(ctrp_signif_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(ctrp_signif_fus$Drug)))

ctrp_fus_boxplot <- ggplot(data = ctrp_fus_filt) +
  facet_grid(FusionPair ~ Drug_f) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "steelblue4", "No" = "slategray3")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  coord_cartesian(ylim = c(0, 1.2)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = ctrp_signif_fus_filt, aes(x = 1.5, y = 1.1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "AUC", x = "Fusion Status")
ctrp_fus_boxplot
# ggsave("./plots/manuscript/ctrp_fus_boxplots.png", ctrp_fus_boxplot, width = 19, height = 5, units = "in")

# GDSC
gdsc_fus_filt <- filter(gdsc_fus, FusionPair %in% c("ABL1-BCR", "BCR-ABL1") & Drug %in% g2p_drugs)
gdsc_fus_filt$Drug_f <- factor(gdsc_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(gdsc_fus$Drug)))
gdsc_signif_fus_filt <- filter(gdsc_signif_fus, Drug %in% g2p_drugs)
gdsc_signif_fus_filt$Drug_f <- factor(gdsc_signif_fus_filt$Drug, levels = intersect(g2p_drug_order, unique(gdsc_signif_fus$Drug)))

gdsc_fus_boxplot <- ggplot(data = gdsc_fus_filt) +
  facet_grid(FusionPair ~ Drug_f) +
  scale_color_manual(name = "Fusion Status", values = c("Yes" = "turquoise4", "No" = "paleturquoise3")) +
  geom_jitter(aes(x = Fusion_Status, y = Score, color = Fusion_Status), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_boxplot(aes(x = Fusion_Status, y = Score), fill = "transparent", outlier.shape = NA, lwd = 0.3) +
  coord_cartesian(ylim = c(0, 1.2)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  geom_text(data = gdsc_signif_fus_filt, aes(x = 1.5, y = 1.1, label = p.signif)) +
  theme(legend.position = "top", legend.direction = "horizontal", axis.text.x = element_text(angle = 45, hjust = 0.8, vjust = 1)) +
  guides(color = guide_legend(override.aes = list(size = 3))) +
  labs(y = "AUC", x = "Fusion Status")
gdsc_fus_boxplot
# ggsave("./plots/manuscript/gdsc_fus_boxplots.png", gdsc_fus_boxplot, width = 12, height = 5, units = "in")
```

```{r direction_boxplots}
# CCLE
ggplot(data = filter(ccle_prot_res, Drug_Gene %in% ccle_signif_prot_res$Drug_Gene)) +
  facet_wrap(~ Drug_Gene, nrow = 1) +
  scale_color_manual(values = c("Mutated" = "springgreen4", "Not mutated" = "palegreen3")) +
  geom_boxplot(mapping = aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), outlier.shape = NA) +
  geom_jitter(aes(x = Direction_Grouper, y = Score, color = Direction_Grouper), width = 0.2, alpha = 0.5, size = 0.6) +
  geom_text(data = ccle_signif_prot_res, aes(x = 1.5, y = 1.0, label = p.signif)) +
  coord_cartesian(ylim = c(0, 1.1)) +
  scale_y_continuous(breaks = seq(0, 1, by = 0.2), labels = seq(0, 1, by = 0.2)) +
  theme(legend.position = "none") +
  labs(y = "AUC", x = "Resistance mutation in associated gene?")
```

## Heatmaps

```{r wilcoxon_heatmap_function}
MakeWilcoxonHeatmap <- function(signif_df, feature, row_names, col_title, color, legend_title, legend_breaks, row_title_txt, radius) {
  if(!is.na(str_match(string = tolower(feature), pattern = "mut"))) {
    signif_grid <- dcast(signif_df, Gene ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid <- signif_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid <- signif_grid[complete.cases(signif_grid), ]
    signif_grid[signif_grid < -log10(0.05)] <- NA
    
    signif_grid_annot <- dcast(signif_df, Gene ~ Drug, value.var = "InG2P", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid_annot <- signif_grid_annot[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid_annot <- signif_grid_annot[complete.cases(signif_grid_annot), ]
    signif_grid_annot[signif_grid_annot == "No"] <- ""
    signif_grid_annot[signif_grid_annot == "Yes"] <- "O"
  }
  if(!is.na(str_match(string = tolower(feature), pattern = "prot"))) {
    signif_grid <- dcast(signif_df, Gene_Protein_Change ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene_Protein_Change)
    signif_grid <- signif_grid[, intersect(g2p_drug_order, colnames(signif_grid))]
    # signif_grid <- signif_grid[complete.cases(signif_grid), ]
    signif_grid[signif_grid < -log10(0.05)] <- NA
    
    signif_grid_annot <- dcast(signif_df, Gene_Protein_Change ~ Drug, value.var = "Direction_Curated", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene_Protein_Change)
    signif_grid_annot <- signif_grid_annot[, intersect(g2p_drug_order, colnames(signif_grid))]
    # signif_grid_annot <- signif_grid_annot[complete.cases(signif_grid_annot), ]
    signif_grid_annot[signif_grid_annot == "resistance"] <- "X"
    signif_grid_annot[signif_grid_annot == "sensitivity"] <- "O"
  }
  if(!is.na(str_match(string = tolower(feature), pattern = "fus"))) {
    signif_grid <- dcast(signif_df, FusionPair ~ Drug, value.var = "p_log10", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-FusionPair)
    signif_grid <- signif_grid[, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid <- signif_grid[complete.cases(signif_grid), ]
    signif_grid[signif_grid < -log10(0.05)] <- NA
    
    signif_grid_annot <- dcast(signif_df, FusionPair ~ Drug, value.var = "InG2P", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-FusionPair)
    signif_grid_annot <- signif_grid_annot[, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid_annot <- signif_grid_annot[complete.cases(signif_grid_annot), ]
    signif_grid_annot[signif_grid_annot == "No"] <- ""
    signif_grid_annot[signif_grid_annot == "Yes"] <- "O"
  }
  col_fun <- colorRamp2(c(legend_breaks[1], legend_breaks[length(legend_breaks)]), c("gray97", color))
  # Legend properties list
  ht_legend_list <- list(title = legend_title, title_position = "topleft", legend_direction = "horizontal", legend_width = unit(2, "in"), border = "gray70", at = legend_breaks, labels = legend_breaks)
  ht <- Heatmap(signif_grid,
                col = col_fun,
                # na_col = "gray97",
                na_col = "white",
                rect_gp = gpar(col = "gray70"),
                heatmap_legend_param = ht_legend_list,
                cluster_rows = FALSE, 
                cluster_columns = FALSE, 
                show_row_names = as.logical(row_names),
                row_names_side = "left",
                row_names_gp = gpar(fontsize = 10),
                column_names_side = "bottom",
                column_names_gp = gpar(fontsize = 10),
                show_column_names = FALSE,
                bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(signif_grid),
                                   rot = 45,
                                   gp = gpar(fontsize = 11),
                                   offset = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.85, "in")),
                  # annotation_height = max_text_width(colnames(signif_grid))),
                row_title = row_title_txt,
                row_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title = col_title, 
                column_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title_side = "bottom",
                cell_fun = function(j, i, x, y, width, height, fill) {
                  # grid.text(signif_grid_annot[i, j], x, y, gp = gpar(fontsize = 20))
                  if(signif_grid_annot[i, j] == "O") {
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "white", col = "black"))
                  }
                  if(signif_grid_annot[i, j] == "X") {
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = radius, default.units = "npc", gp = gpar(fill = "black", col = "black"))
                  }
                }
                )
  return(ht)
}
```

```{r mut_heatmaps, eval=FALSE, fig.height=10, fig.width=3}
ht_ccle <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "-log10(p-value)", legend_breaks = seq(0, 12, by = 3), row_names = TRUE, color = "darkgreen", col_title = "Drugs", row_title_txt = "Genes", radius = 0.03)
png(filename = "./plots/manuscript/ccle_mut_heatmap.png", width = 3, height = 10, units = "in", res = 500)
draw(ht_ccle, heatmap_legend_side = "top")
dev.off()

ht_ctrp <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "-log10(p-value)", legend_breaks = seq(0, 18, by = 3), row_names = TRUE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Genes", radius = 0.008)
png(filename = "./plots/manuscript/ctrp_mut_heatmap.png", width = 9.5, height = 10, units = "in", res = 500)
draw(ht_ctrp, heatmap_legend_side = "top")
dev.off()

ht_gdsc <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "-log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = TRUE, color = "turquoise4", col_title = "Drugs", row_title_txt = "Genes", radius = 0.015)
png(filename = "./plots/manuscript/gdsc_mut_heatmap.png", width = 5, height = 10, units = "in", res = 500)
draw(ht_gdsc, heatmap_legend_side = "top")
dev.off()

ht_ccle2 <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "CCLE -log10(p-value)                                                                     ", legend_breaks = seq(0, 12, by = 3), row_names = TRUE, color = "darkgreen", col_title = "", row_title_txt = "Genes", radius = 0.03)
ht_ctrp2 <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "CTRP -log10(p-value)                                                                    ", legend_breaks = seq(0, 18, by = 3), row_names = FALSE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Genes", radius = 0.008)
ht_gdsc2 <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_g2p_dg, Gene %in% unique(g2p_mut$Gene)), feature = "mutation", legend_title = "GDSC -log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = FALSE, color = "turquoise4", col_title = "", row_title_txt = "Genes", radius = 0.015)
png(filename = "./plots/manuscript/drug_mut_heatmap.png", width = 13, height = 9.5, units = "in", res = 500)
draw(ht_ccle2 + ht_ctrp2 + ht_gdsc2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```

```{r prot_heatmaps, eval=FALSE, fig.height=10, fig.width=3}
ht_ccle <- MakeWilcoxonHeatmap(signif_df = ccle_signif_prot_4plot, feature = "protein changes", legend_title = "-log10(p-value)", legend_breaks = seq(0, 15, by = 3), row_names = TRUE, color = "darkgreen", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.03)
png(filename = "./plots/manuscript/ccle_prot_heatmap.png", width = 3, height = 5.5, units = "in", res = 500)
draw(ht_ccle, heatmap_legend_side = "top")
dev.off()

ht_ctrp <- MakeWilcoxonHeatmap(signif_df = ctrp_signif_prot_4plot, feature = "protein changes", legend_title = "-log10(p-value)", legend_breaks = seq(0, 15, by = 3), row_names = TRUE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.015)
png(filename = "./plots/manuscript/ctrp_prot_heatmap.png", width = 9.5, height = 5.5, units = "in", res = 500)
draw(ht_ctrp, heatmap_legend_side = "top")
dev.off()

ht_gdsc <- MakeWilcoxonHeatmap(signif_df = gdsc_signif_prot_4plot, feature = "protein changes", legend_title = "-log10(p-value)", legend_breaks = seq(0, 8, by = 2), row_names = TRUE, color = "turquoise4", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.015)
png(filename = "./plots/manuscript/gdsc_prot_heatmap.png", width = 5, height = 5.5, units = "in", res = 500)
draw(ht_gdsc, heatmap_legend_side = "top")
dev.off()

ht_ccle2 <- MakeWilcoxonHeatmap(signif_df = ccle_signif_prot_4plot, feature = "protein changes", legend_title = "CCLE -log10(p-value)                                                                     ", legend_breaks = seq(0, 15, by = 3), row_names = TRUE, color = "darkgreen", col_title = "", row_title_txt = "Gene:Protein Change", radius = 0.03)
ht_ctrp2 <- MakeWilcoxonHeatmap(signif_df = ctrp_signif_prot_4plot, feature = "protein changes", legend_title = "CTRP -log10(p-value)                                                                    ", legend_breaks = seq(0, 15, by = 3), row_names = FALSE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Gene:Protein Change", radius = 0.015)
ht_gdsc2 <- MakeWilcoxonHeatmap(signif_df = gdsc_signif_prot_4plot, feature = "protein changes", legend_title = "GDSC -log10(p-value)", legend_breaks = seq(0, 8, by = 2), row_names = FALSE, color = "turquoise4", col_title = "", row_title_txt = "Gene:Protein Change", radius = 0.015)
png(filename = "./plots/manuscript/drug_prot_heatmap.png", width = 13, height = 5, units = "in", res = 500)
draw(ht_ccle2 + ht_ctrp2 + ht_gdsc2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```

```{r fus_heatmaps, eval=FALSE}
ht_ccle <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "-log10(p-value)",  legend_breaks = seq(0, 4, by = 1), row_names = TRUE, color = "darkgreen", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.1)
png(filename = "./plots/manuscript/ccle_fus_heatmap.png", width = 5, height = 3, units = "in", res = 500)
draw(ht_ccle, heatmap_legend_side = "top")
dev.off()

ht_ctrp <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "-log10(p-value)", legend_breaks = seq(0, 10, by = 2), row_names = TRUE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.1)
png(filename = "./plots/manuscript/ctrp_fus_heatmap.png", width = 13, height = 3, units = "in", res = 500)
draw(ht_ctrp, heatmap_legend_side = "top")
dev.off()

ht_gdsc <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "-log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = TRUE, color = "turquoise4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.1)
png(filename = "./plots/manuscript/gdsc_fus_heatmap.png", width = 9, height = 3, units = "in", res = 500)
draw(ht_gdsc, heatmap_legend_side = "top")
dev.off()

ht_ccle2 <- MakeWilcoxonHeatmap(signif_df = filter(ccle_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "CCLE -log10(p-value)                                                                               ", legend_breaks = seq(0, 4, by = 1), row_names = TRUE, color = "darkgreen", col_title = "", row_title_txt = "Fusion Pairs", radius = 0.1)
ht_ctrp2 <- MakeWilcoxonHeatmap(signif_df = filter(ctrp_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "CTRP -log10(p-value)                                                                              ", legend_breaks = seq(0, 10, by = 2), row_names = FALSE, color = "steelblue4", col_title = "Drugs", row_title_txt = "Fusion Pairs", radius = 0.1)
ht_gdsc2 <- MakeWilcoxonHeatmap(signif_df = filter(gdsc_signif_fus, Drug %in% g2p_drugs), feature = "fusion", legend_title = "GDSC -log10(p-value)", legend_breaks = seq(0, 5, by = 1), row_names = FALSE, color = "turquoise4", col_title = "", row_title_txt = "Fusion Pairs", radius = 0.1)
png(filename = "./plots/manuscript/drug_fus_heatmap.png", width = 13, height = 2.4, units = "in", res = 500)
draw(ht_ccle2 + ht_ctrp2 + ht_gdsc2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```


# SPEARMAN: PanCan

--------------------------------------------------------------------------------

## GE/CN

```{r drug_spear_calc, warning=FALSE, eval=FALSE}
# CCLE
ccle_filt <- filter(ccle, Drug %in% g2p_drugs & Gene %in% g2p_genes)
ccle_filt$Gene <- factor(ccle_filt$Gene, ordered = FALSE)
ccle_filt <- unfactorize(ccle_filt)
ccle_ge_res <- ccle_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>% summarize(GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$estimate, GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$p.value)
ccle_ge_res$InG2P <- ifelse(ccle_ge_res$Drug_Gene %in% unique(as.character(g2p_expr$Drug_Gene[!is.na(g2p_expr$Drug_Gene)])), "Yes", "No")
ccle_ge_res$Dataset <- "CCLE"
ccle_cn_res <- ccle_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>% summarize(CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$estimate, CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$p.value)
ccle_cn_res$InG2P <- ifelse(ccle_cn_res$Drug_Gene %in% unique(as.character(g2p_amp$Drug_Gene[!is.na(g2p_amp$Drug_Gene)])), "Yes", "No")
ccle_cn_res$Dataset <- "CCLE"

# CTRP
ctrp_filt <- filter(ctrp, Drug %in% g2p_drugs & Gene %in% g2p_genes)
ctrp_filt$Gene <- factor(ctrp_filt$Gene, ordered = FALSE)
ctrp_filt <- unfactorize(ctrp_filt)
ctrp_ge_res <- ctrp_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>% summarize(GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$estimate, GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$p.value)
ctrp_ge_res$InG2P <- ifelse(ctrp_ge_res$Drug_Gene %in% unique(as.character(g2p_expr$Drug_Gene[!is.na(g2p_expr$Drug_Gene)])), "Yes", "No")
ctrp_ge_res$Dataset <- "CTRP"
ctrp_cn_res <- ctrp_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>% summarize(CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$estimate, CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$p.value)
ctrp_cn_res$InG2P <- ifelse(ctrp_cn_res$Drug_Gene %in% unique(as.character(g2p_amp$Drug_Gene[!is.na(g2p_amp$Drug_Gene)])), "Yes", "No")
ctrp_cn_res$Dataset <- "CTRP"

# GDSC
gdsc_filt <- filter(gdsc, Drug %in% g2p_drugs & Gene %in% g2p_genes)
gdsc_filt$Gene <- factor(gdsc_filt$Gene, ordered = FALSE)
gdsc_filt <- unfactorize(gdsc_filt)
gdsc_ge_res <- gdsc_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>% summarize(GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$estimate, GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$p.value)
gdsc_ge_res$InG2P <- ifelse(gdsc_ge_res$Drug_Gene %in% unique(as.character(g2p_expr$Drug_Gene[!is.na(g2p_expr$Drug_Gene)])), "Yes", "No")
gdsc_ge_res$Dataset <- "GDSC"
gdsc_cn_res <- gdsc_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>% summarize(CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$estimate, CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$p.value)
gdsc_cn_res$InG2P <- ifelse(gdsc_cn_res$Drug_Gene %in% unique(as.character(g2p_amp$Drug_Gene[!is.na(g2p_amp$Drug_Gene)])), "Yes", "No")
gdsc_cn_res$Dataset <- "GDSC"

# All
cn_cor <- rbind(ccle_cn_res, ctrp_cn_res, gdsc_cn_res)
saveRDS(cn_cor, "./data_munging/rds/drug_cn_cor.rds")
ge_cor <- rbind(ccle_ge_res, ctrp_ge_res, gdsc_ge_res)
saveRDS(ge_cor, "./data_munging/rds/drug_ge_cor.rds")
```

```{r load_drug_spear}
g2p_amp_4merge <- select(g2p_amp, Drug_Gene, Direction_Curated)
cn_cor <- readRDS("./data_munging/rds/drug_cn_cor.rds")
cn_cor <- merge(cn_cor, g2p_amp_4merge, by = "Drug_Gene", all.x = TRUE)

g2p_expr_4merge <- select(g2p_expr, Drug_Gene, Direction_Curated)
ge_cor <- readRDS("./data_munging/rds/drug_ge_cor.rds")
ge_cor <- merge(ge_cor, g2p_expr_4merge, by = "Drug_Gene", all.x = TRUE)
```

## Plots

Spearman correlations for gene expression and copy number:
```{r crispr_spear_calc, warning=FALSE}
crispr_spear_ge <- transform(crispr %>% group_by(Gene) %>% drop_na(TPM) %>% summarize(p = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$p.value, p_log10 = -log10(p), corr = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$estimate, Metric = "Gene Expression"), Gene = reorder(Gene, corr))
crispr_spear_cn <- transform(crispr %>% group_by(Gene) %>% drop_na(CN) %>% summarize(p = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$p.value, p_log10 = -log10(p), corr = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$estimate, Metric = "Copy Number"), Gene = reorder(Gene, corr))
c("white", "deepskyblue4")

crispr_spear_cn_plot <- ggplot(data = filter(crispr_spear_cn, Gene %in% g2p_dg_genes)) +
  geom_point(aes(x = Gene, y = corr, fill = p_log10), pch = 21, size = 5) +
  theme(legend.position = "bottom") +
  scale_fill_gradientn(colours = c("white", "green4"),
                       breaks = c(0, 4, 8, 12, 16, 20),
                       labels = c(0, 4, 8, 12, 16, 20),
                       limits = c(0, 21)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10)) +
  labs(title = "Copy Number",
       fill = expression(-log[10](p)),
       x = "Gene",
       y = expression("Spearman Correlation Coefficient"~(r[s])))

crispr_spear_ge_plot <- ggplot(data = filter(crispr_spear_ge, Gene %in% g2p_dg_genes)) +
  geom_point(aes(x = Gene, y = corr, fill = p_log10), pch = 21, size = 5) +
  theme(legend.position = "bottom") +
  scale_fill_gradientn(colours = c("white", "deepskyblue4"),
                       breaks = c(0, 4, 8, 12, 16, 20),
                       labels = c(0, 4, 8, 12, 16, 20),
                       limits = c(0, 21)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, size = 10)) +
  labs(title = "Gene Expression",
       fill = expression(-log[10](p)),
       x = "Gene",
       y = expression("Spearman Correlation Coefficient"~(r[s])))

crispr_spearman_corr_plot <- plot_grid(crispr_spear_cn_plot, crispr_spear_ge_plot, align = "h")
crispr_spearman_corr_plot

# ggsave("./plots/manuscript/crispr_spearman_corr_plot.png", crispr_spearman_corr_plot, device = "png", dpi = 450, width = 20, height = 10, units = "in")
```

```{r corr_heatmap_function}
MakeCorrHeatmap <- function(drug_cor_res, row_names, col_title, color_range, legend_title, metric, radius) {
  if(toupper(metric) == "CN" | toupper(metric) == "COPY NUMBER") {
    signif_grid <- dcast(drug_cor_res, Gene ~ Drug, value.var = "CN_p", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid <- signif_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid <- signif_grid[complete.cases(signif_grid), ]
    signif_grid[signif_grid >= 0.05] <- ""
    signif_grid[signif_grid != ""] <- "*"
    
    corr_grid <- dcast(drug_cor_res, Gene ~ Drug, value.var = "CN_r", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    corr_grid <- corr_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(corr_grid))]
    corr_grid <- corr_grid[complete.cases(corr_grid), ]
  }
  
  if(toupper(metric) == "GE" | toupper(metric) == "GENE EXPRESSION") {
    signif_grid <- dcast(drug_cor_res, Gene ~ Drug, value.var = "GE_p", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    signif_grid <- signif_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
    signif_grid <- signif_grid[complete.cases(signif_grid), ]
    signif_grid[signif_grid >= 0.05] <- ""
    signif_grid[signif_grid != ""] <- "*"
    
    corr_grid <- dcast(drug_cor_res, Gene ~ Drug, value.var = "GE_r", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
    corr_grid <- corr_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(corr_grid))]
    corr_grid <- corr_grid[complete.cases(corr_grid), ]
  }
  
  annot_grid <- dcast(drug_cor_res, Gene ~ Drug, value.var = "Direction_Curated", fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Gene)
  annot_grid <- annot_grid[g2p_gene_order, intersect(g2p_drug_order, colnames(signif_grid))]
  annot_grid <- annot_grid[complete.cases(annot_grid), ]
  # annot_grid[annot_grid == "No"] <- ""
  # annot_grid[annot_grid == "Yes"] <- "G"
  
  col_fun <- colorRamp2(color_range, c("red3", "white", "blue3"))
  
  # Legend properties list
  ht_legend_list <- list(title_position = "topleft", legend_direction = "horizontal", legend_width = unit(3, "in"), border = "gray70", at = seq(color_range[1], color_range[3], by = 0.1), labels = seq(color_range[1], color_range[3], by = 0.1))
  ht <- Heatmap(corr_grid,
                col = col_fun,
                name = legend_title,
                na_col = "gray97",
                rect_gp = gpar(col = "gray70"),
                heatmap_legend_param = ht_legend_list,
                cluster_rows = FALSE, 
                cluster_columns = FALSE, 
                show_row_names = as.logical(row_names),
                row_names_side = "left",
                row_names_gp = gpar(fontsize = 10),
                column_names_side = "bottom",
                column_names_gp = gpar(fontsize = 10),
                show_column_names = FALSE,
                bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(signif_grid),
                                   rot = 45,
                                   gp = gpar(fontsize = 11),
                                   offset = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.85, "in")),
                row_title = "Genes",
                row_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title = col_title, 
                column_title_gp = gpar(fontface = "bold", fontsize = 14),
                column_title_side = "top",
                cell_fun = function(j, i, x, y, width, height, fill) {
                  grid.text(signif_grid[i, j], x = x - width*0.20, y = y, gp = gpar(fontsize = 18))
                  if(annot_grid[i, j] == "sensitivity") {
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = height*0.15, default.units = "npc", gp = gpar(fill = "white", col = "black"))
                    # grid.circle(x = x, y = y, r = radius, gp = gpar(fill = "transparent", col = "black"))
                  }
                  if(annot_grid[i, j] == "resistance") {
                    grid.circle(x = x - width*0.20, y = y - height*0.20, r = height*0.15, default.units = "npc", gp = gpar(fill = "black", col = "black"))
                    # grid.text("☐", x, y, just = c(0.5, 0.425), gp = gpar(fontsize = 23, fontface = "bold"))
                  }
                }
                )
  return(ht)
}
```

```{r drug_spear_plots, eval=FALSE}
# Copy number
ccle_cn_cor <- unfactorize(filter(cn_cor, Dataset == "CCLE" & Gene %in% unique(g2p_amp$Gene)))
ccle_cn_ht <- MakeCorrHeatmap(drug_cor_res = ccle_cn_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.2, 0, 0.2), radius = 0.1, legend_title = "CCLE Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/ccle_cn_heatmap.png", width = 3, height = 3, units = "in", res = 500)
draw(ccle_cn_ht, heatmap_legend_side = "top")
dev.off()

ctrp_cn_cor <- unfactorize(filter(cn_cor, Dataset == "CTRP" & Gene %in% unique(g2p_amp$Gene)))
ctrp_cn_ht <- MakeCorrHeatmap(drug_cor_res = ctrp_cn_cor, row_names = TRUE, col_title = "CTRP", color_range = c(-0.2, 0, 0.2), radius = 0.1, legend_title = "CTRP Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/ctrp_cn_heatmap.png", width = 8, height = 3, units = "in", res = 500)
draw(ctrp_cn_ht, heatmap_legend_side = "top")
dev.off()

gdsc_cn_cor <- unfactorize(filter(cn_cor, Dataset == "GDSC" & Gene %in% unique(g2p_amp$Gene)))
gdsc_cn_ht <- MakeCorrHeatmap(drug_cor_res = gdsc_cn_cor, row_names = TRUE, col_title = "GDSC", color_range = c(-0.2, 0, 0.2), radius = 0.1, legend_title = "GDSC Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/gdsc_cn_heatmap.png", width = 5, height = 3, units = "in", res = 500)
draw(gdsc_cn_ht, heatmap_legend_side = "top")
dev.off()

ccle_cn_ht2 <- MakeCorrHeatmap(drug_cor_res = ccle_cn_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.2, 0, 0.2), radius = 0.025, legend_title = "Copy num. Spearman correlation", metric = "CN")
ctrp_cn_ht2 <- MakeCorrHeatmap(drug_cor_res = ctrp_cn_cor, row_names = FALSE, col_title = "CTRP", color_range = c(-0.2, 0, 0.2), radius = 0.025, legend_title = "Copy num. Spearman correlation", metric = "CN")
gdsc_cn_ht2 <- MakeCorrHeatmap(drug_cor_res = gdsc_cn_cor, row_names = FALSE, col_title = "GDSC", color_range = c(-0.2, 0, 0.2), radius = 0.025, legend_title = "Copy num. Spearman correlation", metric = "CN")
png(filename = "./plots/manuscript/drug_cn_heatmap.png", width = 13, height = 3, units = "in", res = 500)
draw(ccle_cn_ht2 + ctrp_cn_ht2 + gdsc_cn_ht2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()

# Gene expression
ccle_ge_cor <- unfactorize(filter(ge_cor, Dataset == "CCLE" & Gene %in% unique(g2p_expr$Gene)))
ccle_ge_ht <- MakeCorrHeatmap(drug_cor_res = ccle_ge_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "CCLE Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/ccle_ge_heatmap.png", width = 3, height = 2.5, units = "in", res = 500)
draw(ccle_ge_ht, heatmap_legend_side = "top")
dev.off()

ctrp_ge_cor <- unfactorize(filter(ge_cor, Dataset == "CTRP" & Gene %in% unique(g2p_expr$Gene)))
ctrp_ge_ht <- MakeCorrHeatmap(drug_cor_res = ctrp_ge_cor, row_names = TRUE, col_title = "CTRP", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "CTRP Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/ctrp_ge_heatmap.png", width = 8, height = 2.5, units = "in", res = 500)
draw(ctrp_ge_ht, heatmap_legend_side = "top")
dev.off()

gdsc_ge_cor <- unfactorize(filter(ge_cor, Dataset == "GDSC" & Gene %in% unique(g2p_expr$Gene)))
gdsc_ge_ht <- MakeCorrHeatmap(drug_cor_res = gdsc_ge_cor, row_names = TRUE, col_title = "GDSC", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "GDSC Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/gdsc_ge_heatmap.png", width = 5, height = 2.5, units = "in", res = 500)
draw(gdsc_ge_ht, heatmap_legend_side = "top")
dev.off()

ccle_ge_ht2 <- MakeCorrHeatmap(drug_cor_res = ccle_ge_cor, row_names = TRUE, col_title = "CCLE", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "Gene expr. Spearman correlation", metric = "GE")
ctrp_ge_ht2 <- MakeCorrHeatmap(drug_cor_res = ctrp_ge_cor, row_names = FALSE, col_title = "CTRP", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "Gene expr. Spearman correlation", metric = "GE")
gdsc_ge_ht2 <- MakeCorrHeatmap(drug_cor_res = gdsc_ge_cor, row_names = FALSE, col_title = "GDSC", color_range = c(-0.5, 0, 0.5), radius = 0.2, legend_title = "Gene expr. Spearman correlation", metric = "GE")
png(filename = "./plots/manuscript/drug_ge_heatmap.png", width = 13, height = 2.5, units = "in", res = 500)
draw(ccle_ge_ht2 + ctrp_ge_ht2 + gdsc_ge_ht2, gap = unit(0.25, "in"), heatmap_legend_side = "top")
dev.off()
```

# WATERFALL/GSEA: PanCAn

--------------------------------------------------------------------------------

Consider looking at GSEA rankings.

# WILCOXON: Lineages

--------------------------------------------------------------------------------

# SPEARMAN: Lineages

--------------------------------------------------------------------------------

## GE/CN

# SESSION INFORMATION

--------------------------------------------------------------------------------

```{r session_info}
print(sessionInfo())
```