---
title: "19Q2"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the [Broad's Avana CRISPR](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017), the [Cancer Therapeutics Response Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (CTRP) drug screen dataset (v2) (Basu et al., 2013; M. G. Rees et al., 2016; B. Seashore-Ludlow et al., 2015), the [Genomics of Drug Sensitivity in Cancer](https://www.cancerrxgene.org/downloads) (GDSC) drug screen dataset (Garnett et al., 2012; Yang et al., 2012) , and the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) (CCLE) drug screen dataset (J. Barretina et al., 2012; T. C. C. L. E. Consortium & Consortium, 2015).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

Annotation files for mutation status, copy number, and gene expression of the CCLE, GDSC, and CTRP datasets (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/). Note that the group that conducted the GDSC screen also collected their own genomic annotation information for mutation status, copy number, and gene expression, but we did not use it.

# SET UP

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

```{r libraries}
library(circlize)
library(ComplexHeatmap)
library(cowplot)
library(crunch)
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(jsonlite)
library(kableExtra)
library(naniar)
library(openxlsx)
library(reshape2)
library(scales)
library(stringr)
library(tidyverse)
library(VennDiagram)

theme_set(theme_light())
```

```{r general_functions}
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) {
    df[[i]] = as.character(df[[i]])
  }
  return(df)
}
```

# DATA

--------------------------------------------------------------------------------

## Cell line metadata

**D. Charytonowicz cell line converter**

This comprehensive cancer cell line information was curated by both Daniel Charytonowicz and myself.
```{r dan_ccl_database}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
colnames(ccl_converter)[colnames(ccl_converter) == "Broad_ID"] <- "DepMap_ID"
ccl_lineage_converter <- unique(ccl_converter[, c("DepMap_ID", "grouped_general_doid", "group_general_lineage_name")])
lineage_converter <- unique(select(ccl_lineage_converter, grouped_general_doid, group_general_lineage_name))
ccl_histologies <- read.delim("./data_munging/ccl_histologies_metadata.csv", sep = ",", header = TRUE, row.names = 1)
colnames(ccl_histologies) <- c("Cell_line", "CCLE_histology", "TCGA_classification", "CCLE_name", "DepMap_ID")
```

**DepMap metadata**

DepMap 19Q2 update fixed almost all of the inconsistencies of the Primary.Disease column, though there were come capitalization inconsistencies.
```{r depmap_metadata}
ccl_meta <- read.delim("./data_munging/19Q2/sample_info.csv", sep = ",", header = TRUE, na.strings = c("", NA))
ccl_meta <- select(ccl_meta, DepMap_ID, stripped_cell_line_name, CCLE_name, disease)
colnames(ccl_meta) <- c("DepMap_ID", "Cell_line", "CCLE_name", "disease")
# Remove two rows that contain cell lines with two different histologies
# The histologies that are kept are verified through matching in Cellosaurus and Cell Model Passports
# KMH2, haematopoietic_and_lymphoid: https://web.expasy.org/cellosaurus/CVCL_1330, https://cellmodelpassports.sanger.ac.uk/passports/SIDM01018
# MS1, lung: https://web.expasy.org/cellosaurus/CVCL_1429, https://cellmodelpassports.sanger.ac.uk/passports/SIDM00245
ccl_meta <- subset(ccl_meta, CCLE_name != "KMH2_THYROID" & CCLE_name != "MS1_SKIN")
ccl_meta <- subset(ccl_meta, !grepl("MERGED", CCLE_name, fixed = TRUE))
ccl_diseases <- as.character(unique(ccl_meta$disease))[as.character(unique(ccl_meta$disease)) != " "]
CCLE_histology <- tolower(gsub("^[a-z0-9]*_", "", ccl_meta$CCLE_name, ignore.case = TRUE))
ccl_meta <- add_column(ccl_meta, CCLE_histology = CCLE_histology, .after = "CCLE_name")
ccl_meta$disease <- with(ccl_meta, ifelse(disease == " " & CCLE_histology %in% ccl_diseases, as.character(CCLE_histology), as.character(disease)))
ccl_meta$disease <- with(ccl_meta, ifelse(disease == "ovary", "ovarian",
                                   ifelse(disease == " " & CCLE_histology == "vulva", "vulva",
                                   ifelse(disease == " " & CCLE_histology == "stomach", "gastric",
                                   ifelse(disease == " " & CCLE_histology == "upper_aerodigestive_tract", "upper_aerodigestive_tract",
                                   ifelse(disease == " " & CCLE_histology == "large_intestine", "colorectal",
                                   ifelse(disease == " " & CCLE_histology == "biliary_tract", "bile_duct",
                                   ifelse(disease == " " & CCLE_histology == "oesophagus", "esophagus",
                                   ifelse(disease == " " & CCLE_histology == "testis", "embryo", disease)))))))))
```

## G2P

```{r g2p_parsing_functions}
ExtractBiomarkerType <- function(json_string) {
  json_string <- as.character(json_string)
  biomarker_type <- ((str_extract(string = json_string, pattern = "biomarker_type.+") %>% strsplit(split = ","))[[1]][1] %>% str_extract(pattern = "u'.+") %>% str_match("'(.*?)'"))[,2]
  biomarker_type <- ifelse(is.na(biomarker_type), "", biomarker_type)
  return(biomarker_type)
}
```

```{r g2p_parsing, eval=FALSE}
g2p <- read.delim("./data_munging/g2p_full_level_A_dataset_NEW_DEC_2_2018_withGroupedLineages.csv", sep = "\t", header = TRUE)
g2p[, 1:2] <- NULL
g2p <- unfactorize(g2p)
g2p$Drug <- tolower(g2p$Drug)
g2p[g2p == -1] <- ""
g2p$Gene <- ifelse(g2p$Gene == "" & g2p$Drug == "daunorubicin", "MLL", as.character(g2p$Gene))
g2p$Direction <- ifelse(g2p$Direction == 1 | g2p$Direction == "", NA, as.character(g2p$Direction))
g2p$Ref[g2p$Ref == "None" | g2p$Ref == "-" | is.na(g2p$Ref)] <- ""
g2p$Alt[g2p$Alt == "None" | g2p$Alt == "-" | is.na(g2p$Alt)] <- ""
g2p <- g2p[, c("Drug", "DrugID", "Direction", "Gene", "Ref", "Alt", "Start", "End", "Chromosome", "Feature.Names", "Mutation", "JSON", "Evidence.Level", "Phenotype.Description", "Phenotype.Family", "Phenotype.ID", "Sequence.ID", "grouped_general_doid")]
g2p <- add_column(g2p, Ref_Length = nchar(as.character(g2p$Ref)), .after = "Ref")
g2p <- add_column(g2p, Alt_Length = nchar(as.character(g2p$Alt)), .after = "Alt")
g2p <- add_column(g2p, Variant_Length = NA, .after = "Alt_Length")
g2p$Variant_Length <-
  ifelse(g2p$Alt_Length == 1 & g2p$Ref_Length == 1, 1,
  ifelse(g2p$Alt_Length == 2 & g2p$Ref_Length == 2, 2,
  ifelse(g2p$Alt_Length == 3 & g2p$Ref_Length == 3, 3,
  ifelse(g2p$Ref_Length == g2p$Alt_Length & g2p$Ref_Length > 3, g2p$Ref_Length,
  ifelse(g2p$Ref_Length != g2p$Alt_Length, abs(g2p$Ref_Length - g2p$Alt_Length), NA)))))
g2p <- add_column(g2p, Variant_Type = NA, .after = "Alt_Length")
g2p$Variant_Type <-
  ifelse(g2p$Alt_Length == 1 & g2p$Ref_Length == 1, "SNP",
  ifelse(g2p$Alt_Length == 2 & g2p$Ref_Length == 2, "DNP",
  ifelse(g2p$Alt_Length == 3 & g2p$Ref_Length == 3, "TNP",
  ifelse(g2p$Ref_Length == g2p$Alt_Length & g2p$Ref_Length > 3, "ONP",
  ifelse(g2p$Ref_Length > g2p$Alt_Length, "DEL",
  ifelse(g2p$Ref_Length < g2p$Alt_Length, "INS", "None"))))))
g2p <- subset(g2p, !grepl("^SID|^CHEMBL", g2p$DrugID))
g2p <- subset(g2p, !grepl("germline|homozygosity", tolower(g2p$Feature.Names)))
g2p <- add_column(g2p, biomarker_type = tolower(sapply(g2p$JSON, ExtractBiomarkerType, simplify = TRUE)), .after = "Feature.Names")
g2p <- add_column(g2p, biomarker_type_curated = NA, .after = "biomarker_type")

# Manually assign biomarker_types based on a combination of biomarker_type derived from the JSON string and patterns in the Feature.Names
biomarker_type_curated <- 
  ifelse(!is.na(str_match(string = tolower(g2p$biomarker_type), pattern = "copy number variant|amplification")) | !is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "copy number variant|amplification")), "Amplification/CNV",
  ifelse(!is.na(str_match(string = toupper(g2p$Feature.Names), pattern = "[A-Z0-9|A-Z]+-[A-Z0-9|A-Z]+|FUSION")) & is.na(str_match(string = g2p$Feature.Names, pattern = "[0-9]+-[0-9]+")) | g2p$biomarker_type == "fusion" | g2p$biomarker_type == "rearrange", "Fusion",
  ifelse(!is.na(str_match(string = g2p$Feature.Names, pattern = "[A-Z][0-9]+([A-Z]|[*])|[0-9A-Z]+ [A-Z][0-9]+|[A-Z][0-9]+$")), "Mutation",
  ifelse(!is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "expression|methylation")), "Expression",
  ifelse(!is.na(str_match(string = g2p$biomarker_type, pattern = "intron|insertion|deletion|mutant")), "Mutation",
  ifelse(!is.na(str_match(string = tolower(g2p$Feature.Names), pattern = "codon|positive|biallelic inactivation|del|ins|dup|mut|frameshift|missense|variant|[0-9]+fs|>")), "Mutation",
  ifelse(!is.na(str_match(string = g2p$Feature.Names, pattern = "^KIT$|^BRCA$")), "Mutation", NA)))))))
g2p$biomarker_type_curated <- biomarker_type_curated[,1]

# Add association columns
g2p[g2p == ""] <- NA
g2p$Drug_Gene <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene), NA, paste0(g2p$Drug, "_", g2p$Gene))
g2p <- merge(lineage_converter, g2p, by = "grouped_general_doid", all.y = TRUE)
g2p$Drug_Gene_Lin <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene) | is.na(g2p$group_general_lineage_name), NA, paste0(g2p$Drug, "_", g2p$Gene, "_", g2p$group_general_lineage_name))
g2p$Gene_Lin <- ifelse(is.na(g2p$Drug) | !is.na(g2p$Gene) | !is.na(g2p$group_general_lineage_name), paste0(g2p$Gene, "_", g2p$group_general_lineage_name), NA)

# write.csv(g2p, "./data_munging/g2p_db_edited.csv")
```

```{r g2p}
g2p <- read.delim("./data_munging/g2p_db_edited.csv", sep = ",", header = TRUE, row.names = 1)
g2p$Drug_Gene_Lin <- gsub(pattern = " cancer", "", g2p$Drug_Gene_Lin)

g2p_dgl <-  as.character(unique(g2p$Drug_Gene_Lin[!is.na(g2p$Drug_Gene_Lin)]))
g2p_dg <- as.character(unique(g2p$Drug_Gene[!is.na(g2p$Drug_Gene)]))
g2p_gl <- as.character(unique(g2p$Gene_Lin[!is.na(g2p$Gene_Lin)]))
g2p_drugs <- as.character(unique(g2p$Drug[!is.na(g2p$Drug)]))
g2p_genes <- as.character(unique(g2p$Gene[!is.na(g2p$Gene)]))
g2p_lins <- as.character(unique(g2p$group_general_lineage_name[!is.na(g2p$group_general_lineage_name)]))
```

## Omics

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Mutation calls (coding region, germline filtered):
```{r mutation_status, eval=FALSE}
maf_raw <- read.delim("./data_munging/19Q2/CCLE_mutations_20190514.csv.gz", header = TRUE, sep = ",")
maf_raw[, 1:2] <- NULL
colnames(maf_raw)[colnames(maf_raw) == "Hugo_Symbol"] <- "Gene"
maf_raw <- unfactorize(maf_raw)
# Select columns
maf <- subset(maf_raw, select = c("Gene", "DepMap_ID"))
maf$Mutation <- 1
maf_grid <- expand.grid("Gene" = unique(maf$Gene), "DepMap_ID" = unique(maf$DepMap_ID))
maf_df <- merge(maf, maf_grid, by = c("Gene", "DepMap_ID"), all = TRUE)
maf_df[is.na(maf_df)] <- 0

saveRDS(maf_df, "./data_munging/19Q2/maf_df_19Q2.rds", compress = "xz")
```

DepMap 19Q2 fusion calls derived using the [STAR-Fusion pipeline](https://github.com/STAR-Fusion/STAR-Fusion/wiki). Column descriptions:

- *JunctionReads*: the number of RNA-Seq fragments containing a read that aligns as a split read at the site of the putative fusion junction
- *SpanningFrags*: the number of RNA-Seq fragments that encompass the fusion junction such that one read of the pair aligns to a different gene than the other paired-end read of that fragment
    - Those predictions that have very few JunctionReads and/or SpanningReads are going to be enriched for false positives. Note, depending on the site of the fusion breakpoint and length of the reads, it may not be possible to have SpanningFragments and all evidence may show up in the form of JunctionReads
- *FFPM (fusion fragments per million total reads)*: normalized measure of the fusion-supporting RNA-seq fragmentS; a filter of 0.1 sum FFPM (meaning at least 1 fusion-supporting RNA-seq fragment per 10M total reads) tends to be effective at excluding fusion artifacts, and is the current default for filtering fusions from the final output
- *LargeAnchorSupport*: whether there are split reads that provide 'long' (set to length of 25 bases) alignments on both sides of the putative breakpoint
    - Those fusions supported only by split reads (no spanning fragments) and lack LargeAnchorSupport are often highly suspicious and tend to be false positives
    - Those with LargeAnchorSupport are labeled as 'YES_LDAS' (where LDAS = long double anchor support)
- *SpliceType*: whether the proposed breakpoint occurs at reference exon junctions as provided by the reference transcript structure annotations (ex. gencode)
- *LeftBreakEntropy* and *RightBreakEntropy*: the Shannon entropy of the 15 exonic bases flanking the breakpoint.
    - The maximum entropy is 2, representing highest complexity, and the lowest would be zero (involving a 15 base mononucleotide run)
    - Low entropy sites should generally be treated as less confident breakpoints.

```{r fusion_calls, eval=TRUE}
fusions <- read.delim("./data_munging/19Q2/CCLE_fusions.csv.gz", header = TRUE, sep = ",")
fusions$LeftGene <- gsub(" .*", "", fusions$LeftGene)
fusions$RightGene <- gsub(" .*", "", fusions$RightGene)
fusions$X.FusionName <- gsub("--", "-", fusions$X.FusionName)
fusions <- select(fusions, DepMap_ID, X.FusionName)
fusions <- as.data.frame(lapply(fusions, function(x) gsub("IGH@|IGH-@", "IGH", x)))
colnames(fusions) <- c("DepMap_ID", "FusionPair")
fusions <- merge(fusions, ccl_meta, by = "DepMap_ID", all.x = TRUE)
```

DepMap WES CN Data:

- gene_cn - NumericalMatrix
- Gene level copy number data, log2 transformed with a pseudo count 1. This is generated by mapping genes onto the segment level calls.
- Rows: cell lines (Broad IDs)
- Columns: genes (HGNC symbol and Entrez ID)

```{r copy_number, eval=FALSE}
cn <- read.delim("./data_munging/19Q2/CCLE_gene_cn_20190514.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "DepMap_ID"
# Melt
cn_melt <- melt(data = cn, id.vars = "DepMap_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Gene", value.name = "CN")
cn_melt <- unfactorize(cn_melt)

saveRDS(cn_melt, "./data_munging/19Q2/cn_melt_19Q2.rds", compress = "xz")
```

19Q2, CCLE RNAseq gene expression data:

- expression - NumericalMatrix
- RNAseq TPM gene expression data for just protein coding genes using RSEM. Log2 transformed, using a pseudo-count of 1.
- Rows: cell lines (Broad IDs)
- Columns: genes (HGNC symbol and Entrez ID)

```{r gene_expression, eval=FALSE}
ge <- read.delim("./data_munging/19Q2/CCLE_expression_20190514.csv.gz", header = TRUE, sep = ",", check.names = FALSE)
# Remove Entrez gene IDs from colnames, format is "Gene (Entrez ID)"
colnames(ge) <- gsub(" .*", "", colnames(ge))
colnames(ge)[1] <- "DepMap_ID"
# Melt
ge_melt <- melt(ge, id.vars = "DepMap_ID", measure.vars = colnames(ge)[2:ncol(ge)], variable.name = "Gene", value.name = "TPM")
# ge_melt$TPM <- log2(ge_melt$TPM + 1)
ge_melt <- unfactorize(ge_melt)

saveRDS(ge_melt, "./data_munging/19Q2/ge_melt_19Q2.rds", compress = "xz")
```

Load Rds:
```{r load_omics_rds}
maf_df <- readRDS("./data_munging/19Q2/maf_df_19Q2.rds")
cn_melt <- readRDS("./data_munging/19Q2/cn_melt_19Q2.rds")
ge_melt <- readRDS("./data_munging/19Q2/ge_melt_19Q2.rds")
```

# MERGE OMICS

--------------------------------------------------------------------------------

```{r omics_functions}
MergeOmicsG2P <- function(database, maf_df, ge_df, cn_df, gene_list, outfile) {
  # Merge mutation status, CN, and GE omics data with drugscreen results and metadata
  if(database %in% c("CCLE", "CTRP", "GDSC")) {
    filt <- auc %>% filter(Database == database) %>% select(Cell_line, CCLE_histology, TCGA_classification, CCLE_name, DepMap_ID, Drug, AUC_IC50)
    print(paste0("Processing ", database, " data."))
    colnames(filt)[colnames(filt) == "AUC_IC50"] <- "Score"

    grid <- expand.grid("Cell_line" = unique(filt$Cell_line), "Gene" = gene_list)
    data <-  merge(filt, grid, by = "Cell_line", all.y = TRUE)
    print("Merged grid.")
    # Merge metadata
    data <-  merge(data, select(ccl_meta, Cell_line, CCLE_histology, CCLE_name, DepMap_ID, disease), by = c("Cell_line", "CCLE_histology", "CCLE_name", "DepMap_ID"), all.y = TRUE)
    ## Add columns for G2P drug-gene and drug-gene-lineage sorting
    data$Drug_Gene <- paste0(data$Drug, "_", data$Gene)
    data$G2P_Drug_Gene <- ifelse(data$Drug_Gene %in% g2p_dg, "Yes", "No")
    data$Drug_Gene_Lin <- paste0(data$Drug, "_", data$Gene, "_", data$disease)
    data$G2P_Drug_Gene_Lin <- ifelse(data$Drug_Gene_Lin %in% g2p_dgl, "Yes", "No")
    data$Gene_Lin <- paste0(data$Gene, "_", data$Gene, "_", data$disease)
    data$G2P_Gene_Lin <- ifelse(data$Gene_Lin %in% g2p_gl, "Yes", "No")
    print("Merged metadata.")
    # Merge mutation calls
    data <- unfactorize(data)
    data <- merge(data, maf_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
    data$Mutation <- ifelse(is.na(data$Mutation), 0, data$Mutation)
  }
  if(database == "CRISPR") {
    print(paste0("Processing ", database, " data."))
    # Melt CRISPR dataset for merging
    data <- melt(crispr_all, id.vars = "DepMap_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")
    # Merge cell line metadata
    data <- merge(data, select(ccl_meta, Cell_line, CCLE_histology, CCLE_name, DepMap_ID, disease), by = "DepMap_ID", all.y = TRUE)
    print("Merged metadata.")
    # Merge mutation calls
    data <- unfactorize(data)
    data <- merge(data, maf_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
    data$Mutation <- ifelse(is.na(data$Mutation), 0, data$Mutation)
  }

  ## Summarize number of mutant and Wildtype cell lines
  data_muts_summ <- data %>% group_by(Gene) %>% summarize(N_Wildtype = length(Mutation) - sum(Mutation == 1), N_Mutant = sum(Mutation == 1))
  ## Merge test results back into full dataset, which restores information lost in the summarization
  data <- merge(data_muts_summ, data, by = "Gene")
  print("Merged mutation calls.")
  # Merge GE data
  data <- merge(data, ge_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
  print("Merged gene expression data.")
  # Merge CN data
  data <- merge(data, cn_df, by = c("Gene", "DepMap_ID"), all.x = TRUE)
  print("Merged copy number data.")
  # Drop cell lines not in DepMap
  data <- filter(data, !is.na(DepMap_ID) & !(DepMap_ID %in% c("ACH-001182", "ACH-001451")))
  # Save and return dataframe
  data <- unfactorize(data)
  saveRDS(data, outfile, compress = "xz")
  return(data)
}
```

## Drug screens

`r nrow(filter(test, n != 1))`/`r nrow(test)` gene-cell line combinations have more than one mutation in the MAF file. This is corrected in the point mutation filtering done below.
```{r merge_drug_data, eval=FALSE}
# Load in Pozdeyev et al. (2016) data
ccle_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CCLE_drug_response_metrics.csv.gz", sep = ",")
ctrp_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/CTRP_drug_response_metrics.csv.gz", sep = ",")
gdsc_all <- read.delim("./../htscreens_giant_files/pozdeyev_data/GDSC_drug_response_metrics.csv.gz", sep = ",")

auc <- bind_rows(ccle_all, ctrp_all, gdsc_all)

# Harmonize Pozdeyev cell line names with DepMap
auc$Drug <- tolower(auc$Drug_name)
auc$Cell_line <- toupper(gsub("-", "", auc$Cell_line))
auc$Cell_line <- ifelse(auc$Cell_line == "K052", "KO52", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "TI73", "T173", auc$Cell_line)
auc <- merge(ccl_histologies, auc, by = "Cell_line", all.y = TRUE)

ccle <- MergeOmicsG2P(database = "CCLE", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/ccle_data_19Q2.rds")
write.csv.gz(ccle %>% drop_na(Score), "./data_munging/19Q2/ccle_data_19Q2.csv.gz")

ctrp <- MergeOmicsG2P(database = "CTRP", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./../htscreens_giant_files/ctrp_data_19Q2.rds")
write.csv.gz(ctrp %>% drop_na(Score), "./../htscreens_giant_files/ctrp_data_19Q2.csv.gz")

gdsc <- MergeOmicsG2P(database = "GDSC", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/gdsc_data_19Q2.rds")
write.csv.gz(gdsc %>% drop_na(Score), "./../htscreens_giant_files/gdsc_data_19Q2.csv.gz")

# all_genes <- intersect(unique(maf_df$Gene), intersect(unique(cn_melt$Gene), unique(ge_melt$Gene)))
# ccle <- MergeOmicsG2P(database = "CCLE", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = all_genes, outfile = "./../htscreens_giant_files/ccle_data_19Q2.rds")
# ctrp <- MergeOmicsG2P(database = "CTRP", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = all_genes, outfile = "./../htscreens_giant_files/ctrp_data_19Q2.rds")
# gdsc <- MergeOmicsG2P(database = "GDSC", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = all_genes, outfile = "./../htscreens_giant_files/gdsc_data_19Q2.rds")
```

```{r load_drug_rds, eval=TRUE}
ccle <- readRDS("./data_munging/19Q2/ccle_data_19Q2.rds") %>% drop_na(Drug)
ctrp <- readRDS("./../htscreens_giant_files/ctrp_data_19Q2.rds") %>% drop_na(Drug)
gdsc <- readRDS("./data_munging/19Q2/gdsc_data_19Q2.rds") %>% drop_na(Drug)

# Testing
test_ccle <- ccle %>% count(Drug)
test_ctrp <- ctrp %>% count(Drug)
test_gdsc <- gdsc %>% count(Drug)
```

How many datasets screen each G2P drug?
```{r drug_testing, eval=FALSE}
drug_summ <- data.frame(Drug = character(), Dataset_Count = integer(), stringsAsFactors = FALSE)

for(drug in g2p_drugs) {
  counter = 0
  if(drug %in% as.character(unique(ccle$Drug[!is.na(ccle$Drug)]))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(ctrp$Drug[!is.na(ctrp$Drug)]))) { counter <- counter + 1 }
  if(drug %in% as.character(unique(gdsc$Drug[!is.na(gdsc$Drug)]))) { counter <- counter + 1 }
  drug_summ <- rbind(drug_summ, data.frame(Drug = drug, Dataset_Count = counter))
}
```

## CRISPR

```{r merge_crispr_data, eval=FALSE}
crispr_all <- read.delim("./data_munging/19Q2/Achilles_gene_effect_20190514.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr_all) <- gsub(" .*", "", colnames(crispr_all))
colnames(crispr_all)[1] <- "DepMap_ID"

# Melt CRISPR dataset for merging
# crispr_melt <- melt(crispr_all, id.vars = "DepMap_ID", measure.vars = colnames(crispr_all)[2:ncol(crispr_all)], variable.name = "Gene", value.name = "Score")

crispr <- MergeOmicsG2P(database = "CRISPR", maf_df = maf_df, ge_df = ge_melt, cn_df = cn_melt, gene_list = g2p_genes, outfile = "./data_munging/19Q2/crispr_data_19Q2.rds")
write.csv.gz(crispr %>% drop_na(Score), "./../htscreens_giant_files/crispr_data_19Q2.csv.gz")
```

```{r read_crispr_data}
crispr <- readRDS("./data_munging/19Q2/crispr_data_19Q2.rds")
```


# EXPLORATORY FIGURES

--------------------------------------------------------------------------------

## Mutation

## Cell line and drug overlaps

```{r ccl_venn, eval=FALSE}
ccle_ccls <- unique(ccle$DepMap_ID)
ctrp_ccls <- unique(ctrp$DepMap_ID)
gdsc_ccls <- unique(gdsc$DepMap_ID)
crispr_ccls <- unique(crispr$DepMap_ID)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_ccls, ctrp_ccls, gdsc_ccls, crispr_ccls),
             filename = "./plots/manuscript/all_venn_ccls.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1,
             category.names = c("CCLE", "CTRP", "GDSC", "CRISPR"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             # cat.dist = c(0.06, 0.06, 0.035),
             fill = c("springgreen4", "steelblue4", "turquoise4", "darkorchid"))
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_ccls, ctrp_ccls, gdsc_ccls),
             filename = "./plots/manuscript/drug_venn_ccls.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1.4,
             category.names = c("CCLE", "CTRP", "GDSC"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             cat.dist = c(0.06, 0.06, 0.035),
             fill = c("springgreen4", "steelblue4", "turquoise4"))
```

```{r drug_venn, eval=FALSE}
ccle_drugs <- unique(ccle$Drug)
ctrp_drugs <- unique(ctrp$Drug)
gdsc_drugs <- unique(gdsc$Drug)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn.diagram(x = list(ccle_drugs, ctrp_drugs, gdsc_drugs),
             filename = "./plots/manuscript/drug_venn_drugs.png",
             output = TRUE, imagetype = "png",
             height = 2000, width = 2000, resolution = 500,
             lwd = 1, cex = 1.4,
             category.names = c("CCLE", "CTRP", "GDSC"),
             cat.cex = 1.4, cat.fontface = "bold", cat.default.pos = "outer",
             cat.dist = c(0.06, 0.06, 0.035),
             fill = c("palegreen3", "slategray3", "paleturquoise3"))
```

## Histology distributions

**RUN THIS WHEN CRISPR IS DONE MERGING (5/17)**
```{r histology_dists_data, eval=FALSE}
ccle_hists <- unique(select(ccle, DepMap_ID, disease)) %>% count(disease)
ccle_hists$Perc <- ccle_hists$n / sum(ccle_hists$n) * 100
ccle_hists$Dataset <- "CCLE"
ctrp_hists <- unique(select(ctrp, DepMap_ID, disease)) %>% count(disease)
ctrp_hists$Perc <- ctrp_hists$n / sum(ctrp_hists$n) * 100
ctrp_hists$Dataset <- "CTRP"
gdsc_hists <- unique(select(gdsc, DepMap_ID, disease)) %>% count(disease)
gdsc_hists$Perc <- gdsc_hists$n / sum(gdsc_hists$n) * 100
gdsc_hists$Dataset <- "GDSC"
crispr_hists <- unique(select(crispr, DepMap_ID, disease)) %>% count(disease)
crispr_hists$Perc <- crispr_hists$n / sum(crispr_hists$n) * 100
crispr_hists$Dataset <- "CRISPR"

drug_hists <- rbind(ccle_hists, ctrp_hists, gdsc_hists, crispr_hists) %>% drop_na(disease)
drug_hists$Dataset <- factor(drug_hists$Dataset, levels = c("CCLE", "CTRP", "GDSC", "CRISPR"))
drug_hists$disease <- as.character(drug_hists$disease)
drug_hists$disease <- gsub(" cancer", "", drug_hists$disease)
saveRDS(drug_hists, "./data_munging/rds/histology_distributions.rds")
```

```{r histology_dists, fig.height=4.5, fig.width=4, eval=FALSE}
drug_hists <- readRDS("./data_munging/rds/histology_distributions.rds")
png(filename = "./plots/manuscript/all_histology_distributions.png", width = 8, height = 9, units = "in", res = 500)
ggplot(drug_hists, aes(x = Dataset, y = n, fill = disease)) +
  geom_bar(stat = "identity", color = "black") +
  scale_y_continuous(breaks = seq(0, 850, by = 50), labels = seq(0, 850, by = 50)) +
  guides(fill = guide_legend(title = "Histology", ncol = 1)) +
  labs(y = "Count")
dev.off()
```

**RUN THIS WHEN CRISPR IS DONE MERGING (5/17)**
```{r, eval=FALSE}
ccle_ccls <- unique(ccle$DepMap_ID)
ctrp_ccls <- unique(ctrp$DepMap_ID)
gdsc_ccls <- unique(gdsc$DepMap_ID)
crispr_ccls <- unique(crispr$DepMap_ID)

ccls <- data.frame(DepMap_ID = union(ccle_ccls, union(ctrp_ccls, gdsc_ccls)))
all_ccls <- merge(ccls, ccl_meta, by = "DepMap_ID")
all_ccls$disease <- gsub(" cancer", "", all_ccls$disease)

png(filename = "./plots/manuscript/drug_histology_distributions.png", width = 8, height = 4, units = "in", res = 500)
ggplot(all_ccls, aes(x = fct_infreq(factor(disease)), fill = disease)) +
  geom_bar(color = "black") +
  scale_y_continuous(breaks = seq(0, 220, by = 40), labels = seq(0, 220, by = 40)) +
  guides(fill = FALSE) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
  labs(y = "Number of Cell Lines", x = "Histology", caption = paste0(formatC(nrow(all_ccls), format = "d", big.mark = ","), " cell lines represented."))
dev.off()
```

## G2P summary heatmap

```{r dg_heatmap, fig.height=6, fig.width=8, warning=FALSE}
g2p_dg_counts <- g2p %>% count(Drug, Gene) %>% filter(!is.na(Drug), !is.na(Gene))
g2p_dg_counts <- dcast(g2p_dg_counts, Drug ~ Gene, value.var = "n", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
g2p_dg_counts <- g2p_dg_counts[order(rowSums(-g2p_dg_counts)), order(colSums(-g2p_dg_counts))]
# g2p_dg_counts[g2p_dg_counts == 0] <- NA

g2p_gene_order <- colnames(g2p_dg_counts)
g2p_drug_order <- rownames(g2p_dg_counts)

# Create annotation layer
hta_data <- data.frame(Drug = rownames(g2p_dg_counts), CCLE = NA, CTRP = NA, GDSC = NA)
hta_data$CCLE <- ifelse(hta_data$Drug %in% unique(ccle$Drug), "Screened", "Not screened")
hta_data$CTRP <- ifelse(hta_data$Drug %in% unique(ctrp$Drug), "Screened", "Not screened")
hta_data$GDSC <- ifelse(hta_data$Drug %in% unique(gdsc$Drug), "Screened", "Not screened")
rownames(hta_data) <- hta_data$Drug
hta_data$Drug <- NULL

hta_annot <- HeatmapAnnotation(text = anno_text(colnames(hta_data), offset = unit(0.1, "in"), gp = gpar(fontsize = 10)))
hta_legend_list <- list(legend_direction = "horizontal", border = "gray70", nrow = 2)
hta <- Heatmap(hta_data,
               name = "Functional screen status           ",
               col = c("Screened" = "purple4", "Not screened" = "gray97"), 
               rect_gp = gpar(col = "gray70"),
               top_annotation = hta_annot,
               heatmap_legend_param = hta_legend_list,
               row_names_side = "left",
               row_names_gp = gpar(fontsize = 10),
               column_names_side = "top",
               column_names_gp = gpar(fontsize = 10),
               show_column_names = FALSE,
               cluster_rows = FALSE, 
               cluster_columns = FALSE,
               row_title = "Drugs",
               row_title_gp = gpar(fontface = "bold", fontsize = 14),
               column_title = "Dataset",
               column_title_gp = gpar(fontface = "bold", fontsize = 14),
               width = unit(1.5, "in"))

# Define color scale
col_fun <- colorRamp2(c(0, 1, 100), c("gray97", "mistyrose", "purple4"))
# Legend properties list
ht_legend_list <- list(legend_direction = "horizontal", legend_width = unit(2.5, "in"), border = "gray70")
ht <- Heatmap(g2p_dg_counts, 
              name = "Number of G2P interpretations",
              col = col_fun,
              na_col = "gray97",
              rect_gp = gpar(col = "gray70"),
              heatmap_legend_param = ht_legend_list,
              cluster_rows = FALSE, 
              cluster_columns = FALSE, 
              show_row_names = FALSE,
              row_names_side = "left",
              row_names_gp = gpar(fontsize = 10),
              column_names_side = "bottom",
              column_names_gp = gpar(fontsize = 10),
              show_column_names = FALSE,
              bottom_annotation = HeatmapAnnotation(
                  text = anno_text(colnames(g2p_dg_counts),
                                   rot = 45,
                                   gp = gpar(fontsize = 10),
                                   offset = unit(1, "npc"),
                                   just = "right"),
                  annotation_height = unit(0.45, "in")),
              row_title = "Drugs",
              row_title_gp = gpar(fontface = "bold", fontsize = 14),
              column_title = "Genes", 
              column_title_gp = gpar(fontface = "bold", fontsize = 14),
              column_title_side = "bottom")

png(filename = "./plots/manuscript/g2p_summary_heatmap.png", width = 10, height = 9, units = "in", res = 500)
draw(hta + ht, gap = unit(5, "mm"), heatmap_legend_side = "top")
dev.off()
```

## Clustered AUC heatmap

```{r auc_clustered_ht, eval=FALSE}
ccle_auc_grid <- dcast(ccle, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
ctrp_auc_grid <- dcast(ctrp, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
gdsc_auc_grid <- dcast(gdsc, Drug ~ DepMap_ID, value.var = "Score", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)

ccle_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(ccle_auc_grid)), select(ccl_meta, DepMap_ID, disease), by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)
ctrp_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(ctrp_auc_grid)), select(ccl_meta, DepMap_ID, disease), by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)
gdsc_lineages_ha <- merge(data.frame("DepMap_ID" = colnames(gdsc_auc_grid)), select(ccl_meta, DepMap_ID, disease), by = "DepMap_ID", all.x = TRUE) %>% `rownames<-`(.[,1]) %>% select(-DepMap_ID)

ht_legend_list <- list(direction = "horizontal")

ccle_col_fun <- colorRamp2(c(0, 1), c("gray97", "darkgreen"))
ccle_auc_ha <- HeatmapAnnotation(Lineage = ccle_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 5)))
ccle_auc_ht <- Heatmap(ccle_auc_grid,
                       na_col = "black",
                       col = ccle_col_fun,
                       top_annotation = ccle_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

ctrp_col_fun <- colorRamp2(c(0, 1), c("gray97", "steelblue4"))
ctrp_auc_ha <- HeatmapAnnotation(Lineage = ctrp_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 3)))
ctrp_auc_ht <- Heatmap(ctrp_auc_grid,
                       na_col = "black",
                       col = ctrp_col_fun,
                       top_annotation = ctrp_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

gdsc_col_fun <- colorRamp2(c(0, 1), c("gray97", "turquoise4"))
gdsc_auc_ha <- HeatmapAnnotation(Lineage = gdsc_lineages_ha$disease, annotation_legend_param = list(Lineage = list(direction = "horizontal", nrow = 3)))
gdsc_auc_ht <- Heatmap(gdsc_auc_grid,
                       na_col = "black",
                       col = gdsc_col_fun,
                       top_annotation = gdsc_auc_ha,
                       heatmap_legend_param = ht_legend_list,
                       name = "AUC",
                       show_column_names = FALSE, 
                       row_title = "Drugs and Small Molecules",
                       column_title = "Cancer Cell Lines",
                       column_title_side = "top")

png(filename = "./plots/manuscript/ccle_clustered_heatmap.png", width = 10, height = 8, units = "in", res = 500)
draw(ccle_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()

png(filename = "./plots/manuscript/ctrp_clustered_heatmap.png", width = 22, height = 40, units = "in", res = 500)
draw(ctrp_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()

png(filename = "./plots/manuscript/gdsc_clustered_heatmap.png", width = 18, height = 20, units = "in", res = 500)
draw(gdsc_auc_ht, merge_legend = TRUE, heatmap_legend_side = "top", annotation_legend_side = "top")
dev.off()
```


# WILCOXON (MUT)

--------------------------------------------------------------------------------

```{r wilcoxon_functions}
adj_signif <- function(df, alpha) {
  # Takes df from compare_means and creates a signif code column for FDR-adjusted p-vals
  df$p.signif.adj <- ifelse(df$p.adj <= alpha, "*", NA)
  df$p.signif <- ifelse(df$p.signif == "ns", NA, df$p.signif)
  df$p.short <- formatC(df$p, format = "g", digits =  2)
  df$p.adj.short <- formatC(df$p.adj, format = "g", digits =  2)
  return(df)
}

WilcoxonTestDF <- function(dataframe, grouping, outfile) {
  require(ggpubr)
  if("Drug_Gene_Lin" %in% grouping) {
    # Filter out DGLs with too few samples in at least one mutation status category
    count_mut_n <- dataframe %>% count(Drug_Gene_Lin, Mutation) %>% filter(n >= 3)
    count_mut_category <- count_mut_n %>% count(Drug_Gene_Lin) %>% filter(n == 2)
    dgl_keep <- intersect(count_mut_n$Drug_Gene_Lin, count_mut_category$Drug_Gene_Lin)
    dataframe <- filter(dataframe, Drug_Gene_Lin %in% dgl_keep)
    dataframe$Drug_Gene_Lin <- as.character(dataframe$Drug_Gene_Lin)
  }
  signif <- compare_means(Score ~ Mutation, group.by = grouping, data = dataframe, paired = FALSE, method = "wilcox.test", p.adjust.method = "BH")
  signif <- adj_signif(signif, alpha = 0.05)
  signif <- signif[order(signif$p),]
  if("Drug_Gene" %in% grouping) {
    signif$InG2P <- ifelse(signif$Drug_Gene %in% g2p_dg, "Yes", "No")
  }
  if("Drug_Gene_Lin" %in% grouping) {
    signif$InG2P <- ifelse(signif$Drug_Gene_Lin %in% g2p_dgl, "Yes", "No")
  }
  saveRDS(signif, outfile)
  return(signif)
}
```

```{r wilcoxon_tests, eval=FALSE}
crispr <- unfactorize(crispr)
crispr_signif_all_gene <- WilcoxonTestDF(dataframe = crispr, grouping = "Gene", outfile = "./data_munging/rds/crispr_signif_g.rds")
crispr_filt <- filter(crispr, Gene %in% g2p_genes)
crispr_signif_gene <- WilcoxonTestDF(dataframe = crispr_filt, grouping = "Gene", outfile = "./data_munging/rds/crispr_signif_g2p_g.rds")
crispr_signif_lineage <- WilcoxonTestDF(dataframe = crispr_filt, grouping = c("Gene", "disease"), outfile = "./data_munging/rds/crispr_signif_g2p_gl.rds")

ccle_filt <- filter(ccle, Drug %in% g2p_drugs & Gene %in% g2p_genes)
ccle_filt$Gene <- factor(ccle_filt$Gene, ordered = FALSE)
ccle_filt <- unfactorize(ccle_filt)
ccle_signif_dg <- WilcoxonTestDF(dataframe = ccle, grouping = c("Drug", "Gene", "Drug_Gene"), outfile = "./data_munging/rds/ccle_signif_dg.rds")
ccle_signif_g2p_dg <- WilcoxonTestDF(dataframe = ccle_filt, grouping = c("Drug", "Gene", "Drug_Gene"), outfile = "./data_munging/rds/ccle_signif_g2p_dg.rds")
ccle_signif_g2p_dgl <- WilcoxonTestDF(dataframe = ccle_filt, grouping = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), outfile = "./data_munging/rds/ccle_signif_g2p_dgl.rds")

ctrp_filt <- filter(ctrp, Drug %in% g2p_drugs & Gene %in% g2p_genes)
ctrp_filt$Gene <- factor(ctrp_filt$Gene, ordered = FALSE)
ctrp_filt <- unfactorize(ctrp_filt)
ctrp_signif_dg <- WilcoxonTestDF(dataframe = ctrp, grouping = c("Drug", "Gene", "Drug_Gene"), outfile = "./data_munging/rds/ctrp_signif_dg.rds")
ctrp_signif_g2p_dg <- WilcoxonTestDF(dataframe = ctrp_filt, grouping = c("Drug", "Gene", "Drug_Gene"), outfile = "./data_munging/rds/ctrp_signif_g2p_dg.rds")
ctrp_signif_g2p_dgl <- WilcoxonTestDF(dataframe = ctrp_filt, grouping = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), outfile = "./data_munging/rds/ctrp_signif_g2p_dgl.rds")

gdsc_filt <- filter(gdsc, Drug %in% g2p_drugs & Gene %in% g2p_genes)
gdsc_filt$Gene <- factor(gdsc_filt$Gene, ordered = FALSE)
gdsc_filt <- unfactorize(gdsc_filt)
gdsc_signif_dg <- WilcoxonTestDF(dataframe = gdsc, grouping = c("Drug", "Gene", "Drug_Gene"), outfile = "./data_munging/rds/gdsc_signif_dg.rds")
gdsc_signif_g2p_dg <- WilcoxonTestDF(dataframe = gdsc_filt, grouping = c("Drug", "Gene", "Drug_Gene"), outfile = "./data_munging/rds/gdsc_signif_g2p_dg.rds")
gdsc_signif_g2p_dgl <- WilcoxonTestDF(dataframe = gdsc_filt, grouping = c("Drug", "Gene", "disease", "Drug_Gene_Lin"), outfile = "./data_munging/rds/gdsc_signif_g2p_dgl.rds")
```

```{r load_wilcoxon_rds}
# crispr_signif_gene <- readRDS("./data_munging/rds/crispr_signif_g2p_gene.rds")
# crispr_signif_lineage <- readRDS("./data_munging/rds/crispr_signif_g2p_lineage.rds")

# ccle_signif_gene <- readRDS("./data_munging/rds/ccle_signif_g2p_gene.rds")
# ccle_signif_g2p_druggene <- readRDS("./data_munging/rds/ccle_signif_g2p_druggene.rds")
# ccle_signif_lineage <- readRDS("./data_munging/rds/ccle_signif_g2p_lineage.rds")
# 
# ctrp_signif_gene <- readRDS("./data_munging/rds/ctrp_signif_g2p_gene.rds")
# ctrp_signif_g2p_druggene <- readRDS("./data_munging/rds/ctrp_signif_g2p_druggene.rds")
# ctrp_signif_lineage <- readRDS("./data_munging/rds/ctrp_signif_g2p_lineage.rds")
# 
# gdsc_signif_gene <- readRDS("./data_munging/rds/gdsc_signif_g2p_gene.rds")
# gdsc_signif_g2p_druggene <- readRDS("./data_munging/rds/gdsc_signif_g2p_druggene.rds")
# gdsc_signif_lineage <- readRDS("./data_munging/rds/gdsc_signif_g2p_lineage.rds")
```

# SPEARMAN (CN, GE)

--------------------------------------------------------------------------------

```{r drug_spear_calc, warning=FALSE, eval=FALSE}
g2p_druggene_tally <- g2p %>% group_by(Drug_Gene) %>% tally() %>% drop_na()

# CCLE
ccle_ge_res <- ccle_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>% summarize(
                GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$estimate,
                GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$p.value)
ccle_cn_res <- ccle_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>% summarize(
                CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$estimate,
                CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$p.value)
ccle_cor_res <- merge(g2p_druggene_tally, merge(ccle_ge_res, ccle_cn_res, by = c("Drug", "Gene", "Drug_Gene"), all = TRUE), by = "Drug_Gene", all.y = TRUE)
ccle_cor_res$n <- ifelse(is.na(ccle_cor_res$n), 0, ccle_cor_res$n)
ccle_cor_res$InG2P <- ifelse(ccle_cor_res$n != 0, "Yes", "No")
ccle_cor_res$Dataset <- "CCLE"

# CTRP
ctrp_ge_res <- ctrp_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>% summarize(
                GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$estimate,
                GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$p.value)
ctrp_cn_res <- ctrp_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>% summarize(
                CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$estimate,
                CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$p.value)
ctrp_cor_res <- merge(g2p_druggene_tally, merge(ctrp_ge_res, ctrp_cn_res, by = c("Drug", "Gene", "Drug_Gene"), all = TRUE), by = "Drug_Gene", all.y = TRUE)
ctrp_cor_res$n <- ifelse(is.na(ctrp_cor_res$n), 0, ctrp_cor_res$n)
ctrp_cor_res$InG2P <- ifelse(ctrp_cor_res$n != 0, "Yes", "No")
ctrp_cor_res$Dataset <- "CTRP"

# GDSC
gdsc_ge_res <- gdsc_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(TPM) %>% summarize(
                GE_r = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$estimate,
                GE_p = cor.test(y = Score, x = TPM, method = "spearman", use = "na.or.complete")$p.value)
gdsc_cn_res <- gdsc_filt %>% group_by(Drug, Gene, Drug_Gene) %>% drop_na(CN) %>% summarize(
                CN_r = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$estimate,
                CN_p = cor.test(y = Score, x = CN, method = "spearman", use = "na.or.complete")$p.value)
gdsc_cor_res <- merge(g2p_druggene_tally, merge(gdsc_ge_res, gdsc_cn_res, by = c("Drug", "Gene", "Drug_Gene"), all = TRUE), by = "Drug_Gene", all.y = TRUE)
gdsc_cor_res$n <- ifelse(is.na(gdsc_cor_res$n), 0, gdsc_cor_res$n)
gdsc_cor_res$InG2P <- ifelse(gdsc_cor_res$n != 0, "Yes", "No")
gdsc_cor_res$Dataset <- "GDSC"

# All
g2p_cor <- rbind(ccle_cor_res, ctrp_cor_res, gdsc_cor_res)

saveRDS(g2p_cor, "./data_munging/rds/drug_cor.rds")
```

```{r load_drug_spear}
g2p_cor <- readRDS("./data_munging/rds/drug_cor.rds")
```

Spearman correlations for gene expression and copy number:
```{r crispr_spear_calc, warning=FALSE}
crispr_spear_ge <- transform(crispr %>% group_by(Gene) %>% drop_na(TPM) %>% summarize(p = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$p.value, p_log10 = -log10(p), corr = cor.test(y = Score, x = TPM, method = "spearman", use = "complete.obs")$estimate, Metric = "Gene Expression"), Gene = reorder(Gene, corr))
crispr_spear_cn <- transform(crispr %>% group_by(Gene) %>% drop_na(CN) %>% summarize(p = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$p.value, p_log10 = -log10(p), corr = cor.test(y = Score, x = CN, method = "spearman", use = "complete.obs")$estimate, Metric = "Copy Number"), Gene = reorder(Gene, corr))
c("white", "deepskyblue4")

ggplot(data = filter(crispr_spear_cn, Gene %in% g2p_genes)) +
  geom_point(aes(x = Gene, y = corr, fill = p_log10), pch = 21) +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), legend.position = "bottom") +
  scale_fill_gradientn(colours = c("white", "green4"),
                       breaks = c(0, 4, 8, 12, 16, 20),
                       labels = c(0, 4, 8, 12, 16, 20),
                       limits = c(0, 21)) +
  geom_text_repel(data = filter(crispr_spear_cn, Gene %in% g2p_genes & p < 0.05) , mapping = aes(x = Gene, y = corr, label = Gene)) +
  labs(title = "Copy Number",
       fill = expression(-log[10](p)),
       x = "Gene",
       y = expression("Spearman Correlation Coefficient"~(r[s])))
```

# SESSION INFORMATION

--------------------------------------------------------------------------------

```{r session_info}
print(sessionInfo())
```