---
title: "Lineage Analysis of CRISPR and shRNA Screens"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the Broad's Avana CRISPR (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017) and the Broad and Dana-Farber Cancer Institite's Achilles shRNA (MacFarland et al. 2018, Data Science 2018).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

In a previous version of the shRNA screen, DEMETER ([GitHub](https://github.com/cancerdatasci/demeter)) was used to compute a dependency score for each gene by using the depletion values from each shRNA to infer the effect of target knockdown (on-target) and of expressing a given miRNA seed (off-target) in each cell line. More negative values indicate increased dependency, while more positive values indicate lower dependency. Zero represents the avergae dependency across all cell lines.

In the new data release, [DEMETER2](https://depmap.org/R2-D2/) ([GitHub repository](https://github.com/cancerdatasci/demeter2)), developed by McFarland et al. (2018) was used to analyze the Achilles screen. DEMETER2 expands on DEMETER by including parameters for cell-line-specific screen effects and noisy data, correcting for global differences in shRNA levels across cell lines, pooling data acorss cell lines via hierarchical modeling, and using Bayesian inference to compute uncertainty estimates. Now, a score of zero represents no dependency.

All annotation files (copy number, mutation status, and gene expression) (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/).

# Set up

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)

num_cores <- parallel::detectCores()
options(mc.cores = num_cores, expressions = 5e5)
```

Load libraries.
```{r}
library(broom)
library(bsselectR)
library(caret)
library(CePa)
library(circlize)
library(ComplexHeatmap)
library(data.table)
library(devtools)
library(ggpubr)
library(ggsignif)
library(ggfortify)
library(ggord)
library(glmnet)
library(gridExtra)
library(jsonlite)
library(jsonlite)
library(kableExtra)
library(magrittr)
library(matrixStats)
library(NMF)
library(parallel)
library(plyr)
library(reshape2)
library(rowr)
library(tidyverse)
```

# Functions {.tabset .tabset-fade .tabset-pills}

--------------------------------------------------------------------------------

## Significance thresholding

```{r}
adj_signif <- function(df) {
  # Takes df from compare_means and creates a signif code column for adjusted p-vals
  df$p.signif <- ifelse(df$p.signif == "ns", NA, df$p.signif)
  df$p.signif.adj <- ifelse(df$p.adj <= 0.0001, "****",
                     ifelse(df$p.adj <= 0.001, "***",
                     ifelse(df$p.adj <= 0.01, "**",
                     ifelse(df$p.adj <= 0.05, "*", NA))))
  df$p.short <- formatC(df$p, format = "g", digits =  2)
  df$p.adj.short <- formatC(df$p.adj, format = "g", digits =  2)
  
  return(df)
}
```

## Make CRISPR grouped plots

```{r}
makeCRISPRgrob <- function(g, filt_method) {
  if(filt_method == "lvlA") {
    gene <- filter(crispr_data_lvlA, Hugo_Symbol == g)
    sig <- filter(crispr_signif_lvlA, Hugo_Symbol == g)
  }
  if(filt_method == "all") {
    gene <- filter(crispr_data_ptmuts, Hugo_Symbol == g)
    sig <- filter(crispr_signif_nonsilent, Hugo_Symbol == g)
  }
  
  # Mutation status
  crispr_color <- as.character(gene$Color_Nonsilent)
  names(crispr_color) <- gene$Mutation_Status_Nonsilent

  plot_mut <- ggplot(data = gene, aes(x = Mutation_Status_Nonsilent, y = Score, color = Mutation_Status_Nonsilent)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(alpha = 0.3, size = 0.7, position = position_jitter(w = 0.05)) +
    scale_color_manual(values = crispr_color) +
    geom_hline(yintercept = 0, linetype = 2, lwd = 0.3) +
    theme_light() +
    theme(legend.position = "none") +
    labs(x = paste0(g, " Mutation Status"), y = "CERES Score",
         title = "Mutation Status",
         subtitle = paste0("Wilcoxon test:\n- BH-corrected p-value: ", sig$p.adj.short, "\n- Uncorrected p-value: ", sig$p.short))
  
  # Copy number
  lo_cn <- range(gene$Copy_Number[!is.na(gene$Copy_Number)])[1]
  hi_cn <- range(gene$Copy_Number[!is.na(gene$Copy_Number)])[2]
  mid_cn <- (hi_cn - lo_cn) / 2
  plot_cn <- ggplot(data = gene, aes(x = Copy_Number, y = Score, color = Mutation_Status_Nonsilent)) +
    geom_point(size = 0.5, alpha = 0.5) +
    geom_smooth(method = "lm", size = 0.5) +
    geom_hline(yintercept = 0, linetype = 2, lwd = 0.3) +
    scale_color_manual(values = crispr_color) +
    stat_cor(method = "pearson", show.legend = FALSE, label.x = c((lo_cn + mid_cn) / 2, (hi_cn - mid_cn) / 2 + mid_cn), label.y = max(gene$Score)) +
    theme(legend.position = "none") +
    labs(y = "CERES Score", color = "Mutation Status",
         x = paste0(g, " Theoretical Copy number"), title = "Copy Number",
         subtitle = "r: Pearson correlation coeffcient")
  
  # Gene expression
  lo_ge <- range(gene$RPKM_log2[!is.na(gene$RPKM_log2)])[1]
  hi_ge <- range(gene$RPKM_log2[!is.na(gene$RPKM_log2)])[2]
  mid_ge <- (hi_ge - lo_ge) / 2
  plot_ge <- ggplot(data = gene, aes(x = RPKM_log2, y = Score, color = Mutation_Status_Nonsilent)) +
    geom_point(size = 0.5, alpha = 0.5) +
    geom_smooth(method = "lm", size = 0.5) +
    geom_hline(yintercept = 0, linetype = 2, lwd = 0.3) +
    scale_color_manual(values = crispr_color) +
    stat_cor(method = "pearson", show.legend = FALSE, label.x = c((lo_ge + mid_ge) / 2, (hi_ge - mid_ge) / 2 + mid_ge), label.y = max(gene$Score)) +
    theme(legend.position = "none") +
    labs(y = "CERES Score", color = "Mutation Status",
         x = paste0(g, " Gene Expression [log2(RPKM)]"),
         title = "Gene Expression",
         subtitle = "r: Pearson correlation coeffcient")
  
  # Cell line lineage
  plot_tissue <- ggplot(data = gene, aes(x = primary_tissue, y = Score, color = Mutation_Status_Nonsilent)) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = crispr_color) +
    geom_hline(yintercept = 0, linetype = 2, lwd = 0.3) +
    scale_y_continuous(sec.axis = sec_axis(~ .)) +
    coord_flip() +
    theme(legend.position = "none") +
    labs(y = "CERES Score", x = "Primary Tissue",
         color = "Mutation Status", title = g)
  
  # Arrange plots
  plot <- ggarrange(ggarrange(plot_tissue, nrow = 1, labels = c("A")),
                    ggarrange(plot_mut, plot_cn, plot_ge, nrow = 3,
                              labels = c("B", "C", "D"), heights = c(2, 3, 3)),
                    font.label = list(size = 30, face = "bold"),
                    nrow = 1, ncol = 2, widths = c(3, 2))
  return(plot)
}
```

## Make sorted lineage plots

```{r}
makeCRISPRlinplot <- function(g) {
  sig <- filter(crispr_signif_lineage, Hugo_Symbol == g)
  if(nrow(sig) == 0) {
    return(NULL)
  }
  else {
    gene <- filter(crispr_data, Hugo_Symbol == g)
    gene <- merge(gene, sig, by = c("Hugo_Symbol", "group_general_lineage_name"))
    gene <- mutate(gene, group_general_lineage_name = reorder(group_general_lineage_name, p, mean))
    crispr_color <- as.character(gene$Color)
    names(crispr_color) <- gene$Mutation_Status
    score_range <- abs(range(gene$Score)[2] - range(gene$Score)[1])
    round_accuracy <- ifelse(score_range <= 2, 0.25,
                             ifelse(score_range <= 3, 0.5, 1.0))
    
    plot <- ggplot(data = gene, aes(x = group_general_lineage_name, y = Score)) +
      geom_point(alpha = 0.5, mapping = aes(color = Mutation_Status)) +
      scale_color_manual(values = crispr_color) +
      coord_cartesian(y = c(min(gene$Score), round_any(x = max(gene$Score), accuracy = round_accuracy, f = ceiling))) +
      geom_signif(data = gene, mapping = aes(xmin = group_general_lineage_name, xmax = group_general_lineage_name, annotations = paste(p.short, "\n", p.adj.short), y_position = max(gene$Score)), manual = TRUE, tip_length   = 0, size = 0, textsize = 3) +
      theme(legend.position = "top", axis.text.x = element_text(angle = 70, hjust = 1, size = 10)) +
      labs(y = "CERES Score",
           x = "Lineage",
           color = "Mutation Status",
           title = paste0(g, ": CERES score by cell line lineage"),
           subtitle = "Lineages sorted by increasing p-value; labels indicate unadjusted p-value / BH-corrected p-value.\nMutant lines harbor deleterious mutations.")
    return(plot)
  }
}
```

## Make drug screen boxplots
```{r}
makeDrugBoxplots <- function(drug, dataset) {
  if(dataset == "CCLE" | dataset == "ccle") {
    use_data <- filter(ccle_ptmuts_lvlA_filt, Drug == drug)
    use_color <- as.character(use_data$Color_Nonsilent)
    names(use_color) <- use_data$Mutation_Status_Nonsilent
    sig <- filter(ccle_signif_lvlA, Drug == drug)
    label_text <- data.frame(p.signif.adj = sig$p.signif.adj, Hugo_Symbol = sig$Hugo_Symbol)
    label_text <- filter(label_text, Hugo_Symbol %in% genes_lvlA_filt)
    label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = genes_lvlA_filt)
  }
  else if(dataset == "CTRP" | dataset == "ctrp") {
    use_data <- filter(ctrp_ptmuts_lvlA_filt, Drug == drug)
    use_color <- as.character(use_data$Color_Nonsilent)
    names(use_color) <- use_data$Mutation_Status_Nonsilent
    sig <- filter(ctrp_signif_lvlA, Drug == drug)
    label_text <- data.frame(p.signif.adj = sig$p.signif.adj, Hugo_Symbol = sig$Hugo_Symbol)
    label_text <- filter(label_text, Hugo_Symbol %in% genes_lvlA_filt)
    label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = genes_lvlA_filt)
  }
  else if(dataset == "GDSC" | dataset == "gdsc") {
    use_data <- filter(gdsc_ptmuts_lvlA_filt, Drug == drug)
    use_color <- as.character(use_data$Color_Nonsilent)
    names(use_color) <- use_data$Mutation_Status_Nonsilent
    sig <- filter(gdsc_signif_lvlA, Drug == drug)
    label_text <- data.frame(p.signif.adj = sig$p.signif.adj, Hugo_Symbol = sig$Hugo_Symbol)
    label_text <- filter(label_text, Hugo_Symbol %in% genes_lvlA_filt)
    label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = genes_lvlA_filt)
  }
  else { return("Error: Invalid dataset.") }
    
  plot <- ggplot(data = use_data, aes(x = Mutation_Status_Nonsilent, y = AUC)) +
    facet_wrap(~ Hugo_Symbol, drop = FALSE, ncol = length(genes_lvlA_filt)) +
    geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
    geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85)) +
    scale_fill_manual(values = use_color) +
    guides(color = FALSE) +
    geom_text(data = label_text, mapping = aes(x = 1.5, y = max(use_data$AUC), label = p.signif.adj)) +
    theme(legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
    labs(fill = "Mutation Status", y = "AUC", x = "Mutation Status", title = paste0(dataset, ": ", drug))
  return(plot)
}
```


# Data management {.tabset .tabset-fade .tabset-pills}

--------------------------------------------------------------------------------

## D. Charytonowicz cell line converter

This comprehensive cancer cell line information curated by Daniel Charytonowicz.
```{r}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
```

```{r, eval=FALSE, include=FALSE}
# Editing CCL DB
# groupings <- read.delim("./data_munging/cell_line_database_completed_extra_grouped_doids_FINAL.csv", sep = "\t", header = TRUE)
# groupings <- unique(select(groupings, lineage_name, grouped_general_doid, group_general_lineage_name))
# ccl_converter <- read.delim("./data_munging/cell_line_database_FINAL_MASTER.tsv", sep = "\t", row.names = 1, header = TRUE)
# colnames(ccl_converter)[colnames(ccl_converter) == "ccle_cell_line_name"] <- "CCLE_Name"
# colnames(ccl_converter)[colnames(ccl_converter) == "broad_id"] <- "Broad_ID"
# ccl_converter <- merge(ccl_converter, groupings, by = "lineage_name", all.x = TRUE)

# Testing completeness of grouped lineage names
unique(ccl_converter$group_general_lineage_name)
test <- filter(ccl_converter, is.na(group_general_lineage_name))
unique(test$lineage_name)
test2 <- filter(ccl_converter, is.na(grouped_general_doid))
```

## CGC gene list

Select genes from the Cancer Gene Census (CGC). The list was pulled from the [International Cancer Genome Consortium (ICGC) data portal](https://dcc.icgc.org/search/g?filters=%7B%22gene%22:%7B%22curatedSetId%22:%7B%22is%22:%5B%22GS1%22%5D%7D%7D%7D) (Advaced Search > Genes > Curated Gene Set > Cancer Gene Census).
```{r}
cgc <- data_frame("Hugo_Symbol" = read.delim("./data_munging/gene-ids-for-set-Cancer Gene Census.tsv", header = FALSE, sep = "\t")[, 2])
```

## DepMap cell line metadata

```{r}
ccl_info <- read.delim("./data_munging/DepMap-2018q3-celllines.csv", sep = ",", header = TRUE, na.strings = c("", NA))
ccl_info$Primary.Disease <- gsub("\\\\", "", ccl_info$Primary.Disease)
ccl_info$Primary.Disease <- gsub("Ewings", "Ewing's", ccl_info$Primary.Disease)

# From figshare
crispr_meta <- read.delim("./data_munging/sample_info_18Q3_crispr.csv", sep = ",", header = TRUE, na.strings = c("", NA))
colnames(crispr_meta)[7] <- "CCLE_Name"
```

```{r, include=FALSE}
dan_test <- select(ccl_converter, CCLE_Name, Broad_ID, cell_line_name)
dan_test <- dan_test[-which(dan_test$CCLE_Name == ""), ]
ccl_test <- select(ccl_info, CCLE_Name, Broad_ID)

test <- merge(ccl_test, dan_test, by = "CCLE_Name", all = TRUE, suffixes = c("_DepMap", "_Daniel"))
test$BroadIDs_match <- ifelse(as.character(test$Broad_ID_DepMap) == as.character(test$Broad_ID_Daniel), TRUE, FALSE)

test_false <- filter(test, BroadIDs_match == FALSE)
# write.table(ccl_converter, file = "~/Desktop/cell_line_database_v1_20180907.tsv", quote = FALSE, sep = "\t")
```

## Genomic annotations

### CCLE MAF file

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.
```{r}
maf_raw <- read.delim("./data_munging/CCLE_DepMap_18q3_maf_20180718.txt.gz", header = TRUE, sep = "\t")
# Filter for cell lines in CRISPR screen
maf_raw <- filter(maf_raw, Broad_ID %in% unique(crispr_meta$Broad_ID))
colnames(maf_raw)[colnames(maf_raw) == "Tumor_Sample_Barcode"] <- "CCLE_Name"

# Select columns
maf_df <- subset(maf_raw, select = c("Hugo_Symbol", "CCLE_Name", "Broad_ID", "Variant_Classification", "Reference_Allele", "Tumor_Seq_Allele1"))
maf_df$Reference_Allele_Length <- ifelse(maf_df$Reference_Allele == "-", 0, nchar(as.character(maf_df$Reference_Allele)))
maf_df$Tumor_Seq_Allele1_Length <- nchar(as.character(maf_df$Tumor_Seq_Allele1))

# Add Mutation_Status column
maf_df$Mutation_Status_Nonsilent <- ifelse(maf_df$Variant_Classification == "Silent", "Other", "Mutant")
maf_df <- unique(subset(maf_df, select = c("Hugo_Symbol", "CCLE_Name", "Broad_ID", "Mutation_Status_Nonsilent", "Reference_Allele_Length", "Tumor_Seq_Allele1_Length")))

# Whole MAF summary table
maf_summ <- maf_raw[, c("Variant_Classification", "isDeleterious")] %>% group_by(Variant_Classification, isDeleterious) %>% tally()
maf_summ$Percent <- format(round(maf_summ$n / sum(maf_summ$n) * 100, 4), nsmall = 2)
maf_summ$isDeleterious <- ifelse(maf_summ$isDeleterious == "TRUE", "Yes", "No")

knitr::kable(maf_summ, caption = "Distribution of variant classifications in the MAF file") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

```{r, eval=FALSE, include=FALSE}
max(nchar(as.character(maf_raw$Reference_Allele)))
maf_raw %>% group_by(Variant_Type, isTCGAhotspot) %>% tally()

test <- filter(maf_raw, Hugo_Symbol == "KRAS")
test %>% group_by(isDeleterious) %>% tally()
```

### CCLE copy number file

```{r, eval=FALSE}
cn <- read.delim("./data_munging/public_18Q3_gene_cn_v2.csv.gz", sep = ",", check.names = FALSE, header = TRUE)

# Convert log2 ratios (log2(CN/2)) to CN
cn[2:ncol(cn)] <- lapply(cn[2:ncol(cn)], function(x) 2 * (2 ^ x))

# Remove Entrez gene IDs from colnames
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "Broad_ID"

# Melt
cn_melt <- melt(data = cn, id.vars = "Broad_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Hugo_Symbol", value.name = "Copy_Number")

saveRDS(cn_melt, "./data_munging/rds/cn_melt_18Q3.rds", compress = "xz")
```

```{r, eval=FALSE}
cn_melt <- readRDS("./data_munging/rds/cn_melt_18Q3.rds")
```

### CCLE gene expression file

Gene expression data (Reads Per Kilobase of transcript, per Million mapped reads, RPKM).
```{r, eval=FALSE}
ge <- read.delim("./../crispr_lineages_giant_files/CCLE_DepMap_18q3_RNAseq_RPKM_20180718.gct.gz", skip = 2, header = TRUE, sep = "\t", check.names = FALSE)

# Edit columns
ge$Name <- NULL
colnames(ge)[1] <- "Hugo_Symbol"

# Melt
ge_melt <- melt(data = ge, id.vars = "Hugo_Symbol", measure.vars = colnames(ge[2:ncol(ge)]), value.name = "RPKM")
## Split variable column
ge_melt <- with(ge_melt, cbind(Hugo_Symbol, colsplit(variable, pattern = " ", names = c("CCLE_Name", "Broad_ID")), RPKM))
## Remove parentheses around Broad IDs
ge_melt$Broad_ID <- gsub("\\(|\\)", "", ge_melt$Broad_ID)

saveRDS(ge_melt, "./../crispr_lineages_giant_files/ge_melt_18Q3.rds", compress = "xz")
```

```{r, eval=FALSE}
ge_melt <- readRDS("./../crispr_lineages_giant_files/ge_melt_18Q3.rds")
```

## CRISPR

### DepMap dependency probabilities

```{r, eval=FALSE}
dep <- read.delim("./data_munging/gene_dependency_18Q3.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(dep) <- gsub(" .*", "", colnames(dep))
```

### Merge CRISPR data and annotations

The latest CRISPR CERES score data (18Q3, August 2018) was pulled from the [DepMap Data Portal](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017).
```{r, eval=TRUE}
crispr <- read.delim("./data_munging/gene_effect_18Q3.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr) <- gsub(" .*", "", colnames(crispr))
```

Merge annotation data:
```{r, eval=FALSE}
# Melt CRISPR dataset for merging
crispr_melt <- melt(crispr, id.vars = "Broad_ID", measure.vars = colnames(crispr)[2:ncol(crispr)], variable.name = "Hugo_Symbol", value.name = "Score")
# Melt dependency probabilities dataset for merging
dep_melt <- melt(dep, id.vars = "Broad_ID", measure.vars = colnames(dep)[2:ncol(dep)], variable.name = "Hugo_Symbol", value.name = "Dep_Prob")
# Merge dependency probabilities
crispr_melt <- merge(crispr_melt, dep_melt, by = c("Broad_ID", "Hugo_Symbol"), all.x = TRUE)
# Merge cell line metadata
crispr_melt <- merge(crispr_melt, ccl_info, by = "Broad_ID", all.x = TRUE)
crispr_melt <- merge(crispr_melt, crispr_meta, by = c("CCLE_Name", "Broad_ID"), all.x = TRUE)
# Merge mutation annotations
crispr_muts <- merge(crispr_melt, maf_df, by = c("Hugo_Symbol", "CCLE_Name", "Broad_ID"), all.x = TRUE)
crispr_muts$Hugo_Symbol <- factor(crispr_muts$Hugo_Symbol)
crispr_muts <- crispr_muts %>% mutate(Mutation_Status_Nonsilent = if_else(is.na(Mutation_Status_Nonsilent), "Other", Mutation_Status_Nonsilent))
# Summarize number of mutant and Other cell lines
crispr_muts_summ <- crispr_muts %>% group_by(Hugo_Symbol) %>%
  summarize(N_Nonsilent_Other = sum(Mutation_Status_Nonsilent == "Other"),
            N_Nonsilent_Mutant = sum(Mutation_Status_Nonsilent == "Mutant"))
# Merge test results back into full dataset, which restores information lost in the summarization
crispr_data <- merge(crispr_muts_summ, crispr_muts, by = "Hugo_Symbol")
# Add Color columns
crispr_data$Color_Nonsilent <- ifelse(crispr_data$Mutation_Status_Nonsilent == "Other", "thistle", "darkorchid")
crispr_data$Color_Nonsilent <- factor(crispr_data$Color_Nonsilent)
# Cell line lineages
crispr_data <- merge(crispr_data, ccl_converter, by = c("CCLE_Name", "Broad_ID"), all.x = TRUE)
levels(crispr_data$lineage_name) <- sort(levels(crispr_data$lineage_name), decreasing = TRUE)
# Copy number
crispr_data <- merge(crispr_data, cn_melt, by = c("Hugo_Symbol", "Broad_ID"), all.x = TRUE)
# Gene expression (RPKM)
ge_filt <- filter(ge_melt, Hugo_Symbol %in% unique(crispr_data$Hugo_Symbol) & Broad_ID %in% unique(crispr_data$Broad_ID))
crispr_data <- merge(crispr_data, ge_filt, by = c("Hugo_Symbol", "Broad_ID", "CCLE_Name"), all.x = TRUE)
crispr_data$RPKM_log2 <- log2(crispr_data$RPKM + 0.0001)
saveRDS(crispr_data, "./../crispr_lineages_giant_files/crispr_data_18Q3.rds", compress = "xz")
```

```{r}
crispr_data <- readRDS("./../crispr_lineages_giant_files/crispr_data_18Q3.rds")
crispr_ccl <- data.frame("Broad_ID" = crispr_data$Broad_ID)
# write.table(crispr_data, file = "~/Desktop/crispr_data.tsv", quote = FALSE, sep = "\t")
```

```{r, eval=FALSE, include=FALSE}
# Code for testing colors on plots
test <- filter(crispr_data, Hugo_Symbol == "KRAS")
ggplot(data = test, aes(x = Mutation_Status_Nonsilent, y = Score, fill = Mutation_Status_Nonsilent)) +
  geom_boxplot() +
  geom_jitter(aes(color = Mutation_Status_Nonsilent)) +
  theme_light() +
  scale_color_manual(values = c("turquoise4", "paleturquoise3")) +
  scale_fill_manual(values = c("turquoise4", "paleturquoise3"))
```

```{r fig.height=12, fig.width=12, eval=FALSE, include=FALSE}
# Test dimensional scaling
crispr_pca_data <- merge(ccl_converter[, c("Broad_ID", "group_general_lineage_name")], crispr, by = "Broad_ID")
crispr_lin <- crispr_pca_data[, 2]
crispr_pca <- prcomp(data.matrix(crispr_pca_data[, 3:ncol(crispr_pca_data)]), scale. = TRUE, center = TRUE)
plot(crispr_pca, type = "l")
biplot(crispr_pca, obs.scale = 1, var.scale = 1, groups = crispr_lin, ellipse = TRUE, circle = TRUE)
ggord(crispr_pca, crispr_pca_data$group_general_lineage_name)
```

## Drug screens

```{r}
g2p <- read.delim("./data_munging/g2p_full_level_A_dataset_v2_10_10_2018.tsv", sep = "\t", header = TRUE)
genes_lvlA_df <- data.frame(Hugo_Symbol = unique(g2p$Gene))
genes_lvlA <- unique(g2p$Gene)
```

```{r, eval=FALSE}
# Drug metadata
drug_meta <- read.delim("./data_munging/drug_id_drug_name_table_10_12_2018.tsv", sep = "\t", header = FALSE)
colnames(drug_meta) <- c("CID", "Drug")
drug_meta$Drug <- tolower(drug_meta$Drug)
maf_df_lvlA <- filter(maf_df, Hugo_Symbol %in% genes_lvlA)

# CCLE
ccle <- read.delim("./data_munging/data_drug_auc_ccle_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
ccle <- merge(drug_meta, ccle, by = "CID")
ccle_melt <- melt(ccle, id.vars = c("CID", "Drug"), measure.vars = colnames(ccle)[3:ncol(ccle)], variable.name = "accession_id", value.name = "AUC")
ccle_grid <- expand.grid("accession_id" = unique(ccle_melt$accession_id), "Hugo_Symbol" = genes_lvlA)
ccle_data <- merge(ccle_grid, ccle_melt, by = "accession_id", all.x = TRUE)
ccle_data <- merge(ccle_data, ccl_converter, by = "accession_id", all.x = TRUE)
ccle_data <- merge(ccle_data, maf_df_lvlA, by = c("Hugo_Symbol", "CCLE_Name", "Broad_ID"), all.x = TRUE)
ccle_data$Mutation_Status_Nonsilent <- ifelse(is.na(ccle_data$Mutation_Status_Nonsilent), "Other", ccle_data$Mutation_Status_Nonsilent)
ccle_data$Hugo_Symbol <- factor(ccle_data$Hugo_Symbol)
ccle_data$AUC <- as.numeric(ccle_data$AUC)
ccle_data$Color_Nonsilent <- ifelse(ccle_data$Mutation_Status_Nonsilent == "Other", "palegreen3", "springgreen4")
ccle_data$Color_Nonsilent <- factor(ccle_data$Color_Nonsilent)
ccle_data <- filter(ccle_data, !is.na(ccle_data$AUC))
saveRDS(ccle_data, "./data_munging/rds/ccle_data_18Q3.rds", compress = "xz")

# CTRP
ctrp <- read.delim("./data_munging/data_drug_auc_ctrp_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
ctrp <- merge(drug_meta, ctrp, by = "CID")
ctrp_melt <- melt(ctrp, id.vars = c("CID", "Drug"), measure.vars = colnames(ctrp)[3:ncol(ctrp)], variable.name = "accession_id", value.name = "AUC")
ctrp_grid <- expand.grid("accession_id" = unique(ctrp_melt$accession_id), "Hugo_Symbol" = genes_lvlA)
ctrp_data <- merge(ctrp_grid, ctrp_melt, by = "accession_id", all.x = TRUE)
ctrp_data <- merge(ctrp_data, ccl_converter, by = "accession_id", all.x = TRUE)
ctrp_data <- merge(ctrp_data, maf_df_lvlA, by = c("Hugo_Symbol", "CCLE_Name", "Broad_ID"), all.x = TRUE)
ctrp_data$Mutation_Status_Nonsilent <- ifelse(is.na(ctrp_data$Mutation_Status_Nonsilent), "Other", ctrp_data$Mutation_Status_Nonsilent)
ctrp_data$Hugo_Symbol <- factor(ctrp_data$Hugo_Symbol)
ctrp_data$AUC <- as.numeric(ctrp_data$AUC)
ctrp_data$Color_Nonsilent <- ifelse(ctrp_data$Mutation_Status_Nonsilent == "Other", "slategray3", "steelblue4")
ctrp_data$Color_Nonsilent <- factor(ctrp_data$Color_Nonsilent)
ctrp_data <- filter(ctrp_data, !is.na(ctrp_data$AUC))
saveRDS(ctrp_data, "./data_munging/rds/ctrp_data_18Q3.rds", compress = "xz")

# GDSC
gdsc <- read.delim("./data_munging/data_drug_auc_gdsc_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
gdsc <- merge(drug_meta, gdsc, by = "CID")
gdsc_melt <- melt(gdsc, id.vars = c("CID", "Drug"), measure.vars = colnames(gdsc)[3:ncol(gdsc)], variable.name = "accession_id", value.name = "AUC")
gdsc_grid <- expand.grid("accession_id" = unique(gdsc_melt$accession_id), "Hugo_Symbol" = genes_lvlA)
gdsc_data <- merge(gdsc_grid, gdsc_melt, by = "accession_id", all.x = TRUE)
gdsc_data <- merge(gdsc_data, ccl_converter, by = "accession_id", all.x = TRUE)
gdsc_data <- merge(gdsc_data, maf_df_lvlA, by = c("Hugo_Symbol", "CCLE_Name", "Broad_ID"), all.x = TRUE)
gdsc_data$Mutation_Status_Nonsilent <- ifelse(is.na(gdsc_data$Mutation_Status_Nonsilent), "Other", gdsc_data$Mutation_Status_Nonsilent)
gdsc_data$Hugo_Symbol <- factor(gdsc_data$Hugo_Symbol)
gdsc_data$AUC <- as.numeric(gdsc_data$AUC)
gdsc_data$Color_Nonsilent <- ifelse(gdsc_data$Mutation_Status_Nonsilent == "Other", "paleturquoise3", "turquoise4")
gdsc_data$Color_Nonsilent <- factor(gdsc_data$Color_Nonsilent)
gdsc_data <- filter(gdsc_data, !is.na(gdsc_data$AUC))
saveRDS(gdsc_data, "./data_munging/rds/ctrp_data_18Q3.rds", compress = "xz")
```

```{r}
ccle_data <- readRDS("./data_munging/rds/ccle_data_18Q3.rds")
ctrp_data <- readRDS("./data_munging/rds/ctrp_data_18Q3.rds")
gdsc_data <- readRDS("./data_munging/rds/gdsc_data_18Q3.rds")
```

```{r}
test_gdsc <- gdsc_data %>% group_by(Drug, Hugo_Symbol, Mutation_Status_Nonsilent) %>% tally()
```


# Wilcoxon tests: All genes

--------------------------------------------------------------------------------

## CRISPR

Filter for point mutations:
```{r}
crispr_data_ptmuts <- filter(crispr_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
crispr_data_ptmuts$Keep_PtMut_Tests <- ifelse(crispr_data_ptmuts$Reference_Allele_Length == 1 | is.na(crispr_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(crispr_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
crispr_data_ptmuts <- filter(crispr_data_ptmuts, Keep_PtMut_Tests == TRUE)

test_ptmuts <- filter(crispr_data_ptmuts, is.na(group_general_lineage_name))
test_ptmuts <- filter(crispr_data_ptmuts, Hugo_Symbol == "KRAS")
```

### Grouped by gene

```{r, eval=FALSE}
crispr_signif_nonsilent <- compare_means(Score ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol"), data = crispr_data_ptmuts, method = "wilcox.test", p.adjust.method = "BH")
crispr_signif_nonsilent <- adj_signif(crispr_signif_nonsilent)
crispr_signif_nonsilent <- crispr_signif_nonsilent[order(crispr_signif_nonsilent$p),]
saveRDS(crispr_signif_nonsilent, "./data_munging/rds/crispr_signif_ptmuts_nonsilent_gene.rds")

# write.table(crispr_signif_nonsilent, file = "~/Desktop/crispr_signif_ptmuts_nonsilent_gene.csv", quote = FALSE, sep = ",", row.names = FALSE)
```

```{r, eval=TRUE}
crispr_signif_nonsilent <- readRDS("./data_munging/rds/crispr_signif_ptmuts_nonsilent_gene.rds")

knitr::kable(filter(crispr_signif_nonsilent, p < 0.01)[, c("Hugo_Symbol", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "Wilcoxon test results comparing non-silent mutant vs other cell lines, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8}
wilcox_gene_nonsilent_plot <- ggplot(data = crispr_signif_nonsilent) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_gene_nonsilent_plot
```

### Grouped by gene and lineage

```{r, eval=FALSE}
crispr_signif_nonsilent_lineage <- compare_means(Score ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "group_general_lineage_name"), data = crispr_data_ptmuts, method = "wilcox.test", p.adjust.method = "BH")
crispr_signif_nonsilent_lineage <- adj_signif(crispr_signif_nonsilent_lineage)
crispr_signif_nonsilent_lineage <- crispr_signif_nonsilent_lineage[order(crispr_signif_nonsilent_lineage$p),]
saveRDS(crispr_signif_nonsilent_lineage, "./data_munging/rds/crispr_signif_ptmuts_nonsilent_lineage.rds")

# write.table(crispr_signif_nonsilent_lineage, file = "~/Desktop/crispr_signif_ptmuts_nonsilent_lineage.csv", quote = FALSE, sep = ",", row.names = FALSE)
```

```{r, eval=FALSE}
crispr_signif_nonsilent_lineage <- readRDS("./data_munging/rds/crispr_signif_ptmuts_nonsilent_lineage.rds")

knitr::kable(filter(crispr_signif_nonsilent_lineage, p < 0.01)[, c("Hugo_Symbol", "group_general_lineage_name", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "Wilcoxon test results comparing non-silent mutant vs other cell lines by lineage, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8, eval=FALSE}
wilcox_lineage_nonsilent_plot <- ggplot(data = crispr_signif_nonsilent_lineage) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_lineage_nonsilent_plot
```


# G2P project

--------------------------------------------------------------------------------

## Gene mutation frequency

```{r}
crispr_data_lvlA <- filter(crispr_data_ptmuts, Hugo_Symbol %in% genes_lvlA)

crispr_data_lvlA_summ <- unique(crispr_data_lvlA[, c("Hugo_Symbol", "N_Nonsilent_Mutant", "N_Nonsilent_Other")])
crispr_data_lvlA_summ$Fraction_Mutant <- crispr_data_lvlA_summ$N_Nonsilent_Mutant / (crispr_data_lvlA_summ$N_Nonsilent_Mutant + crispr_data_lvlA_summ$N_Nonsilent_Other)
crispr_data_lvlA_summ$Percent_Mutant <- crispr_data_lvlA_summ$Fraction_Mutant * 100
# write.table(crispr_data_lvlA_summ, file = "~/Desktop/crispr_data_lvlA_summ_20181003.tsv", quote = FALSE, sep = "\t")
```

## Wilcoxon tests

### CRISPR: Gene

```{r, eval=FALSE}
crispr_signif_lvlA <- compare_means(Score ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol"), data = crispr_data_lvlA, method = "wilcox.test", p.adjust.method = "BH")
crispr_signif_lvlA <- adj_signif(crispr_signif_lvlA)
crispr_signif_lvlA <- crispr_signif_lvlA[order(crispr_signif_lvlA$p),]
saveRDS(crispr_signif_lvlA, "./data_munging/rds/crispr_signif_lvlA_gene.rds")
```

```{r, eval=TRUE}
crispr_signif_lvlA <- readRDS("./data_munging/rds/crispr_signif_lvlA_gene.rds")

crispr_signif_genes_lvlA <- as.character(crispr_signif_lvlA$Hugo_Symbol[0:20])

knitr::kable(filter(crispr_signif_lvlA, p < 0.1)[, c("Hugo_Symbol", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "Wilcoxon test results for Level A point mutations, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8}
wilcox_gene_lvlA_plot <- ggplot(data = crispr_signif_lvlA) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_gene_lvlA_plot
```

```{r fig.height=7, fig.width=12}
lvlA_genes_signif <- filter(crispr_data_ptmuts, Hugo_Symbol %in% crispr_signif_genes_lvlA)
lvlA_genes_signif <- merge(lvlA_genes_signif, crispr_signif_lvlA, by = "Hugo_Symbol")
lvlA_genes_signif$Hugo_Symbol <- ordered(lvlA_genes_signif$Hugo_Symbol, levels = crispr_signif_genes_lvlA)
lvlA_genes_signif_color <- as.character(lvlA_genes_signif$Color_Nonsilent)
names(lvlA_genes_signif_color) <- lvlA_genes_signif$Mutation_Status_Nonsilent
lvlA_genes_signif <- merge(lvlA_genes_signif, lvlA_genes_signif %>% group_by(Hugo_Symbol) %>% summarize(Text_y = max(Score)), by = "Hugo_Symbol")

lvlA_genes_text <- unique(data.frame(p.signif.adj = lvlA_genes_signif$p.signif.adj, Hugo_Symbol = lvlA_genes_signif$Hugo_Symbol, Text_y = lvlA_genes_signif$Text_y))
lvlA_genes_text$Hugo_Symbol <- factor(lvlA_genes_text$Hugo_Symbol, levels = crispr_signif_genes_lvlA)

lvlA_signif_genes_plot <- ggplot(data = lvlA_genes_signif, aes(x = Mutation_Status_Nonsilent, y = Score)) +
  facet_wrap(~ Hugo_Symbol, nrow = 2, scales = "free_y") +
  geom_jitter(aes(color = Mutation_Status_Nonsilent), alpha = 0.5, width = 0.3) +
  geom_boxplot(color = "black", fill = NA, outlier.shape = NA) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  scale_color_manual(values = lvlA_genes_signif_color) + 
  geom_text(data = lvlA_genes_text, mapping = aes(x = 1.5, y = Text_y, label = p.signif.adj), nudge_y = 0.2) +
  theme(legend.position = "top") +
  labs(y = "CERES Score",
       x = "Lineage",
       color = "Mutation Status",
       title = "CERES scores for select Level A G2P genes",
       subtitle = "Genes with adjusted p-values < 0.3 are shown.\nBH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001\nGenes are sorted by increasing p-value.")
lvlA_signif_genes_plot

# ggsave("./plots_18Q3/manuscript/crispr_g2p_lvlA_signif_genes_plot.png", lvlA_signif_genes_plot, width = 12, height = 7, units = "in")
```

### CRISPR: Gene and lineage

```{r, eval=FALSE}
crispr_signif_lvlA_lineage <- compare_means(Score ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "group_general_lineage_name"), data = crispr_data_lvlA, method = "wilcox.test", p.adjust.method = "BH")
crispr_signif_lvlA_lineage <- adj_signif(crispr_signif_lvlA_lineage)
crispr_signif_lvlA_lineage <- crispr_signif_lvlA_lineage[order(crispr_signif_lvlA_lineage$p),]
saveRDS(crispr_signif_lvlA_lineage, "./data_munging/rds/crispr_signif_lvlA_lineage.rds")

# write.table(crispr_signif_lvlA_lineage, file = "~/Desktop/crispr_signif_lvlA_lineage.csv", quote = FALSE, sep = ",", row.names = FALSE)
```

```{r, eval=TRUE}
crispr_signif_lvlA_lineage <- readRDS("./data_munging/rds/crispr_signif_lvlA_lineage.rds")

knitr::kable(filter(crispr_signif_lvlA_lineage, p < 0.01)[, c("Hugo_Symbol", "group_general_lineage_name", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "Wilcoxon test results comparing non-silent mutant vs other cell lines by lineage, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8, eval=FALSE}
wilcox_lineage_lvlA_plot <- ggplot(data = crispr_signif_lvlA_lineage) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_lineage_lvlA_plot
```

```{r fig.height=15, fig.width=15}
lvlA_signif <- filter(crispr_data_ptmuts, Hugo_Symbol %in% crispr_signif_genes_lvlA)
lvlA_signif <- merge(lvlA_signif, crispr_signif_lvlA_lineage, by = c("Hugo_Symbol", "group_general_lineage_name"))
lvlA_signif$Hugo_Symbol <- ordered(lvlA_signif$Hugo_Symbol, levels = crispr_signif_genes_lvlA)
lvlA_signif_color <- as.character(lvlA_signif$Color_Nonsilent)
names(lvlA_signif_color) <- lvlA_signif$Mutation_Status_Nonsilent
lvlA_signif <- merge(lvlA_signif, lvlA_signif %>% group_by(Hugo_Symbol) %>% summarize(Text_y = max(Score)), by = "Hugo_Symbol")

lvlA_text <- unique(data.frame(p.signif.adj = lvlA_signif$p.signif.adj, group_general_lineage_name = lvlA_signif$group_general_lineage_name, Hugo_Symbol = lvlA_signif$Hugo_Symbol, Text_y = lvlA_signif$Text_y))
lvlA_text$Hugo_Symbol <- factor(lvlA_text$Hugo_Symbol, levels = crispr_signif_genes_lvlA)

lvlA_signif_lineages_plot <- ggplot(data = lvlA_signif, aes(x = group_general_lineage_name, y = Score)) +
  facet_wrap(~ Hugo_Symbol, nrow = 4, scales = "free_y") +
  geom_point(aes(color = Mutation_Status_Nonsilent), alpha = 0.5) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  scale_color_manual(values = lvlA_signif_color) +
  geom_text(data = lvlA_text, mapping = aes(x = group_general_lineage_name, y = Text_y, label = p.signif.adj), nudge_y = 0.1) +
  theme(legend.position = "top", axis.text.x = element_text(angle = 90, hjust = 1, vjust = 1, size = 10)) +
  labs(y = "CERES Score",
       x = "Lineage",
       color = "Mutation Status",
       title = "CERES score by cell line lineage for significant Level A G2P genes",
       subtitle = "Genes with adjusted p-values < 0.3 are shown.\nBH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001\nGenes are sorted by increasing lineage-agnostic p-value.")
lvlA_signif_lineages_plot

# ggsave("./plots_18Q3/manuscript/crispr_g2p_lvlA_signif_lineages_plot.png", lvlA_signif_lineages_plot, width = 12, height = 15, units = "in")
```

```{r}
lvlA_signif_summ <- lvlA_signif %>% group_by(Hugo_Symbol, group_general_lineage_name, Mutation_Status_Nonsilent) %>% tally()
# write.table(lvlA_signif_summ, file = "~/Desktop/lvlA_signif_summ_20181005.tsv", quote = FALSE, sep = "\t")

knitr::kable(lvlA_signif_summ, caption = "Mutant/Other counts for gene-lineage combinations") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

### CCLE: Gene and drug

```{r}
ccle_data_ptmuts <- filter(ccle_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
ccle_data_ptmuts$Keep_PtMut_Tests <- ifelse(ccle_data_ptmuts$Reference_Allele_Length == 1 | is.na(ccle_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(ccle_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
ccle_data_ptmuts <- filter(ccle_data_ptmuts, Keep_PtMut_Tests == TRUE)
ccle_data_lvlA <- filter(ccle_data_ptmuts, Hugo_Symbol %in% genes_lvlA)
```

```{r, eval=FALSE}
ccle_signif_lvlA <- compare_means(AUC ~ Mutation_Status_Nonsilent, group.by = c("Drug", "Hugo_Symbol"), data = ccle_data_lvlA, method = "wilcox.test", p.adjust.method = "BH")
ccle_signif_lvlA <- adj_signif(ccle_signif_lvlA)
ccle_signif_lvlA <- ccle_signif_lvlA[order(ccle_signif_lvlA$p),]
saveRDS(ccle_signif_lvlA, "./data_munging/rds/ccle_signif_lvlA_gene.rds")
```

```{r, eval=TRUE}
ccle_signif_lvlA <- readRDS("./data_munging/rds/ccle_signif_lvlA_gene.rds")

# ccle_signif_genes_lvlA <- as.character(ccle_signif_lvlA$Hugo_Symbol[0:20])

knitr::kable(ccle_signif_lvlA[, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CCLE: Wilcoxon test results for Level A point mutations, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8}
wilcox_ccle_gene_lvlA_plot <- ggplot(data = ccle_signif_lvlA) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_ccle_gene_lvlA_plot
```

### CTRP: Gene and drug

```{r}
ctrp_data_ptmuts <- filter(ctrp_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
ctrp_data_ptmuts$Keep_PtMut_Tests <- ifelse(ctrp_data_ptmuts$Reference_Allele_Length == 1 | is.na(ctrp_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(ctrp_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
ctrp_data_ptmuts <- filter(ctrp_data_ptmuts, Keep_PtMut_Tests == TRUE)
ctrp_data_lvlA <- filter(ctrp_data_ptmuts, Hugo_Symbol %in% genes_lvlA)
```

```{r, eval=FALSE}
ctrp_signif_lvlA <- compare_means(AUC ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "Drug"), data = ctrp_data_lvlA, method = "wilcox.test", p.adjust.method = "BH")
ctrp_signif_lvlA <- adj_signif(ctrp_signif_lvlA)
ctrp_signif_lvlA <- ctrp_signif_lvlA[order(ctrp_signif_lvlA$p),]
saveRDS(ctrp_signif_lvlA, "./data_munging/rds/ctrp_signif_lvlA_gene.rds")
```

```{r, eval=TRUE}
ctrp_signif_lvlA <- readRDS("./data_munging/rds/ctrp_signif_lvlA_gene.rds")

# ctrp_signif_genes_lvlA <- as.character(ctrp_signif_lvlA$Hugo_Symbol[0:20])

knitr::kable(ctrp_signif_lvlA[, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CTRP: Wilcoxon test results for Level A point mutations, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8}
wilcox_ctrp_gene_lvlA_plot <- ggplot(data = ctrp_signif_lvlA) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_ctrp_gene_lvlA_plot
```

### GDSC: Gene and drug

```{r}
gdsc_data_ptmuts <- filter(gdsc_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
gdsc_data_ptmuts$Keep_PtMut_Tests <- ifelse(gdsc_data_ptmuts$Reference_Allele_Length == 1 | is.na(gdsc_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(gdsc_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
gdsc_data_ptmuts <- filter(gdsc_data_ptmuts, Keep_PtMut_Tests == TRUE)
gdsc_data_lvlA <- filter(gdsc_data_ptmuts, Hugo_Symbol %in% genes_lvlA)
```

```{r, eval=FALSE}
gdsc_signif_lvlA <- compare_means(AUC ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "Drug"), data = gdsc_data_lvlA, method = "wilcox.test", p.adjust.method = "BH")
gdsc_signif_lvlA <- adj_signif(gdsc_signif_lvlA)
gdsc_signif_lvlA <- gdsc_signif_lvlA[order(gdsc_signif_lvlA$p),]
saveRDS(gdsc_signif_lvlA, "./data_munging/rds/gdsc_signif_lvlA_gene.rds")
```

```{r, eval=TRUE}
gdsc_signif_lvlA <- readRDS("./data_munging/rds/gdsc_signif_lvlA_gene.rds")

# gdsc_signif_genes_lvlA <- as.character(gdsc_signif_lvlA$Hugo_Symbol[0:20])

knitr::kable(gdsc_signif_lvlA[, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "GDSC: Wilcoxon test results for Level A point mutations, p < 0.01 (BH-adjusted p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=4, fig.width=8}
wilcox_gdsc_gene_lvlA_plot <- ggplot(data = gdsc_signif_lvlA) +
  geom_histogram(aes(x = p, fill = "chartreuse4"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  geom_histogram(aes(x = p.adj, fill = "darkslategray3"), breaks = seq(0, 1, by = 0.025), color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_fill_manual(name = "P-values", values = c("chartreuse4" = "chartreuse4", "darkslategray3" = "darkslategray3"), labels = c("Unadjusted", "BH-adjusted")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1), legend.position = c(0.1, 0.85)) +
  labs(x = "BH-adjusted p-values", y = "Frequency")
wilcox_gdsc_gene_lvlA_plot
```

## Screen boxplots

Mutation status boxplots for genes identified in G2P.
```{r}
genes_lvlA_filt <- c("KRAS", "EGFR", "MET", "ALK", "BRAF", "ERBB2", "ABL1", "ROS1", "KIT", "DPYD", "PDGFRA", "RET", "BRCA1", "UGT1A1", "IDH2", "ESR1", "TPMT", "BRCA2", "PDGFRB", "G6PD", "PML", "TSC1", "FLT3", "ERBB3", "CYP19A1", "AKT1", "PTCH1", "MGMT", "DNMT3A", "MYD88", "PDGFB", "TSC2", "NPM1", "JAK2")
```


### CRISPR

```{r fig.height=7, fig.width=12}
crispr_ptmuts_lvlA_filt <- filter(crispr_data_ptmuts, Hugo_Symbol %in% genes_lvlA_filt)
crispr_ptmuts_lvlA_filt$Hugo_Symbol <- ordered(crispr_ptmuts_lvlA_filt$Hugo_Symbol, levels = genes_lvlA_filt)
crispr_ptmuts_lvlA_filt_color <- as.character(crispr_ptmuts_lvlA_filt$Color_Nonsilent)
names(crispr_ptmuts_lvlA_filt_color) <- crispr_ptmuts_lvlA_filt$Mutation_Status_Nonsilent

label_text <- data.frame(p.signif.adj = crispr_signif_lvlA$p.signif.adj, Hugo_Symbol = crispr_signif_lvlA$Hugo_Symbol)
label_text <- filter(label_text, Hugo_Symbol %in% genes_lvlA_filt)
label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = genes_lvlA_filt)

crispr_ptmuts_lvlA_filt_plot_faceted <- ggplot(data = crispr_ptmuts_lvlA_filt, aes(x = Mutation_Status_Nonsilent, y = Score)) +
  facet_wrap(~ Hugo_Symbol, drop = FALSE, ncol = length(genes_lvlA)) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85)) +
  scale_fill_manual(values = crispr_ptmuts_lvlA_filt_color) +
  guides(color = FALSE) +
  geom_text(data = label_text, mapping = aes(x = 1.5, y = 2, label = p.signif.adj)) +
  theme(legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
  labs(fill = "Mutation Status", y = "CERES Score", x = "Mutation Status")
crispr_ptmuts_lvlA_filt_plot_faceted
# ggsave("./plots_18Q3/manuscript/crispr_ptmuts_lvlA_filt_plot_faceted.png", crispr_ptmuts_lvlA_filt_plot_faceted, device = "png", dpi = 450, width = 20, height = 4, units = "in")
```

### CCLE

```{r fig.height=4, fig.width=20}
ccle_ptmuts_lvlA_filt <- filter(ccle_data_ptmuts, Hugo_Symbol %in% genes_lvlA_filt)
ccle_ptmuts_lvlA_filt$Hugo_Symbol <- ordered(ccle_ptmuts_lvlA_filt$Hugo_Symbol, levels = genes_lvlA_filt)
ccle_drugs <- as.character(unique(ccle_ptmuts_lvlA_filt$Drug))
```

Plot code:
```{r, eval=FALSE}
ccle_lvlA_drugboxplots <- lapply(ccle_drugs, makeDrugBoxplots, dataset = "CCLE")
ccle_lvlA_drugboxplot_paths <- paste0(ccle_drugs, "_ccle_boxplots.png")
pwalk(list(ccle_lvlA_drugboxplot_paths, ccle_lvlA_drugboxplots), ggsave, path = "./plots_18Q3/manuscript/ccle_lvlA_boxplots", dpi = 450, width = 20, height = 4, units = "in")
```

Data management:
```{r, eval=TRUE}
ccle_lvlA_drugboxplots <- paste0(list.files("./plots_18Q3/manuscript/ccle_lvlA_boxplots", full.names = TRUE))
names(ccle_lvlA_drugboxplots) <- str_replace_all(ccle_lvlA_drugboxplots, c("_ccle_boxplots.png" = "", "./plots_18Q3/manuscript/ccle_lvlA_boxplots/" = ""))
bsselect(ccle_lvlA_drugboxplots, type = "img", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275)
```

### CTRP

```{r fig.height=4, fig.width=20}
ctrp_ptmuts_lvlA_filt <- filter(ctrp_data_ptmuts, Hugo_Symbol %in% genes_lvlA_filt)
ctrp_ptmuts_lvlA_filt$Hugo_Symbol <- ordered(ctrp_ptmuts_lvlA_filt$Hugo_Symbol, levels = genes_lvlA_filt)
ctrp_drugs <- as.character(unique(ctrp_ptmuts_lvlA_filt$Drug))
```

Plot code:
```{r, eval=FALSE}
ctrp_lvlA_drugboxplots <- lapply(ctrp_drugs, makeDrugBoxplots, dataset = "CTRP")
ctrp_lvlA_drugboxplot_paths <- paste0(ctrp_drugs, "_ctrp_boxplots.png")
pwalk(list(ctrp_lvlA_drugboxplot_paths, ctrp_lvlA_drugboxplots), ggsave, path = "./plots_18Q3/manuscript/ctrp_lvlA_boxplots", dpi = 450, width = 20, height = 4, units = "in")
```

Data management:
```{r, eval=TRUE}
ctrp_lvlA_drugboxplots <- paste0(list.files("./plots_18Q3/manuscript/ctrp_lvlA_boxplots", full.names = TRUE))
names(ctrp_lvlA_drugboxplots) <- str_replace_all(ctrp_lvlA_drugboxplots, c("_ctrp_boxplots.png" = "", "./plots_18Q3/manuscript/ctrp_lvlA_boxplots/" = ""))
bsselect(ctrp_lvlA_drugboxplots, type = "img", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275)
```

### GDSC

```{r fig.height=4, fig.width=20}
gdsc_ptmuts_lvlA_filt <- filter(gdsc_data_ptmuts, Hugo_Symbol %in% genes_lvlA_filt)
gdsc_ptmuts_lvlA_filt$Hugo_Symbol <- ordered(gdsc_ptmuts_lvlA_filt$Hugo_Symbol, levels = genes_lvlA_filt)
gdsc_drugs <- as.character(unique(gdsc_ptmuts_lvlA_filt$Drug))
```

Plot code:
```{r, eval=FALSE}
gdsc_lvlA_drugboxplots <- lapply(gdsc_drugs, makeDrugBoxplots, dataset = "GDSC")
gdsc_lvlA_drugboxplot_paths <- paste0(gdsc_drugs, "_gdsc_boxplots.png")
pwalk(list(gdsc_lvlA_drugboxplot_paths, gdsc_lvlA_drugboxplots), ggsave, path = "./plots_18Q3/manuscript/gdsc_lvlA_boxplots", dpi = 450, width = 20, height = 4, units = "in")
```

Data management:
```{r, eval=TRUE}
gdsc_lvlA_drugboxplots <- paste0(list.files("./plots_18Q3/manuscript/gdsc_lvlA_boxplots", full.names = TRUE))
names(gdsc_lvlA_drugboxplots) <- str_replace_all(gdsc_lvlA_drugboxplots, c("_gdsc_boxplots.png" = "", "./plots_18Q3/manuscript/gdsc_lvlA_boxplots/" = ""))
bsselect(gdsc_lvlA_drugboxplots, type = "img", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275)
```

## CRISPR: Point mutation mapping

Match specific point mutations.
```{r fig.height=5, fig.width=20}
g2p_indications <- filter(read.delim("./data_munging/data_mutation_associations_appended.csv", sep = "\t", header = TRUE), Evidence.Level == "A")
maf_g2p_indications <- filter(maf_raw, Genome_Change %in% g2p_indications$MutationName)
crispr_g2p_indications <- merge(maf_g2p_indications, crispr_data_ptmuts, by = c("Hugo_Symbol", "CCLE_Name", "Broad_ID"))
dup_g2p_indications <- filter(crispr_g2p_indications[, c("Hugo_Symbol", "CCLE_Name", "Broad_ID")] %>% group_by(Hugo_Symbol, CCLE_Name, Broad_ID) %>% tally(), n > 1)
crispr_g2p_indications <- merge(crispr_g2p_indications, dup_g2p_indications, by = c("Hugo_Symbol", "Broad_ID"), all.x = TRUE)
crispr_g2p_indications <- filter(crispr_g2p_indications, is.na(n))

crispr_g2p_indications_plot <- ggplot(crispr_g2p_indications, aes(x = Hugo_Symbol, y = Score)) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(alpha = 0.5, color = "lightgray", outlier.shape = NA) +
  geom_jitter(width = 0.3, mapping = aes(color = group_general_lineage_name), size = 1) +
  theme_light() +
  theme(legend.title = element_blank()) +
  labs(x = "Gene", y = "CERES Score")
crispr_g2p_indications_plot

crispr_g2p_indications_muts_plot <- ggplot(crispr_g2p_indications, aes(x = Protein_Change, y = Score)) +
  facet_grid(~ Hugo_Symbol, scales = "free_x", space = "free") +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(alpha = 0.5, color = "lightgray", outlier.shape = NA) +
  geom_jitter(width = 0.1, mapping = aes(color = group_general_lineage_name), size = 1) +
  theme_light() +
  theme(legend.title = element_blank(), axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Gene", y = "CERES Score")
crispr_g2p_indications_muts_plot

# ggsave("./plots_18Q3/manuscript/crispr_g2p_indications_plot_test.png", crispr_g2p_indications_plot, width = 12, height = 5, units = "in")
# ggsave("./plots_18Q3/manuscript/crispr_g2p_indications_muts_plot_test.png", crispr_g2p_indications_muts_plot, width = 20, height = 5, units = "in")
```

## Level A: Per-gene grouped plots

Plot code:
```{r, eval=FALSE}
crispr_lvlA_groups <- lapply(as.character(genes_lvlA), makeCRISPRgrob, filt_method = "lvlA")
crispr_lvlA_groups_paths <- paste0(genes_lvlA, "_crispr_grouped.png")
pwalk(list(crispr_lvlA_groups_paths, crispr_lvlA_groups), ggsave, path = "./plots_18Q3/crispr_grouped_plots_lvlA", dpi = 300, width = 12, height = 12, units = "in")
```

Data management:
```{r, eval=TRUE}
crispr_lvlA_bygene <- paste0(list.files("./plots_18Q3/crispr_grouped_plots_lvlA", full.names = TRUE))
names(crispr_lvlA_bygene) <- str_replace_all(crispr_lvlA_bygene, c("_crispr_grouped.png" = "", "./plots_18Q3/crispr_grouped_plots_lvlA/" = ""))
crispr_lvlA_order <- c(intersect(crispr_signif_lvlA$Hugo_Symbol, names(crispr_lvlA_bygene)), setdiff(names(crispr_lvlA_bygene), crispr_signif_lvlA$Hugo_Symbol))
crispr_lvlA_bygene <- crispr_lvlA_bygene[match(crispr_lvlA_order, names(crispr_lvlA_bygene))]
bsselect(crispr_lvlA_bygene, type = "img", selected = "KRAS", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275)
```

<!-- ## All genes: Per-gene grouped plots -->

<!-- Plot code: -->
<!-- ```{r, eval=FALSE} -->
<!-- crispr_all_groups <- lapply(unique(crispr_data_ptmuts$Hugo_Symbol), makeCRISPRgrob, filt_method = "all") -->
<!-- crispr_all_groups_paths <- paste0(unique(crispr_data_ptmuts$Hugo_Symbol), "_crispr_grouped.png") -->
<!-- pwalk(list(crispr_lvlA_groups_paths, crispr_lvlA_groups), ggsave, path = "./plots_18Q3/crispr_grouped_plots", dpi = 300, width = 12, height = 12, units = "in") -->
<!-- ``` -->

<!-- Data management: -->
<!-- ```{r, eval=FALSE} -->
<!-- crispr_all_bygene <- paste0(list.files("./plots_18Q3/crispr_grouped_plots", full.names = TRUE)) -->
<!-- names(crispr_all_bygene) <- str_replace_all(crispr_all_bygene, c("_crispr_grouped.png" = "", "./plots_18Q3/crispr_grouped_plots_lvlA/" = "")) -->
<!-- crispr_all_order <- c(intersect(crispr_signif_nonsilent$Hugo_Symbol, names(crispr_all_bygene)), setdiff(names(crispr_all_bygene), crispr_signif_nonsilent$Hugo_Symbol)) -->
<!-- crispr_all_bygene <- crispr_all_bygene[match(crispr_all_order, names(crispr_all_bygene))] -->
<!-- bsselect(crispr_all_bygene, type = "img", selected = "KRAS", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275) -->
<!-- ``` -->

## JSON testing

```{r}
g2pA <- read.delim("./data_munging/g2p_full_level_A_dataset_v2_10_10_2018.csv", sep = "\t", header = TRUE, na.strings = "-1", quote = "")
# g2pA[g2pA == "-1"] <- NA
g2pA$X <- NULL

g2pA_JSON <- data.frame(JSON = g2pA$JSON)

g2pA_noJSON <- g2pA
g2pA_noJSON$JSON <- NULL

# write.table(g2pA, file = "./data_munging/g2p_full_level_A_dataset_v2_10_10_2018.tsv", quote = FALSE, sep = "\t", row.names = FALSE)
# write.table(g2pA_noJSON, file = "./data_munging/g2pA_noJSON_v2_10_10_2018.tsv", quote = FALSE, sep = "\t", row.names = FALSE)
# write.table(g2pA_JSON, file = "./data_munging/g2pA_JSON_v2_10_10_2018.tsv", quote = FALSE, sep = "\t", row.names = FALSE)
```

```
{
'feature_names': 'ALK positive',
'jax': '{\"responseType\": \"sensitive\", 
        \"efficacyEvidence\": \"Zykadia (ceritinib) is FDA approved for the treatment of ALK-positive metastatic non-small cell lung cancer (NSCLC) in patients previously treated with crizotinib (FDA.gov).\",
        \"molecularProfile\": {\"profileName\": \"ALK positive\",
                               \"id\": 1772},
        \"evidenceType\": \"Actionable\",
        \"indication\": {\"source\": \"DOID\",
                         \"id\": 3908,
                         \"name\": \"non-small cell lung carcinoma\"},
        \"references\": [{\"url\": \"http://www.fda.gov/\",
                          \"id\": 286,
                          \"pubMedId\": null,
                          \"title\": \"FDA.gov\"}],
        \"approvalStatus\": \"FDA approved\",
        \"therapy\": {\"id\": 789,
                      \"therapyName\": \"Ceritinib\"},
        \"id\": 1481}',
'tags': [],
'genes': ['ALK'],
'source_url': 'https://ckb.jax.org/therapy/show/789',
'source': 'jax',
'dev_tags': [],
'gene_identifiers': [{'symbol': 'ALK',
                      'entrez_id': '238',
                      'ensembl_gene_id': 'ENSG00000171094'}],
'association': {'drug_labels': 'CERITINIB',
                'description': 'Zykadia (ceritinib) is FDA approved for the treatment of ALK-positive metastatic non-small cell lung cancer (NSCLC) in patients previously treated with crizotinib (FDA.gov).',
                'publication_url': 'http://www.fda.gov/',
                'source_link': 'https://ckb.jax.org/molecularProfile/show/1772',
                'variant_name': [''],
                'phenotypes': [{'source': 'http://purl.obolibrary.org/obo/DOID_3908',
                                'term': 'non-small cell lung carcinoma',
                                'description': 'non-small cell lung carcinoma',
                                'family': 'lung cancer',
                                'id': 'DOID:3908'}],
                'evidence': [{'info': {'publications': ['http://www.fda.gov/']},
                              'evidenceType': {'sourceName': 'jax'},
                              'description': 'sensitive'}],
                              'environmentalContexts': [{'term': 'CERITINIB',
                                                          'description': 'CERITINIB',
                                                          'taxonomy': {'kingdom': 'Chemical entities',
                                                                       'direct-parent': 'Phenylpiperidines',
                                                                       'class': 'Organoheterocyclic compounds',
                                                                       'subclass': 'Piperidines',
                                                                       'superclass': 'Organic compounds'},
                                                          'source': 'http://rdf.ncbi.nlm.nih.gov/pubchem/compound',
                                                          'usan_stem': 'tyrosine kinase inhibitors',
                                                          'toxicity': 'There is not currently any data on carcinogenicity, effect on human fertility, or on early embryonic development. However, based on its mechanism of action, ceritinib may cause fetal harm when administered to pregnant women and should therefore be administered with effective contraception during treatment. Diarrhea, nausea, vomiting, or abdominal pain occurred in 96% of 255 patients including severe cases in 14% of patients. Drug-induced hepatotoxicity also occurred in 27% of 255 patients, presenting as alanine aminotransferase (ALT) levels greater than 5 times the upper limit of normal (ULN). Severe, life-threatening, or fatal interstitial lung disease (ILD)/pneumonitis, hyperglycaemia, and bradycardia have also been reported.',
                                                          'approved_countries': ['Canada',
                                                                                 'US'],
                                                          'id': 'CID57379345'}],
                'evidence_label': 'A',
                'response_type': 'sensitive',
                'evidence_level': 1},
'features': [{'provenance_rule': 'default_feature',
              'end': 29484957,
              'description': 'ALK  positive',
              'links': ['http://reg.genome.network/refseq/RS000050',
                        'http://reg.genome.network/refseq/RS000026',
                        'http://reg.genome.network/allele/CA425438613',
                        'http://reg.genome.network/refseq/RS000002',
                        'http://myvariant.info/v1/variant/chr2:g.29484957C>G?assembly=hg19',
                        'http://reg.genome.network/refseq/RS001597'],
              'sequence_ontology': {'hierarchy': ['SO:0000400',
                                                  'SO:0000733',
                                                  'SO:0000401',
                                                  'SO:0000119',
                                                  'SO:0000123',
                                                  'SO:0000125'],
                                    'soid': 'SO:0000475',
                                    'root_name': 'sequence_attribute',
                                    'name': 'positively_autoregulated',
                                    'root_soid': 'SO:0000400'},
              'provenance': ['http://myvariant.info/v1/query?q=ALK  positive',
                             'http://reg.genome.network/allele?hgvs=NC_000002.11%3Ag.29484957C%3EG'], 
              'start': 29484957,
              'synonyms': ['NC_000002.11:g.29484957C>G',
                           'NC_000002.10:g.29338461C>G',
                           'LRG_488:g.664476G>C',
                           'chr2:g.29484957C>G',
                           'NG_009445.1:g.664476G>C',
                           'NC_000002.12:g.29262091C>G',
                           'CM000664.1:g.29484957C>G',
                           'CM000664.2:g.29262091C>G'],
              'biomarker_type': 'positive',
              'referenceName': 'GRCh37',
              'geneSymbol': 'ALK',
              'alt': 'G',
              'ref': 'C',
              'chromosome': '2',
              'name': 'ALK  positive'}]
}
```

# Heatmaps

--------------------------------------------------------------------------------

## Data management

The mutation status portion of these heatmaps was made using deleterious and missense mutations as "mutant" and all others as "other."

Make and save heatmap data matrices:
```{r, eval=FALSE}
# Set up heatmap data
chm_scores <- t(crispr)
colnames(chm_scores) <- chm_scores[1, ]
chm_scores <- chm_scores[-1, ]
chm_rows <- rownames(chm_scores)
chm_cols <- colnames(chm_scores)
chm_scores <- apply(chm_scores, 2, as.double)
chm_scores <- apply(chm_scores, 2, function(x) (x - mean(x)) / var(x))
colnames(chm_scores) <- chm_cols
rownames(chm_scores) <- chm_rows
saveRDS(chm_scores, "./data_munging/rds/crispr_heatmap_scores.rds")

# Make full grid for genomic features
crispr_grid_ccls <- unique(crispr_data$Broad_ID)
crispr_grid_genes <- unique(crispr_data$Hugo_Symbol)
crispr_grid <- expand.grid("Broad_ID" = crispr_grid_ccls, "Hugo_Symbol" = crispr_grid_genes)

# Gene expression
ge_filt <- filter(ge_melt, Hugo_Symbol %in% unique(crispr_data$Hugo_Symbol))[, c("Hugo_Symbol", "Broad_ID", "RPKM")]
ge_filt$RPKM <- log2(ge_filt$RPKM + 0.0001)
chm_ge_grid <- merge(crispr_grid, ge_filt, by = c("Broad_ID", "Hugo_Symbol"), all.x = TRUE)
chm_ge <- reshape(chm_ge_grid, idvar = "Hugo_Symbol", timevar = "Broad_ID", direction = "wide")
chm_ge_rows <- chm_ge$Hugo_Symbol
chm_ge$Hugo_Symbol <- NULL
chm_ge_cols <- colnames(chm_ge) %>% gsub("RPKM.", "", .)
chm_ge <- apply(chm_ge, 2, as.double)
chm_ge <- apply(chm_ge, 2, function(x) (x - mean(x, na.rm = TRUE)) / var(x, na.rm = TRUE))
rownames(chm_ge) <- chm_ge_rows
colnames(chm_ge) <- chm_ge_cols
saveRDS(chm_ge, "./data_munging/rds/crispr_heatmap_ge.rds")

# Mutation status
chm_mut_grid <- merge(crispr_grid, maf_df[, c("Broad_ID", "Hugo_Symbol", "Mutation_Status")], by = c("Broad_ID", "Hugo_Symbol"), all.x = TRUE)
chm_mut_grid$Mutation_Status <- ifelse(is.na(chm_mut_grid$Mutation_Status), "Other", chm_mut_grid$Mutation_Status)
chm_mut_grid$Mutation_Status <- ifelse(chm_mut_grid$Mutation_Status == "Other", 0, 1)
chm_mut <- reshape(chm_mut_grid, idvar = "Hugo_Symbol", timevar = "Broad_ID", direction = "wide")
chm_mut_rows <- chm_mut$Hugo_Symbol
chm_mut$Hugo_Symbol <- NULL
chm_mut_cols <- colnames(chm_mut) %>% gsub("Mutation_Status.", "", .)
chm_mut <- matrix(as.numeric(unlist(chm_mut)), nrow = nrow(chm_mut))
rownames(chm_mut) <- chm_mut_rows
colnames(chm_mut) <- chm_mut_cols
saveRDS(chm_mut, "./data_munging/rds/crispr_heatmap_mut.rds")

# Copy number
cn_filt <- filter(cn_melt, Hugo_Symbol %in% unique(crispr_data$Hugo_Symbol))
chm_cn_grid <- merge(crispr_grid, cn_melt, by = c("Broad_ID", "Hugo_Symbol"), all.x = TRUE)
chm_cn <- reshape(chm_cn_grid, idvar = "Hugo_Symbol", timevar = "Broad_ID", direction = "wide")
chm_cn_rows <- chm_cn$Hugo_Symbol
chm_cn$Hugo_Symbol <- NULL
chm_cn_cols <- colnames(chm_cn) %>% gsub("Copy_Number.", "", .)
chm_cn <- apply(chm_cn, 2, as.double)
chm_cn <- apply(chm_cn, 2, function(x) (x - mean(x, na.rm = TRUE)) / var(x, na.rm = TRUE))
rownames(chm_cn) <- chm_cn_rows
colnames(chm_cn) <- chm_cn_cols
saveRDS(chm_cn, "./data_munging/rds/crispr_heatmap_cn.rds")
```

Load heatmap matrices:
```{r, eval=FALSE}
chm_scores <- readRDS("./data_munging/rds/crispr_heatmap_scores.rds")
chm_ge <- readRDS("./data_munging/rds/crispr_heatmap_ge.rds")
chm_mut <- readRDS("./data_munging/rds/crispr_heatmap_mut.rds")
chm_cn <- readRDS("./data_munging/rds/crispr_heatmap_cn.rds")

# Metadata anotation dataframe
chm_annot <- merge(merge(data.frame("Broad_ID" = crispr$Broad_ID), crispr_meta[, c("Broad_ID", "cell_line_SSMD", "cas9_activity", "culture_type", "primary_tissue")], by = "Broad_ID", all.x = TRUE), ccl_info[, c("Broad_ID", "Primary.Disease", "Gender", "Source")], by = "Broad_ID", all.x = TRUE)
rownames(chm_annot) <- chm_annot$Broad_ID
chm_annot$Broad_ID <- NULL
colnames(chm_annot) <- c("Cell_Line_SSMD", "Cas9_Activity", "Culture_Type", "Primary_Tissue", "Primary_Disease", "Gender", "Source")
chm_annot$Cell_Line_SSMD <- as.numeric(chm_annot$Cell_Line_SSMD)
chm_annot$Cas9_Activity <- as.numeric(chm_annot$Cas9_Activity)
```

## All genes, no omics

```{r, eval=FALSE}
chm_annot_list <- HeatmapAnnotation(df = chm_annot, annotation_legend_param = list(
  Cell_Line_SSMD = list(title = "Cell Line SSMD", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
  Cas9_Activity = list(title = "Cas9 Activity", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
  Culture_Type = list(title = "Culture Type", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Culture Type", title_position = "topleft", grid_height = unit(0.3, "in")),
  Primary_Tissue = list(title = "Primary Tissue", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Primary_Tissue", title_position = "topleft", grid_height = unit(0.3, "in")),
  Primary_Disease = list(title = "Primary Disease", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Primary Disease", title_position = "topleft", grid_height = unit(0.3, "in")),
  Gender = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), grid_height = unit(0.3, "in")),
  Source = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Source", title_position = "topleft", grid_height = unit(0.3, "in"))))

chm_all_plot <- Heatmap(chm_scores, name = "CERES Score",
        bottom_annotation = chm_annot_list,
        bottom_annotation_height = unit(5, "in"),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        row_title = "Genes", column_title = "Cancer Cell Lines",
        col = colorRamp2(c(min(chm_scores), 0, max(chm_scores)), c("purple4", "white", "seagreen4")),
        na_col = "black",
        row_title_gp = gpar(fontsize = 30, fontface = "bold"),
        row_names_gp = gpar(fontsize = 5),
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_names_gp = gpar(fontsize = 5),
        row_dend_reorder = TRUE,
        column_dend_reorder = TRUE,
        clustering_distance_rows = "euclidean",
        clustering_distance_columns = "euclidean",
        clustering_method_rows = "complete",
        clustering_method_columns = "complete",
        column_dend_height = unit(2, "in"),
        row_dend_width = unit(4, "in"),
        width = 3)

pdf("./plots_18Q3/crispr_heatmap_euclidean_all.pdf", width = 50, height = 50, paper = "special")
draw(chm_all_plot, heatmap_legend_side = "bottom", annotation_legend_side = "bottom")
seekViewport("annotation_Cell_Line_SSMD")
grid.text("Cell Line SSMD", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Cas9_Activity")
grid.text("Cas9 Activity", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Culture_Type")
grid.text("Culture Type", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Tissue")
grid.text("Primary Tissue", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Disease")
grid.text("Primary Disease", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Gender")
grid.text("Gender", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Source")
grid.text("Source", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
dev.off()

png("./plots_18Q3/crispr_heatmap_euclidean_all.png", width = 50, height = 50, units = "in", res = 96)
draw(chm_all_plot, heatmap_legend_side = "bottom", annotation_legend_side = "bottom")
seekViewport("annotation_Cell_Line_SSMD")
grid.text("Cell Line SSMD", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Cas9_Activity")
grid.text("Cas9 Activity", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Culture_Type")
grid.text("Culture Type", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Tissue")
grid.text("Primary Tissue", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Disease")
grid.text("Primary Disease", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Gender")
grid.text("Gender", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Source")
grid.text("Source", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
dev.off()
```

## CGC genes

Code for heatmap of CGC-filtered genes only

### Euclidean distance

```{r, eval=FALSE}
chm_scores_cgc <- subset(chm_scores, rownames(chm_scores) %in% cgc$Hugo_Symbol)
chm_ge_cgc <- subset(chm_ge, rownames(chm_ge) %in% cgc$Hugo_Symbol)
chm_mut_cgc <- subset(chm_mut, rownames(chm_mut) %in% cgc$Hugo_Symbol)
chm_cn_cgc <- subset(chm_cn, rownames(chm_cn) %in% cgc$Hugo_Symbol)

chm_annot_list <- HeatmapAnnotation(df = chm_annot, annotation_legend_param = list(
  Cell_Line_SSMD = list(title = "Cell Line SSMD", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
  Cas9_Activity = list(title = "Cas9 Activity", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
  Culture_Type = list(title = "Culture Type", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Culture Type", title_position = "topleft", grid_height = unit(0.3, "in")),
  Primary_Tissue = list(title = "Primary Tissue", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Primary_Tissue", title_position = "topleft", grid_height = unit(0.3, "in")),
  Primary_Disease = list(title = "Primary Disease", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Primary Disease", title_position = "topleft", grid_height = unit(0.3, "in")),
  Gender = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), grid_height = unit(0.3, "in")),
  Source = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Source", title_position = "topleft", grid_height = unit(0.3, "in"))))

chm_plot_cgc <- Heatmap(chm_scores_cgc, name = "CERES Score",
        bottom_annotation = chm_annot_list,
        bottom_annotation_height = unit(5, "in"),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        row_title = "Genes", column_title = "Cancer Cell Lines",
        col = colorRamp2(c(min(chm_scores_cgc), 0, max(chm_scores_cgc)), c("purple4", "white", "seagreen4")),
        na_col = "black",
        row_title_gp = gpar(fontsize = 30, fontface = "bold"),
        row_names_gp = gpar(fontsize = 5),
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_names_gp = gpar(fontsize = 5),
        row_dend_reorder = TRUE,
        column_dend_reorder = TRUE,
        clustering_distance_rows = "euclidean",
        clustering_distance_columns = "euclidean",
        clustering_method_rows = "complete",
        clustering_method_columns = "complete",
        column_dend_height = unit(2, "in"),
        row_dend_width = unit(4, "in"),
        width = 3)

chm_plot_cgc_cols <- colnames(chm_scores_cgc)[unlist(column_order(chm_plot_cgc))]
chm_plot_cgc_rows <- rownames(chm_scores_cgc)[unlist(row_order(chm_plot_cgc))]

chm_ge_plot_cgc <- Heatmap(chm_ge_cgc, name = "Gene Expression",
        col = colorRamp2(c(min(chm_ge_cgc[!is.na(chm_ge_cgc)]), 0, max(chm_ge_cgc[!is.na(chm_ge_cgc)])), c("dodgerblue4", "white", "firebrick4")),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        na_col = "black",
        cluster_rows = FALSE, cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 5), show_column_names = FALSE,
        column_title = "Gene Expression",
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_order = chm_plot_cgc_cols,
        row_order = chm_plot_cgc_rows,
        width = 1)
chm_cn_plot_cgc <- Heatmap(chm_cn_cgc, name = "Copy Number",
        col = colorRamp2(c(min(chm_cn_cgc[!is.na(chm_cn_cgc)]), 0, max(chm_cn_cgc[!is.na(chm_cn_cgc)])), c("white", "thistle2", "mediumorchid4")),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        na_col = "black",
        cluster_rows = FALSE, cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 5), show_column_names = FALSE,
        column_title = "Copy Number",
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_order = chm_plot_cgc_cols,
        row_order = chm_plot_cgc_rows,
        width = 1)
chm_mut_plot_cgc <- Heatmap(chm_mut_cgc, name = "Mutation Status",
        na_col = "black",
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), at = c(0, 1), labels = c("Other", "Mutant"), title_position = "topcenter", grid_height = unit(0.3, "in")),
        col = c("cyan3", "darkorchid"),
        cluster_rows = FALSE, cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 5), show_column_names = FALSE,
        column_title = "Mutation Status",
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_order = chm_plot_cgc_cols,
        row_order = chm_plot_cgc_rows,
        width = 1)

chm_all_plot_cgc <- chm_plot_cgc + chm_ge_plot_cgc + chm_cn_plot_cgc + chm_mut_plot_cgc

pdf("./plots_18Q3/crispr_heatmap_euclidean_all_cgc.pdf", width = 50, height = 50, paper = "special")
draw(chm_all_plot_cgc, heatmap_legend_side = "bottom", annotation_legend_side = "bottom")
seekViewport("annotation_Cell_Line_SSMD")
grid.text("Cell Line SSMD", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Cas9_Activity")
grid.text("Cas9 Activity", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Culture_Type")
grid.text("Culture Type", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Tissue")
grid.text("Primary Tissue", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Disease")
grid.text("Primary Disease", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Gender")
grid.text("Gender", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Source")
grid.text("Source", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
dev.off()

png("./plots_18Q3/crispr_heatmap_euclidean_all_cgc.png", width = 50, height = 50, units = "in", res = 96)
draw(chm_all_plot_cgc, heatmap_legend_side = "bottom", annotation_legend_side = "bottom")
seekViewport("annotation_Cell_Line_SSMD")
grid.text("Cell Line SSMD", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Cas9_Activity")
grid.text("Cas9 Activity", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Culture_Type")
grid.text("Culture Type", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Tissue")
grid.text("Primary Tissue", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Disease")
grid.text("Primary Disease", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Gender")
grid.text("Gender", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Source")
grid.text("Source", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
dev.off()
```


### Pearson correlation

```{r, eval=FALSE}
chm_scores_cgc <- subset(chm_scores, rownames(chm_scores) %in% cgc$Hugo_Symbol)
chm_ge_cgc <- subset(chm_ge, rownames(chm_ge) %in% cgc$Hugo_Symbol)
chm_mut_cgc <- subset(chm_mut, rownames(chm_mut) %in% cgc$Hugo_Symbol)
chm_cn_cgc <- subset(chm_cn, rownames(chm_cn) %in% cgc$Hugo_Symbol)

chm_annot_list <- HeatmapAnnotation(df = chm_annot, annotation_legend_param = list(
  Cell_Line_SSMD = list(title = "Cell Line SSMD", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
  Cas9_Activity = list(title = "Cas9 Activity", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
  Culture_Type = list(title = "Culture Type", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Culture Type", title_position = "topleft", grid_height = unit(0.3, "in")),
  Primary_Tissue = list(title = "Primary Tissue", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Primary_Tissue", title_position = "topleft", grid_height = unit(0.3, "in")),
  Primary_Disease = list(title = "Primary Disease", title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Primary Disease", title_position = "topleft", grid_height = unit(0.3, "in")),
  Gender = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), grid_height = unit(0.3, "in")),
  Source = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), nrow = 8, title = "Source", title_position = "topleft", grid_height = unit(0.3, "in"))))

chm_plot_cgc <- Heatmap(chm_scores_cgc, name = "CERES Score",
        bottom_annotation = chm_annot_list,
        bottom_annotation_height = unit(5, "in"),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        row_title = "Genes", column_title = "Cancer Cell Lines",
        col = colorRamp2(c(min(chm_scores_cgc), 0, max(chm_scores_cgc)), c("purple4", "white", "seagreen4")),
        na_col = "black",
        row_title_gp = gpar(fontsize = 30, fontface = "bold"),
        row_names_gp = gpar(fontsize = 5),
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_names_gp = gpar(fontsize = 5),
        row_dend_reorder = TRUE,
        column_dend_reorder = TRUE,
        clustering_distance_rows = "pearson",
        clustering_distance_columns = "pearson",
        clustering_method_rows = "complete",
        clustering_method_columns = "complete",
        column_dend_height = unit(2, "in"),
        row_dend_width = unit(4, "in"),
        width = 3)

chm_plot_cgc_cols <- colnames(chm_scores_cgc)[unlist(column_order(chm_plot_cgc))]
chm_plot_cgc_rows <- rownames(chm_scores_cgc)[unlist(row_order(chm_plot_cgc))]

chm_ge_plot_cgc <- Heatmap(chm_ge_cgc, name = "Gene Expression",
        col = colorRamp2(c(min(chm_ge_cgc[!is.na(chm_ge_cgc)]), 0, max(chm_ge_cgc[!is.na(chm_ge_cgc)])), c("dodgerblue4", "white", "firebrick4")),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        na_col = "black",
        cluster_rows = FALSE, cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 5), show_column_names = FALSE,
        column_title = "Gene Expression",
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_order = chm_plot_cgc_cols,
        row_order = chm_plot_cgc_rows,
        width = 1)
chm_cn_plot_cgc <- Heatmap(chm_cn_cgc, name = "Copy Number",
        col = colorRamp2(c(min(chm_cn_cgc[!is.na(chm_cn_cgc)]), 0, max(chm_cn_cgc[!is.na(chm_cn_cgc)])), c("white", "thistle2", "mediumorchid4")),
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), title_position = "topcenter", legend_height = unit(2, "in")),
        na_col = "black",
        cluster_rows = FALSE, cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 5), show_column_names = FALSE,
        column_title = "Copy Number",
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_order = chm_plot_cgc_cols,
        row_order = chm_plot_cgc_rows,
        width = 1)
chm_mut_plot_cgc <- Heatmap(chm_mut_cgc, name = "Mutation Status",
        na_col = "black",
        heatmap_legend_param = list(title_gp = gpar(fontsize = 20), labels_gp = gpar(fontsize = 15), at = c(0, 1), labels = c("Other", "Mutant"), title_position = "topcenter", grid_height = unit(0.3, "in")),
        col = c("cyan3", "darkorchid"),
        cluster_rows = FALSE, cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 5), show_column_names = FALSE,
        column_title = "Mutation Status",
        column_title_gp = gpar(fontsize = 30, fontface = "bold"),
        column_order = chm_plot_cgc_cols,
        row_order = chm_plot_cgc_rows,
        width = 1)

chm_all_plot_cgc <- chm_plot_cgc + chm_ge_plot_cgc + chm_cn_plot_cgc + chm_mut_plot_cgc

pdf("./plots_18Q3/crispr_heatmap_pearson_all_cgc.pdf", width = 50, height = 50, paper = "special")
draw(chm_all_plot_cgc, heatmap_legend_side = "bottom", annotation_legend_side = "bottom")
seekViewport("annotation_Cell_Line_SSMD")
grid.text("Cell Line SSMD", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Cas9_Activity")
grid.text("Cas9 Activity", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Culture_Type")
grid.text("Culture Type", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Tissue")
grid.text("Primary Tissue", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Primary_Disease")
grid.text("Primary Disease", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Gender")
grid.text("Gender", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
seekViewport("annotation_Source")
grid.text("Source", unit(1, "npc") + unit(1, "mm"), 0.5, gp = gpar(fontsize = 15), default.units = "npc", just = "left")
dev.off()
```



# Per-gene plots

--------------------------------------------------------------------------------

## Grouped plots

```{r fig.height=12, fig.width=12, include=FALSE, eval=FALSE}
makeCRISPRgrob("KRAS", filt_method = "all")
```

Plot code:
```{r, eval=FALSE}
crispr_lvlA_genes <- unique(crispr_data_cgc$Hugo_Symbol)
crispr_all_groups <- lapply(crispr_all_genes, makeCRISPRgrob)
crispr_all_groups_paths <- paste0(crispr_all_genes, "_crispr_grouped.png")
pwalk(list(crispr_all_groups_paths, crispr_all_groups), ggsave, path = "./plots_18Q3/crispr_grouped_plots_cgc", dpi = 300, width = 12, height = 12, units = "in")
```

Data management:
```{r, eval=FALSE}
crispr_all_bygene <- paste0(list.files("./plots_18Q3/crispr_grouped_plots_cgc", full.names = TRUE))
names(crispr_all_bygene) <- str_replace_all(crispr_all_bygene, c("_crispr_grouped.png" = "", "./plots_18Q3/crispr_grouped_plots_cgc/" = ""))
crispr_bygene_order <- c(intersect(crispr_signif$Hugo_Symbol, names(crispr_all_bygene)), setdiff(names(crispr_all_bygene), crispr_signif$Hugo_Symbol))
crispr_all_bygene <- crispr_all_bygene[match(crispr_bygene_order, names(crispr_all_bygene))]
bsselect(crispr_all_bygene, type = "img", selected = "KRAS", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275)
```

## Sorted lineage plots

```{r fig.height=12, fig.width=12, include=FALSE, eval=FALSE}
makeCRISPRlinplot("KRAS")
```

Plot code:
```{r, eval=FALSE}
crispr_data_cgc <- filter(crispr_data, Hugo_Symbol %in% cgc$Hugo_Symbol)
crispr_all_genes <- unique(crispr_data_cgc$Hugo_Symbol)
crispr_all_lins <- lapply(crispr_all_genes, makeCRISPRlinplot)
crispr_all_lins_paths <- paste0(crispr_all_genes, "_crispr_lineage.png")
pwalk(list(crispr_all_lins_paths, crispr_all_lins), ggsave, path = "./plots_18Q3/crispr_lineage_plots_cgc", dpi = 300, width = 12, height = 12, units = "in")
```

Data management:
```{r, eval=FALSE}
crispr_all_bygene <- paste0(list.files("./plots_18Q3/crispr_lineage_plots_cgc", full.names = TRUE))
names(crispr_all_bygene) <- str_replace_all(crispr_all_bygene, c("_crispr_lineage.png" = "", "./plots_18Q3/crispr_lineage_plots_cgc/" = ""))
crispr_bygene_order <- c(intersect(crispr_signif$Hugo_Symbol, names(crispr_all_bygene)), setdiff(names(crispr_all_bygene), crispr_signif$Hugo_Symbol))
crispr_all_bygene <- crispr_all_bygene[match(crispr_bygene_order, names(crispr_all_bygene))]
bsselect(crispr_all_bygene, type = "img", selected = "KRAS", live_search = TRUE, show_tick = TRUE, height = 300, frame_height = 275)
```


# Report #2

--------------------------------------------------------------------------------

##  Metadata

```{r, eval=FALSE}
tissue_summ <- crispr_meta %>% group_by(primary_tissue) %>% tally()
disease_summ <- ccl_info %>% group_by(Primary.Disease) %>% tally()
```

```{r fig.height=12, fig.width=3, eval=FALSE}
ggplot(data = tissue_summ) + 
  aes(x = 0, y = n, fill = primary_tissue, label = primary_tissue) +
  geom_histogram(color = "black", stat = "identity") +
  geom_text(position = position_stack(vjust = 0.5)) +
  scale_y_continuous(breaks = seq(0, 500, by = 25), labels = seq(0, 500, by = 25)) +
  coord_cartesian(ylim = c(0, 500)) +
  theme(legend.position = "none", axis.text.x = element_blank(), axis.ticks.x = element_blank(), axis.title.x = element_blank()) +
  labs(y = "Count")
```

```{r, include=FALSE, eval=FALSE}
length(intersect(unique(crispr_data$Hugo_Symbol), unique(maf_df$Hugo_Symbol)))
length(intersect(unique(crispr_data$Hugo_Symbol), unique(cn_melt$Hugo_Symbol)))
length(intersect(unique(crispr_data$Hugo_Symbol), unique(ge_melt$Hugo_Symbol)))

length(intersect(unique(crispr_data$Broad_ID), unique(maf_df$Broad_ID)))
length(intersect(unique(crispr_data$Broad_ID), unique(cn_melt$Broad_ID)))
length(intersect(unique(crispr_data$Broad_ID), unique(ge_melt$Broad_ID)))

setdiff(unique(crispr_data$Broad_ID), unique(maf_df$Broad_ID))
setdiff(unique(crispr_data$Broad_ID), unique(cn_melt$Broad_ID))
setdiff(unique(crispr_data$Broad_ID), unique(ge_melt$Broad_ID))

length(intersect(intersect(unique(crispr_data$Broad_ID), unique(cn_melt$Broad_ID)), unique(ge_melt$Broad_ID)))
setdiff(crispr_data$Broad_ID, intersect(intersect(unique(crispr_data$Broad_ID), unique(cn_melt$Broad_ID)), unique(ge_melt$Broad_ID)))
```

## Scores

Distribution of scores across cell lines and primary tissues. Each data point represents a single cell line.
```{r fig.height=4, fig.width=12}
ceres_summ <- crispr_data %>% group_by(primary_tissue) %>% summarize(Max = max(Score),
                                                                     Min = min(Score))
score_summ_plot <- ggplot(data = crispr_data, aes(x = primary_tissue, color = primary_tissue)) +
  geom_boxplot(mapping = aes(y = Score), outlier.shape = NA) +
  geom_point(data = ceres_summ, mapping = aes(x = primary_tissue, y = Min)) +
  geom_point(data = ceres_summ, mapping = aes(x = primary_tissue, y = Max)) +
  scale_y_continuous(breaks = seq(-4, 6, by = 1), labels = seq(-4, 6, by = 1)) +
  theme(axis.text.x = element_text(angle = 30, hjust = 1), legend.position = "none") +
  labs(x = "Primary Tissue", y = "CERES Score")
score_summ_plot
# ggsave(filename = "./plots_18Q3/crispr/ceres_score_summ_plot.pdf", plot = score_summ_plot, width = 12, height = 4, device = "pdf")
```

## Mutations

```{r}
maf_df_filt <- filter(maf_df, Hugo_Symbol %in% unique(crispr_data$Hugo_Symbol))
# maf_summ_filt <- maf_df_filt %>% group_by(Mutation_Status) %>% tally()
# maf_summ_filt$Percent <- format(round(maf_summ_filt$n / sum(maf_summ_filt$n) * 100, 4), nsmall = 2)
# maf_summ_filt$Percent <- as.numeric(as.character(maf_summ_filt$Percent))
```

Distribution of mutations across cell lines and primary tissues. Each data point represents a single cell line.
```{r fig.height=4, fig.width=10}
maf_ccl_annot <- merge(maf_df %>% group_by(Broad_ID) %>% tally(), crispr_meta, by = "Broad_ID", all.y = TRUE)

maf_ccl_annot_plot <- ggplot(data = maf_ccl_annot, aes(x = primary_tissue, y = n, color = primary_tissue)) +
  geom_boxplot(outlier.shape = NA, fill = NA) +
  geom_jitter(alpha = 0.7, size = 1, position = position_jitter(w = 0.25)) +
  scale_y_continuous(breaks = seq(0, 9000, by = 1000), labels = seq(0, 9000, by = 1000)) +
  theme(axis.ticks.x = element_blank(), axis.title.x = element_blank(), axis.text.x = element_blank(), legend.position = "none") +
  labs(x = "Primary Tissue", y = "Number of Mutations")
maf_ccl_annot_plot
# ggsave(filename = "./plots_18Q3/crispr/maf_ccl_annot_plot.pdf", plot = maf_ccl_annot_plot, width = 12, height = 4, device = "pdf")
```

```{r fig.height=8, fig.width=12}
# ggsave(filename = "./plots_18Q3/crispr/score_mut_summ.pdf", plot = grid.draw(rbind(ggplotGrob(maf_ccl_annot_plot), ggplotGrob(score_summ_plot), size = "first")), width = 12, height = 8, device = "pdf")
```

```{r fig.height=6, fig.width=12}
maf_gene_annot <- maf_df_filt %>% group_by(Hugo_Symbol) %>% tally()

ggplot(data = maf_gene_annot, aes(x = n)) +
  geom_histogram(breaks = seq(0, 600, by = 20), fill = "darkslategray3", color = "black", alpha = 0.7) +
  coord_cartesian(ylim = c(0, 13000)) +
  scale_y_continuous(breaks = seq(0, 13500, by = 500), labels = formatC(seq(0, 13500, by = 500), format = "d")) +
  scale_x_continuous(breaks = seq(0, 600, by = 20), labels = seq(0, 600, by = 20)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  stat_bin(breaks = seq(0, 600, by = 20), geom = "text", colour = "black", size = 2.5, aes(label = ..count..), vjust = -0.2, angle = 45, hjust = -0.1) +
  labs(x = "Number of Mutations", y = "Frequency", title = paste0("Distribution of mutational load for ", length(unique(maf_gene_annot$Hugo_Symbol)), " genes in the CCLE MAF file across all ", length(unique(crispr_ccl$Broad_ID)), " cell lines in the CRISPR screen"))
```

## Dependency probabilities

```{r fig.height=6, fig.width=12}
drop_prob_100 <- ggplot(data = crispr_data, aes(x = Dep_Prob)) +
  geom_histogram(breaks = seq(0, 1, by = 0.05), fill = "darkslategray3", color = "black", alpha = 0.7) +
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), labels = seq(0, 1, by = 0.05)) +
  scale_y_continuous(breaks = seq(0, 5500000, by = 500000), labels = formatC(seq(0, 5500000, by = 500000), format = "d")) +
  stat_bin(breaks = seq(0, 1, by = 0.05), geom = "text", colour = "black", size = 3.5, aes(label = ..count..), vjust = -1) +
  labs(title = "Distribution of all dependency probabilities", x = "Probability of Real Dependency", y = "Frequency")

dep_prob_5 <- ggplot(data = filter(crispr_data, Dep_Prob >= 0.95), aes(x = Dep_Prob)) +
  geom_histogram(mapping = aes(fill = primary_tissue), breaks = seq(0.95, 1, by = 0.005), color = "black") +
  scale_y_continuous(breaks = seq(0, 60000, by = 10000), labels = formatC(seq(0, 60000, by = 10000), format = "d")) +
  theme(legend.position = "bottom") +
  guides(fill = guide_legend(nrow = 4, byrow = FALSE)) +
  stat_bin(breaks = seq(0.95, 1, by = 0.005), geom = "text", color = "black", size = 3.5, aes(label = ..count..), vjust = -1) +
  labs(title = "Distribution of all dependency probabilities > 0.95", x = "Probability of Real Dependency", y = "Frequency", fill = "Primary Tissue")
```

## Per-gene plots

```{r fig.height=18, fig.width=5, eval=FALSE}
sig_genes <- c("KRAS", "TP53", "NRAS", "BRAF", "PIK3CA", "PTEN", "CTNNB1")
crispr_sig_genes <- filter(crispr_data, Hugo_Symbol %in% sig_genes)
crispr_sig_genes$Hugo_Symbol <- factor(crispr_sig_genes$Hugo_Symbol, levels = c("KRAS", "TP53", "NRAS", "BRAF", "PIK3CA", "PTEN", "CTNNB1"))

crispr_color <- as.character(crispr_sig_genes$Color)
names(crispr_color) <- crispr_sig_genes$Mutation_Status

# Mutation status
plot_mut <- ggplot(data = crispr_sig_genes, aes(x = Mutation_Status, y = Score, color = Mutation_Status)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(alpha = 0.3, size = 0.7, position = position_jitter(w = 0.05)) +
  facet_wrap(~ Hugo_Symbol, nrow = 7, scales = "free_y") +
  scale_color_manual(values = crispr_color) +
  geom_hline(yintercept = 0, linetype = 2, lwd = 0.3) +
  theme_light() +
  theme(legend.position = "none") +
  labs(x = "Mutation Status", y = "CERES Score",
       title = "Mutation Status")
# plot_mut

# Copy number
plot_cn <- ggplot(data = crispr_sig_genes, aes(x = Copy_Number, y = Score, color = Mutation_Status)) +
  geom_point(size = 0.5, alpha = 0.5) +
  geom_smooth(method = "lm", size = 0.5) +
  scale_color_manual(values = crispr_color) +
  facet_wrap(~ Hugo_Symbol, nrow = 7, scales = "free") +
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc = "right", label.y.npc = "top") +
  theme_light() +
  theme(legend.position = "none", axis.title.y = element_blank()) +
  labs(y = "CERES Score", color = "Mutation Status",
       x = "Theoretical Copy number", title = "Copy Number")
# plot_cn

# Gene expression
plot_ge <- ggplot(data = crispr_sig_genes, aes(x = RPKM_log2, y = Score, color = Mutation_Status)) +
  geom_point(size = 0.5, alpha = 0.5) +
  geom_smooth(method = "lm", size = 0.5) +
  facet_wrap(~ Hugo_Symbol, nrow = 7, scales = "free_y") +
  scale_color_manual(values = crispr_color) +
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc = "left", label.y.npc = "top") +
  theme_light() +
  theme(legend.position = "none", axis.title.y = element_blank()) +
  labs(y = "CERES Score", color = "Mutation Status",
       x = "Gene Expression [log2(RPKM)]",
       title = "Gene Expression")
# plot_ge

# ggsave(filename = "./plots_18Q3/crispr/omics_sig_genes.pdf", plot = grid.draw(cbind(ggplotGrob(plot_mut), ggplotGrob(plot_ge), ggplotGrob(plot_cn), size = "last")), width = 15, height = 18, device = "pdf")
```



# shRNA

--------------------------------------------------------------------------------

## Data and annotations

DepMap cell line metadata:
```{r, eval=FALSE}
# From figshare
shrna_meta <- read.delim("./data_munging/sample_info_18Q3_shrna.csv", sep = ",", header = TRUE, na.strings = c("", NA))
colnames(shrna_meta)[1] <- "CCLE_Name"
```

The Achilles shRNA DEMETER score data was pulled from the [CTD^2^ Data Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (Tsherniak et al 2017).
```{r, eval=FALSE}
shrna <- read.table("./data_munging/D2_combined_gene_dep_scores.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
colnames(shrna)[1] <- "Hugo_Symbol"
# Remove Entrez gene IDs from gene names
shrna$Hugo_Symbol <- gsub(" .*", "", shrna$Hugo_Symbol)

# Melt shRNA dataset for merging
shrna_melt <- melt(shrna , id.vars = "Hugo_Symbol", measure.vars = colnames(shrna)[2:ncol(shrna)], variable.name = "CCLE_Name", value.name = "Score")
shrna_melt <- drop_na(shrna_melt)
```

Merge annotation data:
```{r, eval=FALSE}
# Merge cell line metadata
shrna_melt <- merge(shrna_melt, ccl_info, by = "CCLE_Name", all.x = TRUE)
shrna_melt <- merge(shrna_melt, shrna_meta, by = "CCLE_Name", all.x = TRUE)

# Merge mutation annotations
shrna_muts <- merge(shrna_melt, maf_df, by = c("Hugo_Symbol", "CCLE_Name", "Broad_ID"), all.x = TRUE)
shrna_muts <- shrna_muts %>% mutate(Mutation_Status = if_else(is.na(Mutation_Status), "Other", Mutation_Status))
shrna_muts$Hugo_Symbol <- factor(shrna_muts$Hugo_Symbol)

# Summarize number of mutant and Other cell lines
shrna_muts_summ <- shrna_muts %>% group_by(Hugo_Symbol) %>%
  summarize(N_Other = sum(Mutation_Status == "Other"),
            N_Mutant = sum(Mutation_Status == "Mutant"))

# Merge test results back into full dataset, which restores information lost in the summarization
shrna_data <- merge(shrna_muts_summ, shrna_muts, by = "Hugo_Symbol")

# Add Color column
shrna_data$Color <- ifelse(shrna_data$Mutation_Status == "Other", "cyan3", "darkorchid")
shrna_data$Color <- factor(shrna_data$Color)

# Cell line lineages
shrna_data <- merge(shrna_data, ccl_converter, by = c("CCLE_Name", "Broad_ID"), all.x = TRUE)
levels(shrna_data$lineage_name) <- sort(levels(shrna_data$lineage_name), decreasing = TRUE)

# Copy number
shrna_data <- merge(shrna_data, cn_melt, by = c("Hugo_Symbol", "Broad_ID"), all.x = TRUE)

# Gene expression (RPKM)
ge_filt <- filter(ge_melt, Hugo_Symbol %in% unique(shrna_data$Hugo_Symbol))
shrna_data <- merge(shrna_data, ge_filt, by = c("Hugo_Symbol", "Broad_ID", "CCLE_Name"), all.x = TRUE)

saveRDS(shrna_data, "./../crispr_lineages_giant_files/shrna_data_18Q3.rds", compress = "xz")
```

```{r, eval=FALSE}
shrna_data <- readRDS("./../crispr_lineages_giant_files/shrna_data_18Q3.rds")
shrna_ccl <- data.frame("Broad_ID" = shrna_data$Broad_ID)
```

## Wilcoxon tests

### Grouped by gene

```{r, eval=FALSE}
shrna_signif <- compare_means(Score ~ Mutation_Status, group.by = c("Hugo_Symbol"), data = shrna_data, method = "wilcox.test", p.adjust.method = "BH")
shrna_signif <- adj_signif(shrna_signif)
shrna_signif <- shrna_signif[order(shrna_signif$p),]
saveRDS(shrna_signif, "./data_munging/rds/shrna_signif.rds")
```

```{r, eval=FALSE}
shrna_signif <- readRDS("./data_munging/rds/shrna_signif.rds")

knitr::kable(filter(shrna_signif, p < 0.01)[, c("Hugo_Symbol", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "shRNA Screen: Wilcoxon Test Results Comparing Mutant and Other Cell Lines, p < 0.1  (Benjamini-Hochberg-corrected p-values: * p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

### Grouped by gene and lineage

```{r, eval=FALSE}
shrna_signif_lineage <- compare_means(Score ~ Mutation_Status, group.by = c("Hugo_Symbol", "lineage_name"), data = shrna_data, method = "wilcox.test", p.adjust.method = "BH")
shrna_signif_lineage <- adj_signif(shrna_signif_lineage)
shrna_signif_lineage <- mutate(shrna_signif_lineage, lineage_name = reorder(lineage_name, p.adj, mean))
saveRDS(shrna_signif_lineage, "./../crispr_lineages_giant_files/shrna_signif_lineage.rds")
```

```{r, eval=FALSE}
shrna_signif_lineage <- readRDS("./../crispr_lineages_giant_files/shrna_signif_lineage.rds")
```

# References

--------------------------------------------------------------------------------

Barretina, J., Caponigro, G., Stransky, N., Venkatesan, K., Margolin, A. A., Kim, S., … Garraway, L. A. (2012). The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity. Nature, 483(7391), 603–607. https://doi.org/10.1038/nature11003

Broad Institute Cancer Dependency Map; Cancer Data Science (2018): Cancer Dependency Map, CRISPR Avana dataset 18Q3 (Avana_public_18Q3). figshare. Fileset. doi:10.6084/m9.figshare.6931364.v1

Consortium, T. C. C. L. E., & Consortium, T. G. of D. S. in C. (2015). Pharmacogenomic agreement between two cancer cell line data sets. Nature, 528(7580), 84–87. https://doi.org/10.1038/nature15736

Data Science, Cancer (2018): DEMETER2 data. figshare. Fileset. doi:10.6084/m9.figshare.6025238.v2

Doench, J. G., Fusi, N., Sullender, M., Hegde, M., Vaimberg, E. W., Donovan, K. F., … Root, D. E. (2016). Optimized sgRNA design to maximize activity and minimize off-target effects of CRISPR-Cas9. Nature Biotechnology, 34(2), 184–191. https://doi.org/10.1038/nbt.3437

Meyers, R. M., Bryan, J. G., McFarland, J. M., Weir, B. A., Sizemore, A. E., Xu, H., … Tsherniak, A. (2017). Computational correction of copy-number effect improves specificity of CRISPR-Cas9 essentiality screens in cancer cells. Nature Genetics, 49(12), 1779–1784. https://doi.org/10.1038/ng.3984

McFarland, J. M., Ho, Z. V., Kugener, G., Dempster, J. M., Montgomery, P. G., Bryan, J. G., … Tsherniak, A. (2018). Improved estimation of cancer dependencies from large-scale RNAi screens using model-based normalization and data integration. https://doi.org/10.1101/305656


# Session information

--------------------------------------------------------------------------------

```{r}
print(sessionInfo())
```




