---
output              :
  word_document     :
    reference_docx  : ./word_reference_template.docx
    toc             : true
    toc_depth       : 2
    fig_caption     : true
    df_print        : kable
    # citation_package: natbib
header-includes:
- \usepackage{caption}
title               : "Manuscript Draft"
author              :
- Sally E. Claridge
- Daniel M. Charytonowicz
- Adam A. Margolin
date                : "`r format(Sys.time(), '%d %B %Y')`"
bibliography        : manuscript.bib
fontsize            : 12pt
geometry            : margin=1in
linkcolor           : blue
urlcolor            : blue
citecolor           : blue
floatsintext        : yes
figurelist          : no
tablelist           : no
footnotelist        : no
linenumbers         : yes
abstract            : Multiple high-throughput functional screens in cancer cell lines have generated large amounts of information on drug efficacy in a variety of genomic contexts and cancer lineages. Reported inconsistencies between datasets have led us to investigate whether these large-scale screens reproduce established clinical drug-gene associations and if genomic features particular to specific genes improve said reproducibility. We evaluated three large-scale, small-molecule drug screens within the context of clinical interpretations derived from a new cancer variant annotation resource published by the Variant Interpretation for Cancer Consortium (VICC). We identified low levels of concordance between the three drug screen datasets and clinical drug-gene associations using mutation status, gene expression, and copy number as genomic indicators. Less than half of the clinical drug-gene cancer associations from the VICC resource were identified in these three drug screen datasets, suggesting a barrier to translating findings from these large-scale screens into the clinic.
---

\newpage

```{r setup, include=FALSE}
library(ggpubr)
library(tidyverse)
library(kableExtra)
library(reshape2)
# install.packages("tinytex")
# tinytex::install_tinytex()
# install.packages("citr")
# devtools::install_github("benmarwick/wordcountaddin", type = "source", dependencies = TRUE)
# update wordcount in Addins
```

```{r analysis-preferences, include=FALSE}
# Seed for random number generation
set.seed(42)
knitr::opts_chunk$set(cache.extra = knitr::rand_seed)
```

# Introduction

Cancer cell lines are a long-standing model for systematic testing of candidate therapeutics, beginning with the National Cancer Institute 60 (NCI60) assay from the late 1980s [@alley_feasibility_1988; @shoemaker_nci60_2006; @stinson_morphological_1992], which has been used to screen over 100,000 compounds as of 2010 [@holbeck_analysis_2010]. Since then, numerous small-molecule and gene essentiality screens of various scales and study aims have been conducted in cell lines, from grouping drugs by therapeutic target similarity [@greshock_molecular_2010] to screening only near-haploid cell lines to generate genome-level insights into gene essentiality [@wang_identification_2015] to broadly identifying cancer dependencies with large-scale screens in multiple cancer types [@mcdonald_project_2017; @mcfarland_improved_2018; @meyers_computational_2017; @patel_identification_2017] or select lineages [@heiser_integrated_2009; @marcotte_essential_2012; @patel_identification_2017]. Despite their widespread use, cancer cell lines are known to have issues with inconsistent naming conventions and contamination [@yu_resource_2015], and some cell lines have been shown to vary widely at the genetic level and in response to drug treatment across strains [@ben-david_genetic_2018]. Additionally, comparisons between cancer cell lines and tumors indicate that cell lines have higher numbers of genomic aberrations [@domcke_evaluating_2013; @mouradov_colorectal_2014; @neve_collection_2006] and tend to be hypermethylated [@paz_systematic_2003; @varley_dynamic_2013], which could prove an impediment to translating cell line discoveries into the clinic.

There have also been debates over consistency between drug screen datasets, namely the Broad Institute’s and Novartis Institutes for Biomedical Research’s Cancer Cell Line Encyclopedia (CCLE) [@barretina_cancer_2012; @consortium_pharmacogenomic_2015] and the Genomics of Drug Sensitivity in Cancer (GDSC) from the Cancer Genome Project at the Wellcome Sanger Institute and the Center for Molecular Therapeutics at Massachusetts General Hospital Cancer Center [@garnett_systematic_2012; @yang_genomics_2012]. The GDSC has also been referred to as the Cancer Genome Project (CGP) and the Sanger dataset. Studies have shown that drug-gene interactions matched between CCLE and GDSC exhibited poor correlation and inconsistencies [@haibe-kains_inconsistency_2013; @jang_systematic_2014], prompting other groups to join the debate on how best correct for experimental and methodological variation between the original drug screens and subsequent computational analysis [@consortium_pharmacogenomic_2015; @geeleher_cancer_2016; @geeleher_consistency_2016; @hatzis_enhancing_2014; @haverty_reproducible_2016; @safikhani_revisiting_2017; @safikhani_safikhani_2016]

The prevalence and importance of cancer cell lines in large-scale therapeutic research and the apparent inconsistencies between the GDSC and CCLE encouraged us to compare the results from these functional screens to clinical drug-gene associations. An outstanding question concerning studies based on cancer cell lines is whether these the cell line systems can accurately model tumor dynamics or recapitulate clinical cancer vulnerabilities, and many large-scale grants and clinical trials are fundamentally anchored by results from screens conducted in cancer cell lines. Thus, our goal was to evaluate how well these functional screens recapitulate known drug, gene, and tumor type associations that are currently used in clinical decision-making.

# Results

## Curating clinical biomarkers

To curate an evidence-based list of these clinical biomarkers, we utilized a new oncological data aggregation project conducted by the Variant Interpretation for Cancer Consortium (VICC; https://cancervariants.org/), a Driver Project for the Global Alliance for Genomics and Health (GA4GH) [@global_alliance_for_genomics_and_health_federated_2016]. The VICC has curated annotations of known cancer variants at varying levels of evidence from multiple resources [@wagner_harmonized_2018]: the Precision Medicine Knowledgebase [@huang_cancer_2016], MolecularMatch (https://www.molecularmatch.com/index.html), OncoKB [@chakravarty_oncokb:_2017], Jackson Labs Clinical Knowledgebase [@patterson_clinical_2016], Clinical Interpretations of Variance in Cancers (CIViC) [@griffith_civic_2017], and the Cancer Genome Interpreter (CGI) [@tamborero_cancer_2018]. These harmonized variants are hosted on an ElasticSearch (Kibana v6.0) platform called Genotype to Phenotype (G2P, https://search.cancervariants.org/#*) that allows users to query and filter aggregated evidence from the various databases listed above as well as the GA4GH beacon service (http://beacon-network.org/#/), which yields access to genetic mutation data from over 200 datasets as of 2016 [@global_alliance_for_genomics_and_health_federated_2016]. For drug screen datasets, we focus on three well-known, large-scale drug projects: the CCLE and GDSC, which were mentioned above, and the Center for the Science of Therapeutics at the Broad Institute’s Cancer Therapeutics Response Portal (CTRP) dataset [@basu_interactive_2013; @rees_correlating_2016; @seashore-ludlow_harnessing_2015]. We also analyzed the results from a large-scale CRISPR/Cas9 screen conducted by the Broad Institute’s Cancer Dependency Map project (DepMap) using the Avana knockout library [@doench_optimized_2016; @meyers_computational_2017], which will address whether gene essentiality screens via CRISPR yield results comparable to functional drug screens.

## Do small-molecule screens in cancer cell lines recapitulate clinical drug-gene associations?

```{r functions, echo=FALSE, include=FALSE}
e2exp_converter <- function(string, p_type) {
  # Convert p-vals in scientific format to latex-equation style p-vals with exponent
  e <- formatC(as.numeric(string), format = "e", digits = 1)
  if(p_type == "pk") {
    if(grepl("(\\d+)e-([1-9]+)", e)) {
      exp <- sub("(.+)e-(\\d+)", "$p_k = \\1^{-\\2}$", e, perl = TRUE)
    }
    if(!grepl("(\\d+)e-([1-9]+)", e)) {
      exp <- sub("(.+)e-0(\\d)", "$p_k = \\1^{-\\2}$", e, perl = TRUE)
    }
  }
  if(p_type == "p") {
    if(grepl("(\\d+)e-([1-9]+)", e)) {
      exp <- sub("(.+)e-(\\d+)", "$p = \\1^{-\\2}$", e, perl = TRUE)
    }
    if(!grepl("(\\d+)e-([1-9]+)", e)) {
      exp <- sub("(.+)e-0(\\d)", "$p = \\1^{-\\2}$", e, perl = TRUE)
    }
  }
  if(p_type == "padj") {
    if(grepl("(\\d+)e-([1-9]+)", e)) {
      exp <- sub("(.+)e-(\\d+)", "$p_{adj} = \\1^{-\\2}$", e, perl = TRUE)
    }
    if(!grepl("(\\d+)e-([1-9]+)", e)) {
      exp <- sub("(.+)e-0(\\d)", "$p_{adj} = \\1^{-\\2}$", e, perl = TRUE)
    }
  }
  return(exp)
}

drugstring2intext_converter <- function(string, type) {
  # Convert drug-gene/drug-gene-lineage strings to have italic gene name
  if(type == "dg") {
    d_g <- strsplit(string, split = "_")
    text <- paste0(d_g[[1]][1], "-*", d_g[[1]][2], "*")
  }
  if(type == "dgl") {
    d_g_l <- strsplit(string, split = "_")
    text <- paste0(d_g_l[[1]][1], "-*", d_g_l[[1]][2], "*-", d_g_l[[1]][3])
  }  
  return(text)
}
```

```{r g2p-data, echo=FALSE, include=FALSE}
ccl_meta <- read.delim("./data_munging/19Q2/sample_info_edit_20191008.csv", sep = ",", header = TRUE, row.names = 1)
ccl_meta <- ccl_meta %>% select(DepMap_ID, Cell_line, disease)
ccl_unique <- data.frame(disease = unique(ccl_meta$disease))

g2p <- read.delim("./data_munging/g2p_db_edited_20190718.csv", sep = ",", header = TRUE, row.names = 1)

g2p_dgl <-  as.character(unique(g2p$Drug_Gene_Lin[!is.na(g2p$Drug_Gene_Lin)]))
g2p_dg <- as.character(unique(g2p$Drug_Gene[!is.na(g2p$Drug_Gene)]))
g2p_drugs <- as.character(unique(g2p$Drug[!is.na(g2p$Drug)]))
g2p_genes <- as.character(unique(g2p$Gene[!is.na(g2p$Gene)]))
g2p_dg_genes <- unique(sapply(strsplit(g2p_dgl, "_", fixed = TRUE), function(x) x[2]))
g2p_lins <- as.character(unique(g2p$group_general_lineage_name[!is.na(g2p$group_general_lineage_name)]))

g2p$biomarker_type_curated <- ifelse(is.na(g2p$biomarker_type_curated), "None", as.character(g2p$biomarker_type_curated))
g2p_split <- filter(g2p, !is.na(Drug)) %>% split(g2p$biomarker_type_curated)
g2p_amp <- g2p_split[[1]]
g2p_expr <- g2p_split[[2]]
g2p_fus <- g2p_split[[3]]
g2p_fus$FusionPair <- sapply(X = g2p_fus$Feature.Names, function(x) strsplit(as.character(x)," ")[[1]][1])
g2p_fus$FusionPair <- ifelse(g2p_fus$FusionPair == "Fusions", as.character(g2p_fus$Gene), as.character(g2p_fus$FusionPair))
g2p_fus$FusionPair <- ifelse(g2p_fus$FusionPair == "BCR-ABL", "BCR-ABL1", as.character(g2p_fus$FusionPair))
g2p_fus <- g2p_fus[grep("-", g2p_fus$FusionPair), ]
g2p_fus$Drug_FusionPair <- ifelse(is.na(g2p_fus$Drug) | is.na(g2p_fus$FusionPair), NA, paste0(g2p_fus$Drug, "_", g2p_fus$FusionPair))
g2p_fus$Drug_Fusion_Lineage <- ifelse(is.na(g2p_fus$Drug) | is.na(g2p_fus$FusionPair) | is.na(g2p_fus$disease), NA, paste0(g2p_fus$Drug, "_", g2p_fus$FusionPair, "_", g2p_fus$disease))
g2p_mut <- g2p_split[[4]] %>% drop_na(Direction_Curated)
g2p_mut_genes <- unique(sapply(strsplit(as.character(g2p_mut$Drug_Gene)[!is.na(g2p_mut$Drug_Gene)], "_", fixed = TRUE), function(x) x[2]))

g2p_prot <- unique(filter(g2p, !is.na(Gene_Protein_Change) & !is.na(Drug)) %>% select(disease, Drug, Direction_Curated, Gene, Protein_Change, Gene_Protein_Change))
g2p_prot$Drug_Gene_Protein_Change <- with(g2p_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change), NA, paste0(Drug, "_", Gene_Protein_Change)))
g2p_prot$Drug_Gene_Protein_Change_Lineage <- with(g2p_prot, ifelse(is.na(Drug) | is.na(Gene_Protein_Change) | is.na(disease), NA, paste0(Drug, "_", Gene_Protein_Change, "_", disease)))

# Counts
g2p_dg_counts <- g2p %>% count(Drug, Gene) %>% filter(!is.na(Drug), !is.na(Gene))
g2p_dg_counts <- dcast(g2p_dg_counts, Drug ~ Gene, value.var = "n", fun.aggregate = mean, fill = 0) %>% `rownames<-`(.[,1]) %>% select(-Drug)
g2p_dg_counts <- g2p_dg_counts[order(rowSums(-g2p_dg_counts)), order(colSums(-g2p_dg_counts))]
g2p_gene_order <- colnames(g2p_dg_counts)
g2p_drug_order <- rownames(g2p_dg_counts)
```

G2P designed a set of standards for stratifying their database’s cancer variant interpretations, ranging from preclinical data at the low-evidence end (level D) to clinically actionable interpretations at the high-evidence end (level A). To test only these clinically actionable associations, we filtered the G2P dataset for only level-A G2P interpretations, of which there were `r formatC(nrow(g2p), big.mark = ",", big.interval = 3L)` (see Methods). 

![Summary of G2P data content and concordance with drug sensitivity screens. (left) The presence or absence of small molecule drugs analyzed in GDSC, CTRP, and CCLE that were identified with level-A evidence in the G2P database. (right) The number of level-A interpretations for each drug-gene combination in the G2P database. Both axes are ordered by decreasing number of G2P associations.](./plots/manuscript/g2p_summary_heatmap.png)

![Overlap of cancer cell lines in our four datasets.](./plots/manuscript/all_venn_ccls.png) 

### Mutation status

![Mutation status.](./plots/manuscript/all_mut_heatmap.png)

![Protein changes.](./plots/manuscript/all_prot_heatmap.png)

Taking these results as a whole, we observe little concordance between G2P associations and the correlation of mutation status and AUC for said G2P association, suggesting that mutation status alone is not a robust predicto r of clinically actionable genetic targets. This led us to examine the correlations between AUC and gene expression and AUC and copy number, two other genomic features that are known to associate with cancer and drug response. We computed Spearman correlation coefficients ($r_s$) between AUC of each drug-gene association and gene expression (RPKM) and copy number (log$_2$ ratio) of the gene in the association...**need to complete this thought/analysis**

### Fusions

![Fusion status.](./plots/manuscript/all_fus_heatmap.png)

### Copy number

![Correlation between putative copy number and AUC.](./plots/manuscript/all_cn_heatmap.png)

### Gene expression

![Correlation between gene expression and AUC.](./plots/manuscript/all_ge_heatmap.png)




## Do these pancancer drug-gene associations penetrate to the lineage-specific level?

- What do these correlations with mutation status, gene expression, and copy number look like when you constrict to a specific lineage?
- When you restrict, do the correlations become more distinct or are the lineage-specific results approximately the same as the pancancer results?

In the `nrow(g2p_filt)` level-A G2P interpretations, there was a wide variation in the specificity of cancer description, ranging from “Waldenström macroglobulinemia” and “hyper eosinophilic advanced syndrome” on the more detailed end to “cancer” on the broad end. Similarly, the drug screen datasets had a wide range in lineage specificity, e.g. CCLE, GTRP, and GDSC all had lineages labeled “leukemia” and “T-cell childhood acute lymphocytic leukemia” and the CRISPR dataset had cell lines labeled “Epstein-Barr virus-related Burkitt lymphoma” and “lymphoma.” To more easily conduct comparative analyses in a cancer-specific manner, we developed a more general lineage grouping method derived from the Human Disease Ontology identity codes (DOIDs, http://disease-ontology.org/) [@schriml2019] assigned to each cell line, yielding ___ unique lineage groupings. This custom lineage grouping was included in our harmonized cancer cell line database (see Methods).

## Are CRISPR/Cas9 gene essentiality results comparable to those of functional drug screens?


# Discussion

**NUMBERS WRONG HERE**

Level-A G2P interpretations correspond to clinical evidence that suggests efficacy of gene targets and/or drugs, and this work questions whether these relationships manifest themselves in in vitro drug screens. We have demonstrated that on its own, mutation status is not a robust predictor of drug efficacy and that copy number and gene expression fare better as predictors. However, for these three genomic features in all three datasets, only 11.1% (mutation status, 1 of 9, CCLE) to 47.9% (gene expression, 23 of 48, CTRP) (Table 2), of the level-A G2P drug-gene associations in the respective dataset were significant at an uncorrected $\alpha = 0.05$. This suggests that while the drug screens successfully identify some clinically relevant drug-gene associations, many associations are also missed, which raises the question of the extent to which clinical researchers can rely on in vitro drug screens when attempting to develop and select therapeutics for cancer patients.

These results also highlight the need to account for the complex differences inherent between data generated in clinical scenarios and data captured in highly controlled, artificial in vitro environments. Potential sources of variation and error include the fact that cells growing in a laboratory as opposed to those, even of the same tissue type, growing in a multicellular organism are exposed to differing sets of stressors and signals that can significantly impact intracellular signaling pathways, irrespective of shared genomic profiles. These differences can include, but are not be limited to, immunologic reactions, both adaptive and intrinsic (e.g. cytokine signaling, inflammation), endocrine (e.g. stress hormones), and nervous system stimulation, all of which can have significant downstream implications on cellular behavior in the context of therapeutic efficacy. Similarly, a laboratory environment and in vitro cell culture introduce abnormal growth conditions with respect to extracellular matrix composition, nutrient availability, and cell density, all of which have the potential to alter cellular signaling and thus render a drug ineffective during screening despite would-be in vivo activity.

In the era of precision medicine, the ultimate boon would be that the unique omics signatures of a patient’s individual cancer can be used to guide treatment. High-throughput, in vitro screens of targeted cancer agents against cell lines with known omics profiles are beneficial for testing hypotheses concerning the mechanisms of action for these agents and they allow for scalability and systematic screening. However, extrapolating the potential downstream effects of inhibiting a major growth pathway (e.g. MAPK, IP-DAG) to predict the clinical prognosis and progression of a multicellular tumor mass growing in a complex environment, compounded with the cross-reacting effects of tumor genomic heterogeneity, make it a questionable statement that one can safely predict clinical consequences from suppressing a single pathway in a model system, as evidenced by our findings of fewer than half of the known clinical associations in the three drug screen datasets. The benefits of these cancer cell line models, in contrast, lie in their ability to assess the big-picture effects of these agents. In order to fully understand therapeutic drug response, a higher degree of granularity is needed. Additional effort to combine more cell line information such as gene expression, epigenetic profiles, and proteomic data with mutational profiles is essential to improving the efficacy and validity of in vitro cell line screens. Further benefit could be derived from conducting these screens in environments that are more in line with the clinical scenarios we are trying to predict. This would include things such a 3-dimensional cell culture, coculturing with stromal cells to mimic the tumor microenvironment, as well as other modalities that can attempt to better replicate in vivo environments.

As a next step to improve the informational granularity of the analysis presented here, our subsequent goal will be to assess the extent to which these drug-gene associations are specific to individual cancer cell line lineages, which are available and annotated in the data sets analyzed within this report. By evaluating how many clinically actionable associations are identified in these large-scale functional screens, we can begin to address best practices for translating discoveries from these tumor models into clinical trials. 

# Methods

**Need to update all methods for data processing, these method are current up to ~May 2019**

## Drug screen dataset retrieval

**12/4 updated**

We utilized small molecule screening data from the Cancer Therapeutics Response Portal (CTRP, Broad Institute) [@basu_interactive_2013; @rees_correlating_2016; @seashore-ludlow_harnessing_2015], the Genomics of Drug Sensitivity in Cancer (GDSC, Wellcome Sanger Institute and Massachusetts General Hospital Cancer Center), [@garnett_systematic_2012; @yang_genomics_2012], and the Cancer Cell Line Encyclopedia (CCLE, Broad Institute and Novartis Institutes for Biomedical Research) [@barretina_cancer_2012; @consortium_pharmacogenomic_2015]. To ensure metric consistency, normalized area under the IC50 curve (AUC) for all three datasets were taken from the Quantitative Analysis of Pharmacogenomics in Cancer (http://tanlab.ucdenver.edu/QAPC/) (QAPC) [@pozdeyev_integrating_2016].

## DepMap data retrieval and processing

**12/4 updated**

CRISPR/Cas9 and mutation calls, fusions, gene expression, and copy number data from the CCLE were all retrieved from the Broad Institute’s Cancer Dependency Map (DepMap) Public 19Q2 release via the DepMap data portal (https://depmap.org/portal/download/) [@broad_depmap_depmap_2019-1]. For cell lines with no CCLE annotation, the DepMap group drew raw copy number and mutation data from whole exome sequencing data produced by the Wellcome Trust Sanger Institute [Catalogue Of Somatic Mutations In Cancer (COSMIC, https://cancer.sanger.ac.uk/cell_lines) [@bamford_cosmic_2004; @forbes_cosmic:_2017]; European Genome-phenome Archive [@lappalainen_european_2015], accession number EGAD00001001039] and processed the data following the CCLE pipelines to ensure consistency.

### CRISPR/Cas9 screen

Gene effect scores were inferred from a CRISPR/Cas9 (clustered regularly interspaced short palindromic repeats/CRISPR-associated 9) screen using the Broad Institute’s Avana knockout library [@doench_optimized_2016]. The Broad Institute’s data processing and screening methods are available from the figshare record [@broad_depmap_depmap_2019-1] and the original publication of CERES, the algorithm that computes inferred gene dependency scores [@meyers_computational_2017].

### Cell line harmonization

**12/4 updated**

In an effort to compare CCLE, CTRP, GDSC and CRISPR results and include lineage specificity, it was necessary to consolidate cell lines screened in each dataset. Given that unique identifiers can be used between studies, it was necessary to standardize the identity of each cell line with a common source. To achieve this, a freely available cell line database called Cellosaurus (v26.0) [@bairoch_cellosaurus_2018] was downloaded and merged with the Broad Institute’s DepMap cell line data to create a harmonized cancer cell line database containing synonymous identifiers that enabled consolidation of all cell lines used in CCLE, CTRP, GDSC and CRISPR into a common framework. Subsequent modifications to this database included the merging of DepMap IDs used in the 19Q2 DepMap data release and harmonization with the consensus cell line names from @pozdeyev_integrating_2016. We curated missing synonyms and identifiers, and we manually annotated the granularity of lineage labeling for all cell lines analyzed in this study based on their Human Disease Ontology identity codes (DOIDs, http://disease-ontology.org/) [@schriml_human_2019], resulting in `r length(levels(g2p$disease))` unique lineages.

### Mutation calls

**12/4 EDIT THIS**

To preclude variation in genomic feature calls across the datasets, we used identical omics annotations from CCLE for all cell lines screened in the three datasets. The CCLE called substitutions using MuTect [@cibulskis_sensitive_2013] and annotated variants using Oncotator [@ramos_oncotator:_2015] and indels using Indelocator (https://software.broadinstitute.org/cancer/cga/indelocator).

We had three different mutation types: binarily encoded mutation status (regardless of quantity of harbored mutations per gene in a given cell line), specific protein changes, and fusion status.

We filtered mutation calls for point mutations, defined as single-nucleotide insertions, deletions, and substitutions, regardless of result, i.e. frameshift, missense, or nonsense mutation. For this analysis, cell lines harboring non-silent point mutations were considered "mutant" for the gene in question. All other mutations and genes without annotation in the Mutation Annotation Format (MAF) file were considered "wildtype." "Mutant" and "wildtype" calls were binarily encoded per-gene/per-cell line, regardless of quantity of harbored mutations per gene in a given cell line.

### Copy number data

**12/4 EDIT THIS**

The CCLE generated genomic copy number (CN) data using the Affymetrix Genome-Wide Human SNP Array 6.0 and GenePattern pipeline [@thecancergenomeatlasresearchnetwork2008] and normalized segmented CN log2-ratios for 23,299 genes across 1,098 cell lines using circular binary segmentation [@olshen2004] (dataset: public_18Q4_gene_cn.csv, accessed 15 November 2018).

### Gene expression data

**12/4 EDIT THIS**

The CCLE reports gene expression in RPKM (reads per kilobase per million mapped reads) for 54,356 genes across 1,156 cell lines, generated on the GeneChip Human Genome U133 Plus 2.0 Array (dataset: CCLE_DepMap_18q3_RNAseq_RPKM_20180718.gct, accessed 18 July 2018).

## G2P cancer variants

**12/4 updated**

Cancer variants with the highest level of evidence (i.e. level A) in the Genotype to Phenotype (G2P) database were retrieved from the Variant Interpretation for Cancer Consortium (VICC) portal (https://search.cancervariants.org/#*) [@wagner_harmonized_2018] using a customized JSON-query script. All JSON queries were passed to the available application program interface (API) where a request was made for all drug-gene associations with level-A evidence via the G2P evidence label, i.e. `association.evidence_label`. From manual inspection using G2P’s front-end Kibana interface, it was known that, at the time of the last query, 27 November 2018, there were 1,297 known level-A associations. Due to the limited request processing capabilities of the G2P JSON API, queries were batched into packets of 10 data points, for a total of 130 requests made in succession. Each returned request was processed as JSON object and searched to identify any existing key:value pairs for the following variables: evidence level, mutation, gene, chromosome, start, end, ref, alt, direction, phenotype description, phenotype family, phenotype ID, drug, drug ID, feature names, and sequence ID. For any evidence point where a given key:value pair was not found or unavailable, a value of -1 was assigned. One of the 1,297 level-A entries was irretrievable, yielding a final dataset of 1,296 level-A G2P interpretations, which were stored in a pandas data table and exported in CSV format. We split each interpretation by row to isolate drug associations with singular lineages, protein changes, and/or mutation types. The VICC’s methods for the harvesting and harmonizing of cancer variants is available in a GitHub repository from Oregon Health & Science University (https://github.com/ohsu-comp-bio/g2p-aggregator).

All compounds screened within the GDSC, CTRP, and CCLE datasets are considered "small molecules" and are respectively linked to Compound (CID) numbers used for indexing in the PubChem database (https://pubchem.ncbi.nlm.nih.gov/). As a result, non-small molecules, which include proteins and biologics (i.e. monoclonal antibodies), that had level-A G2P evidence could not be included in our comparisons. Thus, G2P interpretations for which the compound did not have a matching CID code were excluded from further analysis as were interpretations that did not have an associated compound. It should be noted that cisplatin, a small molecule, was excluded due to its being indexed with a ChEMBLdb identifier (https://www.ebi.ac.uk/chembl/), which we did not manually re-index to a CID to avoid mischaracterization.

## Computational

**12/4 EDIT THIS**
```{r, include=FALSE}
r_v <- paste0("v", getRversion())
# rstudio_v <- paste0("v", RStudio.Version()$version)
ggplot_v <- paste0("v", packageVersion("ggplot2"))
ggpubr_v <- paste0("v", packageVersion("ggpubr"))
```

For *Figure 1*, all data were imported and manipulated using pandas (v0.23.0) [@mckinney2011] and plotted using seaborn (v0.90) [@waskom2012], which sat on top of matplotlib (v2.2.2) [@hunter2007] and was executed on Python (v2.7.15) (Python Software Foundation, https://www.python.org/) in an IPython (v5.5.0) [@perez2007] kernel within a localhost Jupyter (v5.2.2) notebook [@kluyver2016]. All other analyses were conducted in R (`r r_v`) [@rdevelopmentcoreteam2010, https://www.R-project.org/] in an RStudio environment (v1.1.447) [@rstudio2012]. All other plots were drawn using the ggplot2 (`r ggplot_v`) R package [@wickham2016].

### Statistics

**12/4 EDIT THIS**

Wilcoxon tests were conducted using the `compare_means` function in the ggpubr (`r ggpubr_v`) R package [@kassambara2018]. `compare_means` does Benjamini-Hochberg p-value adjustment using the `p.adjust` function from the `stats` base R package.

For computing Spearman correlation coefficients, we used the `cor.test` function from the `stats` package. If there were 3 or fewer data points, then the correlation was not done and a value of `NA` was supplied. 

# Conflicts of interest?

# Grant support?

# References

