---
title: "Manuscript"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the Broad's Avana CRISPR (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017) and the Broad and Dana-Farber Cancer Institite's Achilles shRNA (MacFarland et al. 2018, Data Science 2018).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

All annotation files (copy number, mutation status, and gene expression) (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/).

# Set up

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)

# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

Edited `.Renviron` to avoid "Error: memory exhausted (limit reached?)" when melting the GE dataset, as per the suggestion from [StackOverflow](https://stackoverflow.com/questions/51295402/r-on-macos-error-vector-memory-exhausted-limit-reached)
```
cd ~
touch .Renviron
open .Renviron

# Added following line to .Renviron:
R_MAX_VSIZE=100Gb
```

```{r}
library(bsselectR)
library(data.table)
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(htmlwidgets)
library(kableExtra)
library(plotly)
library(reshape2)
library(scales)
library(tidyverse)
```

# Functions {.tabset .tabset-fade .tabset-pills}

--------------------------------------------------------------------------------

## Significance thresholding

```{r}
adj_signif <- function(df) {
  # Takes df from compare_means and creates a signif code column for adjusted p-vals
  df$p.signif <- ifelse(df$p.signif == "ns", NA, df$p.signif)
  df$p.signif.adj <- ifelse(df$p.adj <= 0.0001, "****",
                     ifelse(df$p.adj <= 0.001, "***",
                     ifelse(df$p.adj <= 0.01, "**",
                     ifelse(df$p.adj <= 0.05, "*", NA))))
  df$p.short <- formatC(df$p, format = "g", digits =  2)
  df$p.adj.short <- formatC(df$p.adj, format = "g", digits =  2)
  
  return(df)
}
```

## Wilcoxon tests by drug

```{r}
WilcoxonByDrug <- function(drug, dataset, data_name, grouping_var) {
  if(grouping_var == "lineage") {
    grouper <- c("Hugo_Symbol", "group_general_lineage_name")
  }
  else if(grouping_var == "gene") {
    grouper <- "Hugo_Symbol"
  }
  use_data <- filter(dataset, Drug == drug)
  sig <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = grouper, data = use_data, method = "wilcox.test", p.adjust.method = "BH")
  sig <- adj_signif(sig)
  sig <- sig[order(sig$p),]
  sig$Drug <- drug
  sig$Dataset <- toupper(data_name)
  return(sig)
}


# WilcoxonByDrugAllGenes <- function(melt_file, grid_file, data_name) {
#   druglist <- as.character(unique(melt_file$Drug))
#   sig_df <- NULL
#   for(i in 1:length(druglist)) {
#     print(paste0("Current drug in process: ", druglist[i], " (drug ", i, " of ", length(druglist), ")."))
#     flush.console()
#     melted <- filter(melt_file, Drug == druglist[i])
#     use_data <- merge(grid_file, melted, by = "accession_id", all.x = TRUE)
#     use_data <- merge(use_data, ccl_converter, by = "accession_id", all.x = TRUE)
#     use_data <- merge(use_data, maf_df, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
#     use_data$Mutation_Status_Nonsilent <- ifelse(is.na(use_data$Mutation_Status_Nonsilent), "Wildtype", use_data$Mutation_Status_Nonsilent)
#     use_data$Hugo_Symbol <- factor(use_data$Hugo_Symbol)
#     use_data$AUC <- as.numeric(use_data$AUC)
#     use_data <- filter(use_data, !is.na(use_data$AUC))
#     use_data$Drug_Gene <- paste0(use_data$Drug, "_", use_data$Hugo_Symbol)
#     use_data_ptmuts <- filter(use_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
#     use_data_ptmuts$Keep_PtMut_Tests <- ifelse(use_data_ptmuts$Reference_Allele_Length == 1 | is.na(use_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(use_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
#     use_data_ptmuts <- filter(use_data_ptmuts, Keep_PtMut_Tests == TRUE)
#     
#     sig <- compare_means(AUC ~ Mutation_Status_Nonsilent, group.by = "Hugo_Symbol", data = use_data_ptmuts, method = "wilcox.test", p.adjust.method = "BH")
#     sig <- adj_signif(sig)
#     sig <- sig[order(sig$p),]
#     sig <- filter(sig, p <= 0.05)
#     sig$Drug <- druglist[i]
#     sig$Dataset <- toupper(data_name)
#     sig_df <- rbind(sig_df, sig)
#   }
#   return(sig_df)
# }
```

## Pearson correlation by lineage
```{r}
PearsonByLineage <- function(lin, dataset, variable) {
  use_data <- filter(dataset, group_general_lineage_name == lin)
  if(variable == "GE") {
    use_data <- filter(use_data, !is.na(RPKM))
    use_data_summ <- use_data %>% count(Drug_Gene)
    use_data_summ <- filter(use_data_summ, n >= 3) # cor.test requires at least 3 non-NA data points
    use_data_filt <- filter(use_data, Drug_Gene %in% as.character(unique(use_data_summ$Drug_Gene)))
    if(nrow(use_data_filt) > 0) {
      pearson_res <- use_data_filt %>% group_by(Drug_Gene) %>% summarize(Pearson_Corr_GE = cor.test(y = AUC_zscore, x = RPKM, method = "pearson", use = "complete.obs")$estimate, Pearson_Pval_GE = cor.test(y = AUC_zscore, x = RPKM, method = "pearson", use = "complete.obs")$p.value)
      pearson_res <- merge(g2p_druggene_p_tally, pearson_res, all.y = TRUE)
      pearson_res$group_general_lineage_name <- lin
      pearson_res$Pearson_GE_Abs <- abs(pearson_res$Pearson_Corr_GE)
      pearson_res$Pearson_GE_Sign <- ifelse(sign(pearson_res$Pearson_Corr_GE) == 1, "+",
                                              ifelse(sign(pearson_res$Pearson_Corr_GE) == -1, "-", "0"))
      return(pearson_res)
    }
  }
  else if(variable == "CN") {
    use_data <- filter(use_data, !is.na(CN))
    use_data_summ <- use_data %>% count(Drug_Gene)
    use_data_summ <- filter(use_data_summ, n >= 3) # cor.test requires at least 3 non-NA data points
    use_data_filt <- filter(use_data, Drug_Gene %in% as.character(unique(use_data_summ$Drug_Gene)))
    if(nrow(use_data_filt) > 0) {
      pearson_res <- use_data_filt %>% group_by(Drug_Gene) %>% summarize(Pearson_Corr_CN = cor.test(y = AUC_zscore, x = CN, method = "pearson", use = "complete.obs")$estimate, Pearson_Pval_CN = cor.test(y = AUC_zscore, x = CN, method = "pearson", use = "complete.obs")$p.value)
      pearson_res <- merge(g2p_druggene_p_tally, pearson_res, all.y = TRUE)
      pearson_res$group_general_lineage_name <- lin
      pearson_res$Pearson_CN_Abs <- abs(pearson_res$Pearson_Corr_CN)
      pearson_res$Pearson_CN_Sign <- ifelse(sign(pearson_res$Pearson_Corr_CN) == 1, "+",
                                              ifelse(sign(pearson_res$Pearson_Corr_CN) == -1, "-", "0"))
      return(pearson_res)
    }
  }
}
```

## Save widget to new directory
```{r}
saveWidgetFix <- function (widget , file, ...) {
  ## A wrapper to saveWidget which compensates for arguable BUG in
  ## saveWidget which requires `file` to be in current working
  ## directory.
  wd <- getwd()
  on.exit(setwd(wd))
  outDir <- dirname(file)
  file <- basename(file)
  setwd(outDir);
  saveWidget(widget, file = file,...)
}
```

## Make G2P lineage mutation status plotly
```{r}
MutPlotlyByLineage <- function(lineage_name) {
  use_data <- filter(g2p_druggenelin_all_summ, group_general_lineage_name == lineage_name)
  use_data$Dataset <- factor(use_data$Dataset, levels = c("CCLE", "CTRP", "GDSC"))
  ticker <- 0
  
  use_data_ccle <- filter(use_data, Dataset == "CCLE")
  use_data_ccle_summ <- use_data_ccle %>% count(InG2P)
  ccle_lin_p_ttest <- data.frame(statistic = NA,
                                 parameter =  NA,
                                 p.value = NA)
  if(nrow(use_data_ccle_summ) <= 1) {
    ccle_lin_p_ttest_res <-  NA
    ticker <- ticker + 1
    }
  else if(use_data_ccle_summ[use_data_ccle_summ$InG2P == "Yes",]$nn < 2) { ccle_lin_p_ttest_res <-  NA }
  else if(use_data_ccle_summ[use_data_ccle_summ$InG2P == "Yes",]$nn >= 2) {
    ccle_lin_p_ttest <- t.test(use_data_ccle$p ~ use_data_ccle$InG2P)
    ccle_lin_p_ttest_res <- paste0("t = ", round(ccle_lin_p_ttest$statistic, 3), ", df = ", round(ccle_lin_p_ttest$parameter, 3), ", p = ", round(ccle_lin_p_ttest$p.value, 3))
  }
  
  use_data_ctrp <- filter(use_data, Dataset == "CTRP")
  use_data_ctrp_summ <- use_data_ctrp %>% count(InG2P)
  ctrp_lin_p_ttest <- data.frame(statistic = NA,
                                 parameter =  NA,
                                 p.value = NA)
  if(nrow(use_data_ctrp_summ) <= 1) {
    ctrp_lin_p_ttest_res <-  NA
    ticker <- ticker + 1
    }
  else if(use_data_ctrp_summ[use_data_ctrp_summ$InG2P == "Yes",]$nn < 2) { ctrp_lin_p_ttest_res <-  NA }
  else if(use_data_ctrp_summ[use_data_ctrp_summ$InG2P == "Yes",]$nn >= 2) {
    ctrp_lin_p_ttest <- t.test(use_data_ctrp$p ~ use_data_ctrp$InG2P)
    ctrp_lin_p_ttest_res <- paste0("t = ", round(ctrp_lin_p_ttest$statistic, 3), ", df = ", round(ctrp_lin_p_ttest$parameter, 3), ", p = ", round(ctrp_lin_p_ttest$p.value, 3))
  }
  
  use_data_gdsc <- filter(use_data, Dataset == "GDSC")
  use_data_gdsc_summ <- use_data_gdsc %>% count(InG2P)
  gdsc_lin_p_ttest <- data.frame(statistic = NA,
                                 parameter =  NA,
                                 p.value = NA)
  if(nrow(use_data_gdsc_summ) <= 1) {
    gdsc_lin_p_ttest_res <-  NA
    ticker <- ticker + 1
    }
  else if(use_data_gdsc_summ[use_data_gdsc_summ$InG2P == "Yes",]$nn < 2) { gdsc_lin_p_ttest_res <-  NA }
  else if(use_data_gdsc_summ[use_data_gdsc_summ$InG2P == "Yes",]$nn >= 2) {
    gdsc_lin_p_ttest <- t.test(use_data_gdsc$p ~ use_data_gdsc$InG2P)
    gdsc_lin_p_ttest_res <- paste0("t = ", round(gdsc_lin_p_ttest$statistic, 3), ", df = ", round(gdsc_lin_p_ttest$parameter, 3), ", p = ", round(gdsc_lin_p_ttest$p.value, 3))
  }
  
  y_value <- ifelse(ticker == 3, 10, max(use_data$n) + 1)

  summ_text <- data.frame(Dataset = c("CCLE", "CTRP", "GDSC"),
                          Test_Results_p = c(ccle_lin_p_ttest_res, ctrp_lin_p_ttest_res, gdsc_lin_p_ttest_res),
                          t_statistic = c(round(ccle_lin_p_ttest$statistic, 3), round(ctrp_lin_p_ttest$statistic, 3), round(gdsc_lin_p_ttest$statistic, 3)),
                          df = c(round(ccle_lin_p_ttest$parameter, 3), round(ctrp_lin_p_ttest$parameter, 3), round(gdsc_lin_p_ttest$parameter, 3)),
                          p_value = c(round(ccle_lin_p_ttest$p.value, 3), round(ctrp_lin_p_ttest$p.value, 3), round(gdsc_lin_p_ttest$p.value, 3)),
                          group_general_lineage_name = lineage_name)

  plot_point <- ggplot(data = use_data) +
    facet_wrap(~ Dataset, drop = FALSE) +
    coord_cartesian(xlim = c(0, 1), ylim = c(0, y_value)) +
    scale_x_continuous(breaks = seq(0, 1, by = 0.1), labels = seq(0, 1, by = 0.1)) +
    scale_y_continuous(breaks = pretty_breaks()) +
    geom_point(mapping = aes(label = Drug_Gene, color = InG2P, x = p, y = n), alpha = 0.5, position = position_jitter(width = 0, height = 0.15)) +
    scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
    geom_text_repel(data = subset(use_data, InG2P == "Yes"), mapping = aes(x = p, y = n, label = Drug_Gene), size = 3, segment.size = 0.2, segment.color = "grey50") +
    geom_text(data = summ_text, mapping = aes(x = 0.5, y = y_value, label = Test_Results_p), nudge_y = 0.1) +
    labs(x = "Drug-gene association p-value between mutation groups", y = "Number of level A G2P associations") +
    theme(legend.position = "none", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank())

  plot_box <- ggplot(data = use_data) +
    facet_wrap(~ Dataset, drop = FALSE) +
    coord_cartesian(ylim = c(0, 1)) +
    scale_y_continuous(breaks = seq(0, 1, by = 0.1), labels = seq(0, 1, by = 0.1)) +
    geom_violin(mapping = aes(color = InG2P, y = p, x = InG2P), alpha = 0.5) +
    geom_boxplot(mapping = aes(color = InG2P, y = p, x = InG2P), alpha = 0.75, width = 0.25) +
    scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
    coord_flip() +
    labs(x = "In G2P?", title = toupper(lineage_name)) +
    theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

  plot <- subplot(plot_box, plot_point, nrows = 2, titleX = TRUE, titleY = TRUE) %>% layout(title = toupper(lineage_name), margin = list(t = 75, l = 75), xaxis5 = list(title = "Gene-drug association p-value between mutation groups"), yaxis = list(title = "Level A G2P\nassociation", domain = c(0.8, 1)), yaxis2 = list(title = "Number of level A G2P associations", domain = c(0, 0.79)))

  saveWidgetFix(widget = plot, file = paste0("./plots_18Q3/manuscript/plotlys_mut_by_lineage/", lineage_name, "_mut_plotly.html"))
  
  plot_point <- ggplot_gtable(ggplot_build(plot_point))
  plot_box <- ggplot_gtable(ggplot_build(plot_box))
  p_maxWidth <-  grid::unit.pmax(plot_point$widths, plot_box$widths)
  plot_point$widths <- as.list(p_maxWidth)
  plot_box$widths <- as.list(p_maxWidth)
  
  mut_plot <- cowplot::plot_grid(plot_box,
                     plot_point,
                     nrow = 2,
                     rel_heights = c(1, 4))
  ggsave(paste0("./plots_18Q3/manuscript/pngs_mut_by_lineage/", lineage_name, "_mut_plot.png"), mut_plot, device = "png", dpi = 450, width = 12, height = 6, units =   "in")
  
  return(summ_text)
}
```

## Make G2P lineage Pearson plotlys
```{r}
PearsonPlotlyByLineage <- function(lineage) {
  print(paste0("Lineage: ", toupper(lineage)))
  use_data <- filter(signif_all_lin, group_general_lineage_name == lineage)
  use_data$Dataset <- factor(use_data$Dataset, levels = c("CCLE", "CTRP", "GDSC"))
  use_data$Pearson_GE_Label <- paste0(round(use_data$Pearson_Pval_GE, 3), "\nDrug_Gene: ", use_data$Drug_Gene)
  use_data$Pearson_CN_Label <- paste0(round(use_data$Pearson_Pval_CN, 3), "\nDrug_Gene: ", use_data$Drug_Gene)
  ticker_ge <- 0
  ticker_cn <- 0
  ticker <- 0
  
  use_data_ccle <- filter(use_data, Dataset == "CCLE")
  use_data_ccle_ge_summ <- use_data_ccle[!is.na(use_data_ccle$Pearson_GE_Abs),] %>% count(InG2P)
  use_data_ccle_cn_summ <- use_data_ccle[!is.na(use_data_ccle$Pearson_CN_Abs),] %>% count(InG2P)
  ccle_ge_ttest <- data.frame(statistic = NA, parameter = NA, p.value = NA)
  ccle_cn_ttest <- data.frame(statistic = NA, parameter = NA, p.value = NA)
  if(nrow(use_data_ccle_ge_summ) <= 1) {
    ccle_ge_res <-  NA
    ticker_ge <- ticker_ge + 1
  }
  else if(use_data_ccle_ge_summ[use_data_ccle_ge_summ$InG2P == "Yes",]$nn < 2) {
    ccle_ge_res <-  NA
  }
  else if(use_data_ccle_ge_summ[use_data_ccle_ge_summ$InG2P == "Yes",]$nn >= 2) {
    ccle_ge_ttest <- t.test(use_data_ccle$Pearson_GE_Abs ~ use_data_ccle$InG2P)
    ccle_ge_res <- paste0("t = ", round(ccle_ge_ttest$statistic, 3), ", df = ", round(ccle_ge_ttest$parameter, 3), ", p = ", round(ccle_ge_ttest$p.value, 3))
  }
  if(nrow(use_data_ccle_cn_summ) <= 1) {
    ccle_cn_res <-  NA
    ticker_cn <- ticker_cn + 1
  }
  else if(use_data_ccle_cn_summ[use_data_ccle_cn_summ$InG2P == "Yes",]$nn < 2) {
    ccle_cn_res <-  NA
  }
  else if(use_data_ccle_cn_summ[use_data_ccle_cn_summ$InG2P == "Yes",]$nn >= 2) {
    ccle_cn_ttest <- t.test(use_data_ccle$Pearson_CN_Abs ~ use_data_ccle$InG2P)
    ccle_cn_res <- paste0("t = ", round(ccle_cn_ttest$statistic, 3), ", df = ", round(ccle_cn_ttest$parameter, 3), ", p = ", round(ccle_cn_ttest$p.value, 3))
  }
  print("CCLE testing: DONE")
  
  use_data_ctrp <- filter(use_data, Dataset == "CTRP")
  use_data_ctrp_ge_summ <- use_data_ctrp[!is.na(use_data_ctrp$Pearson_GE_Abs),] %>% count(InG2P)
  use_data_ctrp_cn_summ <- use_data_ctrp[!is.na(use_data_ctrp$Pearson_CN_Abs),] %>% count(InG2P)
  ctrp_ge_ttest <- data.frame(statistic = NA, parameter = NA, p.value = NA)
  ctrp_cn_ttest <- data.frame(statistic = NA, parameter = NA, p.value = NA)
  if(nrow(use_data_ctrp_ge_summ) <= 1) {
    ctrp_ge_res <-  NA
    ticker_ge <- ticker_ge + 1
  }
  else if(use_data_ctrp_ge_summ[use_data_ctrp_ge_summ$InG2P == "Yes",]$nn < 2) {
    ctrp_ge_res <-  NA
  }
  else if(use_data_ctrp_ge_summ[use_data_ctrp_ge_summ$InG2P == "Yes",]$nn >= 2) {
    ctrp_ge_ttest <- t.test(use_data_ctrp$Pearson_GE_Abs ~ use_data_ctrp$InG2P)
    ctrp_ge_res <- paste0("t = ", round(ctrp_ge_ttest$statistic, 3), ", df = ", round(ctrp_ge_ttest$parameter, 3), ", p = ", round(ctrp_ge_ttest$p.value, 3))
  }
  if(nrow(use_data_ctrp_cn_summ) <= 1) {
    ctrp_cn_res <-  NA
    ticker_cn <- ticker_cn + 1
  }
  else if(use_data_ctrp_cn_summ[use_data_ctrp_cn_summ$InG2P == "Yes",]$nn < 2) {
    ctrp_cn_res <-  NA
  }
  else if(use_data_ctrp_cn_summ[use_data_ctrp_cn_summ$InG2P == "Yes",]$nn >= 2) {
    ctrp_cn_ttest <- t.test(use_data_ctrp$Pearson_CN_Abs ~ use_data_ctrp$InG2P)
    ctrp_cn_res <- paste0("t = ", round(ctrp_cn_ttest$statistic, 3), ", df = ", round(ctrp_cn_ttest$parameter, 3), ", p = ", round(ctrp_cn_ttest$p.value, 3))
  }
  print("CTRP testing: DONE")
  
  use_data_gdsc <- filter(use_data, Dataset == "GDSC")
  use_data_gdsc_ge_summ <- use_data_gdsc[!is.na(use_data_gdsc$Pearson_GE_Abs),] %>% count(InG2P)
  use_data_gdsc_cn_summ <- use_data_gdsc[!is.na(use_data_gdsc$Pearson_CN_Abs),] %>% count(InG2P)
  gdsc_ge_ttest <- data.frame(statistic = NA, parameter = NA, p.value = NA)
  gdsc_cn_ttest <- data.frame(statistic = NA, parameter = NA, p.value = NA)
  if(nrow(use_data_gdsc_ge_summ) <= 1) {
    gdsc_ge_res <-  NA
    ticker_ge <- ticker_ge + 1
  }
  else if(use_data_gdsc_ge_summ[use_data_gdsc_ge_summ$InG2P == "Yes",]$nn < 2) {
    gdsc_ge_res <-  NA
  }
  else if(use_data_gdsc_ge_summ[use_data_gdsc_ge_summ$InG2P == "Yes",]$nn >= 2) {
    gdsc_ge_ttest <- t.test(use_data_gdsc$Pearson_GE_Abs ~ use_data_gdsc$InG2P)
    gdsc_ge_res <- paste0("t = ", round(gdsc_ge_ttest$statistic, 3), ", df = ", round(gdsc_ge_ttest$parameter, 3), ", p = ", round(gdsc_ge_ttest$p.value, 3))
  }
  if(nrow(use_data_gdsc_cn_summ) <= 1) {
    gdsc_cn_res <-  NA
    ticker_cn <- ticker_cn + 1
  }
  else if(use_data_gdsc_cn_summ[use_data_gdsc_cn_summ$InG2P == "Yes",]$nn < 2) {
    gdsc_cn_res <-  NA
  }
  else if(use_data_gdsc_cn_summ[use_data_gdsc_cn_summ$InG2P == "Yes",]$nn >= 2) {
    gdsc_cn_ttest <- t.test(use_data_gdsc$Pearson_CN_Abs ~ use_data_gdsc$InG2P)
    gdsc_cn_res <- paste0("t = ", round(gdsc_cn_ttest$statistic, 3), ", df = ", round(gdsc_cn_ttest$parameter, 3), ", p = ", round(gdsc_cn_ttest$p.value, 3))
  }
  print("GDSC testing: DONE")
  
  y_value_ge <- ifelse(ticker_ge == 3, 10, max(use_data$n) + 1)
  y_value_cn <- ifelse(ticker_cn == 3, 10, max(use_data$n) + 1)
  summ_text <- data.frame(Dataset = c("CCLE", "CTRP", "GDSC"),
                          Test_Results_GE = c(ccle_ge_res, ctrp_ge_res, gdsc_ge_res),
                          t_statistic_GE = c(round(ccle_ge_ttest$statistic, 3), round(ctrp_ge_ttest$statistic, 3), round(gdsc_ge_ttest$statistic, 3)),
                          df_GE = c(round(ccle_ge_ttest$parameter, 3), round(ctrp_ge_ttest$parameter, 3), round(gdsc_ge_ttest$parameter, 3)),
                          p_value_GE = c(round(ccle_ge_ttest$p.value, 3), round(ctrp_ge_ttest$p.value, 3), round(gdsc_ge_ttest$p.value, 3)),
                          Test_Results_CN = c(ccle_cn_res, ctrp_cn_res, gdsc_cn_res),
                          t_statistic_CN = c(round(ccle_cn_ttest$statistic, 3), round(ctrp_cn_ttest$statistic, 3), round(gdsc_cn_ttest$statistic, 3)),
                          df_CN = c(round(ccle_cn_ttest$parameter, 3), round(ctrp_cn_ttest$parameter, 3), round(gdsc_cn_ttest$parameter, 3)),
                          p_value_CN = c(round(ccle_cn_ttest$p.value, 3), round(ctrp_cn_ttest$p.value, 3), round(gdsc_cn_ttest$p.value, 3)),
                          group_general_lineage_name = lineage)
  
  ## Gene expression
  if(sum(!is.na(use_data$Pearson_GE_Abs)) > 0) {
    plot_ge_point <- ggplot(data = use_data) +
      facet_wrap(~ Dataset, drop = FALSE) +
      scale_y_continuous(breaks = pretty_breaks()) +
      geom_point(mapping = aes(label = Pearson_GE_Label, color = InG2P, fill = InG2P, shape = Pearson_GE_Sign, x = Pearson_GE_Abs, y = n), alpha = 0.5, position = position_jitter(width = 0, height = 0.15)) +
      scale_shape_manual(values = c(25, 24), labels = c("Negative", "Positive")) +
      scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
      scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
      geom_text_repel(data = subset(use_data, InG2P == "Yes"), aes(label = Drug_Gene, x = Pearson_GE_Abs, y = n), size = 3, segment.size = 0.2, segment.color = "grey50") +
      geom_text(data = summ_text, mapping = aes(x = 0.5, y = y_value_ge, label = Test_Results_GE), nudge_y = 1) +
      labs(y = "Number of level A G2P associations", x = "Pearson correlation coefficient between drug-gene AUC and gene expression") +
      theme(legend.position = "none", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank())
    plot_ge_box <- ggplot(data = use_data) +
      facet_wrap(~ Dataset, drop = FALSE) +
      geom_violin(mapping = aes(color = InG2P, y = Pearson_GE_Abs, x = InG2P), alpha = 0.5) +
      geom_boxplot(mapping = aes(color = InG2P, y = Pearson_GE_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
      scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
      coord_flip() +
      labs(x = "In G2P?", title = toupper(lineage)) +
      theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))
    plot_ge <- subplot(plot_ge_box, plot_ge_point, nrows = 2, titleX = TRUE, titleY = TRUE) %>%
               layout(title = lineage, margin = list(t = 75, l = 75),
                      xaxis5 = list(title = "Pearson correlation coefficient between drug-gene AUC and gene expression"),
                      yaxis = list(title = "In G2P?", domain = c(0.8, 1)),
                      yaxis2 = list(title = "Number of level A G2P associations", domain = c(0, 0.79)))
    saveWidgetFix(widget = plot_ge, file = paste0("./plots_18Q3/manuscript/plotlys_ge_by_lineage/", lineage, "_ge_plotly.html"))
    print("Gene expression plotly: DONE")
    
    png_ge_point <- ggplot_gtable(ggplot_build(plot_ge_point))
    png_ge_box <- ggplot_gtable(ggplot_build(plot_ge_box))
    p_maxWidth <-  grid::unit.pmax(png_ge_point$widths, png_ge_box$widths)
    png_ge_point$widths <- as.list(p_maxWidth)
    png_ge_box$widths <- as.list(p_maxWidth)
    
    png_ge <- cowplot::plot_grid(png_ge_box, png_ge_point, nrow = 2, rel_heights = c(1, 4))
    png_ge
    ggsave(paste0("./plots_18Q3/manuscript/pngs_ge_by_lineage/", lineage, "_ge_plot.png"), png_ge, device = "png", dpi = 450, width = 12, height = 6, units = "in")
    print("Gene expression plot: DONE")
  }
  else if(sum(!is.na(use_data$Pearson_GE_Abs)) == 0) { print("Gene expression plots: None") }
  
  ## Copy number
  if(sum(!is.na(use_data$Pearson_CN_Abs)) > 0) {
    plot_cn_point <- ggplot(data = use_data) +
      facet_wrap(~ Dataset, drop = FALSE) +
      scale_y_continuous(breaks = pretty_breaks()) +
      geom_point(mapping = aes(label = Pearson_CN_Label, color = InG2P, fill = InG2P, shape = Pearson_CN_Sign, x = Pearson_CN_Abs, y = n), alpha = 0.5, position = position_jitter(width = 0, height = 0.15)) +
      scale_shape_manual(values = c(25, 24), labels = c("Negative", "Positive")) +
      scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
      scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
      geom_text_repel(data = subset(use_data, InG2P == "Yes"), aes(label = Drug_Gene, x = Pearson_CN_Abs, y = n), size = 3, segment.size = 0.2, segment.color = "grey50") +
      geom_text(data = summ_text, mapping = aes(x = 0.5, y = y_value_cn, label = Test_Results_CN), nudge_y = 1) +
      labs(y = "Number of level A G2P associations", x = "Pearson correlation coefficient between drug-gene AUC and copy number") +
      theme(legend.position = "none", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank())
    plot_cn_box <- ggplot(data = use_data) +
      facet_wrap(~ Dataset, drop = FALSE) +
      geom_violin(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5) +
      geom_boxplot(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
      scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
      coord_flip() +
      labs(x = "In G2P?", title = toupper(lineage)) +
      theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))
    plot_cn <- subplot(plot_cn_box, plot_cn_point, nrows = 2, titleX = TRUE, titleY = TRUE) %>%
               layout(title = lineage, margin = list(t = 75, l = 75),
                      xaxis5 = list(title = "Pearson correlation coefficient between drug-gene AUC and copy number"),
                      yaxis = list(title = "In G2P?", domain = c(0.8, 1)),
                      yaxis2 = list(title = "Number of level A G2P associations", domain = c(0, 0.79)))
    saveWidgetFix(widget = plot_cn, file = paste0("./plots_18Q3/manuscript/plotlys_cn_by_lineage/", lineage, "_cn_plotly.html"))
    print("Copy number plotly: DONE")
    
    png_cn_point <- ggplot_gtable(ggplot_build(plot_cn_point))
    png_cn_box <- ggplot_gtable(ggplot_build(plot_cn_box))
    p_maxWidth <-  grid::unit.pmax(png_cn_point$widths, png_cn_box$widths)
    png_cn_point$widths <- as.list(p_maxWidth)
    png_cn_box$widths <- as.list(p_maxWidth)
    
    png_cn <- cowplot::plot_grid(png_cn_box, png_cn_point, nrow = 2, rel_heights = c(1, 4))
    png_cn
    ggsave(paste0("./plots_18Q3/manuscript/pngs_cn_by_lineage/", lineage, "_cn_plot.png"), png_cn, device = "png", dpi = 450, width = 12, height = 6, units = "in")
    print("Copy number plot: DONE")
  }
  else if(sum(!is.na(use_data$Pearson_CN_Abs)) == 0) { print("Copy number plots: None") }
  
  return(summ_text)
}
```


## Make per-drug boxplots
```{r}
makeDrugBoxplots <- function(drug, dataset) {
  if(dataset == "CCLE" | dataset == "ccle") {
    use_data <- filter(ccle_data_ptmuts, Drug == drug)
    use_color <- as.character(use_data$Color_Nonsilent)
    names(use_color) <- use_data$Mutation_Status_Nonsilent
    sig <- ccle_signif_g2p_ByDrug[[drug]]
    label_text <- data.frame(p.signif = sig$p.signif, p.signif.adj = sig$p.signif.adj, Hugo_Symbol = sig$Hugo_Symbol)
    label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = g2p_genes)
  }
  else if(dataset == "CTRP" | dataset == "ctrp") {
    use_data <- filter(ctrp_data_ptmuts, Drug == drug)
    use_color <- as.character(use_data$Color_Nonsilent)
    names(use_color) <- use_data$Mutation_Status_Nonsilent
    sig <- ctrp_signif_g2p_ByDrug[[drug]]
    label_text <- data.frame(p.signif = sig$p.signif, p.signif.adj = sig$p.signif.adj, Hugo_Symbol = sig$Hugo_Symbol)
    label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = g2p_genes)
  }
  else if(dataset == "GDSC" | dataset == "gdsc") {
    use_data <- filter(gdsc_data_ptmuts, Drug == drug)
    use_color <- as.character(use_data$Color_Nonsilent)
    names(use_color) <- use_data$Mutation_Status_Nonsilent
    sig <- gdsc_signif_g2p_ByDrug[[drug]]
    label_text <- data.frame(p.signif = sig$p.signif, p.signif.adj = sig$p.signif.adj, Hugo_Symbol = sig$Hugo_Symbol)
    label_text$Hugo_Symbol <- factor(label_text$Hugo_Symbol, levels = g2p_genes)
  }
  else { return("Error: Invalid dataset.") }
    
  plot <- ggplot(data = use_data, aes(x = Mutation_Status_Nonsilent, y = AUC)) +
    facet_wrap(~ Hugo_Symbol, drop = FALSE, nrow = 1) +
    geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85), outlier.shape = 3, outlier.size = 0.5) +
    scale_fill_manual(values = use_color) +
    guides(color = FALSE) +
    geom_text(data = label_text, mapping = aes(x = 1.5, y = max(use_data$AUC), label = p.signif), nudge_y = 0.2) +
    theme(strip.text.x = element_text(size = 10, angle = 45), legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
    labs(fill = "Mutation Status", y = "AUC", x = "Mutation Status", title = paste0(toupper(dataset), ": ", drug))
  return(plot)
}
```

# Data management {.tabset .tabset-fade .tabset-pills}

--------------------------------------------------------------------------------

## D. Charytonowicz cell line converter

This comprehensive cancer cell line information curated by Daniel Charytonowicz.
```{r}
ccl_converter <- read.delim("./data_munging/cell_line_database_v1_20180911.tsv", row.names = 1, sep = "\t", header = TRUE)
lineage_converter <- unique(ccl_converter[, c("grouped_general_doid", "group_general_lineage_name")])
colnames(ccl_converter)[colnames(ccl_converter) == "Broad_ID"] <- "DepMap_ID"
```

## DepMap cell line metadata

```{r}
ccl_info <- read.delim("./data_munging/18Q4/DepMap-2018q4-celllines_20181115.csv", sep = ",", header = TRUE, na.strings = c("", NA))
ccl_info$Primary.Disease <- gsub("\\\\", "", ccl_info$Primary.Disease)
ccl_info$Primary.Disease <- gsub("Ewings", "Ewing's", ccl_info$Primary.Disease)

# From figshare
crispr_meta <- read.delim("./data_munging/18Q4/sample_info_18Q4_crispr_20181115.csv", sep = ",", header = TRUE, na.strings = c("", NA))
colnames(crispr_meta)[7] <- "CCLE_Name"
```

## Genomic annotations

### CCLE MAF file

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Merged mutation calls (coding region, germline filtered):
```{r, eval=FALSE}
maf_raw <- read.delim("./data_munging/18Q4/depmap_18Q4_mutation_calls_20181114.csv.gz", header = TRUE, sep = ",", row.names = 1)
colnames(maf_raw)[colnames(maf_raw) == "Tumor_Sample_Barcode"] <- "CCLE_Name"

# Select columns
maf_df <- subset(maf_raw, select = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID", "Variant_Classification", "Reference_Allele", "Tumor_Seq_Allele1", "Genome_Change"))
maf_df$Reference_Allele_Length <- ifelse(maf_df$Reference_Allele == "-", 0, nchar(as.character(maf_df$Reference_Allele)))
maf_df$Tumor_Seq_Allele1_Length <- nchar(as.character(maf_df$Tumor_Seq_Allele1))

# Add Mutation_Status column
maf_df$Mutation_Status_Nonsilent <- ifelse(maf_df$Variant_Classification == "Silent", "Wiltype", "Mutant")
maf_df <- unique(maf_df)
saveRDS(maf_df, "./data_munging/rds/maf_df_18Q4.rds", compress = "xz")
```

```{r, eval=FALSE}
maf_df <- readRDS("./data_munging/rds/maf_df_18Q4.rds")
```


### CCLE copy number file

DepMap WES CN Data:
```{r, eval=FALSE}
cn <- read.delim("./data_munging/18Q4/public_18Q4_gene_cn_20181115.csv.gz", sep = ",", check.names = FALSE, header = TRUE)

# Remove Entrez gene IDs from colnames
# Format: Hugo_Symbol (Entrez ID)
colnames(cn) <- gsub(" .*", "", colnames(cn))

# Melt
cn_melt <- melt(data = cn, id.vars = "DepMap_ID", measure.vars = colnames(cn[2:ncol(cn)]), variable.name = "Hugo_Symbol", value.name = "CN")

saveRDS(cn_melt, "./data_munging/rds/cn_melt_18Q4.rds", compress = "xz")
```

```{r, eval=FALSE}
cn_melt <- readRDS("./data_munging/rds/cn_melt_18Q4.rds")
```

### CCLE gene expression file

18Q3, Gene expression data (Reads Per Kilobase of transcript, per Million mapped reads, RPKM).
```{r, eval=FALSE}
ge_18Q3 <- read.delim("./../htscreen_lineages_giant_files/CCLE_DepMap_18q3_RNAseq_RPKM_20180718.gct.gz", skip = 2, header = TRUE, sep = "\t", check.names = FALSE)

# Edit columns
ge_18Q3$Name <- NULL
colnames(ge_18Q3)[1] <- "Hugo_Symbol"

# Melt
ge_melt_18Q3 <- melt(data = ge_18Q3, id.vars = "Hugo_Symbol", measure.vars = colnames(ge_18Q3[2:ncol(ge_18Q3)]), value.name = "RPKM")
## Split variable column
ge_melt_18Q3 <- with(ge_melt_18Q3, cbind(Hugo_Symbol, colsplit(variable, pattern = " ", names = c("CCLE_Name", "DepMap_ID")), RPKM))
## Remove parentheses around Broad IDs
ge_melt_18Q3$DepMap_ID <- gsub("\\(|\\)", "", ge_melt_18Q3$DepMap_ID)

saveRDS(ge_melt_18Q3, "./../htscreen_lineages_giant_files/ge_melt_18Q3.rds", compress = "xz")
```

```{r, eval=FALSE}
ge_melt_18Q3 <- readRDS("./../htscreen_lineages_giant_files/ge_melt_18Q3.rds")
```

18Q4, CCLE RNAseq gene expression data (log2(TPM+1)):
```{r, eval=FALSE}
ge_18Q4 <- read.delim("./../htscreen_lineages_giant_files/CCLE_depMap_18Q4_TPM_v2_20181115.csv.gz", header = TRUE, sep = ",", check.names = FALSE)

# Remove Entrez gene IDs from colnames
# Format: Hugo_Symbol (Entrez ID)
colnames(ge_18Q4) <- gsub(" .*", "", colnames(ge_18Q4))
colnames(ge_18Q4)[1] <- "DepMap_ID"

# Melt
ge_melt_18Q4_all <- melt(ge_18Q4, id.vars = "DepMap_ID", measure.vars = colnames(ge_18Q4)[2:ncol(ge_18Q4)], variable.name = "Hugo_Symbol", value.name = "TPM")
ge_melt_18Q4 <- unique(ge_melt_18Q4)

saveRDS(ge_melt_18Q4, "./data_munging/rds/ge_melt_18Q4.rds", compress = "xz")
```

```{r, eval=FALSE}
ge_melt_18Q4 <- readRDS("./data_munging/rds/ge_melt_18Q4.rds")
```

## G2P

```{r}
g2p <- read.delim("./data_munging/g2p_full_level_A_dataset_withGroupedLineages.csv", sep = "\t", header = TRUE, row.names = 1)
g2p <- merge(g2p, lineage_converter, by = "grouped_general_doid", all.x = TRUE)
g2p$Drug <- tolower(g2p$Drug)
g2p[g2p == -1] <- NA
g2p$Ref[g2p$Ref == "None"] <- NA
g2p$Alt[g2p$Alt == "None"] <- NA
g2p$Drug_Gene <- ifelse(is.na(g2p$Drug) | is.na(g2p$Gene), NA, paste0(g2p$Drug, "_", g2p$Gene))
g2p$position <- paste0("g.chr", g2p$Chromosome, ":", g2p$End, g2p$Ref, ">", g2p$Alt)
g2p$position <- ifelse(grepl("NA", g2p$position), NA, g2p$position)
g2p <- subset(g2p, grepl("^CID", g2p$DrugID))


g2p_drug_genes <- unique(g2p$Drug_Gene[!is.na(g2p$Drug_Gene)])
g2p_genes <- unique(sapply(strsplit(g2p_drug_genes, "_", fixed = TRUE), function(x) x[2]))
g2p_drugs <- unique(sapply(strsplit(g2p_drug_genes, "_", fixed = TRUE), function(x) x[1]))

g2p_genes
g2p_drugs
# Order and genes taken from pink heatmapo from Daniel's analyses
g2p_genes <- c("EGFR", "KRAS", "ERBB2", "ERBB3", "MET", "ALK", "ROS1", "PDGFRA", "ABL1", "KIT", "PDGFRB", "PDGFB", "BRAF", "G6PD", "RET", "BRCA1", "BRCA2", "UGT1A1", "TSC1", "TSC2", "AKT1", "IDH2", "DPYD", "ESR1", "CYP19A1", "TPMT", "FLT3", "NPM1", "DNMT3A", "PML", "PTCH1", "JAK2", "MYD88", "MGMT")
```


## CRISPR

### DepMap dependency probabilities

```{r, eval=FALSE}
dep <- read.delim("./data_munging/18Q4/gene_dependency_18Q4_20181115.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(dep) <- gsub(" .*", "", colnames(dep))
```

### Merge CRISPR data and annotations

The latest CRISPR CERES score data (18Q3, August 2018) was pulled from the [DepMap Data Portal](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017).
```{r, eval=FALSE}
crispr <- read.delim("./data_munging/18Q4/gene_effect_18Q4_20181115.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)
# Remove Entrez gene IDs from colnames
colnames(crispr) <- gsub(" .*", "", colnames(crispr))
```

Merge annotation data:
```{r, eval=FALSE}
# Melt CRISPR dataset for merging
crispr_melt <- melt(crispr, id.vars = "DepMap_ID", measure.vars = colnames(crispr)[2:ncol(crispr)], variable.name = "Hugo_Symbol", value.name = "Score")
# Melt dependency probabilities dataset for merging
dep_melt <- melt(dep, id.vars = "DepMap_ID", measure.vars = colnames(dep)[2:ncol(dep)], variable.name = "Hugo_Symbol", value.name = "Dep_Prob")
# Merge dependency probabilities
crispr_melt <- merge(crispr_melt, dep_melt, by = c("DepMap_ID", "Hugo_Symbol"), all.x = TRUE)
# Merge cell line metadata
crispr_melt <- merge(crispr_melt, ccl_info, by = "DepMap_ID", all.x = TRUE)
crispr_melt <- merge(crispr_melt, crispr_meta, by = c("CCLE_Name", "DepMap_ID"), all.x = TRUE)
# Merge mutation annotations
maf_filt_crispr <- filter(maf_df, Hugo_Symbol %in% as.character(unique(crispr$Hugo_Symbol)))
crispr_muts <- merge(crispr_melt, maf_filt_crispr, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
crispr_muts$Hugo_Symbol <- factor(crispr_muts$Hugo_Symbol)
crispr_muts <- crispr_muts %>% mutate(Mutation_Status_Nonsilent = if_else(is.na(Mutation_Status_Nonsilent), "Wildtype", Mutation_Status_Nonsilent))
# Summarize number of mutant and Wildtype cell lines
crispr_muts_summ <- crispr_muts %>% group_by(Hugo_Symbol) %>%
  summarize(N_Nonsilent_Wildtype = sum(Mutation_Status_Nonsilent == "Wildtype"),
            N_Nonsilent_Mutant = sum(Mutation_Status_Nonsilent == "Mutant"))
# Merge test results back into full dataset, which restores information lost in the summarization
crispr_data <- merge(crispr_muts_summ, crispr_muts, by = "Hugo_Symbol")
# Add Color columns
crispr_data$Color_Nonsilent <- ifelse(crispr_data$Mutation_Status_Nonsilent == "Wildtype", "thistle", "darkorchid")
crispr_data$Color_Nonsilent <- factor(crispr_data$Color_Nonsilent)
# Cell line lineages
crispr_data <- merge(crispr_data, ccl_converter, by = c("CCLE_Name", "DepMap_ID"), all.x = TRUE)
levels(crispr_data$lineage_name) <- sort(levels(crispr_data$lineage_name), decreasing = TRUE)
# Copy number
crispr_data <- merge(crispr_data, cn_melt, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
# Gene expression (RPKM)
ge_crispr_18Q3 <- filter(ge_melt_18Q3, Hugo_Symbol %in% as.character(unique(crispr_melt$Hugo_Symbol)))
ge_crispr_18Q4 <- filter(ge_melt_18Q4_test, Hugo_Symbol %in% as.character(unique(crispr_melt$Hugo_Symbol)))
crispr_data <- merge(crispr_data, ge_crispr_18Q3, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
crispr_data <- merge(crispr_data, ge_crispr_18Q4, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
saveRDS(crispr_data, "./../htscreen_lineages_giant_files/crispr_data_18Q4.rds", compress = "xz")
```

```{r}
crispr_data <- readRDS("./../htscreen_lineages_giant_files/crispr_data_18Q4.rds")

crispr_data_ptmuts <- filter(crispr_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
crispr_data_ptmuts$Keep_PtMut_Tests <- ifelse(crispr_data_ptmuts$Reference_Allele_Length == 1 | is.na(crispr_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(crispr_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
crispr_data_ptmuts <- filter(crispr_data_ptmuts, Keep_PtMut_Tests == TRUE)

# Filter CRISPR for G2P genes in G2P
crispr_data_g2p <- filter(crispr_data_ptmuts, Hugo_Symbol %in% g2p_genes)
crispr_data_g2p$Hugo_Symbol <- factor(crispr_data_g2p$Hugo_Symbol, levels = g2p_genes)
```

## Drug screens


```{r, eval=FALSE}
# Drug metadata
drug_meta <- read.delim("./data_munging/drug_id_drug_name_table_10_12_2018.tsv", sep = "\t", header = FALSE)
colnames(drug_meta) <- c("CID", "Drug")
drug_meta$Drug <- tolower(drug_meta$Drug)
maf_df_g2p <- filter(maf_df, Hugo_Symbol %in% g2p_genes)
ge_filt_18Q3 <- filter(ge_melt_18Q3, Hugo_Symbol %in% g2p_genes)
# ge_filt_18Q4 <- filter(ge_melt_18Q4, Hugo_Symbol %in% g2p_genes)
cn_filt <- filter(cn_melt, Hugo_Symbol %in% g2p_genes)

# CCLE
ccle <- read.delim("./data_munging/data_drug_auc_ccle_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
ccle <- merge(drug_meta, ccle, by = "CID")
ccle_melt <- melt(ccle, id.vars = c("CID", "Drug"), measure.vars = colnames(ccle)[3:ncol(ccle)], variable.name = "accession_id", value.name = "ActArea")
ccle_grid <- expand.grid("accession_id" = unique(ccle_melt$accession_id), "Hugo_Symbol" = g2p_genes)
ccle_data <- merge(ccle_grid, ccle_melt, by = "accession_id", all.x = TRUE)
ccle_data <- merge(ccle_data, ccl_converter, by = "accession_id", all.x = TRUE)
ccle_data <- merge(ccle_data, maf_df_g2p, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
ccle_data$Mutation_Status_Nonsilent <- ifelse(is.na(ccle_data$Mutation_Status_Nonsilent), "Wildtype", ccle_data$Mutation_Status_Nonsilent)
ccle_data <- merge(ccle_data, ge_filt_18Q3, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
# ccle_data <- merge(ccle_data, ge_filt_18Q4, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
ccle_data <- merge(ccle_data, cn_filt, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
ccle_data$Hugo_Symbol <- factor(ccle_data$Hugo_Symbol)
ccle_data$ActArea <- as.numeric(ccle_data$ActArea)
# CCLE AUC is actually AOC
# assuming scale is 0-8, transform to range between 0 and 1
# get AUC by 1 - AOC
ccle_data$AUC <- 1 - (ccle_data$ActArea / 8)
ccle_data <- filter(ccle_data, !is.na(ccle_data$AUC))
ccle_data$AUC_zscore <- (ccle_data$AUC - mean(ccle_data$AUC)) / sd(ccle_data$AUC)
ccle_data$Color_Nonsilent <- ifelse(ccle_data$Mutation_Status_Nonsilent == "Wildtype", "palegreen3", "springgreen4")
ccle_data$Color_Nonsilent <- factor(ccle_data$Color_Nonsilent)
ccle_data$Drug_Gene <- paste0(ccle_data$Drug, "_", ccle_data$Hugo_Symbol)
saveRDS(ccle_data, "./data_munging/rds/ccle_data_18Q4_g2pgenes.rds", compress = "xz")

# CTRP
ctrp <- read.delim("./data_munging/data_drug_auc_ctrp_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
ctrp <- merge(drug_meta, ctrp, by = "CID")
ctrp_melt <- melt(ctrp, id.vars = c("CID", "Drug"), measure.vars = colnames(ctrp)[3:ncol(ctrp)], variable.name = "accession_id", value.name = "AUC")
ctrp_grid <- expand.grid("accession_id" = unique(ctrp_melt$accession_id), "Hugo_Symbol" = g2p_genes)
ctrp_data <- merge(ctrp_grid, ctrp_melt, by = "accession_id", all.x = TRUE)
ctrp_data <- merge(ctrp_data, ccl_converter, by = "accession_id", all.x = TRUE)
ctrp_data <- merge(ctrp_data, maf_df_g2p, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
ctrp_data$Mutation_Status_Nonsilent <- ifelse(is.na(ctrp_data$Mutation_Status_Nonsilent), "Wildtype", ctrp_data$Mutation_Status_Nonsilent)
ctrp_data <- merge(ctrp_data, ge_filt_18Q3, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
# ctrp_data <- merge(ctrp_data, ge_filt_18Q4, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
ctrp_data <- merge(ctrp_data, cn_filt, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
ctrp_data$Hugo_Symbol <- factor(ctrp_data$Hugo_Symbol)
ctrp_data$AUC <- as.numeric(ctrp_data$AUC)
ctrp_data <- filter(ctrp_data, !is.na(ctrp_data$AUC))
ctrp_data$AUC_zscore <- (ctrp_data$AUC - mean(ctrp_data$AUC)) / sd(ctrp_data$AUC)
ctrp_data$Color_Nonsilent <- ifelse(ctrp_data$Mutation_Status_Nonsilent == "Wildtype", "slategray3", "steelblue4")
ctrp_data$Color_Nonsilent <- factor(ctrp_data$Color_Nonsilent)
ctrp_data$Drug_Gene <- paste0(ctrp_data$Drug, "_", ctrp_data$Hugo_Symbol)
saveRDS(ctrp_data, "./data_munging/rds/ctrp_data_18Q4_g2pgenes.rds", compress = "xz")

# GDSC
gdsc <- read.delim("./data_munging/data_drug_auc_gdsc_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
gdsc <- merge(drug_meta, gdsc, by = "CID")
gdsc_melt <- melt(gdsc, id.vars = c("CID", "Drug"), measure.vars = colnames(gdsc)[3:ncol(gdsc)], variable.name = "accession_id", value.name = "AUC")
gdsc_grid <- expand.grid("accession_id" = unique(gdsc_melt$accession_id), "Hugo_Symbol" = g2p_genes)
gdsc_data <- merge(gdsc_grid, gdsc_melt, by = "accession_id", all.x = TRUE)
gdsc_data <- merge(gdsc_data, ccl_converter, by = "accession_id", all.x = TRUE)
gdsc_data <- merge(gdsc_data, maf_df_g2p, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
gdsc_data$Mutation_Status_Nonsilent <- ifelse(is.na(gdsc_data$Mutation_Status_Nonsilent), "Wildtype", gdsc_data$Mutation_Status_Nonsilent)
gdsc_data <- merge(gdsc_data, ge_filt_18Q3, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
# gdsc_data <- merge(gdsc_data, ge_filt_18Q4, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
gdsc_data <- merge(gdsc_data, cn_filt, by = c("Hugo_Symbol", "DepMap_ID"), all.x = TRUE)
gdsc_data$Hugo_Symbol <- factor(gdsc_data$Hugo_Symbol)
gdsc_data$AUC <- as.numeric(gdsc_data$AUC)
gdsc_data <- filter(gdsc_data, !is.na(gdsc_data$AUC))
gdsc_data$AUC_zscore <- (gdsc_data$AUC - mean(gdsc_data$AUC)) / sd(gdsc_data$AUC)
gdsc_data$Color_Nonsilent <- ifelse(gdsc_data$Mutation_Status_Nonsilent == "Wildtype", "paleturquoise3", "turquoise4")
gdsc_data$Color_Nonsilent <- factor(gdsc_data$Color_Nonsilent)
gdsc_data$Drug_Gene <- paste0(gdsc_data$Drug, "_", gdsc_data$Hugo_Symbol)
saveRDS(gdsc_data, "./data_munging/rds/gdsc_data_18Q4_g2pgenes.rds", compress = "xz")
```



```{r}
ccle_data <- readRDS("./data_munging/rds/ccle_data_18Q4_g2pgenes.rds")
ccle_data_ptmuts <- filter(ccle_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
ccle_data_ptmuts$Keep_PtMut_Tests <- ifelse(ccle_data_ptmuts$Reference_Allele_Length == 1 | is.na(ccle_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(ccle_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
ccle_data_ptmuts <- filter(ccle_data_ptmuts, Keep_PtMut_Tests == TRUE)
ccle_data_ptmuts$Dataset <- "CCLE"
ccle_data_ptmuts$Hugo_Symbol <- ordered(ccle_data_ptmuts$Hugo_Symbol, levels = g2p_genes)

ctrp_data <- readRDS("./data_munging/rds/ctrp_data_18Q4_g2pgenes.rds")
ctrp_data_ptmuts <- filter(ctrp_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
ctrp_data_ptmuts$Keep_PtMut_Tests <- ifelse(ctrp_data_ptmuts$Reference_Allele_Length == 1 | is.na(ctrp_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(ctrp_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
ctrp_data_ptmuts <- filter(ctrp_data_ptmuts, Keep_PtMut_Tests == TRUE)
ctrp_data_ptmuts$Dataset <- "CTRP"
ctrp_data_ptmuts$Hugo_Symbol <- ordered(ctrp_data_ptmuts$Hugo_Symbol, levels = g2p_genes)

gdsc_data <- readRDS("./data_munging/rds/gdsc_data_18Q4_g2pgenes.rds")
gdsc_data_ptmuts <- filter(gdsc_data, Reference_Allele_Length == 1 | Reference_Allele_Length == 0 | is.na(Reference_Allele_Length))
gdsc_data_ptmuts$Keep_PtMut_Tests <- ifelse(gdsc_data_ptmuts$Reference_Allele_Length == 1 | is.na(gdsc_data_ptmuts$Reference_Allele_Length), TRUE, ifelse(gdsc_data_ptmuts$Tumor_Seq_Allele1_Length == 1, TRUE, FALSE))
gdsc_data_ptmuts <- filter(gdsc_data_ptmuts, Keep_PtMut_Tests == TRUE)
gdsc_data_ptmuts$Dataset <- "GDSC"
gdsc_data_ptmuts$Hugo_Symbol <- ordered(gdsc_data_ptmuts$Hugo_Symbol, levels = g2p_genes)

# Filter drug screen datasets for G2P drug-gene associations
ccle_data_g2p <- filter(ccle_data_ptmuts, Drug_Gene %in% g2p_drug_genes)
ctrp_data_g2p <- filter(ctrp_data_ptmuts, Drug_Gene %in% g2p_drug_genes)
gdsc_data_g2p <- filter(gdsc_data_ptmuts, Drug_Gene %in% g2p_drug_genes)

# Filter drug screen datasets for G2P drugs and genes
ccle_data_g2p_grid <- filter(ccle_data_ptmuts, Drug %in% g2p_drugs)
ctrp_data_g2p_grid <- filter(ctrp_data_ptmuts, Drug %in% g2p_drugs)
gdsc_data_g2p_grid <- filter(gdsc_data_ptmuts, Drug %in% g2p_drugs)
```

# Mutation Status Wilcoxon tests

--------------------------------------------------------------------------------

## CRISPR

### G2P genes

```{r, eval=FALSE}
crispr_signif_g2p_gene <- compare_means(Score ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol"), data = crispr_data_g2p, method = "wilcox.test", p.adjust.method = "BH")
crispr_signif_g2p_gene <- adj_signif(crispr_signif_g2p_gene)
crispr_signif_g2p_gene <- crispr_signif_g2p_gene[order(crispr_signif_g2p_gene$p),]
saveRDS(crispr_signif_g2p_gene, "./data_munging/rds/crispr_signif_g2p_gene.rds")
```

```{r}
crispr_signif_g2p_gene <- readRDS("./data_munging/rds/crispr_signif_g2p_gene.rds")

crispr_signif_g2p_20gene <- as.character(crispr_signif_g2p_gene$Hugo_Symbol[0:20])

knitr::kable(crispr_signif_g2p_gene[, c("Hugo_Symbol", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "Wilcoxon test results for Level A point mutations (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=7, fig.width=12}
crispr_data_signif_g2p_gene_filt <- filter(crispr_data_ptmuts, Hugo_Symbol %in% crispr_signif_g2p_20gene)
crispr_data_signif_g2p_gene <- merge(crispr_data_signif_g2p_gene_filt, crispr_signif_g2p_gene, by = "Hugo_Symbol")
crispr_data_signif_g2p_gene$Hugo_Symbol <- ordered(crispr_data_signif_g2p_gene$Hugo_Symbol, levels = crispr_signif_g2p_20gene)
crispr_data_signif_g2p_gene_color <- as.character(crispr_data_signif_g2p_gene$Color_Nonsilent)
names(crispr_data_signif_g2p_gene_color) <- crispr_data_signif_g2p_gene$Mutation_Status_Nonsilent
crispr_data_signif_g2p_gene <- merge(crispr_data_signif_g2p_gene, crispr_data_signif_g2p_gene %>% group_by(Hugo_Symbol) %>% summarize(Text_y = max(Score)), by = "Hugo_Symbol")

crispr_data_signif_g2p_gene_text <- unique(data.frame(p.signif = crispr_data_signif_g2p_gene$p.signif, p.signif.adj = crispr_data_signif_g2p_gene$p.signif.adj,  Hugo_Symbol = crispr_data_signif_g2p_gene$Hugo_Symbol, Text_y = crispr_data_signif_g2p_gene$Text_y))

crispr_data_signif_g2p_gene_plot <- ggplot(data = crispr_data_signif_g2p_gene, aes(x = Mutation_Status_Nonsilent, y = Score)) +
  facet_wrap(~ Hugo_Symbol, drop = FALSE, nrow = 2, scales = "free_y") +
  geom_jitter(aes(color = Mutation_Status_Nonsilent), alpha = 0.5, width = 0.3) +
  geom_boxplot(color = "black", fill = NA, outlier.shape = NA) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  scale_color_manual(values = crispr_data_signif_g2p_gene_color) + 
  geom_text(data = crispr_data_signif_g2p_gene_text, mapping = aes(x = 1.5, y = Text_y, label = p.signif), nudge_y = 0.2) +
  theme(legend.position = "top", axis.text.x = element_blank(), axis.ticks.x = element_blank(), axis.title.x = element_blank()) +
  labs(y = "CERES Score",
       color = "Mutation Status",
       title = "CERES scores for select Level A G2P genes",
       subtitle = "Genes are sorted by increasing p-value.")
crispr_data_signif_g2p_gene_plot

# ggsave("./plots_18Q3/manuscript/crispr_data_signif_g2p_gene_plot.png", crispr_data_signif_g2p_gene_plot, width = 12, height = 7, units = "in")
```

### Gene

```{r, eval=FALSE}
crispr_signif_g2p_lineage <- compare_means(Score ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "group_general_lineage_name"), data = crispr_data_g2p, method = "wilcox.test", p.adjust.method = "BH")
crispr_signif_g2p_lineage <- adj_signif(crispr_signif_g2p_lineage)
crispr_signif_g2p_lineage <- crispr_signif_g2p_lineage[order(crispr_signif_g2p_lineage$p),]
saveRDS(crispr_signif_g2p_lineage, "./data_munging/rds/crispr_signif_g2p_lineage.rds")

# write.table(crispr_signif_g2p_lineage, file = "~/Desktop/crispr_signif_g2p_lineage.csv", quote = FALSE, sep = ",", row.names = FALSE)
```

```{r}
crispr_signif_g2p_lineage <- readRDS("./data_munging/rds/crispr_signif_g2p_lineage.rds")

knitr::kable(filter(crispr_signif_g2p_lineage, p < 0.1)[, c("Hugo_Symbol", "group_general_lineage_name", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "Wilcoxon test results comparing non-silent mutant vs wildtype cell lines by lineage, p < 0.1 (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

```{r fig.height=15, fig.width=15}
crispr_data_signif_g2p_lineage <- merge(crispr_data_signif_g2p_gene_filt, crispr_signif_g2p_lineage, by = c("Hugo_Symbol", "group_general_lineage_name"))
crispr_data_signif_g2p_lineage$Hugo_Symbol <- ordered(crispr_data_signif_g2p_lineage$Hugo_Symbol, levels = crispr_signif_g2p_20gene)
crispr_data_signif_g2p_lineage_color <- as.character(crispr_data_signif_g2p_lineage$Color_Nonsilent)
names(crispr_data_signif_g2p_lineage_color) <- crispr_data_signif_g2p_lineage$Mutation_Status_Nonsilent
crispr_data_signif_g2p_lineage <- merge(crispr_data_signif_g2p_lineage, crispr_data_signif_g2p_lineage %>% group_by(Hugo_Symbol) %>% summarize(Text_y = max(Score)), by = "Hugo_Symbol")

crispr_data_signif_g2p_lineage_text <- unique(data.frame(p.signif = crispr_data_signif_g2p_lineage$p.signif, p.signif.adj = crispr_data_signif_g2p_lineage$p.signif.adj,  group_general_lineage_name = crispr_data_signif_g2p_lineage$group_general_lineage_name, Hugo_Symbol = crispr_data_signif_g2p_lineage$Hugo_Symbol, Text_y = crispr_data_signif_g2p_lineage$Text_y))

crispr_data_signif_g2p_lineage_plot <- ggplot(data = crispr_data_signif_g2p_lineage, aes(x = group_general_lineage_name, y = Score)) +
  facet_wrap(~ Hugo_Symbol, nrow = 4, scales = "free_y") +
  geom_point(aes(color = Mutation_Status_Nonsilent), alpha = 0.5) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  scale_color_manual(values = crispr_data_signif_g2p_lineage_color) +
  geom_text(data = crispr_data_signif_g2p_lineage_text, mapping = aes(x = group_general_lineage_name, y = Text_y, label = p.signif), angle = 90, vjust = 0.8, nudge_y = 0.1) +
  theme(legend.position = "top", axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.4, size = 10)) +
  labs(y = "CERES Score",
       x = "Lineage",
       color = "Mutation Status",
       title = "CERES score by cell line lineage for significant Level A G2P genes",
       subtitle = "* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001\nGenes are sorted by increasing lineage-agnostic p-value.")
crispr_data_signif_g2p_lineage_plot

# ggsave("./plots_18Q3/manuscript/crispr_crispr_data_signif_g2p_lineage_plot.png", crispr_data_signif_g2p_lineage_plot, width = 12, height = 15, units = "in")
```

```{r}
crispr_data_signif_g2p_lineage_summ <- crispr_data_signif_g2p_lineage %>% group_by(Hugo_Symbol, group_general_lineage_name, Mutation_Status_Nonsilent) %>% tally()
# write.table(crispr_data_signif_g2p_lineage_summ, file = "~/Desktop/crispr_data_signif_g2p_lineage_summ_20181005.tsv", quote = FALSE, sep = "\t")

knitr::kable(crispr_data_signif_g2p_lineage_summ, caption = "Mutant/Wildtype counts for gene-lineage combinations") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

## CCLE

### G2P gene and drug

```{r, eval=FALSE}
# G2P drug-gene associations only
ccle_signif_g2p <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = "Drug_Gene", data = ccle_data_g2p, method = "wilcox.test", p.adjust.method = "BH")
ccle_signif_g2p <- adj_signif(ccle_signif_g2p)
ccle_signif_g2p <- ccle_signif_g2p[order(ccle_signif_g2p$p),]
saveRDS(ccle_signif_g2p, "./data_munging/rds/ccle_signif_g2p_gene.rds")

# Grid comparisons of G2P drugs and genes
ccle_signif_g2p_grid <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "Drug"), data = ccle_data_g2p_grid, method = "wilcox.test", p.adjust.method = "BH")
ccle_signif_g2p_grid <- adj_signif(ccle_signif_g2p_grid)
ccle_signif_g2p_grid <- ccle_signif_g2p_grid[order(ccle_signif_g2p_grid$p),]
ccle_signif_g2p_grid$Drug_Gene <- paste0(ccle_signif_g2p_grid$Drug, "_", ccle_signif_g2p_grid$Hugo_Symbol)
saveRDS(ccle_signif_g2p_grid, "./data_munging/rds/ccle_signif_g2p_grid.rds")
```

```{r}
ccle_signif_g2p <- readRDS("./data_munging/rds/ccle_signif_g2p_gene.rds")

knitr::kable(ccle_signif_g2p[, c("Drug_Gene", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CCLE: Wilcoxon test results for level A G2P drug-gene associations (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")

ccle_signif_g2p_grid <- readRDS("./data_munging/rds/ccle_signif_g2p_grid.rds")

knitr::kable(ccle_signif_g2p_grid[, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CCLE: Wilcoxon test results for point mutations in level A G2P genes and drugs (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

### G2P gene, drug, and lineage

```{r, eval=FALSE}
ccle_signif_g2p_lineage <- lapply(unique(ccle_data_g2p_grid$Drug), WilcoxonByDrug, dataset = ccle_data_g2p_grid, data_name = "ccle", grouping_var = "lineage")
names(ccle_signif_g2p_lineage) <- unique(ccle_signif_g2p_lineage$Drug)
ccle_signif_g2p_lineage <- rbindlist(ccle_signif_g2p_lineage, use.names = TRUE)
ccle_signif_g2p_lineage <- ccle_signif_g2p_lineage[order(ccle_signif_g2p_lineage$p),]
saveRDS(ccle_signif_g2p_lineage, "./data_munging/rds/ccle_signif_g2p_lineage.rds")
```

```{r}
ccle_signif_g2p_lineage <- readRDS("./data_munging/rds/ccle_signif_g2p_lineage.rds")

knitr::kable(ccle_signif_g2p_lineage[1:500, c("Hugo_Symbol", "Drug", "group_general_lineage_name", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CCLE: Wilcoxon test results for level A G2P drug-gene associations across lineages (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

## CTRP

### G2P gene and drug

```{r, eval=FALSE}
ctrp_signif_g2p <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = "Drug_Gene", data = ctrp_data_g2p, method = "wilcox.test", p.adjust.method = "BH")
ctrp_signif_g2p <- adj_signif(ctrp_signif_g2p)
ctrp_signif_g2p <- ctrp_signif_g2p[order(ctrp_signif_g2p$p),]
saveRDS(ctrp_signif_g2p, "./data_munging/rds/ctrp_signif_g2p_gene.rds")

# Grid comparisons of G2P drugs and genes
ctrp_signif_g2p_grid <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "Drug"), data = ctrp_data_g2p_grid, method = "wilcox.test", p.adjust.method = "BH")
ctrp_signif_g2p_grid <- adj_signif(ctrp_signif_g2p_grid)
ctrp_signif_g2p_grid <- ctrp_signif_g2p_grid[order(ctrp_signif_g2p_grid$p),]
ctrp_signif_g2p_grid$Drug_Gene <- paste0(ctrp_signif_g2p_grid$Drug, "_", ctrp_signif_g2p_grid$Hugo_Symbol)
saveRDS(ctrp_signif_g2p_grid, "./data_munging/rds/ctrp_signif_g2p_grid.rds")
```

```{r}
ctrp_signif_g2p <- readRDS("./data_munging/rds/ctrp_signif_g2p_gene.rds")

knitr::kable(ctrp_signif_g2p[, c("Drug_Gene", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CTRP: Wilcoxon test results for level A G2P drug-gene associations (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")

ctrp_signif_g2p_grid <- readRDS("./data_munging/rds/ctrp_signif_g2p_grid.rds")

knitr::kable(ctrp_signif_g2p_grid[, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CTRP: Wilcoxon test results for point mutations in level A G2P genes and drugs (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

### G2P gene, drug, and lineage

```{r, eval=FALSE}
ctrp_signif_g2p_lineage <- lapply(unique(ctrp_data_g2p_grid$Drug), WilcoxonByDrug, dataset = ctrp_data_g2p_grid, data_name = "ctrp", grouping_var = "lineage")
names(ctrp_signif_g2p_lineage) <- unique(ctrp_signif_g2p_lineage$Drug)
ctrp_signif_g2p_lineage <- rbindlist(ctrp_signif_g2p_lineage, use.names = TRUE)
ctrp_signif_g2p_lineage <- ctrp_signif_g2p_lineage[order(ctrp_signif_g2p_lineage$p),]
saveRDS(ctrp_signif_g2p_lineage, "./data_munging/rds/ctrp_signif_g2p_lineage.rds")
```

```{r}
ctrp_signif_g2p_lineage <- readRDS("./data_munging/rds/ctrp_signif_g2p_lineage.rds")

knitr::kable(ctrp_signif_g2p_lineage[1:500, c("Hugo_Symbol", "Drug", "group_general_lineage_name", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CTRP: Wilcoxon test results for level A G2P drug-gene associations across lineages (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

## GDSC

### G2P gene and drug

```{r, eval=FALSE}
gdsc_signif_g2p <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = "Drug_Gene", data = gdsc_data_g2p, method = "wilcox.test", p.adjust.method = "BH")
gdsc_signif_g2p <- adj_signif(gdsc_signif_g2p)
gdsc_signif_g2p <- gdsc_signif_g2p[order(gdsc_signif_g2p$p),]
saveRDS(gdsc_signif_g2p, "./data_munging/rds/gdsc_signif_g2p_gene.rds")

# Grid comparisons of G2P drugs and genes
gdsc_signif_g2p_grid <- compare_means(AUC_zscore ~ Mutation_Status_Nonsilent, group.by = c("Hugo_Symbol", "Drug"), data = gdsc_data_g2p_grid, method = "wilcox.test", p.adjust.method = "BH")
gdsc_signif_g2p_grid <- adj_signif(gdsc_signif_g2p_grid)
gdsc_signif_g2p_grid <- gdsc_signif_g2p_grid[order(gdsc_signif_g2p_grid$p),]
gdsc_signif_g2p_grid$Drug_Gene <- paste0(gdsc_signif_g2p_grid$Drug, "_", gdsc_signif_g2p_grid$Hugo_Symbol)
saveRDS(gdsc_signif_g2p_grid, "./data_munging/rds/gdsc_signif_g2p_grid.rds")
```

```{r}
gdsc_signif_g2p <- readRDS("./data_munging/rds/gdsc_signif_g2p_gene.rds")

knitr::kable(gdsc_signif_g2p[, c("Drug_Gene", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "GDSC: Wilcoxon test results for level A G2P drug-gene associations (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")

gdsc_signif_g2p_grid <- readRDS("./data_munging/rds/gdsc_signif_g2p_grid.rds")

knitr::kable(gdsc_signif_g2p_grid[, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "GDSC: Wilcoxon test results for point mutations in level A G2P genes and drugs (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

### G2P gene, drug, and lineage

```{r, eval=FALSE}
gdsc_signif_g2p_lineage <- lapply(unique(gdsc_data_g2p_grid$Drug), WilcoxonByDrug, dataset = gdsc_data_g2p_grid, data_name = "gdsc", grouping_var = "lineage")
names(gdsc_signif_g2p_lineage) <- unique(gdsc_signif_g2p_lineage$Drug)
gdsc_signif_g2p_lineage <- rbindlist(gdsc_signif_g2p_lineage, use.names = TRUE)
gdsc_signif_g2p_lineage <- gdsc_signif_g2p_lineage[order(gdsc_signif_g2p_lineage$p),]
saveRDS(gdsc_signif_g2p_lineage, "./data_munging/rds/gdsc_signif_g2p_lineage.rds")
```

```{r}
gdsc_signif_g2p_lineage <- readRDS("./data_munging/rds/gdsc_signif_g2p_lineage.rds")

knitr::kable(gdsc_signif_g2p_lineage[1:500, c("Hugo_Symbol", "Drug", "group_general_lineage_name", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "GDSC: Wilcoxon test results for level A G2P drug-gene associations across lineages (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

## Drug screens: All CRISPR genes

Drug metadata:
```{r}
drug_meta <- read.delim("./data_munging/drug_id_drug_name_table_10_12_2018.tsv", sep = "\t", header = FALSE)
colnames(drug_meta) <- c("CID", "Drug")
drug_meta$Drug <- tolower(drug_meta$Drug)
```

CCLE:

user   system  elapsed 
1483.399  262.649 1901.698 

```{r, eval=FALSE}
ccle <- read.delim("./data_munging/data_drug_auc_ccle_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
ccle <- merge(drug_meta, ccle, by = "CID")
ccle_melt <- melt(ccle, id.vars = c("CID", "Drug"), measure.vars = colnames(ccle)[3:ncol(ccle)], variable.name = "accession_id", value.name = "AUC")
ccle_grid <- expand.grid("accession_id" = unique(ccle_melt$accession_id), "Hugo_Symbol" = as.character(unique(crispr_data$Hugo_Symbol)))
system.time({ ccle_data_allgenes <- WilcoxonByDrugAllGenes(ccle_melt, ccle_grid, "CCLE") })
saveRDS(ccle_data_allgenes, "./data_munging/rds/ccle_data_18Q3_allgenes.rds", compress = "xz")
```

```{r}
ccle_data_allgenes <- readRDS("./data_munging/rds/ccle_data_18Q3_allgenes.rds")
ccle_data_allgenes <- ccle_data_allgenes[order(ccle_data_allgenes$p),]

knitr::kable(ccle_data_allgenes[1:1000, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CCLE: by-drug Wilcoxon test results for point mutations in genes targeted in CRISPR screen (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

CTRP:
```{r, eval=FALSE}
ctrp <- read.delim("./data_munging/data_drug_auc_ctrp_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
ctrp <- merge(drug_meta, ctrp, by = "CID")
ctrp_melt <- melt(ctrp, id.var  s = c("CID", "Drug"), measure.vars = colnames(ctrp)[3:ncol(ctrp)], variable.name = "accession_id", value.name = "AUC")
ctrp_grid <- expand.grid("accession_id" = unique(ctrp_melt$accession_id), "Hugo_Symbol" = as.character(unique(crispr_data$Hugo_Symbol)))
ctrp_data_allgenes <- WilcoxonByDrugAllGenes(ctrp_melt, ctrp_grid, "CTRP")
saveRDS(ctrp_data_allgenes, "./data_munging/rds/ctrp_data_18Q3_allgenes.rds", compress = "xz")
```

```{r}
ctrp_data_allgenes <- readRDS("./data_munging/rds/ctrp_data_18Q3_allgenes.rds")
ctrp_data_allgenes <- ctrp_data_allgenes[order(ctrp_data_allgenes$p),]

knitr::kable(ctrp_data_allgenes[1:1000, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "CTRP: by-drug Wilcoxon test results for point mutations in genes targeted in CRISPR screen (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

GDSC:
```{r, eval=FALSE}
gdsc <- read.delim("./data_munging/data_drug_auc_gdsc_10_12_2018.csv", sep = "\t", header  = TRUE, row.names = 1, check.names = FALSE)
gdsc <- merge(drug_meta, gdsc, by = "CID")
gdsc_melt <- melt(gdsc, id.vars = c("CID", "Drug"), measure.vars = colnames(gdsc)[3:ncol(gdsc)], variable.name = "accession_id", value.name = "AUC")
gdsc_grid <- expand.grid("accession_id" = unique(gdsc_melt$accession_id), "Hugo_Symbol" = as.character(unique(crispr_data$Hugo_Symbol)))
system.time({ gdsc_data_allgenes <- WilcoxonByDrugAllGenes(gdsc_melt, gdsc_grid, "GDSC") })
saveRDS(gdsc_data_allgenes, "./data_munging/rds/gdsc_data_18Q3_allgenes.rds", compress = "xz")
```

```{r}
gdsc_data_allgenes <- readRDS("./data_munging/rds/gdsc_data_18Q3_allgenes.rds")
gdsc_data_allgenes <- gdsc_data_allgenes[order(gdsc_data_allgenes$p),]

knitr::kable(gdsc_data_allgenes[1:1000, c("Hugo_Symbol", "Drug", "p", "p.adj", "p.format", "p.signif", "p.signif.adj")], caption = "GDSC: by-drug Wilcoxon test results for point mutations in genes targeted in CRISPR screen (* p <= 0.05, ** p <= 0.01, *** p <= 0.001, **** p <= 0.0001)") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```


# Dataset summaries

--------------------------------------------------------------------------------

## CRISPR: Mutation status counts

```{r}
crispr_data_g2p_summ <- unique(crispr_data_g2p[, c("Hugo_Symbol", "N_Nonsilent_Mutant", "N_Nonsilent_Wildtype")])
crispr_data_g2p_summ$Percent_Mutant <- crispr_data_g2p_summ$N_Nonsilent_Mutant / (crispr_data_g2p_summ$N_Nonsilent_Mutant + crispr_data_g2p_summ$N_Nonsilent_Wildtype) * 100
rownames(crispr_data_g2p_summ) <- NULL
# write.table(crispr_data_g2p_summ, file = "~/Desktop/crispr_data_g2p_summ_20181003.tsv", quote = FALSE, sep = "\t")

knitr::kable(crispr_data_g2p_summ, caption = "CRISPR: Level A G2P genes mutation status summary") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

## Drugscreen: G2P evidence

### Lineage-agnostic

#### Mutation status

```{r}
g2p_druggene_p_tally <- g2p %>% group_by(Drug_Gene) %>% tally() %>% drop_na()
```

```{r fig.height=4, fig.width=8}
g2p_druggene_p_summ <- merge(g2p_druggene_p_tally, ccle_signif_g2p[, c("Drug_Gene", "p")], by = "Drug_Gene", all.x = TRUE)
g2p_druggene_p_summ <- merge(g2p_druggene_p_summ, ctrp_signif_g2p[, c("Drug_Gene", "p")], by = "Drug_Gene", all.x = TRUE, suffixes = c("_CCLE", "_CTRP"))
g2p_druggene_p_summ <- merge(g2p_druggene_p_summ, gdsc_signif_g2p[, c("Drug_Gene", "p")], by = "Drug_Gene", all.x = TRUE)
colnames(g2p_druggene_p_summ)[colnames(g2p_druggene_p_summ) == "p_CCLE"] <- "CCLE"
colnames(g2p_druggene_p_summ)[colnames(g2p_druggene_p_summ) == "p_CTRP"] <- "CTRP"
colnames(g2p_druggene_p_summ)[colnames(g2p_druggene_p_summ) == "p"] <- "GDSC"
g2p_druggene_p_summ_melt <- melt(g2p_druggene_p_summ, id.vars = c("Drug_Gene", "n"), measure.vars = c("CCLE", "CTRP", "GDSC"), variable.name = "Dataset", value.name = "p")

g2p_druggene_p_summ_plot <- ggplot(data = g2p_druggene_p_summ_melt, aes(x = p, y = n, shape = Dataset, color = Dataset)) +
  geom_point(size = 2) +
  scale_color_manual(values = c("green3", "darkslategray4", "darkturquoise")) +
  labs(x = "Gene-drug association p-value by mutation status", y = "Number of level A G2P associations")
g2p_druggene_p_summ_plot
# ggsave("./plots_18Q3/manuscript/g2p_druggene_p_summ_plot.png", g2p_druggene_p_summ_plot, device = "png", dpi = 450, width = 8, height = 4, units = "in")
```

```
for_dan <- with(g2p_druggene_p_summ, cbind(n, CCLE, CTRP, GDSC, colsplit(Drug_Gene, pattern = "_", names = c("Drug", "Hugo_Symbol"))))

write.table(for_dan, file = "~/Desktop/for_dan.tsv", sep = "\t")
```


```{r}
ccle_druggene_summ <- merge(g2p_druggene_p_tally, ccle_signif_g2p_grid[, c("Drug_Gene", "p")], by = "Drug_Gene", all = TRUE)
ccle_druggene_summ$n <- ifelse(is.na(ccle_druggene_summ$n), 0, ccle_druggene_summ$n)
ccle_druggene_summ$InG2P <- ifelse(ccle_druggene_summ$n != 0, "Yes", "No")
ccle_druggene_summ$Dataset <- "CCLE"
ccle_p_ttest <- t.test(ccle_druggene_summ$p ~ ccle_druggene_summ$InG2P)
ccle_p_ttest_res <- paste0("t = ", round(ccle_p_ttest$statistic, 3), ", df = ", round(ccle_p_ttest$parameter, 3), ", p = ", round(ccle_p_ttest$p.value, 3))

ctrp_druggene_summ <- merge(g2p_druggene_p_tally, ctrp_signif_g2p_grid[, c("Drug_Gene", "p")], by = "Drug_Gene", all = TRUE)
ctrp_druggene_summ$n <- ifelse(is.na(ctrp_druggene_summ$n), 0, ctrp_druggene_summ$n)
ctrp_druggene_summ$InG2P <- ifelse(ctrp_druggene_summ$n != 0, "Yes", "No")
ctrp_druggene_summ$Dataset <- "CTRP"
ctrp_p_ttest <- t.test(ctrp_druggene_summ$p ~ ctrp_druggene_summ$InG2P)
ctrp_p_ttest_res <- paste0("t = ", round(ctrp_p_ttest$statistic, 3), ", df = ", round(ctrp_p_ttest$parameter, 3), ", p = ", round(ctrp_p_ttest$p.value, 3))

gdsc_druggene_summ <- merge(g2p_druggene_p_tally, gdsc_signif_g2p_grid[, c("Drug_Gene", "p")], by = "Drug_Gene", all = TRUE)
gdsc_druggene_summ$n <- ifelse(is.na(gdsc_druggene_summ$n), 0, gdsc_druggene_summ$n)
gdsc_druggene_summ$InG2P <- ifelse(gdsc_druggene_summ$n != 0, "Yes", "No")
gdsc_druggene_summ$Dataset <- "GDSC"
gdsc_p_ttest <- t.test(gdsc_druggene_summ$p ~ gdsc_druggene_summ$InG2P)
gdsc_p_ttest_res <- paste0("t = ", round(gdsc_p_ttest$statistic, 3), ", df = ", round(gdsc_p_ttest$parameter, 3), ", p = ", round(gdsc_p_ttest$p.value, 3))

g2p_druggene_all_summ <- rbind(ccle_druggene_summ, ctrp_druggene_summ, gdsc_druggene_summ)
g2p_druggene_all_summ_text <- data.frame(Dataset = c("CCLE", "CTRP", "GDSC"), Test_Results_p = c(ccle_p_ttest_res, ctrp_p_ttest_res, gdsc_p_ttest_res))
g2p_druggene_all_summ$Drug_Gene_Ordered <- factor(g2p_druggene_all_summ$Drug_Gene, levels = as.character(unique(g2p_druggene_all_summ$Drug_Gene)), labels = sapply(strsplit(x = as.character(unique(g2p_druggene_all_summ$Drug_Gene)), split = "_"), function(x) paste0(x[1], "\n", x[2])))
```

```{r fig.height=5, fig.width=12}
plot_p_point <- ggplot(data = g2p_druggene_all_summ, aes(x = p, y = n, label = Drug_Gene)) +
  facet_wrap(~ Dataset, scales = "free_x") +
  geom_point(mapping = aes(color = InG2P), alpha = 0.5) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_y_continuous(breaks = seq(0, 110, by = 10), labels = seq(0, 110, by = 10)) +
  geom_text_repel(data = subset(g2p_druggene_all_summ, InG2P == "Yes" & n > 30), size = 3, segment.size = 0.2, segment.color = "grey50") +
  geom_text(data = g2p_druggene_all_summ_text, mapping = aes(x = 0.5, y = 110, label = Test_Results_p)) +
  labs(x = "Gene-drug association p-value between mutation groups", y = "Number of level A G2P associations") +
  theme(legend.position = "none", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank())

plot_p_box <- ggplot(data = g2p_druggene_all_summ) +
  facet_wrap(~ Dataset, scales = "free_x") +
  geom_violin(mapping = aes(color = InG2P, y = p, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = p, x = InG2P), alpha = 0.75, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "In G2P?") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

plot_p_point <- ggplot_gtable(ggplot_build(plot_p_point))
plot_p_box <- ggplot_gtable(ggplot_build(plot_p_box))
p_maxWidth <-  grid::unit.pmax(plot_p_point$widths, plot_p_box$widths)
plot_p_point$widths <- as.list(p_maxWidth)
plot_p_box$widths <- as.list(p_maxWidth)

g2p_druggene_mut_plot <- cowplot::plot_grid(plot_p_box,
                   plot_p_point,
                   nrow = 2,
                   rel_heights = c(1, 4))
g2p_druggene_mut_plot
# ggsave("./plots_18Q3/manuscript/g2p_druggene_mut_plot.png", g2p_druggene_mut_plot, device = "png", dpi = 450, width = 12, height = 5, units = "in")
```

```{r, eval=FALSE}
plot_p_point <- ggplot(data = g2p_druggene_all_summ) +
  facet_wrap(~ Dataset, scales = "free_x") +
  geom_point(mapping = aes(label = Drug_Gene, color = InG2P, x = p, y = n), alpha = 0.5) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_y_continuous(breaks = seq(0, 110, by = 10), labels = seq(0, 110, by = 10)) +
  geom_text(data = g2p_druggene_all_summ_text, mapping = aes(x = 0.5, y = 110, label = Test_Results_p)) +
  labs(x = "Gene-drug association p-value between mutation groups", y = "Number of level A G2P associations") +
  theme(legend.position = "none", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank())

plot_p_box <- ggplot(data = g2p_druggene_all_summ) +
  facet_wrap(~ Dataset, scales = "free_x") +
  geom_violin(mapping = aes(color = InG2P, y = p, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = p, x = InG2P), alpha = 0.75, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "Level A G2P\nassociation") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

plot <- subplot(plot_p_box, plot_p_point, nrows = 2, titleX = TRUE, titleY = TRUE) %>% layout(margin = list(l = 75), xaxis5 = list(title = "Gene-drug association p-value between mutation groups"), yaxis = list(title = "In G2P?", domain = c(0.8, 1)), yaxis2 = list(title = "Number of level A G2P associations", domain = c(0, 0.79)))

saveWidgetFix(widget = plot, file = "./plots_18Q3/manuscript/g2p_druggene_mut_plot.html")
```

#### GE and CN

```{r}
ccle_pearson_res <- ccle_data_g2p_grid %>% group_by(Drug_Gene) %>%
                    summarize(Pearson_Corr_GE = cor.test(y = AUC, x = RPKM, method = "pearson", use = "complete.obs")$estimate,
                              Pearson_Pval_GE = cor.test(y = AUC, x = RPKM, method = "pearson", use = "complete.obs")$p.value,
                              Pearson_Corr_CN = cor.test(y = AUC, x = CN, method = "pearson", use = "complete.obs")$estimate,
                              Pearson_Pval_CN = cor.test(y = AUC, x = CN, method = "pearson", use = "complete.obs")$p.value)
ccle_pearson_res <- merge(g2p_druggene_p_tally, ccle_pearson_res, by = "Drug_Gene", all.y = TRUE)
ccle_pearson_res$n <- ifelse(is.na(ccle_pearson_res$n), 0, ccle_pearson_res$n)
ccle_pearson_res$InG2P <- ifelse(ccle_pearson_res$n != 0, "Yes", "No")
ccle_pearson_res$Dataset <- "CCLE"
ccle_pearson_ttest_GE <- t.test(ccle_pearson_res$Pearson_Corr_GE ~ ccle_pearson_res$InG2P)
ccle_pearson_ttest_res_GE <- paste0("t = ", round(ccle_pearson_ttest_GE$statistic, 3), ", df = ", round(ccle_pearson_ttest_GE$parameter, 3), ", p = ", round(ccle_pearson_ttest_GE$p.value, 3))
ccle_pearson_ttest_CN <- t.test(ccle_pearson_res$Pearson_Corr_CN ~ ccle_pearson_res$InG2P)
ccle_pearson_ttest_res_CN <- paste0("t = ", round(ccle_pearson_ttest_CN$statistic, 3), ", df = ", round(ccle_pearson_ttest_CN$parameter, 3), ", p = ", round(ccle_pearson_ttest_CN$p.value, 3))

ctrp_pearson_res <- ctrp_data_g2p_grid %>% group_by(Drug_Gene) %>%
                    summarize(Pearson_Corr_GE = cor.test(y = AUC, x = RPKM, method = "pearson", use = "complete.obs")$estimate,
                              Pearson_Pval_GE = cor.test(y = AUC, x = RPKM, method = "pearson", use = "complete.obs")$p.value,
                              Pearson_Corr_CN = cor.test(y = AUC, x = CN, method = "pearson", use = "complete.obs")$estimate,
                              Pearson_Pval_CN = cor.test(y = AUC, x = CN, method = "pearson", use = "complete.obs")$p.value)
ctrp_pearson_res <- merge(g2p_druggene_p_tally, ctrp_pearson_res, by = "Drug_Gene", all.y = TRUE)
ctrp_pearson_res$n <- ifelse(is.na(ctrp_pearson_res$n), 0, ctrp_pearson_res$n)
ctrp_pearson_res$InG2P <- ifelse(ctrp_pearson_res$n != 0, "Yes", "No")
ctrp_pearson_res$Dataset <- "CTRP"
ctrp_pearson_ttest_GE <- t.test(ctrp_pearson_res$Pearson_Corr_GE ~ ctrp_pearson_res$InG2P)
ctrp_pearson_ttest_res_GE <- paste0("t = ", round(ctrp_pearson_ttest_GE$statistic, 3), ", df = ", round(ctrp_pearson_ttest_GE$parameter, 3), ", p = ", round(ctrp_pearson_ttest_GE$p.value, 3))
ctrp_pearson_ttest_CN <- t.test(ctrp_pearson_res$Pearson_Corr_CN ~ ctrp_pearson_res$InG2P)
ctrp_pearson_ttest_res_CN <- paste0("t = ", round(ctrp_pearson_ttest_CN$statistic, 3), ", df = ", round(ctrp_pearson_ttest_CN$parameter, 3), ", p = ", round(ctrp_pearson_ttest_CN$p.value, 3))

gdsc_pearson_res <- gdsc_data_g2p_grid %>% group_by(Drug_Gene) %>%
                    summarize(Pearson_Corr_GE = cor.test(y = AUC, x = RPKM, method = "pearson", use = "complete.obs")$estimate,
                              Pearson_Pval_GE = cor.test(y = AUC, x = RPKM, method = "pearson", use = "complete.obs")$p.value,
                              Pearson_Corr_CN = cor.test(y = AUC, x = CN, method = "pearson", use = "complete.obs")$estimate,
                              Pearson_Pval_CN = cor.test(y = AUC, x = CN, method = "pearson", use = "complete.obs")$p.value)
gdsc_pearson_res <- merge(g2p_druggene_p_tally, gdsc_pearson_res, by = "Drug_Gene", all.y = TRUE)
gdsc_pearson_res$n <- ifelse(is.na(gdsc_pearson_res$n), 0, gdsc_pearson_res$n)
gdsc_pearson_res$InG2P <- ifelse(gdsc_pearson_res$n != 0, "Yes", "No")
gdsc_pearson_res$Dataset <- "GDSC"
gdsc_pearson_ttest_GE <- t.test(gdsc_pearson_res$Pearson_Corr_GE ~ gdsc_pearson_res$InG2P)
gdsc_pearson_ttest_res_GE <- paste0("t = ", round(gdsc_pearson_ttest_GE$statistic, 3), ", df = ", round(gdsc_pearson_ttest_GE$parameter, 3), ", p = ", round(gdsc_pearson_ttest_GE$p.value, 3))
gdsc_pearson_ttest_CN <- t.test(gdsc_pearson_res$Pearson_Corr_CN ~ gdsc_pearson_res$InG2P)
gdsc_pearson_ttest_res_CN <- paste0("t = ", round(gdsc_pearson_ttest_CN$statistic, 3), ", df = ", round(gdsc_pearson_ttest_CN$parameter, 3), ", p = ", round(gdsc_pearson_ttest_CN$p.value, 3))

g2p_druggene_pearson <- rbind(ccle_pearson_res, ctrp_pearson_res, gdsc_pearson_res)
g2p_druggene_pearson$Pearson_GE_Abs <- abs(g2p_druggene_pearson$Pearson_Corr_GE)
g2p_druggene_pearson$Pearson_GE_Sign <- ifelse(sign(g2p_druggene_pearson$Pearson_Corr_GE) == 1, "+",
                                           ifelse(sign(g2p_druggene_pearson$Pearson_Corr_GE) == -1, "-", "0"))
g2p_druggene_pearson$Pearson_CN_Abs <- abs(g2p_druggene_pearson$Pearson_Corr_CN)
g2p_druggene_pearson$Pearson_CN_Sign <- ifelse(sign(g2p_druggene_pearson$Pearson_Corr_CN) == 1, "+",
                                           ifelse(sign(g2p_druggene_pearson$Pearson_Corr_CN) == -1, "-", "0"))
g2p_druggene_pearson_text <- data.frame(Dataset = c("CCLE", "CTRP", "GDSC"),
                                        Test_Results_GE = c(ccle_pearson_ttest_res_GE, ctrp_pearson_ttest_res_GE, gdsc_pearson_ttest_res_GE),
                                        Test_Results_CN = c(ccle_pearson_ttest_res_CN, ctrp_pearson_ttest_res_CN, gdsc_pearson_ttest_res_CN))
```

Gene expression:
```{r fig.height=5.5, fig.width=12}
plot_ge_point <- ggplot(data = g2p_druggene_pearson, aes(x = Pearson_GE_Abs, y = n, label = Drug_Gene)) +
  facet_wrap(~ Dataset) +
  geom_point(mapping = aes(fill = InG2P, color = InG2P, shape = Pearson_GE_Sign), alpha = 0.5, size = 1.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_y_continuous(breaks = seq(0, 110, by = 10), labels = seq(0, 110, by = 10)) +
  scale_x_continuous(breaks = pretty_breaks()) +
  scale_shape_manual(values = c(25, 24), labels = c("Negative", "Positive")) +
  geom_text_repel(data = subset(g2p_druggene_pearson, InG2P == "Yes" & n > 10), size = 3, segment.size = 0.2, segment.color = "grey50") +
  geom_text(data = g2p_druggene_pearson_text, mapping = aes(x = 0.3, y = 110, label = Test_Results_GE)) +
  labs(x = "Pearson correlation coefficient between drug-gene AUC and gene expression", y = "Number of level A G2P associations") +
  theme(legend.position = "bottom", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank()) +
  guides(color = FALSE, fill = FALSE) +
  labs(shape = "Direction of correlation:")

plot_ge_box <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_violin(mapping = aes(color = InG2P, y = Pearson_GE_Abs, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = Pearson_GE_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "In G2P?") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

plot_ge_point <- ggplot_gtable(ggplot_build(plot_ge_point))
plot_ge_box <- ggplot_gtable(ggplot_build(plot_ge_box))
p_maxWidth <-  grid::unit.pmax(plot_ge_point$widths, plot_ge_box$widths)
plot_ge_point$widths <- as.list(p_maxWidth)
plot_ge_box$widths <- as.list(p_maxWidth)

g2p_druggene_ge_plot <- cowplot::plot_grid(plot_ge_box, plot_ge_point, nrow = 2, rel_heights = c(1, 4))
g2p_druggene_ge_plot
# ggsave("./plots_18Q3/manuscript/g2p_druggene_ge_plot.png", g2p_druggene_ge_plot, device = "png", dpi = 450, width = 12, height = 5.5, units = "in")
```

```{r, eval=FALSE}
plot_ge_point <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_point(mapping = aes(label = Drug_Gene, fill = InG2P, color = InG2P, shape = Pearson_GE_Sign, x = Pearson_GE_Abs, y = n, size = Pearson_Pval_GE), alpha = 0.5, size = 1.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_shape_manual(values = c(25, 24), labels = c("Negative", "Positive")) +
  scale_y_continuous(breaks = seq(0, 110, by = 10), labels = seq(0, 110, by = 10)) +
  scale_x_continuous(breaks = pretty_breaks()) +
  geom_text(data = g2p_druggene_pearson_text, mapping = aes(x = 0.3, y = 110, label = Test_Results_GE)) +
  labs(x = "Pearson correlation coefficient between drug-gene AUC and gene expression", y = "Number of level A G2P associations") +
  theme(legend.position = "bottom", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank()) +
  guides(color = FALSE, fill = FALSE) +
  labs(shape = NULL)

plot_ge_box <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_violin(mapping = aes(color = InG2P, y = Pearson_GE_Abs, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = Pearson_GE_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "In G2P?") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

plot <- subplot(plot_ge_box, plot_ge_point, nrows = 2, titleX = TRUE, titleY = TRUE) %>% layout(margin = list(l = 75), xaxis5 = list(title = "Pearson correlation coefficient between drug-gene AUC and gene expression"), yaxis = list(title = "In G2P?", domain = c(0.8, 1)), yaxis2 = list(title = "Number of level A G2P associations", domain = c(0, 0.79)))

saveWidgetFix(widget = plot, file = "./plots_18Q3/manuscript/g2p_druggene_ge_plot.html")
```

Copy number:
```{r fig.height=5.5, fig.width=12}
plot_cn_point <- ggplot(data = g2p_druggene_pearson, aes(x = Pearson_CN_Abs, y = n, label = Drug_Gene)) +
  facet_wrap(~ Dataset) +
  geom_point(mapping = aes(fill = InG2P, color = InG2P, shape = Pearson_CN_Sign), alpha = 0.5, size = 1.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_y_continuous(breaks = seq(0, 110, by = 10), labels = seq(0, 110, by = 10)) +
  scale_x_continuous(breaks = pretty_breaks()) +
  scale_shape_manual(values = c(25, 24), labels = c("Negative", "Positive")) +
  geom_text_repel(data = subset(g2p_druggene_pearson, InG2P == "Yes" & n > 10), size = 3, segment.size = 0.2, segment.color = "grey50") +
  geom_text(data = g2p_druggene_pearson_text, mapping = aes(x = 0.3, y = 110, label = Test_Results_CN)) +
  labs(x = "Pearson correlation coefficient between drug-gene AUC and copy number", y = "Number of level A G2P associations") +
  theme(legend.position = "bottom", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank()) +
  guides(color = FALSE, fill = FALSE) +
  labs(shape = "Direction of correlation:")

plot_cn_box <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_violin(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "In G2P?") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

plot_cn_point <- ggplot_gtable(ggplot_build(plot_cn_point))
plot_cn_box <- ggplot_gtable(ggplot_build(plot_cn_box))
p_maxWidth <-  grid::unit.pmax(plot_cn_point$widths, plot_cn_box$widths)
plot_cn_point$widths <- as.list(p_maxWidth)
plot_cn_box$widths <- as.list(p_maxWidth)

g2p_druggene_cn_plot <- cowplot::plot_grid(plot_cn_box, plot_cn_point, nrow = 2, rel_heights = c(1, 4))
g2p_druggene_cn_plot
# ggsave("./plots_18Q3/manuscript/g2p_druggene_cn_plot.png", g2p_druggene_cn_plot, device = "png", dpi = 450, width = 12, height = 5.5, units = "in")
```

```{r, eval=FALSE}
plot_cn_point <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_point(mapping = aes(label = Drug_Gene, fill = InG2P, color = InG2P, shape = Pearson_CN_Sign, x = Pearson_CN_Abs, y = n), alpha = 0.5, size = 1.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_shape_manual(values = c(25, 24), labels = c("Negative", "Positive")) +
  scale_y_continuous(breaks = seq(0, 110, by = 10), labels = seq(0, 110, by = 10)) +
  scale_x_continuous(breaks = pretty_breaks()) +
  geom_text(data = g2p_druggene_pearson_text, mapping = aes(x = 0.3, y = 110, label = Test_Results_GE)) +
  labs(x = "Pearson correlation coefficient between drug-gene AUC and copy number", y = "Number of level A G2P associations") +
  theme(legend.position = "bottom", plot.margin = unit(c(0.05, 0.5, 0.5, 0.5), "cm"), strip.background = element_blank(), strip.text.x = element_blank()) +
  guides(color = FALSE, fill = FALSE) +
  labs(shape = NULL)

plot_cn_box <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_violin(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "In G2P?") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))

plot <- subplot(plot_cn_box, plot_cn_point, nrows = 2, titleX = TRUE, titleY = TRUE) %>% layout(margin = list(l = 75), xaxis5 = list(title = "Pearson correlation coefficient between drug-gene AUC and copy number"), yaxis = list(title = "In G2P?", domain = c(0.8, 1)), yaxis2 = list(title = "Number of level A G2P associations", domain = c(0, 0.79)))

saveWidgetFix(widget = plot, file = "./plots_18Q3/manuscript/g2p_druggene_cn_plot.html")
```


### By lineage

#### Mutation Status

```{r}
g2p_druggenelin_p_tally <- g2p %>% group_by(Drug_Gene, group_general_lineage_name) %>% tally()
```

```{r}
ccle_druggenelin_summ <- ccle_signif_g2p_lineage
ccle_druggenelin_summ$Drug_Gene <- paste0(ccle_druggenelin_summ$Drug, "_", ccle_druggenelin_summ$Hugo_Symbol)
ccle_druggenelin_summ <- merge(g2p_druggenelin_p_tally, ccle_druggenelin_summ[, c("Drug_Gene", "group_general_lineage_name", "p")], by = c("Drug_Gene", "group_general_lineage_name"), all.y = TRUE)
ccle_druggenelin_summ$n <- ifelse(is.na(ccle_druggenelin_summ$n), 0, ccle_druggenelin_summ$n)
ccle_druggenelin_summ$InG2P <- ifelse(ccle_druggenelin_summ$n != 0, "Yes", "No")
ccle_druggenelin_summ$Dataset <- "CCLE"

ctrp_druggenelin_summ <- ctrp_signif_g2p_lineage
ctrp_druggenelin_summ$Drug_Gene <- paste0(ctrp_druggenelin_summ$Drug, "_", ctrp_druggenelin_summ$Hugo_Symbol)
ctrp_druggenelin_summ <- merge(g2p_druggenelin_p_tally, ctrp_druggenelin_summ[, c("Drug_Gene", "group_general_lineage_name", "p")], by = c("Drug_Gene", "group_general_lineage_name"), all.y = TRUE)
ctrp_druggenelin_summ$n <- ifelse(is.na(ctrp_druggenelin_summ$n), 0, ctrp_druggenelin_summ$n)
ctrp_druggenelin_summ$InG2P <- ifelse(ctrp_druggenelin_summ$n != 0, "Yes", "No")
ctrp_druggenelin_summ$Dataset <- "CTRP"

gdsc_druggenelin_summ <- gdsc_signif_g2p_lineage
gdsc_druggenelin_summ$Drug_Gene <- paste0(gdsc_druggenelin_summ$Drug, "_", gdsc_druggenelin_summ$Hugo_Symbol)
gdsc_druggenelin_summ <- merge(g2p_druggenelin_p_tally, gdsc_druggenelin_summ[, c("Drug_Gene", "group_general_lineage_name", "p")], by = c("Drug_Gene", "group_general_lineage_name"), all.y = TRUE)
gdsc_druggenelin_summ$n <- ifelse(is.na(gdsc_druggenelin_summ$n), 0, gdsc_druggenelin_summ$n)
gdsc_druggenelin_summ$InG2P <- ifelse(gdsc_druggenelin_summ$n != 0, "Yes", "No")
gdsc_druggenelin_summ$Dataset <- "GDSC"

g2p_druggenelin_all_summ <- rbind(ccle_druggenelin_summ, ctrp_druggenelin_summ, gdsc_druggenelin_summ)
g2p_druggenelin_all_summ$InG2P <- factor(g2p_druggenelin_all_summ$InG2P, levels = c("Yes", "No"))

# test <- filter(filter(g2p_druggenelin_all_summ, group_general_lineage_name == "breast cancer"), Dataset == "CCLE")
# test <- test[-39,]

```

```{r, eval=FALSE, results = 'hide'}
g2p_druggenelin_p_ttest_res <- lapply(as.character(unique(g2p_druggenelin_all_summ$group_general_lineage_name)), MutPlotlyByLineage)
g2p_druggenelin_p_ttest_res_list <- rbindlist(g2p_druggenelin_p_ttest_res, use.names = TRUE)
g2p_druggenelin_p_ttest_res_list <- g2p_druggenelin_p_ttest_res_list[order(g2p_druggenelin_p_ttest_res_list$p_value),]
saveRDS(g2p_druggenelin_p_ttest_res_list, "./data_munging/rds/g2p_druggenelin_p_ttest_res.rds")
# test <- filter(g2p_druggenelin_all_summ, Dataset == "CCLE")
# test <- filter(g2p_druggenelin_all_summ, group_general_lineage_name == "biliary tract cancer")
# test_summ <- test %>% count(InG2P)
# test2 <- filter(g2p_druggenelin_all_summ, Dataset == "CTRP")
# test2_summ <- test2 %>% count(group_general_lineage_name, InG2P)
# test3 <- filter(g2p_druggenelin_all_summ, Dataset == "GDSC")
# test3_summ <- test3 %>% count(group_general_lineage_name, InG2P)
```

```{r}
g2p_druggenelin_p_ttest_res_list <- readRDS("./data_munging/rds/g2p_druggenelin_p_ttest_res.rds")

knitr::kable(g2p_druggenelin_p_ttest_res_list, caption = "Mutation status: T-test results for level A G2P drug-gene associations across lineages") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

#### GE and CN

```{r, eval=FALSE}
ccle_lineages4pearson <- as.character(unique(ccle_data_g2p_grid$group_general_lineage_name))[!is.na(as.character(unique(ccle_data_g2p_grid$group_general_lineage_name)))]
ccle_signif_ge <- lapply(ccle_lineages4pearson, PearsonByLineage, variable = "GE", dataset = ccle_data_g2p_grid)
ccle_signif_ge_all_lin <- rbindlist(ccle_signif_ge, use.names = TRUE)
ccle_signif_ge_all_lin$Dataset <- "CCLE"
ccle_signif_cn <- lapply(ccle_lineages4pearson, PearsonByLineage, variable = "CN", dataset = ccle_data_g2p_grid)
ccle_signif_cn_all_lin <- rbindlist(ccle_signif_cn, use.names = TRUE)
ccle_signif_cn_all_lin$Dataset <- "CCLE"
ccle_signif_all_lin <- merge(ccle_signif_ge_all_lin, ccle_signif_cn_all_lin, by = c("Drug_Gene", "group_general_lineage_name", "n", "Dataset"), all = TRUE)

ctrp_lineages4pearson <- as.character(unique(ctrp_data_g2p_grid$group_general_lineage_name))[!is.na(as.character(unique(ctrp_data_g2p_grid$group_general_lineage_name)))]
ctrp_signif_ge <- lapply(ctrp_lineages4pearson, PearsonByLineage, variable = "GE", dataset = ctrp_data_g2p_grid)
ctrp_signif_ge_all_lin <- rbindlist(ctrp_signif_ge, use.names = TRUE)
ctrp_signif_ge_all_lin$Dataset <- "CTRP"
ctrp_signif_cn <- lapply(ctrp_lineages4pearson, PearsonByLineage, variable = "CN", dataset = ctrp_data_g2p_grid)
ctrp_signif_cn_all_lin <- rbindlist(ctrp_signif_cn, use.names = TRUE)
ctrp_signif_cn_all_lin$Dataset <- "CTRP"
ctrp_signif_all_lin <- merge(ctrp_signif_ge_all_lin, ctrp_signif_cn_all_lin, by = c("Drug_Gene", "group_general_lineage_name", "n", "Dataset"), all = TRUE)

gdsc_lineages4pearson <- as.character(unique(gdsc_data_g2p_grid$group_general_lineage_name))[!is.na(as.character(unique(gdsc_data_g2p_grid$group_general_lineage_name)))]
gdsc_signif_ge <- lapply(gdsc_lineages4pearson, PearsonByLineage, variable = "GE", dataset = gdsc_data_g2p_grid)
gdsc_signif_ge_all_lin <- rbindlist(gdsc_signif_ge, use.names = TRUE)
gdsc_signif_ge_all_lin$Dataset <- "GDSC"
gdsc_signif_cn <- lapply(gdsc_lineages4pearson, PearsonByLineage, variable = "CN", dataset = gdsc_data_g2p_grid)
gdsc_signif_cn_all_lin <- rbindlist(gdsc_signif_cn, use.names = TRUE)
gdsc_signif_cn_all_lin$Dataset <- "GDSC"
gdsc_signif_all_lin <- merge(gdsc_signif_ge_all_lin, gdsc_signif_cn_all_lin, by = c("Drug_Gene", "group_general_lineage_name", "n", "Dataset"), all = TRUE)

signif_all_lin <- rbind(ccle_signif_all_lin, ctrp_signif_all_lin, gdsc_signif_all_lin)
signif_all_lin$n <- ifelse(is.na(signif_all_lin$n), 0, signif_all_lin$n)
signif_all_lin$InG2P <- ifelse(signif_all_lin$n != 0, "Yes", "No")
saveRDS(signif_all_lin, "./data_munging/rds/signif_all_lin.rds")
```

```{r}
signif_all_lin <- readRDS("./data_munging/rds/signif_all_lin.rds")
```

```{r, results='hide', eval=FALSE}
druggenelin_pearson_ttest_res <- lapply(as.character(unique(signif_all_lin$group_general_lineage_name)), PearsonPlotlyByLineage)
druggenelin_pearson_ttest_res_list <- rbindlist(druggenelin_pearson_ttest_res, use.names = TRUE)
saveRDS(druggenelin_pearson_ttest_res_list, "./data_munging/rds/druggenelin_pearson_ttest_res_list.rds")
```

```{r}
druggenelin_pearson_ttest_res_list <- readRDS("./data_munging/rds/druggenelin_pearson_ttest_res_list.rds")

knitr::kable(druggenelin_pearson_ttest_res_list[order(druggenelin_pearson_ttest_res_list$p_value_GE), c("Dataset", "group_general_lineage_name", "t_statistic_GE", "df_GE", "p_value_GE")], caption = "Gene expression: T-test results for level A G2P drug-gene/gene expression associations across lineages") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")

knitr::kable(druggenelin_pearson_ttest_res_list[order(druggenelin_pearson_ttest_res_list$p_value_CN), c("Dataset", "group_general_lineage_name", "t_statistic_CN", "df_CN", "p_value_CN")], caption = "Copy number: T-test results for level A G2P drug-gene/gene expression associations across lineages") %>% kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive")) %>% scroll_box(width = "900px", height = "450px")
```

### All

```{r fig.height=20, fig.width=12}
all_druggenelin_mut <- g2p_druggenelin_all_summ
all_druggenelin_mut$Sign <- "None"
all_druggenelin_mut$Measure <- "Mutation Status"
colnames(all_druggenelin_mut)[colnames(all_druggenelin_mut) == "p"] <- "Statistic"

all_druggenelin_ge <- select(signif_all_lin, Drug_Gene, group_general_lineage_name, n, Pearson_GE_Abs, InG2P, Dataset, Pearson_GE_Sign)
all_druggenelin_ge$Measure <- "Gene Expression"
colnames(all_druggenelin_ge)[colnames(all_druggenelin_ge) == "Pearson_GE_Abs"] <- "Statistic"
colnames(all_druggenelin_ge)[colnames(all_druggenelin_ge) == "Pearson_GE_Sign"] <- "Sign"

all_druggenelin_cn <- select(signif_all_lin, Drug_Gene, group_general_lineage_name, n, Pearson_CN_Abs, InG2P, Dataset, Pearson_CN_Sign)
all_druggenelin_cn$Measure <- "Copy Number"
colnames(all_druggenelin_cn)[colnames(all_druggenelin_cn) == "Pearson_CN_Abs"] <- "Statistic"
colnames(all_druggenelin_cn)[colnames(all_druggenelin_cn) == "Pearson_CN_Sign"] <- "Sign"

all_druggenelin_summ <- rbind(all_druggenelin_mut, all_druggenelin_ge, all_druggenelin_cn)
all_druggenelin_summ$Sign <- ifelse(is.na(all_druggenelin_summ$Sign), "None", all_druggenelin_summ$Sign)

all_druggenelin_summ_plot <- ggplot(data = all_druggenelin_summ, aes(x = InG2P, y = Statistic)) +
  facet_grid(group_general_lineage_name ~ Measure * Dataset) +
  geom_jitter(mapping = aes(shape = Sign, fill = InG2P, color = InG2P), width = 0.4, alpha = 0.5, size = 1) +
  geom_boxplot(width = 0.25, fill = NA, outlier.shape = NA) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_fill_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  scale_shape_manual(values = c(25, 24, 4), labels = c("Negative", "Positive", "NA")) +
  theme(legend.position = "top") +
  guides(color = FALSE, fill = FALSE)
  
ggsave("./plots_18Q3/manuscript/all_druggenelin_summ_plot.png", all_druggenelin_summ_plot, device = "png", dpi = 450, width = 12, height = 40, units = "in")
```

```
plot_cn_box <- ggplot(data = g2p_druggene_pearson) +
  facet_wrap(~ Dataset) +
  geom_violin(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5) +
  geom_boxplot(mapping = aes(color = InG2P, y = Pearson_CN_Abs, x = InG2P), alpha = 0.5, width = 0.25) +
  scale_color_manual(values = c("Yes" = "darkorchid4", "No" = "lightpink")) +
  coord_flip() +
  labs(x = "In G2P?") +
  theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.margin = unit(c(0.5, 0.5, 0.05, 0.5), "cm"))
```


## Cell line overlap

```{r, eval=FALSE}
crispr_ccl <- unique(select(crispr_data, DepMap_ID, CCLE_Name))
ccle_ccl <- unique(select(ccle_data, DepMap_ID, CCLE_Name, accession_id))
ctrp_ccl <- unique(select(ctrp_data, DepMap_ID, CCLE_Name, accession_id))
gdsc_ccl <- unique(select(gdsc_data, DepMap_ID, CCLE_Name, accession_id))

# all_ccl <- merge(ccle_ccl, ctrp_ccl, by = c("DepMap_ID", "CCLE_Name", "accession_id"), all = TRUE)
```


# Boxplots

--------------------------------------------------------------------------------

Mutation status boxplots for genes identified in G2P.

## G2P genes

Boxplots ordered by gene based on frequency of G2P associations.

### CRISPR

```{r fig.height=5, fig.width=20}
crispr_data_g2p$Hugo_Symbol <- factor(crispr_data_g2p$Hugo_Symbol, levels = g2p_genes)
crispr_data_g2p_color <- as.character(crispr_data_g2p$Color_Nonsilent)
names(crispr_data_g2p_color) <- crispr_data_g2p$Mutation_Status_Nonsilent

crispr_label_text <- data.frame(p.signif.adj = crispr_signif_g2p_gene$p.signif.adj, p.signif = crispr_signif_g2p_gene$p.signif, Hugo_Symbol = crispr_signif_g2p_gene$Hugo_Symbol)
crispr_label_text <- filter(crispr_label_text, Hugo_Symbol %in% g2p_genes)
crispr_label_text$Hugo_Symbol <- factor(crispr_label_text$Hugo_Symbol, levels = g2p_genes)

crispr_data_g2p_plot <- ggplot(data = crispr_data_g2p, aes(x = Mutation_Status_Nonsilent, y = Score)) +
  facet_wrap(~ Hugo_Symbol, drop = FALSE, nrow = 1) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85)) +
  scale_fill_manual(values = crispr_data_g2p_color) +
  guides(color = FALSE) +
  geom_text(data = crispr_label_text, mapping = aes(x = 1.5, y = 2, label = p.signif), nudge_y = 0.1) +
  theme(strip.text.x = element_text(size = 10, angle = 45), legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
  labs(title = "CRISPR", fill = "Mutation Status", y = "CERES Score", x = "Mutation Status")
crispr_data_g2p_plot

# ggsave("./plots_18Q3/manuscript/crispr_data_g2p_plot.png", crispr_data_g2p_plot, device = "png", dpi = 450, width = 20, height = 5, units = "in")
```

### CCLE

Plot code:
```{r, eval=FALSE}
ccle_drugs <- as.character(unique(ccle_data_ptmuts$Drug))
ccle_g2p_drugboxplots <- lapply(ccle_drugs, makeDrugBoxplots, dataset = "CCLE")
ccle_g2p_drugboxplots_paths <- paste0(ccle_drugs, "_ccle_drugboxplots.png")
pwalk(list(ccle_g2p_drugboxplots_paths, ccle_g2p_drugboxplots), ggsave, path = "./plots_18Q3/manuscript/ccle_g2p_drugboxplots", dpi = 450, width = 20, height = 5, units = "in")
```

Data management:
```{r, eval=FALSE}
ccle_g2p_drugboxplots <- paste0(list.files("./plots_18Q3/manuscript/ccle_g2p_drugboxplots", full.names = TRUE))
names(ccle_g2p_drugboxplots) <- str_replace_all(ccle_g2p_drugboxplots, c("_ccle_drugboxplots.png" = "", "./plots_18Q3/manuscript/ccle_g2p_drugboxplots/" = ""))
bsselect(ccle_g2p_drugboxplots, type = "img", live_search = TRUE, show_tick = TRUE, height = 100)
```

### CTRP

Plot code:
```{r, eval=FALSE}
ctrp_drugs <- as.character(unique(ctrp_data_ptmuts$Drug))
ctrp_g2p_drugboxplots <- lapply(ctrp_drugs, makeDrugBoxplots, dataset = "ctrp")
ctrp_g2p_drugboxplots_paths <- paste0(ctrp_drugs, "_ctrp_drugboxplots.png")
pwalk(list(ctrp_g2p_drugboxplots_paths, ctrp_g2p_drugboxplots), ggsave, path = "./plots_18Q3/manuscript/ctrp_g2p_drugboxplots", dpi = 450, width = 20, height = 5, units = "in")
```

Data management:
```{r, eval=FALSE}
ctrp_g2p_drugboxplots <- paste0(list.files("./plots_18Q3/manuscript/ctrp_g2p_drugboxplots", full.names = TRUE))
names(ctrp_g2p_drugboxplots) <- str_replace_all(ctrp_g2p_drugboxplots, c("_ctrp_drugboxplots.png" = "", "./plots_18Q3/manuscript/ctrp_g2p_drugboxplots/" = ""))
bsselect(ctrp_g2p_drugboxplots, type = "img", live_search = TRUE, show_tick = TRUE, height = 100)
```

### GDSC

Plot code:
```{r, eval=FALSE}
gdsc_drugs <- as.character(unique(gdsc_data_ptmuts$Drug))
gdsc_g2p_drugboxplots <- lapply(gdsc_drugs, makeDrugBoxplots, dataset = "gdsc")
gdsc_g2p_drugboxplots_paths <- paste0(gdsc_drugs, "_gdsc_drugboxplots.png")
pwalk(list(gdsc_g2p_drugboxplots_paths, gdsc_g2p_drugboxplots), ggsave, path = "./plots_18Q3/manuscript/gdsc_g2p_drugboxplots", dpi = 450, width = 20, height = 5, units = "in")
```

Data management:
```{r, eval=FALSE}
gdsc_g2p_drugboxplots <- paste0(list.files("./plots_18Q3/manuscript/gdsc_g2p_drugboxplots", full.names = TRUE))
names(gdsc_g2p_drugboxplots) <- str_replace_all(gdsc_g2p_drugboxplots, c("_gdsc_drugboxplots.png" = "", "./plots_18Q3/manuscript/gdsc_g2p_drugboxplots/" = ""))
bsselect(gdsc_g2p_drugboxplots, type = "img", live_search = TRUE, show_tick = TRUE, height = 100)
```


## G2P level A drug-gene associations only

### CCLE

```{r fig.height=4, fig.width=12}
ccle_g2p_order <- as.character(ccle_signif_g2p$Drug_Gene)
ccle_data_g2p$Drug_Gene_Ordered <- factor(ccle_data_g2p$Drug_Gene, levels = ccle_g2p_order, labels = sapply(strsplit(x = ccle_g2p_order, split = "_"), function(x) paste0(x[1], "\n", x[2])))
ccle_signif_g2p$Drug_Gene_Ordered <- factor(ccle_signif_g2p$Drug_Gene, levels = ccle_g2p_order, labels = sapply(strsplit(x = ccle_g2p_order, split = "_"), function(x) paste0(x[1], "\n", x[2])))
ccle_data_g2p_color <- as.character(ccle_data_g2p$Color_Nonsilent)
names(ccle_data_g2p_color) <- ccle_data_g2p$Mutation_Status_Nonsilent

ccle_label_text <- data.frame(p.signif.adj = ccle_signif_g2p$p.signif.adj, p.signif = ccle_signif_g2p$p.signif, Drug_Gene_Ordered = ccle_signif_g2p$Drug_Gene_Ordered)

ccle_data_g2p_plot <- ggplot(data = ccle_data_g2p, aes(x = Mutation_Status_Nonsilent, y = AUC)) +
  facet_wrap(~ Drug_Gene_Ordered, drop = FALSE, nrow = 1) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85), outlier.shape = 3, outlier.size = 0.5) +
  scale_fill_manual(values = ccle_data_g2p_color) +
  guides(color = FALSE) +
  geom_text(data = ccle_label_text, mapping = aes(x = 1.5, y = 7, label = p.signif), nudge_y = 0.1) +
  theme(legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
  labs(title = "CCLE", fill = "Mutation Status", y = "AUC", x = "Mutation Status")
ccle_data_g2p_plot

# ggsave("./plots_18Q3/manuscript/ccle_data_g2p_plot.png", ccle_data_g2p_plot, device = "png", dpi = 450, width = 12, height = 4, units = "in")
```

### CTRP

```{r fig.height=7, fig.width=12}
ctrp_g2p_order <- as.character(ctrp_signif_g2p$Drug_Gene)
ctrp_data_g2p$Drug_Gene_Ordered <- factor(ctrp_data_g2p$Drug_Gene, levels = ctrp_g2p_order, labels = sapply(strsplit(x = ctrp_g2p_order, split = "_"), function(x) paste0(x[1], "\n", x[2])))
ctrp_signif_g2p$Drug_Gene_Ordered <- factor(ctrp_signif_g2p$Drug_Gene, levels = ctrp_g2p_order, labels = sapply(strsplit(x = ctrp_g2p_order, split = "_"), function(x) paste0(x[1], "\n", x[2])))
ctrp_data_g2p_color <- as.character(ctrp_data_g2p$Color_Nonsilent)
names(ctrp_data_g2p_color) <- ctrp_data_g2p$Mutation_Status_Nonsilent

ctrp_label_text <- data.frame(p.signif.adj = ctrp_signif_g2p$p.signif.adj, p.signif = ctrp_signif_g2p$p.signif, Drug_Gene_Ordered = ctrp_signif_g2p$Drug_Gene_Ordered)

ctrp_data_g2p_plot <- ggplot(data = ctrp_data_g2p, aes(x = Mutation_Status_Nonsilent, y = AUC)) +
  facet_wrap(~ Drug_Gene_Ordered, drop = FALSE, nrow = 3) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85), outlier.shape = 3, outlier.size = 0.5) +
  scale_fill_manual(values = ctrp_data_g2p_color) +
  guides(color = FALSE) +
  geom_text(data = ctrp_label_text, mapping = aes(x = 1.5, y = 22.5, label = p.signif), nudge_y = 0.1) +
  theme(legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
  labs(title = "CTRP", fill = "Mutation Status", y = "AUC", x = "Mutation Status")
ctrp_data_g2p_plot

# ggsave("./plots_18Q3/manuscript/ctrp_data_g2p_plot.png", ctrp_data_g2p_plot, device = "png", dpi = 450, width = 12, height = 7, units = "in")
```

### GDSC

```{r fig.height=7, fig.width=12}
gdsc_g2p_order <- as.character(gdsc_signif_g2p$Drug_Gene)
gdsc_data_g2p$Drug_Gene_Ordered <- factor(gdsc_data_g2p$Drug_Gene, levels = gdsc_g2p_order, labels = sapply(strsplit(x = gdsc_g2p_order, split = "_"), function(x) paste0(x[1], "\n", x[2])))
gdsc_signif_g2p$Drug_Gene_Ordered <- factor(gdsc_signif_g2p$Drug_Gene, levels = gdsc_g2p_order, labels = sapply(strsplit(x = gdsc_g2p_order, split = "_"), function(x) paste0(x[1], "\n", x[2])))
gdsc_data_g2p_color <- as.character(gdsc_data_g2p$Color_Nonsilent)
names(gdsc_data_g2p_color) <- gdsc_data_g2p$Mutation_Status_Nonsilent

gdsc_label_text <- data.frame(p.signif.adj = gdsc_signif_g2p$p.signif.adj, p.signif = gdsc_signif_g2p$p.signif, Drug_Gene_Ordered = gdsc_signif_g2p$Drug_Gene_Ordered)

gdsc_data_g2p_plot <- ggplot(data = gdsc_data_g2p, aes(x = Mutation_Status_Nonsilent, y = AUC)) +
  facet_wrap(~ Drug_Gene_Ordered, drop = FALSE, nrow = 3) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(mapping = aes(fill = Mutation_Status_Nonsilent), position = position_dodge(0.85), outlier.shape = 3, outlier.size = 0.5) +
  scale_fill_manual(values = gdsc_data_g2p_color) +
  guides(color = FALSE) +
  geom_text(data = gdsc_label_text, mapping = aes(x = 1.5, y = 1, label = p.signif), nudge_y = 0.1) +
  theme(legend.position = "top", axis.ticks.x = element_blank(), axis.text.x = element_blank(), axis.title.x = element_blank()) +
  labs(title = "GDSC", fill = "Mutation Status", y = "AUC", x = "Mutation Status")
gdsc_data_g2p_plot

# ggsave("./plots_18Q3/manuscript/gdsc_data_g2p_plot.png", gdsc_data_g2p_plot, device = "png", dpi = 450, width = 12, height = 7, units = "in")
```

# CRISPR: Point mutation mapping

--------------------------------------------------------------------------------

Match specific point mutations.
```{r fig.height=5, fig.width=20, eval=FALSE}
g2p_indications <- filter(read.delim("./data_munging/data_mutation_associations_appended.csv", sep = "\t", header = TRUE), Evidence.Level == "A")
maf_g2p_indications <- filter(maf_raw, Genome_Change %in% g2p_indications$MutationName)
crispr_g2p_indications <- merge(crispr_data_ptmuts, maf_g2p_indications, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
crispr_g2p_indications$Mutation_Status_Nonsilent <- ifelse(is.na(crispr_g2p_indications$Mutation_Status_Nonsilent), "Wildtype", crispr_g2p_indications$Mutation_Status_Nonsilent)
dup_g2p_indications <- filter(crispr_g2p_indications[, c("Hugo_Symbol", "CCLE_Name", "DepMap_ID")] %>% group_by(Hugo_Symbol, CCLE_Name, DepMap_ID) %>% tally(), n > 1)
crispr_g2p_indications <- merge(crispr_g2p_indications, dup_g2p_indications, by = c("Hugo_Symbol", "CCLE_Name", "DepMap_ID"), all.x = TRUE)
crispr_g2p_indications$Genome_Change <- factor(crispr_g2p_indications$Genome_Change, levels = unique(crispr_g2p_indications$Genome_Change))
crispr_g2p_indications_nonNA <- filter(crispr_g2p_indications, is.na(n))

crispr_g2p_indications_plot <- ggplot(crispr_g2p_indications, aes(x = Hugo_Symbol, y = Score)) +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(alpha = 0.5, color = "lightgray", outlier.shape = NA) +
  geom_jitter(width = 0.3, mapping = aes(color = Mutation_Status_Nonsilent), size = 1) +
  theme_light() +
  theme(legend.title = element_blank()) +
  labs(x = "Gene", y = "CERES Score")
crispr_g2p_indications_plot

crispr_g2p_indications_muts_plot <- ggplot(crispr_g2p_indications, aes(x = Protein_Change, y = Score)) +
  facet_grid(~ Hugo_Symbol, scales = "free_x", space = "free") +
  geom_hline(yintercept = 0, lty = 2, color = "darkgray") +
  geom_boxplot(alpha = 0.5, color = "lightgray", outlier.shape = NA) +
  geom_jitter(width = 0.1, mapping = aes(color = Mutation_Status_Nonsilent), size = 1) +
  theme_light() +
  theme(legend.title = element_blank(), axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Gene", y = "CERES Score")
crispr_g2p_indications_muts_plot

# ggsave("./plots_18Q3/manuscript/crispr_g2p_indications_plot_test.png", crispr_g2p_indications_plot, width = 12, height = 5, units = "in")
# ggsave("./plots_18Q3/manuscript/crispr_g2p_indications_muts_plot_test.png", crispr_g2p_indications_muts_plot, width = 20, height = 5, units = "in")
```



# References

--------------------------------------------------------------------------------

Barretina, J., Caponigro, G., Stransky, N., Venkatesan, K., Margolin, A. A., Kim, S., … Garraway, L. A. (2012). The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity. Nature, 483(7391), 603–607. https://doi.org/10.1038/nature11003

Broad Institute Cancer Dependency Map; Cancer Data Science (2018): Cancer Dependency Map, CRISPR Avana dataset 18Q3 (Avana_public_18Q3). figshare. Fileset. doi:10.6084/m9.figshare.6931364.v1

Consortium, T. C. C. L. E., & Consortium, T. G. of D. S. in C. (2015). Pharmacogenomic agreement between two cancer cell line data sets. Nature, 528(7580), 84–87. https://doi.org/10.1038/nature15736

Data Science, Cancer (2018): DEMETER2 data. figshare. Fileset. doi:10.6084/m9.figshare.6025238.v2

Doench, J. G., Fusi, N., Sullender, M., Hegde, M., Vaimberg, E. W., Donovan, K. F., … Root, D. E. (2016). Optimized sgRNA design to maximize activity and minimize off-target effects of CRISPR-Cas9. Nature Biotechnology, 34(2), 184–191. https://doi.org/10.1038/nbt.3437

Meyers, R. M., Bryan, J. G., McFarland, J. M., Weir, B. A., Sizemore, A. E., Xu, H., … Tsherniak, A. (2017). Computational correction of copy-number effect improves specificity of CRISPR-Cas9 essentiality screens in cancer cells. Nature Genetics, 49(12), 1779–1784. https://doi.org/10.1038/ng.3984

McFarland, J. M., Ho, Z. V., Kugener, G., Dempster, J. M., Montgomery, P. G., Bryan, J. G., … Tsherniak, A. (2018). Improved estimation of cancer dependencies from large-scale RNAi screens using model-based normalization and data integration. https://doi.org/10.1101/305656


# Session information

--------------------------------------------------------------------------------

```{r}
print(sessionInfo())
```

