---
title: "Omix"
author: "Sally Claridge"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  html_document:
    theme: cerulean
    number_sections: true
    toc: true
    toc_depth: 3
    toc_float:
      collapsed: true
    code_folding: hide
    pandoc_args: [
      "+RTS", "-K2048m",
      "-RTS"
    ]
---

Analysis of the [Broad's Avana CRISPR](https://depmap.org/portal/download/) (Broad Institute Cancer Dependency Map 2018, Meyers et al. 2017), the [Cancer Therapeutics Response Portal](https://ocg.cancer.gov/programs/ctd2/data-portal) (CTRP) drug screen dataset (v2) (Basu et al., 2013; M. G. Rees et al., 2016; B. Seashore-Ludlow et al., 2015), the [Genomics of Drug Sensitivity in Cancer](https://www.cancerrxgene.org/downloads) (GDSC) drug screen dataset (Garnett et al., 2012; Yang et al., 2012) , and the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) (CCLE) drug screen dataset (J. Barretina et al., 2012; T. C. C. L. E. Consortium & Consortium, 2015).

The Avana screen produced results using [CERES](https://depmap.org/ceres/) (Meyers et al. 2017) ([GitHub](https://github.com/cancerdatasci/ceres)), which generates gene dependency scores from sgRNA depletion scores from gene essentiality screens and eliminates bias arising from the effect of copy number variation on Cas9 DNA cleavage. The lower the CERES score, the higher the likelihood that the gene is essential in the associated cell line. Scores are scaled per cell line such that a score of 0 is the median effect of nonessential genes and -1 is the median effect of common core essential genes.

Annotation files for mutation status, copy number, and gene expression of the CCLE and CTRP datasets (Consortium and Consortium 2015, Barretina et al. 2012) were downloaded from the [DepMap Data Portal](https://depmap.org/portal/download/). Note that the group that conducted the GDSC screen also collected their own genomic annotation information for mutation status, copy number, and gene expression, but we did not use it.

# NOTES

--------------------------------------------------------------------------------

- Quality check: cluster TCGA samples and cell linesâ€”verify that same tumor types cluster together rather than clustering by cell lines and tumor samples (Kaiqiao)
- Calculate Pearson correlations between DepMap and TCGA feature sets and create heatmaps (CN, GE, mutation frequency), see Barretina et al. (2016) supplement for example methods and plots
- Look at Barretina et al. supplement for PCA and mutation load correlations)


# SET UP

--------------------------------------------------------------------------------

```{r setup, include=FALSE}
set.seed(1234)
setwd("/Users/claris01/Box\ Sync/SallyProjects/htscreens/omix")

knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
# num_cores <- parallel::detectCores()
options(mc.cores = parallel::detectCores(), expressions = 5e5)
```

Edited `.Renviron` to avoid "Error: memory exhausted (limit reached?)" when melting the GE dataset, as per the suggestion from [StackOverflow](https://stackoverflow.com/questions/51295402/r-on-macos-error-vector-memory-exhausted-limit-reached):
```
cd ~
touch .Renviron
open .Renviron
# Added following line to .Renviron:
R_MAX_VSIZE=100Gb
```

```{r load_libraries}
library(circlize)
library(ComplexHeatmap)
library(crunch)
library(ggpubr)
library(ggrepel)
library(ggsignif)
library(kableExtra)
library(naniar)
library(openxlsx)
library(reshape2)
library(tidyverse)
library(VennDiagram)

theme_set(theme_light())
```

```{r functions}
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) {
    df[[i]] = as.character(df[[i]])
  }
  return(df)
}
```


# DATA MANAGEMENT {.tabset .tabset-fade .tabset-pills}

--------------------------------------------------------------------------------

Note that the new DepMap 19Q1 release made multiple changes to variable names:

- For their universal ccancer cell line nomenclature, they reverted back to `Broad_ID` from 18Q3 rather than `DepMap_ID` that was used in 18Q4
- The `Primary.Disease` column of the cell line metadata file is more consistently named (`Subtype.Disease` now has the specific disease annotations), though there are a couple capitalization inconsistencies
- MAF file no longer has `CCLE_Name` column and has a duplicate column of the Broad IDs (i.e. `DepMap_ID` and `Tumor_Sample_Barcode`)

And changes to the data processing pipelines (the following is taken verbatim from the [DepMap data portal](https://depmap.org/portal/download/)):

- **Expression**: We use the GTEx pipeline. Since last release, there have been some updates to the versions of different components of the pipeline and minor bug fixes on GTEx's end.
- **Mutation**: In this release, we additionally process DepMap WES samples through the CGA mutation calling pipeline adapted for cell lines and available on FireCloud. We have added a column for the allele counts from this pipeline run (CGA_WES_AC). Previously, the WES_AC column was generated from running the Van Allen mutation calling pipeline for SNPs + Mutect2 for INDELs. This column is now named VA_WES_AC.
- **Copy Number**: We have moved to using the newest version of the GATK pipeline.

## Cell line metadata

DepMap 19Q1 update fixed almost all of the inconsistencies of the Primary.Disease column, though there were come capitalization inconsistencies.
```{r ccl_info_processing, eval=FALSE}
ccl_info <- read.delim("./data/DepMap/DepMap-2019q1-celllines_v2_20190305.csv", sep = ",", header = TRUE, na.strings = c("", NA))

ccl_info <- subset(ccl_info, Primary.Disease != "unknown" & Primary.Disease != "primary cells" & !grepl("MERGED", CCLE_Name, fixed = TRUE))
ccl_info$Cell_line <- as.character(lapply(ccl_info$CCLE_Name, function(x) strsplit(as.character(x), split = "_")[[1]][1]))
ccl_info$Histology <- tolower(gsub("^[a-z0-9]*_", "", ccl_info$CCLE_Name, ignore.case = TRUE))

# Convert Primary.Disease entries to all lower case to fix factorization issue
ccl_info$Primary.Disease <- as.factor(tolower(ccl_info$Primary.Disease))

# Harmonize ID column name
colnames(ccl_info)[colnames(ccl_info) == "DepMap_ID"] <- "Broad_ID"

# Remove two rows that contain cell lines with two different histologies
# The histologies that are kept are verified through matching in Cellosaurus and Cell Model Passports
# KMH2, haematopoietic_and_lymphoid: https://web.expasy.org/cellosaurus/CVCL_1330, https://cellmodelpassports.sanger.ac.uk/passports/SIDM01018
# MS1, lung: https://web.expasy.org/cellosaurus/CVCL_1429, https://cellmodelpassports.sanger.ac.uk/passports/SIDM00245
ccl_info <- subset(ccl_info, CCLE_Name != "KMH2_THYROID" & CCLE_Name != "MS1_SKIN")

# Add TCGA labels
tcga_hist_converter_depmap <- filter(read.xlsx("./data/TCGA/Histology_DepMap_Xena_04052019.xlsx", sheet = 3, startRow = 1, colNames = TRUE, rowNames = FALSE), data_source == "DepMap")
colnames(tcga_hist_converter_depmap) <- c("Broad_ID", "TCGA_classification", "data_source")

tcga_hist_converter_tcga <- filter(read.xlsx("./data/TCGA/Histology_DepMap_Xena_04052019.xlsx", sheet = 3, startRow = 1, colNames = TRUE, rowNames = FALSE), data_source == "Xena_TCGA")
colnames(tcga_hist_converter_tcga) <- c("sample", "TCGA_classification", "data_source")

ccl_info <- merge(ccl_info, tcga_hist_converter_depmap, by = "Broad_ID", all.x = TRUE)

# Using Pozdeyev filtering
ccl_histologies <- select(ccl_info, Cell_line, Histology, TCGA_classification, CCLE_Name, Broad_ID) %>% drop_na(Cell_line)

# Account for cell line aliases and missing histologies
auc_na_histologies <- read.delim("./data/pozdeyev2016/auc_na_histologies.csv", sep = ",", row.names = 1)
auc_na_histologies$accession_id <- NULL
auc_na_histologies$Notes <- NULL

auc_na_histologies$Histology <- ifelse(auc_na_histologies$Histology == "", NA, as.character(auc_na_histologies$Histology))
auc_na_histologies$CCLE_Name <- ifelse(auc_na_histologies$CCLE_Name  == "", NA, as.character(auc_na_histologies$CCLE_Name ))
auc_na_histologies$Broad_ID <- ifelse(auc_na_histologies$Broad_ID == "", NA, as.character(auc_na_histologies$Broad_ID))
auc_na_histologies <- unfactorize(auc_na_histologies)

ccl_histologies <- subset(ccl_histologies, !(Broad_ID %in% auc_na_histologies$Broad_ID))
ccl_histologies <- unfactorize(ccl_histologies)
# Bind DepMap histologies and curated histologies
ccl_histologies <- bind_rows(ccl_histologies, auc_na_histologies)
write.csv(ccl_histologies, "./data/ccl_histologies_metadata.csv")
```

Load cell line metadata file:
```{r}
ccl_histologies <- read.delim("./data/ccl_histologies_metadata.csv", sep = ",", header = TRUE, row.names = 1)
```

For mutation calling, get paired gene name and cell line fields in a data frame. For this analysis, we don't care about how many mutations there are per gene or what type of mutations there are, so I didn't save more information. I took unique gene-cell line combinations since we only cared about mutation presence/absence. Add a `Mutation_Status` column denoting all entries in MAF files as mutations present in the associated cell lines.

Mutation calls (coding region, germline filtered):
```{r mutation, eval=FALSE}
maf_raw <- read.delim("./../data_munging/19Q1/depmap_19Q1_mutation_calls_20190305.csv.gz", header = TRUE, sep = ",", row.names = 1)
# Harmonize ID column name
colnames(maf_raw)[colnames(maf_raw) == "DepMap_ID"] <- "Broad_ID"
colnames(maf_raw)[colnames(maf_raw) == "Hugo_Symbol"] <- "Gene"
# Merge CCL metadata
maf_raw <- merge(select(ccl_histologies, Broad_ID, TCGA_classification), maf_raw, by = "Broad_ID", all.y = TRUE)
# Select columns
maf_df <- subset(maf_raw, select = c("Gene", "Broad_ID", "TCGA_classification", "Variant_Classification", "Reference_Allele", "Tumor_Seq_Allele1", "Genome_Change"))
maf_df$Reference_Allele <- ifelse(maf_df$Reference_Allele == "-", 0, nchar(as.character(maf_df$Reference_Allele)))
maf_df$Tumor_Seq_Allele1 <- nchar(as.character(maf_df$Tumor_Seq_Allele1))
maf_df <- maf_df[maf_df$Variant_Classification != "Silent",]
# Add Mutation_Status column
maf_df$Mutation_Status <- 1

saveRDS(maf_df, "./../data_munging/19Q1/maf_df_19Q1.rds", compress = "xz")
```

DepMap WES CN Data:
```{r copy_number, eval=FALSE}
cn <- read.delim("./../data_munging/19Q1/public_19Q1_gene_cn_20190305.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
# Remove Entrez gene IDs from colnames, format is "Hugo_Symbol (Entrez ID)"
colnames(cn) <- gsub(" .*", "", colnames(cn))
colnames(cn)[1] <- "Broad_ID"
# Merge CCL metadata with TCGA labels
cn <- merge(select(ccl_histologies, Broad_ID, TCGA_classification), cn, by = "Broad_ID", all.y = TRUE)
# Melt
cn_melt <- melt(data = cn, id.vars = c("Broad_ID", "TCGA_classification"), measure.vars = colnames(cn[3:ncol(cn)]), variable.name = "Gene", value.name = "CN")
saveRDS(cn_melt, "./../data_munging/19Q1/cn_melt_19Q1.rds", compress = "xz")
```

19Q1, CCLE RNAseq gene expression data (log2(TPM+1)):
```{r gene_expression, eval=FALSE}
ge <- read.delim("./../../htscreens_giant_files/CCLE_depMap_19Q1_TPM_20190305.csv.gz", header = TRUE, sep = ",", check.names = FALSE)
# Remove Entrez gene IDs from colnames, format is "Hugo_Symbol (Entrez ID)"
colnames(ge) <- gsub(" .*", "", colnames(ge))
colnames(ge)[1] <- "Broad_ID"
# Merge CCL metadata with TCGA labels
ge <- merge(select(ccl_histologies, Broad_ID, TCGA_classification), ge, by = "Broad_ID", all.y = TRUE)
# Melt
ge_melt <- melt(ge, id.vars = c("Broad_ID", "TCGA_classification"), measure.vars = colnames(ge)[3:ncol(ge)], variable.name = "Gene", value.name = "TPM")
saveRDS(ge_melt, "./../data_munging/19Q1/ge_melt_19Q1.rds", compress = "xz")
```

Load omics RDS:
```{r load_omics_rds}
maf_df <- readRDS("./../data_munging/19Q1/maf_df_19Q1.rds")
cn_melt <- readRDS("./../data_munging/19Q1/cn_melt_19Q1.rds")
ge_melt <- readRDS("./../data_munging/19Q1/ge_melt_19Q1.rds")
```

## CRISPR

```{r crispr, eval=TRUE}
# Sample info from figshare
crispr_meta <- read.delim("./data/DepMap/sample_info_20190306.csv", sep = ",", header = TRUE, na.strings = c("", NA))
colnames(crispr_meta)[colnames(crispr_meta) == "CCLE_name"] <- "CCLE_Name"

# crispr <- read.delim("./../data_munging/19Q1/gene_effect_corrected_20190305.csv.gz", sep = ",", check.names = FALSE, header = TRUE)
```


# POZDEYEV ET AL (2016) SUMMARY

--------------------------------------------------------------------------------

For plotting:

- `c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")`
- `c("springgreen4", "steelblue4", "turquoise4")`

## Data wrangling

```{r pozdeyev_processing, eval=FALSE}
ccle_poz <- read.delim("./data/pozdeyev2016/CCLE_drug_response_metrics.csv.gz", sep = ",")
# ccle_test <- merge(select(ccle_poz, Drug_name, Cell_line, AUC_IC50), select(filter(auc, Database == "CCLE"), Drug_name, Cell_line, AUC_IC50), by = c("Drug_name", "Cell_line"), suffixes = c("_unmatched", "_matched_all_pairs"), all = TRUE)

ctrp_poz <- read.delim("./../../htscreens_giant_files/omix/pozdeyev2016/CTRP_drug_response_metrics.csv.gz", sep = ",")
# ctrp_test <- merge(select(ctrp_poz, Drug_name, Cell_line, AUC_IC50), select(filter(auc, Database == "CTRP"), Drug_name, Cell_line, AUC_IC50), by = c("Drug_name", "Cell_line"), suffixes = c("_unmatched", "_matched_all_pairs"), all = TRUE)

gdsc_poz <- read.delim("./data/pozdeyev2016/GDSC_drug_response_metrics.csv.gz", sep = ",")
# gdsc_test <- merge(select(gdsc_poz, Drug_name, Cell_line, AUC_IC50), select(filter(auc, Database == "GDSC"), Drug_name, Cell_line, AUC_IC50), by = c("Drug_name", "Cell_line"), suffixes = c("_unmatched", "_matched_all_pairs"), all = TRUE)

auc <- bind_rows(ccle_poz, ctrp_poz, gdsc_poz)

# Harmonize Pozdeyev cell line names with DepMap
auc$Cell_line <- toupper(gsub("-", "", auc$Cell_line))
auc$Cell_line <- ifelse(auc$Cell_line == "K052", "KO52", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "OVCAR3", "NIHOVCAR3", auc$Cell_line)
auc$Cell_line <- ifelse(auc$Cell_line == "TI73", "T173", auc$Cell_line)
auc <- merge(ccl_histologies, auc, by = "Cell_line", all.y = TRUE)

auc$WasDrugScreened <- 1
ccls_not_drugscreened <- filter(ccl_histologies, Cell_line %in% setdiff(ccl_histologies$Cell_line, auc$Cell_line))
ccls_not_drugscreened$WasDrugScreened <- 0

auc$AUC_IC50 <- as.numeric(as.character(auc$AUC_IC50))
# auc$sample_ID <- with(auc, paste(Database, Cell_line, Histology, sep = "_"))
auc$Drug_name_Database <- with(auc, paste(Drug_name, Database, sep = "_"))

num_auc_per_db <- auc %>% count(Database)

# Identify cell lines without histology indications. Write to file to edit for merging back to ccl_histologies file
# test <- filter(auc, is.na(Histology))[, 1:10]
# auc_na_hists <- unique(filter(auc, is.na(Histology))$Cell_line)
# write.csv(auc_na_hists, "./data/pozdeyev2016/auc_na_histologies.csv")

# Create merged dataset with sample_ids in the rows and drugs across the columns
auc_wide <- dcast(auc, Cell_line + Histology + TCGA_classification + CCLE_Name + Broad_ID + WasDrugScreened ~ Drug_name_Database, value.var = "AUC_IC50", fun.aggregate = mean, fill = 1000000)
auc_wide[auc_wide == 1000000] <- ""
auc_wide[auc_wide == ""] <- NA

auc_all <- bind_rows(auc_wide, ccls_not_drugscreened)

auc_all <- add_column(auc_all, InCCLE = ifelse(auc_all$Cell_line %in% unique(ccle_poz$Cell_line), 1, 0), .after = "Broad_ID")
auc_all <- add_column(auc_all, InCTRP = ifelse(auc_all$Cell_line %in% unique(ctrp_poz$Cell_line), 1, 0), .after = "Broad_ID")
auc_all <- add_column(auc_all, InGDSC = ifelse(auc_all$Cell_line %in% unique(gdsc_poz$Cell_line), 1, 0), .after = "Broad_ID")

auc_all <- add_column(auc_all, WasCRISPRScreened = ifelse(auc_all$Broad_ID %in% crispr_meta$Broad_ID, 1, 0), .after = "WasDrugScreened")
auc_all <- add_column(auc_all, HasCopyNumberData = ifelse(auc_all$Broad_ID %in% unique(cn_melt$Broad_ID)[!is.na(unique(cn_melt$Broad_ID))], 1, 0), .after = "WasCRISPRScreened")
auc_all <- add_column(auc_all, HasGeneExpressionData = ifelse(auc_all$Broad_ID %in% unique(ge_melt$Broad_ID)[!is.na(unique(ge_melt$Broad_ID))], 1, 0), .after = "WasCRISPRScreened")
auc_all <- add_column(auc_all, HasMutationData = ifelse(auc_all$Broad_ID %in% unique(maf_df$Broad_ID)[!is.na(unique(maf_df$Broad_ID))], 1, 0), .after = "WasCRISPRScreened")

# auc_all[, -c(1:4)] <- apply(auc_all[, -c(1:4)], MARGIN = 2, FUN = as.numeric)
auc_all <- subset(auc_all, !(WasDrugScreened == 0 & WasCRISPRScreened == 0 & HasMutationData == 0 & HasGeneExpressionData == 0 & HasCopyNumberData == 0))
write.csv.gz(auc_all, file = "./data/pozdeyev2016/merged_auc_v20190408.5.csv.gz", row.names = FALSE, na = "NA")

# test <- as.data.frame(table(auc_all$Cell_line))
# test <- filter(auc_all, Cell_line %in% c("KMH2", "MS1", "TT"))[, 1:12]
```

```{r pozdeyev_summaries}
auc_all <- read.delim("./data/pozdeyev2016/merged_auc_v20190408.5.csv.gz", sep = ",", check.names = FALSE)

# Unique drugs across all datasets
length(unique(sapply(strsplit(as.character(colnames(auc_all)[13:length(colnames(auc_all))]), split = "_"), function(x) x[1])))

auc_all_summary <- t(auc_all %>% select(InGDSC, InCTRP, InCCLE, HasCopyNumberData, HasMutationData, HasGeneExpressionData, WasDrugScreened, WasCRISPRScreened) %>% summarise_each(funs(count = sum)))
auc_all_summary

auc_all %>% count(Histology)
```

## Exploratory plots

```{r exploratory_plots, eval=FALSE}
# Data subsetting for easier tracking throughout plots
pgx_ccls <- filter(auc_all, WasDrugScreened == 1)
crispr_ccls <- filter(auc_all, WasCRISPRScreened == 1)
unscreened_ccls <- filter(auc_all, WasCRISPRScreened == 0 & WasDrugScreened == 0)
ge_ccls <- filter(auc_all, HasGeneExpressionData == 1)
cn_ccls <- filter(auc_all, HasCopyNumberData == 1)
mut_ccls <- filter(auc_all, HasMutationData == 1)
ccle_ccls <- filter(auc_all, InCCLE == 1)
ctrp_ccls <- filter(auc_all, InCTRP == 1)
gdsc_ccls <- filter(auc_all, InGDSC == 1)

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_pgx_crispr_overlap <- venn.diagram(list("CRISPR" = unique(crispr_ccls$Cell_line), "Drug Screen" = unique(pgx_ccls$Cell_line), "Unscreened" = unique(unscreened_ccls$Cell_line)), resolution = 5000, main = "Overlap of cancer cell lines (CCLs) screened via small-molecule and CRISPR", main.fontface = "bold", lty = rep("blank", 3), fill = c("mediumpurple1", "turquoise4", "slategray3"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 180), cat.dist = c(0.01, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
png("./figures/venn_pgx_crispr_overlap.png", res = 500, units = "in", width = 6, height = 6)
grid.newpage()
grid.draw(venn_pgx_crispr_overlap)
dev.off()

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_ccl_overlap <- venn.diagram(list("CCLE" = unique(ccle_ccls$Cell_line), "CTRP" = unique(ctrp_ccls$Cell_line), "GDSC" = unique(gdsc_ccls$Cell_line)), resolution = 5000, main = "Overlap of cancer cell lines (CCLs) screened in each dataset", main.fontface = "bold", lty = rep("blank", 3), fill = c("springgreen4", "steelblue4", "turquoise4"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 180), cat.dist = c(0.01, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
png("./figures/venn_ccl_overlap.png", res = 500, units = "in", width = 6, height = 6)
grid.newpage()
grid.draw(venn_ccl_overlap)
dev.off()

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_omics_overlap <- venn.diagram(list("Gene Expression" = unique(ge_ccls$Cell_line), "Copy Number" = unique(cn_ccls$Cell_line), "Mutation" = unique(mut_ccls$Cell_line)), resolution = 5000, main = "Overlapping omic annotations in cancer cell lines (CCLs)", main.fontface = "bold", lty = rep("blank", 3), fill = c("yellow2", "orange", "tomato"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 180), cat.dist = c(0.01, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
png("./figures/venn_omics_overlap.png", res = 500, units = "in", width = 6, height = 6)
grid.newpage()
grid.draw(venn_omics_overlap)
dev.off()

bar_ccl_per_hist <- ggplot(data = unique(select(auc_all, Cell_line, Histology))) +
  geom_bar(aes(x = Histology, fill = Histology), alpha = 0.7) +
  guides(fill = "none") +
  scale_y_continuous(breaks = seq(0, 300, by = 25), labels = seq(0, 300, by = 25)) +
  labs(y = "Number of cell lines", title = "Number of unique cell lines per histology across all cell lines") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
bar_ccl_per_hist
ggsave("./figures/barplot_ccl_per_histology.png", bar_ccl_per_hist, device = "png", dpi = 450, width = 12, height = 6, units = "in")

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_drugs_overlap <- venn.diagram(list("CCLE" = unique(ccle_poz$Drug_name), "CTRP" = unique(ctrp_poz$Drug_name), "GDSC" = unique(gdsc_poz$Drug_name)), resolution = 5000, main = "Overlap of cancer cell lines (CCLs) screened in each dataset", main.fontface = "bold", lty = rep("blank", 3), fill = c("springgreen4", "steelblue4", "turquoise4"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 180), cat.dist = c(0.01, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
png("./figures/venn_drugs_overlap.png", res = 500, units = "in", width = 6, height = 6)
grid.newpage()
grid.draw(venn_drugs_overlap)
dev.off()

bar_ccl_per_hist_per_db <- ggplot(data = unique(select(auc, Database, Cell_line, Histology))) +
  facet_wrap(~ Database) +
  geom_bar(aes(x = Histology, fill = Histology), alpha = 0.7) +
  guides(fill = "none") +
  # coord_cartesian(ylim = c(0, 130)) +
  scale_y_continuous(breaks = seq(0, 160, by = 20), labels = seq(0, 160, by = 20)) +
  labs(y = "Number of cell lines", title = "Number of unique cell lines per histology, grouped by database") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
bar_ccl_per_hist_per_db
ggsave("./figures/barplot_ccl_per_histology_per_db.png", bar_ccl_per_hist_per_db, device = "png", dpi = 450, width = 12, height = 8, units = "in")

bar_ccl_per_hist <- ggplot(data = unique(select(auc_all, Cell_line, Histology))) +
  geom_bar(aes(x = Histology, fill = Histology), alpha = 0.7) +
  guides(fill = "none") +
  scale_y_continuous(breaks = seq(0, 375, by = 25), labels = seq(0, 375, by = 25)) +
  labs(y = "Number of cell lines", title = "Number of unique cell lines per histology across all annotated cell lines") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
bar_ccl_per_hist
ggsave("./figures/barplot_ccl_per_histology.png", bar_ccl_per_hist, device = "png", dpi = 450, width = 12, height = 8, units = "in")

histo_aucic50_per_db <- ggplot(data = auc) +
  facet_wrap(~ Database, scales = "free_y") +
  geom_histogram(aes(x = AUC_IC50, fill = Database), color = "black", alpha = 0.7, binwidth = 0.025, boundary = 0, closed = "left") +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  guides(fill = "none") +
  labs(x = "AUC IC50", y = "Frequency", title = "Distributions of AUC IC50 values for all drugs in each dataset (unmatched)")
histo_aucic50_per_db
ggsave("./figures/histogram_aucic50_per_db.png", histo_aucic50_per_db, device = "png", dpi = 450, width = 12, height = 6, units = "in")
```

## README text

DATA SOURCE

- Pozdeyev, N., Yoo, M., Mackie, R., Schweppe, R. E., Tan, A. C., & Haugen, B. R. (2016). Integrating heterogeneous drug sensitivity data from cancer pharmacogenomic studies. Oncotarget, 7(32), 51619â€“51625. https://doi.org/10.18632/oncotarget.10010
- Online visualization tool: http://tanlab.ucdenver.edu/QAPC/
- Zipped file containing their R scripts and full datasets can be downloaded here: http://tanlab.ucdenver.edu/QAPC/Downloads/Drug_Sensitvity_Metrics.zip

COLUMN DESCRIPTIONS

For latest version of merged_auc dataframe:

- Cell_line: the name of the cancer cell line (CCL), harmonized across datasets by Pozdeyev et al. (2016)
- Histology: the general histology/tissue type of the CCL according to the CCLE annotations (see DepMap-2019q1-celllines_v2.csv file from the DepMap Data Portal)
- TCGA_classification: curated TCGA labels derived from CCLE, GDSC, or manual assignment by Kaiqiao Li and Sally Claridge; note that some TCGA project labels have been combined into more general categories, e.g. COAD and READ are grouped into COADREAD
- CCLE_Name: the cell line name using CCLE nomenclature, which is a concatenation of the cell line name with the CCLE-annotated histology in all caps, separated by underscores
- Broad_ID: the custom identifiers (formatted as ACH-######) used by the Broad's Cancer Dependency Map project to ease cell line name harmonization efforts 
- InCCLE: 1 means the cell line was screened in CCLE, 0 means it was not
- InCTRP: 1 means the cell line was screened in CTRP, 0 means it was not
- InGDSC: 1 means the cell line was screened in GDSC, 0 means it was not
- WasDrugScreened: 1 means the cell was screened in at least one pharmacogenomic screens, and 0 means it was not screened in any of the pharmacogenomic screens
- WasCRISPRScreened: 1 means the cell line was screened in the Avana CRISPR screen, and 0 means it was not
- HasMutationData: 1 means the cell line had at least one recorded mutation in the MAF file (see file version below), and 0 means it was not annotated in the MAF file
- HasGeneExpressionData: 1 means gene expression data is available for the cell line (see file version below), and 0 means it does not have gene expression information
- HasCopyNumberData: 1 means copy number data is available for the cell line (see file version below), and 0 means it does not have copy number information
- All other columns: compound and drug names that were screened in at least one CCL in at least one database, formatted as Drug_DatabaseName
    - NA values indictate that compound wasn't screened in the respective CCL in the respective source dataset
    - Values are in AUC_IC50, the area under rhe dose response curve calculated from IC50 model 

CHANGE LOG FOR MERGED_AUC DATAFRAME

- merged_auc_v20190319.1.csv.gz
- merged_auc_v20190319.2.csv.gz
    - NAs in Histology column (excluding most of the cell lines indicated as being from a "collaborator" in CTRP) hand-curated by searches in DepMap, COSMIC, Cellosaurus, ATCC, canSAR Black, and Cell Model Passports
    - merged_auc_v20190327.3.csv.gz
    - Reformatted to have only unique cell lines in rows
    - Cancer cell lines screened in more than one database have multiple AUC columns (column names are formatted as Drug_DatabaseName)
    - Added information on binary presence/absencee of omic annotations for each cell line (i.e. mutations, copy number, gene expression)
    - Added binary columns indicating presence in CCLE, CTRP, GDSC, drug screens in general (in 1, 2, or 3), and the Avana CRISPR screen
    - Added cell lines that have at least one type of omic data but are not in the CRISPR or pharmacogenomic screens
- merged_auc_v20190402.4.csv.gz
    - Fixed duplicate cell line rows derived from manually curating some of the CCL histologies that had been added in merged_auc_v20190319.2.csv.gz
    - Fixed HasMutationData, HasCopyNumberData, and HasGeneExpression data columns
    - Reordered columns
- merged_auc_v20190408.5.csv.gz
    - Removed duplicate cell line entries for KMH2 and MS1 that had incorrect histology entries
    - Added TCGA classifications for all cell lines possible (classifications were manually curated by Kaiqiao Li and Sally Claridge by combining CCLE and GDSC TCGA annotations and two-fold manual comparison)
    - Changed CCLE_ccl_name column to CCLE_Name


# DEPMAP/TCGA: Heatmaps

--------------------------------------------------------------------------------

Cancer cell lines and tumor samples are directly incomparable by nature, so the correlation matrices are asymmetric. The top left shows how well the features of the DepMap cancer cell lines correlate with those of the TCGA tumor samples, and the bottom right shows the reverse. The diagonal of the heatmap shows the comparison of matched TCGA classifications.

## GE

Data wrangling and processing:
```{r ge_heatmap_data, eval=FALSE, warning=FALSE}
# DepMap gene expression
ge_depmap <- read.delim("./../../htscreens_giant_files/omix/GE/GE_DepMap_filter.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
colnames(ge_depmap)[colnames(ge_depmap) == "cancer"] <- "TCGA_classification"
# Mean center scale
ge_depmap[, 4:ncol(ge_depmap)] <- apply(ge_depmap[, 4:ncol(ge_depmap)], MARGIN = 2, FUN = scale, center = TRUE, scale = TRUE)
# Melt
ge_depmap_melt <- melt(ge_depmap, id.vars = c("sample_ID", "TCGA_classification"), measure.vars = colnames(ge_depmap)[4:length(colnames(ge_depmap))], variable.name = "Gene", value.name = "TPM_scaled")
# Average scaled TPM per gene per TCGA classification
ge_depmap_avg <- ge_depmap_melt %>% group_by(TCGA_classification, Gene) %>% summarise(Mean_TPM = mean(TPM_scaled))
# Make wide
ge_depmap_wide <- dcast(data = ge_depmap_avg, formula = Gene ~ TCGA_classification, fun.aggregate = mean)
rownames(ge_depmap_wide) <- ge_depmap_wide$Gene
ge_depmap_wide$Gene <- NULL
colnames(ge_depmap_wide) <- paste(colnames(ge_depmap_wide), "DepMap", sep = "_")

# TCGA gene expression
ge_tcga <- read.delim("./../../htscreens_giant_files/omix/GE/GE_Xena_filter.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
colnames(ge_tcga)[colnames(ge_tcga) == "cancer"] <- "TCGA_classification"
# Mean center scale
ge_tcga[, 4:ncol(ge_tcga)] <- apply(ge_tcga[, 4:ncol(ge_tcga)], MARGIN = 2, FUN = scale, center = TRUE, scale = TRUE)
# Melt
ge_tcga_melt <- melt(ge_tcga, id.vars = c("sample_ID", "TCGA_classification"), measure.vars = colnames(ge_tcga)[4:length(colnames(ge_tcga))], variable.name = "Gene", value.name = "TPM_scaled")
# Average scaled TPM per gene per TCGA classification
ge_tcga_avg <- ge_tcga_melt %>% group_by(TCGA_classification, Gene) %>% summarise(Mean_TPM = mean(TPM_scaled))
# Make wide
ge_tcga_wide <- dcast(data = ge_tcga_avg, formula = Gene ~ TCGA_classification, fun.aggregate = mean)
rownames(ge_tcga_wide) <- ge_tcga_wide$Gene
ge_tcga_wide$Gene <- NULL
colnames(ge_tcga_wide) <- paste(colnames(ge_tcga_wide), "TCGA", sep = "_")

# Correlate all against all
ge_wide <- bind_cols(ge_depmap_wide, ge_tcga_wide)
ge_corr <- ge_wide %>% as.matrix %>% cor(method = "spearman") %>% as.data.frame %>% rownames_to_column(var = "var1") %>% gather(var2, value, -var1)
# Filter self-comparisons and eliminate duplicate correlations
ge_corr <- filter(ge_corr, !(grepl("DepMap", var1) & grepl("DepMap", var2)) & !(grepl("TCGA", var1) & grepl("TCGA", var2)) & grepl("TCGA", var2))
# Data management
colnames(ge_corr) <- c("DepMap", "TCGA", "r")
ge_corr <- data.frame(apply(ge_corr, 2, function(x) gsub("_TCGA|_DepMap","", x)))
ge_corr$r <- as.numeric(as.character(ge_corr$r))
# Set up coprrelation dataframe for plotting
ge_corr_wide <- dcast(ge_corr, DepMap ~ TCGA)
rownames(ge_corr_wide) <- ge_corr_wide$DepMap
ge_corr_wide$DepMap <- NULL
```

Pseudo-figure caption: This is comparison of gene expression profiles between DepMap cancer cell lines and TCGA tumor samples, both classified by TCGA project. In each dataset and for each TCGA classification, gene expression (in TPM) for each of the 5,000 most variable genes shared between the two datasets were averaged across samples. The dataset averages per TCGA classification were correlated against each other, and these Pearson/Spearman correlations (r) between TCGA and DepMap gene expression feature sets are shown in the heatmap.

Make heatmaps:
```{r ge_heatmap, eval=FALSE, fig.height=4, fig.width=3.75}
col_fun <- colorRamp2(c(-0.5, 0, 0.5, 1), c("black", "red3", "yellow", "white"))
ge_ht4order <- Heatmap(ge_corr_wide, cluster_rows = FALSE)

ge_ht_legend_list <- list(legend_direction = "horizontal", title_position = "lefttop", legend_width = unit(1.5, "in"), border = "black")

ge_ht <- Heatmap(ge_corr_wide, 
                 name = "Expression correlation",
                 col = col_fun, 
                 rect_gp = gpar(col = "black"),
                 heatmap_legend_param = ge_ht_legend_list,
                 row_order = rev(column_order(ge_ht4order)), 
                 show_column_dend = FALSE, 
                 cluster_rows = FALSE, 
                 row_names_side = "left",
                 row_title = "DepMap cancer cell lines",
                 column_title = "TCGA tumor samples", 
                 column_title_side = "bottom")
png(filename = "./figures/ge_depmap_tcga_heatmap_spearman.png", width = 7.5, height = 8, units = "in", res = 500)
draw(ge_ht, heatmap_legend_side = "top")
dev.off()
```

## CN

Data wrangling and processing:
```{r cn_heatmap_data, eval=FALSE, warning=FALSE}
# TCGA copy number
cn_tcga <- read.delim("./data/CN/CN_GDC_filter.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
colnames(cn_tcga)[colnames(cn_tcga) == "cancer"] <- "TCGA_classification"
# Melt
cn_tcga_melt <- melt(cn_tcga, id.vars = c("sample_ID", "TCGA_classification"), measure.vars = colnames(cn_tcga)[4:length(colnames(cn_tcga))], variable.name = "Gene", value.name = "CN")
# Average scaled TPM per gene per TCGA classification
cn_tcga_avg <- cn_tcga_melt %>% group_by(TCGA_classification, Gene) %>% summarise(Mean_CN = mean(CN))
# Make wide
cn_tcga_wide <- dcast(data = cn_tcga_avg, formula = Gene ~ TCGA_classification, fun.aggregate = mean)
rownames(cn_tcga_wide) <- cn_tcga_wide$Gene
cn_tcga_wide$Gene <- NULL
colnames(cn_tcga_wide) <- paste(colnames(cn_tcga_wide), "TCGA", sep = "_")

# DepMap copy number (18Q4)
cn_depmap <- read.delim("./data/CN/CN_DepMap_filter.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
colnames(cn_depmap)[colnames(cn_depmap) == "cancer"] <- "TCGA_classification"
# Melt
cn_depmap_melt <- melt(cn_depmap, id.vars = c("sample_ID", "TCGA_classification"), measure.vars = colnames(cn_depmap)[4:length(colnames(cn_depmap))], variable.name = "Gene", value.name = "CN")
# Average CN levels per gene per TCGA classification
cn_depmap_avg <- cn_depmap_melt %>% group_by(TCGA_classification, Gene) %>% summarise(Mean_CN = mean(CN))
# Make wide
cn_depmap_wide <- dcast(data = cn_depmap_avg, formula = Gene ~ TCGA_classification, fun.aggregate = mean)
rownames(cn_depmap_wide) <- cn_depmap_wide$Gene
cn_depmap_wide$Gene <- NULL
colnames(cn_depmap_wide) <- paste(colnames(cn_depmap_wide), "DepMap", sep = "_")

# Correlate all against all (18Q4)
cn_wide <- bind_cols(cn_depmap_wide, cn_tcga_wide)
cn_corr <- cn_wide %>% as.matrix %>% cor(method = "pearson") %>% as.data.frame %>% rownames_to_column(var = "var1") %>% gather(var2, value, -var1)
# Filter self-comparisons and eliminate duplicate correlations
cn_corr <- filter(cn_corr, !(grepl("DepMap", var1) & grepl("DepMap", var2)) & !(grepl("TCGA", var1) & grepl("TCGA", var2)) & grepl("TCGA", var2))
# Data management
colnames(cn_corr) <- c("DepMap", "TCGA", "r")
cn_corr <- data.frame(apply(cn_corr, 2, function(x) gsub("_TCGA|_DepMap","", x)))
cn_corr$r <- as.numeric(as.character(cn_corr$r))
# Set up coprrelation dataframe for plotting
cn_corr_wide <- dcast(cn_corr, DepMap ~ TCGA)
rownames(cn_corr_wide) <- cn_corr_wide$DepMap
cn_corr_wide$DepMap <- NULL
```

Pseudo-figure caption: This is comparison of thresholded copy number profiles between DepMap cancer cell lines and TCGA tumor samples, both classified by TCGA project. Levels are -1 for copy number loss, 0 for neutral, and 1 for gain. In each dataset and for each TCGA classification, these numeric copy number levels across all genes shared between the two datasets were averaged, and the datasets were correlated against each other. Pearson/Spearman correlations (r) between TCGA and DepMap copy number feature sets are shown in the heatmap.

Make heatmaps:
```{r cn_heatmaps, eval=FALSE, fig.height=4, fig.width=3.75}
col_fun <- colorRamp2(c(0, 0.25, 0.5, 0.75), c("black", "red3", "yellow", "white"))
cn_ht_legend_list <- list(legend_direction = "horizontal", title_position = "lefttop", legend_width = unit(1.5, "in"), border = "black")

# 18Q4
cn_ht4order <- Heatmap(cn_corr_wide, cluster_rows = FALSE)
cn_ht <- Heatmap(cn_corr_wide, 
                 name = "Copy number correlation",
                 col = col_fun, 
                 rect_gp = gpar(col = "black"),
                 heatmap_legend_param = cn_ht_legend_list,
                 row_order = rev(column_order(cn_ht4order)), 
                 show_column_dend = FALSE, 
                 cluster_rows = FALSE, 
                 row_names_side = "left",
                 row_title = "DepMap cancer cell lines",
                 column_title = "TCGA tumor samples", 
                 column_title_side = "bottom")
# png(filename = "./figures/cn_depmap_tcga_heatmap_pearson.png", width = 7.5, height = 8, units = "in", res = 500)
draw(cn_ht, heatmap_legend_side = "top")
# dev.off()
```

## Mut

Data wrangling and processing:
```{r mut_heatmap_data, eval=FALSE}
# TCGA
mut_tcga <- read.delim("./data/mutation/mutation_Xena_filter.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
mut_tcga <- unfactorize(mut_tcga)
mut_tcga[, 4:ncol(mut_tcga)] <- apply(mut_tcga[, 4:ncol(mut_tcga)], 2, function(x) ifelse(x > 1, 1, x))
colnames(mut_tcga)[colnames(mut_tcga) == "cancer"] <- "TCGA_classification"
# Melt
mut_tcga_melt <- melt(mut_tcga, id.vars = c("sample_ID", "TCGA_classification"), measure.vars = colnames(mut_tcga)[4:length(colnames(mut_tcga))], variable.name = "Gene", value.name = "Mutation_Status")
# Average CN levels per gene per TCGA classification
mut_tcga_prop <- mut_tcga_melt %>% group_by(TCGA_classification, Gene) %>% summarise(Proportion_Mut = sum(Mutation_Status)/n())
# Make wide
mut_tcga_wide <- dcast(data = mut_tcga_prop, formula = Gene ~ TCGA_classification, fun.aggregate = mean)
rownames(mut_tcga_wide) <- mut_tcga_wide$Gene
mut_tcga_wide$Gene <- NULL
colnames(mut_tcga_wide) <- paste(colnames(mut_tcga_wide), "TCGA", sep = "_")

# DepMap
mut_depmap <- read.delim("./data/mutation/mutation_DepMap_filter.csv.gz", sep = ",", header = TRUE, check.names = FALSE)
mut_depmap <- unfactorize(mut_depmap)
mut_depmap[, 4:ncol(mut_depmap)] <- apply(mut_depmap[, 4:ncol(mut_depmap)], 2, function(x) ifelse(x > 1, 1, x))
colnames(mut_depmap)[colnames(mut_depmap) == "cancer"] <- "TCGA_classification"
# Melt
mut_depmap_melt <- melt(mut_depmap, id.vars = c("sample_ID", "TCGA_classification"), measure.vars = colnames(mut_depmap)[4:length(colnames(mut_depmap))], variable.name = "Gene", value.name = "Mutation_Status")
# Average CN levels per gene per TCGA classification
mut_depmap_prop <- mut_depmap_melt %>% group_by(TCGA_classification, Gene) %>% summarise(Proportion_Mut = sum(Mutation_Status)/n())
# Make wide
mut_depmap_wide <- dcast(data = mut_depmap_prop, formula = Gene ~ TCGA_classification, fun.aggregate = mean)
rownames(mut_depmap_wide) <- mut_depmap_wide$Gene
mut_depmap_wide$Gene <- NULL
colnames(mut_depmap_wide) <- paste(colnames(mut_depmap_wide), "DepMap", sep = "_")

# Correlate all against all (18Q4)
mut_wide <- bind_cols(mut_depmap_wide, mut_tcga_wide)
mut_corr <- mut_wide %>% as.matrix %>% cor(method = "spearman") %>% as.data.frame %>% rownames_to_column(var = "var1") %>% gather(var2, value, -var1)
# Filter self-comparisons and eliminate duplicate correlations
mut_corr <- filter(mut_corr, !(grepl("DepMap", var1) & grepl("DepMap", var2)) & !(grepl("TCGA", var1) & grepl("TCGA", var2)) & grepl("TCGA", var2))
# Data management
colnames(mut_corr) <- c("DepMap", "TCGA", "r")
mut_corr <- data.frame(apply(mut_corr, 2, function(x) gsub("_TCGA|_DepMap","", x)))
mut_corr$r <- as.numeric(as.character(mut_corr$r))
# Set up coprrelation dataframe for plotting
mut_corr_wide <- dcast(mut_corr, DepMap ~ TCGA)
rownames(mut_corr_wide) <- mut_corr_wide$DepMap
mut_corr_wide$DepMap <- NULL
```

Pseudo-figure caption: This is comparison of binary gene-level mutation frequency profiles between DepMap cancer cell lines and TCGA tumor samples, both classified by TCGA project. In each dataset and for each TCGA classification, the proportion of samples containing at least one mutation was calculated for all genes shared between the two datasets, and the proportions were correlated against each other. Pearson/Spearman correlations (r) between TCGA and DepMap copy number feature sets are shown in the heatmap.

Genes in the TCGA mutation dataset were analyzed if they were mutated in >=4% annotated samples and had a mutaiton in at least one cancer cell line from DepMap.

Make heatmaps:
```{r cn_heatmaps, eval=FALSE, fig.height=4, fig.width=3.75}
col_fun <- colorRamp2(c(0, 0.15, 0.35, 0.5), c("black", "red3", "yellow", "white"))
mut_ht_legend_list <- list(legend_direction = "horizontal", title_position = "lefttop", legend_width = unit(1.5, "in"), border = "black")

# 18Q4
mut_ht4order <- Heatmap(mut_corr_wide, cluster_rows = FALSE)
mut_ht <- Heatmap(mut_corr_wide, 
                 name = "Mutation correlation",
                 col = col_fun, 
                 rect_gp = gpar(col = "black"),
                 heatmap_legend_param = mut_ht_legend_list,
                 row_order = rev(column_order(mut_ht4order)), 
                 show_column_dend = FALSE, 
                 cluster_rows = FALSE, 
                 row_names_side = "left",
                 row_title = "DepMap cancer cell lines",
                 column_title = "TCGA tumor samples", 
                 column_title_side = "bottom")
# png(filename = "./figures/mut_depmap_tcga_heatmap_spearman.png", width = 7.5, height = 8, units = "in", res = 500)
draw(mut_ht, heatmap_legend_side = "top")
# dev.off()
```













# OLD

--------------------------------------------------------------------------------

## Data management

```{r, eval=FALSE}
tcga_hist_converter <- read.xlsx("./data/TCGA/cell_line_annotation.xlsx", sheet = 1, startRow = 1, colNames = TRUE, rowNames = FALSE)
tcga_hist_converter <- add_column(.data = tcga_hist_converter, TCGA_manual = NA, .after = "TCGA_combine")
tcga_hist_converter <- tcga_hist_converter[with(tcga_hist_converter, order(primary_DepMap, subtype_DepMap, primary_site_CTRP, primary_hist_CTRP, subtype_CTRP, tissue1_GDSC)),]

# For manual curation of TCGA labels
ccls_for_manual_tcga_curation <- filter(tcga_hist_converter, is.na(TCGA_combine))
ccls_for_manual_tcga_curation <- ccls_for_manual_tcga_curation[with(ccls_for_manual_tcga_curation, order(primary_DepMap, subtype_DepMap, primary_site_CTRP, primary_hist_CTRP, subtype_CTRP, tissue1_GDSC)),]
write.csv(ccls_for_manual_tcga_curation, "./data/TCGA/cell_line_annotation_noTCGA.csv")
tcga_hist_converter <- filter(tcga_hist_converter, !is.na(TCGA_combine))
ccls_curated <- read.delim("./data/TCGA/cell_line_annotation_noTCGA_curated.csv", sep = ",", row.names = 1)
tcga_histsconverter <- bind_rows(tcga_hist_converter, ccls_curated)
tcga_hist_converter$TCGA_combine <- with(tcga_hist_converter, ifelse(is.na(TCGA_combine), as.character(TCGA_manual), TCGA_combine))
# write.csv(tcga_hist_converter, "./data/TCGA/cell_line_annotation_curated_sally_20190404.csv")
```

## Pozdeyev et al (2016) matched data

```{r, eval=FALSE}
auc_matched <- read.delim("./data_munging/pharmacogenomics_raw_files/celline_stack.csv.gz", sep = ",")
auc_matched <- merge(ccl_histologies, auc_matched , by = "Cell_line", all.y = TRUE)
num_auc_per_db_matched <- auc_matched %>% count(Database)
ccl_per_db_matched <- unique(select(auc_matched , Database, Cell_line))
num_ccl_per_db_matched <- unique(select(auc_matched , Database, Cell_line)) %>% count(Database)
# only CCLs represetented in 2 or 3 databases were included
num_db_per_ccl_matched <- unique(select(auc_matched , Database, Cell_line)) %>% count(Cell_line)
```

Plots:
```{r, eval=FALSE}
futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_ccl_matched_poz <- venn.diagram(list("CCLE" = unique(filter(ccl_per_db_matched, Database == "CCLE"))$Cell_line, "CTRP" = unique(filter(ccl_per_db_matched, Database == "CTRP"))$Cell_line, "GDSC" = unique(filter(ccl_per_db_matched, Database == "GDSC"))$Cell_line), resolution = 5000, main = "Overlap of cancer cell lines (CCLs) screened in matched_all_pairs dataset", main.fontface = "bold", lty = rep("blank", 3), fill = c("springgreen4", "steelblue4", "turquoise4"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 180), cat.dist = c(0.01, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
png("./gcn_denise/pozdeyev_matched/pozdeyev_venn_ccl_matched.png", res = 500, units = "in", width = 6, height = 6)
grid.newpage()
grid.draw(venn_ccl_matched_poz)
dev.off()

num_ccl_per_db_matched_bar <- ggplot(data = unique(select(auc_matched , Database, Cell_line))) +
  geom_bar(aes(x = Database, fill = Database), alpha = 0.7) +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) + 
  guides(fill = "none") +
  scale_y_continuous(breaks = seq(0, 700, by = 50), labels = seq(0, 700, by = 50)) +
  geom_text(data = num_ccl_per_db_matched, aes(x = Database, y = n + 20, label = n)) +
  labs(y = "Number of cell lines", title = "Number of unique cell lines per dataset")
num_ccl_per_db_matched_bar
ggsave("./gcn_denise/pozdeyev_matched/pozdeyev_ccl_per_db_matched.png", num_ccl_per_db_matched_bar, device = "png", dpi = 450, width = 12, height = 8, units = "in")

num_ccl_per_hist_per_db_matched_bar <- ggplot(data = unique(select(auc_matched , Database, Cell_line, Histology))) +
  facet_wrap(~ Database, scales = "free_y") +
  geom_bar(aes(x = Histology, fill = Histology), alpha = 0.7) +
  guides(fill = "none") +
  coord_cartesian(ylim = c(0, 130)) +
  scale_y_continuous(breaks = seq(0, 130, by = 10), labels = seq(0, 130, by = 10)) +
  labs(y = "Number of cell lines", title = "Number of unique cell lines per histology, grouped by database") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
num_ccl_per_hist_per_db_matched_bar
ggsave("./gcn_denise/pozdeyev_matched/pozdeyev_ccl_per_histology_per_db_matched.png", num_ccl_per_hist_per_db_matched_bar, device = "png", dpi = 450, width = 12, height = 8, units = "in")

num_ccl_per_hist_matched_bar <- ggplot(data = unique(select(auc_matched , Cell_line, Histology))) +
  geom_bar(aes(x = Histology, fill = Histology), alpha = 0.7) +
  guides(fill = "none") +
  coord_cartesian(ylim = c(0, 130)) +
  scale_y_continuous(breaks = seq(0, 130, by = 10), labels = seq(0, 130, by = 10)) +
  labs(y = "Number of cell lines", title = "Number of unique cell lines per histology across all three databases") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
num_ccl_per_hist_matched_bar
ggsave("./gcn_denise/pozdeyev_matched/pozdeyev_ccl_per_histology_matched.png", num_ccl_per_hist_matched_bar, device = "png", dpi = 450, width = 12, height = 8, units = "in")

aucic50_per_db_matched <- ggplot(data = auc_matched ) +
  facet_wrap(~ Database) +
  geom_histogram(aes(x = AUC_IC50, fill = Database), color = "black", alpha = 0.7, binwidth = 0.025, boundary = 0, closed = "left") +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  guides(fill = "none") +
  geom_text(data = num_auc_per_db_matched, aes(x = 0.5, y = 4750, label = paste0("n = ", formatC(n, format = "d", big.mark = ",")))) +
  scale_y_continuous(breaks = seq(0, 5000, by = 500), labels = seq(0, 5000, by = 500)) +
  labs(x = "AUC IC50", y = "Frequency", title = "Distributions of AUC IC50 values for all drugs in each dataset (filtered by matching)")
aucic50_per_db_matched
ggsave("./gcn_denise/pozdeyev_matched/pozdeyev_aucic50_per_db_matched.png", aucic50_per_db_matched, device = "png", dpi = 450, width = 12, height = 6, units = "in")
```

## Denise's GCN dataframes

Code for Denise's graph convolutional networks analyses (see meeting notes from 6 March 2019).
- Convert drug/compound names to all capitalized for harmonization.
- Convert all cell line histologies to lower case with words separated by an underscore
- Maintain same rows across all four data frames, filling with NAs as necessary
- For each dataframe, maintain same columns for all three source datasets
- Columns in final dataframes: `source`, `source_ccl_name`, `source_histology`, `sample_ID`, and `genes/compounds/mutations[p]`

### AUC

Column descriptions for `pgx_all_auc`:

- `source`: the dataset from which the data comes from (either CCLE, CTRP, or GDSC)
- `source_ccl_name`: the name of the cancer cell line (CCL) included in the source dataset
    - Note that all CCLs (excluding matched normal tissues and "merged" CCLs) from the CCLE dataset (N = 1,645) were included in `pgx_all_auc` for imputation
- `source_histology`: the general histology/tissue type of the CCL
- `sample_ID`: custom identifiers for our datasets that are concatenations of the `source`, `source_ccl_name`, and `source_histology`
- `CCLE_ccl_name`: the cell line name using CCLE nomenclature, which is a concatenation of the cell line name with the CCLE-annotated histology in all caps, separated by underscores
- `Broad_ID`: the custom identifiers (formatted as ACH-######) used by the Broad's Cancer Dependency Map project to ease cell line name harmonization efforts 
- All other columns: compound and drug names that were screened in at least one CCL in at least one source dataset
    - NA values indictate that compound wasn't screened in the respective CCL in the respective source dataset

```{r, eval=FALSE}
ccl_info$sample_ID <- with(ccl_info, paste(source, source_ccl_name, source_histology, sep = "_"))

ccle <- read.delim("./data_munging/pharmacogenomics_raw_files/CCLE_NP24.2009_Drug_data_2015.02.24_20190307.csv", sep = ",", header  = TRUE, check.names = FALSE)
ctrp <- read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.data.curves_post_qc.txt.gz", sep = "\t", header  = TRUE, check.names = FALSE)
gdsc <- read.delim("./data_munging/pharmacogenomics_raw_files/GDSC_v17.3_fitted_dose_response_20190306.csv.gz", sep = ",", header  = TRUE, check.names = FALSE)

# CCLE
## Note that ActArea is NOT an AUC value
ccle_filt <- ccle[, c("CCLE Cell Line Name", "ActArea", "Compound")]
ccle_filt$source_ccl_name <- sapply(strsplit(as.character(ccle_filt$`CCLE Cell Line Name`), split = "_"), function(x) x[1])
ccle_filt$source_histology <- tolower(gsub("^[a-z0-9]*_", "", ccle_filt$`CCLE Cell Line Name`, ignore.case = TRUE))
ccle_filt$source_histology <- ifelse(ccle_filt$`CCLE Cell Line Name` == ccle_filt$source_ccl_name, NA, ccle_filt$source_histology)
colnames(ccle_filt)[colnames(ccle_filt) == "CCLE Cell Line Name"] <- "CCLE_ccl_name"
ccle_filt <- merge(select(ccl_info, source, source_ccl_name, source_histology, sample_ID, CCLE_ccl_name, Broad_ID), ccle_filt, by = c("source_ccl_name", "source_histology", "CCLE_ccl_name"), all = TRUE)
ccle_filt$compound_name <- toupper(ccle_filt$Compound)
colnames(ccle_filt)[colnames(ccle_filt) == "ActArea"] <- "AUC"
ccle_auc_long <- select(ccle_filt, source, source_ccl_name, source_histology, sample_ID, CCLE_ccl_name, Broad_ID, compound_name, AUC)
ccle_auc <- dcast(ccle_auc_long, source + source_ccl_name + source_histology + sample_ID + CCLE_ccl_name + Broad_ID ~ compound_name, value.var = "AUC", fun.aggregate = mean, fill = 10000)
ccle_auc[, 7:ncol(ccle_auc)] <- ccle_auc[, 7:ncol(ccle_auc)] %>% replace_with_na_all(condition = ~.x == 10000)

# CTRP
## area_under_curve: integrated area under the sigmoid-fit concentration-response curve
ctrp_filt <- ctrp[, c("experiment_id", "area_under_curve", "master_cpd_id")]
ctrp_meta_ccl <- select(read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.meta.per_cell_line_20190306.txt"), master_ccl_id, ccl_name, ccle_primary_site)
ctrp_meta_exp <- select(read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.meta.per_experiment_20190306.txt"), experiment_id, master_ccl_id)
ctrp_meta_cpd <- select(read.delim("./data_munging/pharmacogenomics_raw_files/CTRP_v20.meta.per_compound_20190306.txt"), master_cpd_id, cpd_name)
ctrp_keep <- merge(merge(ctrp_meta_ccl, ctrp_meta_exp, by = "master_ccl_id", all = TRUE), merge(ctrp_meta_cpd, ctrp_filt, by = "master_cpd_id", all = TRUE), by = "experiment_id", all = TRUE)
ctrp_keep$ccl_name <- gsub("-", "", ctrp_keep$ccl_name, ignore.case = TRUE)
ctrp_keep$cpd_name <- toupper(ctrp_keep$cpd_name)
ctrp_keep$ccle_primary_site <- ifelse(as.character(ctrp_keep$ccle_primary_site) == "", NA, as.character(ctrp_keep$ccle_primary_site))
ctrp_auc_long <- select(ctrp_keep, ccl_name, ccle_primary_site, cpd_name, area_under_curve)
colnames(ctrp_auc_long) <- c("source_ccl_name", "source_histology", "compound_name", "AUC")
ctrp_auc_long$source <- "CTRP"
ctrp_auc_long$sample_ID <- with(ctrp_auc_long, paste(source, source_ccl_name, source_histology, sep = "_"))
ctrp_auc_long <- merge(ctrp_auc_long, select(ccl_info, source_ccl_name, source_histology, CCLE_ccl_name, Broad_ID), by = c("source_ccl_name", "source_histology"), all.x = TRUE)
ctrp_auc_long_div <- ctrp_auc_long
ctrp_auc_long_div$AUC <- ctrp_auc_long_div$AUC / 100
ctrp_auc <- dcast(ctrp_auc_long, source + source_ccl_name + source_histology + sample_ID + CCLE_ccl_name + Broad_ID ~ compound_name, value.var = "AUC", fun.aggregate = mean, fill = 10000)
ctrp_auc[, 7:ncol(ctrp_auc)] <- ctrp_auc[, 7:ncol(ctrp_auc)] %>% replace_with_na_all(condition = ~.x == 10000)
                                                                                       
# GDSC
gdsc_meta_ccl <- select(read.delim("./data_munging/pharmacogenomics_raw_files/GDSC_Cell_Lines_Details.csv", sep = ","), Line, Site)
## For following 4 Lines, Site == "NS"
## Assigned new Sites by looking up cell lines in GDSC cell line portal (https://www.cancerrxgene.org/translation/CellLine#t-all)
gdsc_meta_ccl$Site[gdsc_meta_ccl$Line == "COR-L321"] <- "lung"
gdsc_meta_ccl$Site[gdsc_meta_ccl$Line == "SU-DHL-8"] <- "haematopoietic_and_lymphoid_tissue"
gdsc_meta_ccl$Site[gdsc_meta_ccl$Line == "CRO-AP3"] <- "haematopoietic_and_lymphoid_tissue"
gdsc_meta_ccl$Site[gdsc_meta_ccl$Line == "Hep_3B2_1-7"] <- "liver"
## For tollowing line, Site did not match CCLE notation, so I changed it to CCLE notation
gdsc_meta_ccl$Site[gdsc_meta_ccl$Line == "SW13"] <- "adrenal_cortex"
gdsc_filt <- gdsc[, c("CELL_LINE_NAME", "DRUG_NAME", "AUC")]
gdsc_filt$DRUG_NAME <- toupper(gdsc_filt$DRUG_NAME)
colnames(gdsc_filt)[colnames(gdsc_filt) ==  "CELL_LINE_NAME"] <- "Line"
gdsc_filt <- merge(gdsc_filt, gdsc_meta_ccl, by = "Line", all = TRUE)
gdsc_filt$Line <- gsub("-", "", gdsc_filt$Line, ignore.case = TRUE)
gdsc_auc_long <- select(gdsc_filt, Line, Site, DRUG_NAME, AUC)
colnames(gdsc_auc_long) <- c("source_ccl_name", "source_histology", "compound_name", "AUC")
gdsc_auc_long$source <- "GDSC"
gdsc_auc_long$sample_ID <- with(gdsc_auc_long, paste(source, source_ccl_name, source_histology, sep = "_"))
gdsc_auc_long$source_histology <- as.character(gdsc_auc_long$source_histology)
gdsc_auc_long <- merge(gdsc_auc_long, select(ccl_info, source_ccl_name, source_histology, CCLE_ccl_name, Broad_ID), by = c("source_ccl_name", "source_histology"), all.x = TRUE)
gdsc_auc <- dcast(gdsc_auc_long, source + source_ccl_name + source_histology + sample_ID + CCLE_ccl_name + Broad_ID ~ compound_name, value.var = "AUC", fun.aggregate = mean, fill = 10000)
gdsc_auc[, 7:ncol(gdsc_auc)] <- gdsc_auc[, 7:ncol(gdsc_auc)] %>% replace_with_na_all(condition = ~.x == 10000)

# Bind all AUC
pgx_auc_list <- list(ccle_auc, ctrp_auc, gdsc_auc)
pgx_auc_names <- unique(unlist(lapply(pgx_auc_list, names)))
pgx_all_auc <- do.call(rbind, c(lapply(pgx_auc_list, function(x) data.frame(c(x, sapply(setdiff(pgx_auc_names, names(x)), function(y) NA)), check.names = FALSE))))
# saveRDS(pgx_all_auc, "./data_munging/19Q1/pgx_all_auc.rds", compress = "xz")
# write.csv.gz(pgx_all_auc, "./data_munging/19Q1/pgx_all_auc.csv.gz", row.names = FALSE)
```

Compare histology representation for CCLE, CTRP, and GDSC:
```{r, eval=FALSE}
sort(unique(na.omit(ccle_auc_long$source_histology)))
sort(unique(na.omit(ctrp_auc_long$source_histology)))
sort(unique(na.omit(gdsc_auc_long$source_histology)))

setdiff(unique(na.omit(ccle_auc_long$source_histology)), unique(na.omit(gdsc_auc_long$source_histology)))
setdiff(unique(na.omit(gdsc_auc_long$source_histology)), unique(na.omit(ctrp_auc_long$source_histology)))
```

### Summary figures

```{r, eval=FALSE}
hist_auc_ccle <- ggplot(data = ccle_auc_long) +
  geom_histogram(mapping = aes(x = AUC, fill = source), color = "black", binwidth = 0.25, boundary = 0, closed = "left") +
  scale_x_continuous(breaks = seq(0, 8, by = 1), labels = seq(0, 8, by = 1)) +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  labs(title = "Distributions of AUC/ActArea values for all drug-cell line combinations", subtitle = "CCLE", y = "Frequency", x = "ActArea") +
  guides(fill = FALSE)
hist_auc_ctrp <- ggplot(data = ctrp_auc_long_div) +
  geom_histogram(mapping = aes(x = AUC, fill = source), color = "black", binwidth = 0.0125, boundary = 0, closed = "left") +
  scale_x_continuous(breaks = seq(0, 0.3, by = 0.025), labels = seq(0, 0.3, by = 0.025)) +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  labs(y = "Frequency", x = "AUC", subtitle = "CTRP", caption = "CTRP AUC values were divided by 100 prior to plotting.") +
  guides(fill = FALSE)
hist_auc_gdsc <- ggplot(data = gdsc_auc_long) +
  geom_histogram(mapping = aes(x = AUC, fill = source), color = "black", binwidth = 0.025, boundary = 0, closed = "left") +
  scale_x_continuous(breaks = seq(0, 1, by = 0.1), labels = seq(0, 1, by = 0.1)) +
  scale_fill_manual(values = c("CCLE" = "springgreen4", "CTRP" = "steelblue4", "GDSC" = "turquoise4")) +
  labs(y = "Frequency", x = "AUC", subtitle = "GDSC") +
  guides(fill = FALSE)
hist_auc_ccle <- ggplot_gtable(ggplot_build(hist_auc_ccle))
hist_auc_ctrp <- ggplot_gtable(ggplot_build(hist_auc_ctrp))
hist_auc_gdsc <- ggplot_gtable(ggplot_build(hist_auc_gdsc))
p_maxWidth <-  grid::unit.pmax(hist_auc_ccle$widths[2:3], hist_auc_ctrp$widths[2:3], hist_auc_gdsc$widths[2:3])
hist_auc_ccle$widths[2:3] <- p_maxWidth
hist_auc_ctrp$widths[2:3] <- p_maxWidth
hist_auc_gdsc$widths[2:3] <- p_maxWidth
hist_auc_all <- cowplot::plot_grid(hist_auc_ccle, hist_auc_ctrp, hist_auc_gdsc, nrow = 3, rel_heights = c(9, 9, 8))
hist_auc_all
# ggsave("./gcn_denise/hist_auc_all.png", hist_auc_all, device = "png", dpi = 450, width = 12, height = 8, units = "in")

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_ccl <- venn.diagram(list("CCLE" = unique(na.omit(ccl_info$source_ccl_name)), "CTRP" = unique(ctrp_auc_long$source_ccl_name), "GDSC" = unique(gdsc_auc_long$source_ccl_name)), resolution = 5000, main = "Overlap of cancer cell lines (CCLs) screened in each dataset", main.fontface = "bold", lty = rep("blank", 3), fill = c("springgreen4", "steelblue4", "turquoise4"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 0), cat.dist = c(0.01, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
# png("./gcn_denise/venn_ccl.png")
# grid.newpage()
# grid.draw(venn_ccl)
# dev.off()

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_histology <- venn.diagram(list("CCLE" = unique(na.omit(ccle_auc_long$source_histology)), "CTRP" = unique(na.omit(ctrp_auc_long$source_histology)), "GDSC" = unique(na.omit(gdsc_auc_long$source_histology))), resolution = 5000, main = "Overlap of cancer cell line histologies screened in each dataset", main.fontface = "bold", lty = rep("blank", 3), fill = c("springgreen4", "steelblue4", "turquoise4"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(0, 0, 0), cat.dist = c(0.001, 0.001, -0.01), cat.cex = 1.5, cex = 1, filename = NULL)
# png("./gcn_denise/venn_histology.png")
# grid.newpage()
# grid.draw(venn_histology)
# dev.off()

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")
venn_compound <- venn.diagram(list("CCLE" = unique(na.omit(ccle_auc_long$compound_name)), "CTRP" = unique(na.omit(ctrp_auc_long$compound_name)), "GDSC" = unique(na.omit(gdsc_auc_long$compound_name))), resolution = 5000, main = "Overlap of compounds screened in each dataset", main.fontface = "bold", lty = rep("blank", 3), fill = c("springgreen4", "steelblue4", "turquoise4"), alpha = c(0.5, 0.5, 0.5), cat.pos = c(-30, 30, 180), cat.dist = c(0.001, 0.001, 0.001), cat.cex = 1.5, cex = 1, filename = NULL)
# png("./gcn_denise/venn_compound.png")
# grid.newpage()
# grid.draw(venn_compound)
# dev.off()
```



# SESSION INFO

--------------------------------------------------------------------------------

```{r}
sessionInfo()
```

